<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 4 Enrichment Analysis | A Tutorial for Proteomics Data Analysis with R/Bioconductor</title>
  <meta name="description" content="This is a comprehensive tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 4 Enrichment Analysis | A Tutorial for Proteomics Data Analysis with R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a comprehensive tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 4 Enrichment Analysis | A Tutorial for Proteomics Data Analysis with R/Bioconductor" />
  
  <meta name="twitter:description" content="This is a comprehensive tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

<meta name="author" content="Tyler Sagendorf" />


<meta name="date" content="2021-09-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="volcano-plots.html"/>
<link rel="next" href="sea.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=".">A Tutorial for Proteomics Data Analysis in R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Welcome!</a></li>
<li class="chapter" data-level="2" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>2</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="number-of-features-identified-in-each-sample.html"><a href="number-of-features-identified-in-each-sample.html"><i class="fa fa-check"></i><b>2.1</b> Number of Features Identified in Each Sample</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="number-of-features-identified-in-each-sample.html"><a href="number-of-features-identified-in-each-sample.html#lollipop-plot"><i class="fa fa-check"></i><b>2.1.1</b> Lollipop Plot</a></li>
<li class="chapter" data-level="2.1.2" data-path="number-of-features-identified-in-each-sample.html"><a href="number-of-features-identified-in-each-sample.html#boxplot"><i class="fa fa-check"></i><b>2.1.2</b> Boxplot</a></li>
<li class="chapter" data-level="2.1.3" data-path="number-of-features-identified-in-each-sample.html"><a href="number-of-features-identified-in-each-sample.html#density-plot"><i class="fa fa-check"></i><b>2.1.3</b> Density Plot</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="estimate-blood-contamination.html"><a href="estimate-blood-contamination.html"><i class="fa fa-check"></i><b>2.2</b> Estimate Blood Contamination</a></li>
<li class="chapter" data-level="2.3" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>2.3</b> PCA</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="pca.html"><a href="pca.html#overview"><i class="fa fa-check"></i><b>2.3.1</b> Overview</a></li>
<li class="chapter" data-level="2.3.2" data-path="pca.html"><a href="pca.html#pca-plots"><i class="fa fa-check"></i><b>2.3.2</b> PCA Plots</a></li>
<li class="chapter" data-level="2.3.3" data-path="pca.html"><a href="pca.html#list-top-contributing-features"><i class="fa fa-check"></i><b>2.3.3</b> List Top-Contributing Features</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="DEA.html"><a href="DEA.html"><i class="fa fa-check"></i><b>3</b> Differential Expression Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="dea-with-limma.html"><a href="dea-with-limma.html"><i class="fa fa-check"></i><b>3.1</b> DEA with limma</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="dea-with-limma.html"><a href="dea-with-limma.html#dichotomous-predictors"><i class="fa fa-check"></i><b>3.1.1</b> Dichotomous Predictors</a></li>
<li class="chapter" data-level="3.1.2" data-path="dea-with-limma.html"><a href="dea-with-limma.html#continuous-predictors"><i class="fa fa-check"></i><b>3.1.2</b> Continuous Predictors</a></li>
<li class="chapter" data-level="3.1.3" data-path="dea-with-limma.html"><a href="dea-with-limma.html#reference-level-comparisons"><i class="fa fa-check"></i><b>3.1.3</b> Reference Level Comparisons</a></li>
<li class="chapter" data-level="3.1.4" data-path="dea-with-limma.html"><a href="dea-with-limma.html#custom-contrasts"><i class="fa fa-check"></i><b>3.1.4</b> Custom Contrasts</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="p-value-histograms.html"><a href="p-value-histograms.html"><i class="fa fa-check"></i><b>3.2</b> p-value Histograms</a></li>
<li class="chapter" data-level="3.3" data-path="volcano-plots.html"><a href="volcano-plots.html"><i class="fa fa-check"></i><b>3.3</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="enrich.html"><a href="enrich.html"><i class="fa fa-check"></i><b>4</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sea.html"><a href="sea.html"><i class="fa fa-check"></i><b>4.1</b> SEA</a></li>
<li class="chapter" data-level="4.2" data-path="gsea.html"><a href="gsea.html"><i class="fa fa-check"></i><b>4.2</b> GSEA</a></li>
<li class="chapter" data-level="4.3" data-path="databases.html"><a href="databases.html"><i class="fa fa-check"></i><b>4.3</b> Databases</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="databases.html"><a href="databases.html#gene-ontology"><i class="fa fa-check"></i><b>4.3.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="4.3.2" data-path="databases.html"><a href="databases.html#reactome"><i class="fa fa-check"></i><b>4.3.2</b> Reactome</a></li>
<li class="chapter" data-level="4.3.3" data-path="databases.html"><a href="databases.html#kegg"><i class="fa fa-check"></i><b>4.3.3</b> KEGG</a></li>
<li class="chapter" data-level="4.3.4" data-path="databases.html"><a href="databases.html#pfam"><i class="fa fa-check"></i><b>4.3.4</b> Pfam</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Tutorial for Proteomics Data Analysis with R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="enrich" class="section level1" number="4">
<h1><span class="header-section-number">Section 4</span> Enrichment Analysis</h1>
<!---
TODO
* Explain why I am talking about genes instead of proteins.
* Create a better hook for this chapter. The transition feels too abrupt.
--->
<p>While single-gene analysis (Section <a href="DEA.html#DEA">3</a>) is a useful tool, it is not without its own set of shortcomings. For instance, there may be no genes that pass the significance threshold after correcting for multiple hypothesis testing. Alternatively, there may be many genes that are statistically significant, and interpreting this list can be tedious and “prone to investigator bias toward a hypothesis of interest” <span class="citation">(<a href="#ref-maleki_gene_2020" role="doc-biblioref">Maleki et al., 2020</a>)</span>. Another issue is that single-gene analysis fails “to detect biological processes…that are distributed across an entire network of genes and subtle at the level of individual genes.” That is, it is liable to miss effects that are the result of small changes in many related genes. In order to address these and other issues, enrichment analysis (also called over-representation analysis) analyzes <strong>gene sets</strong>—“groups of genes that share [a] common biological function, chromosomal location,” involvement in a pathway, etc.—rather than individual genes <span class="citation">(<a href="#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al., 2005</a>)</span>.</p>
<!---
TODO:
* Talk about "General principle of enrichment analysis"
--->
<p><span class="citation"><a href="#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al.</a> (<a href="#ref-huang_bioinformatics_2009" role="doc-biblioref">2009</a>)</span> classifies enrichment analysis algorithms into three groups: singular enrichment analysis (SEA), gene set enrichment analysis (GSEA), and modular enrichment analysis (MEA).</p>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-huang_bioinformatics_2009" class="csl-entry">
Huang, D. W., Sherman, B. T. and Lempicki, R. A., Bioinformatics Enrichment Tools: Paths Toward the Comprehensive Functional Analysis of Large Gene Lists, <em>Nucleic Acids Research</em>, vol. <strong>37</strong>, no. 1, pp. 1–13, accessed September 7, 2021, from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615629/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615629/</a>, January 2009. DOI: <a href="https://doi.org/10.1093/nar/gkn923">10.1093/nar/gkn923</a>
</div>
<div id="ref-maleki_gene_2020" class="csl-entry">
Maleki, F., Ovens, K., Hogan, D. J. and Kusalik, A. J., Gene <span>Set</span> <span>Analysis</span>: <span>Challenges</span>, <span>Opportunities</span>, and <span>Future</span> <span>Research</span>, <em>Frontiers in Genetics</em>, vol. <strong>11</strong>, p. 654, accessed September 7, 2021, from <a href="https://www.frontiersin.org/article/10.3389/fgene.2020.00654">https://www.frontiersin.org/article/10.3389/fgene.2020.00654</a>, 2020. DOI: <a href="https://doi.org/10.3389/fgene.2020.00654">10.3389/fgene.2020.00654</a>
</div>
<div id="ref-subramanian_gene_2005" class="csl-entry">
Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., et al., Gene Set Enrichment Analysis: <span>A</span> Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles, <em>Proceedings of the National Academy of Sciences of the United States of America</em>, vol. <strong>102</strong>, no. 43, pp. 15545–50, accessed September 7, 2021, from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/</a>, October 2005. DOI: <a href="https://doi.org/10.1073/pnas.0506580102">10.1073/pnas.0506580102</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="volcano-plots.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sea.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
