<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1.1 Global Proteomics Data | Proteomics Data Analysis in R/Bioconductor</title>
  <meta name="description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="1.1 Global Proteomics Data | Proteomics Data Analysis in R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1.1 Global Proteomics Data | Proteomics Data Analysis in R/Bioconductor" />
  
  <meta name="twitter:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

<meta name="author" content="Tyler Sagendorf" />


<meta name="date" content="2022-01-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="isobaric-quantification-pipelines.html"/>
<link rel="next" href="phosphoproteomics-data.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=".">Proteomics Data Analysis in R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="isobaric-quantification-pipelines.html"><a href="isobaric-quantification-pipelines.html"><i class="fa fa-check"></i><b>1</b> Isobaric Quantification Pipelines</a>
<ul>
<li class="chapter" data-level="1.1" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html"><i class="fa fa-check"></i><b>1.1</b> Global Proteomics Data</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#read-ms-gf-data"><i class="fa fa-check"></i><b>1.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="1.1.2" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#correct-isotope-selection-error"><i class="fa fa-check"></i><b>1.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.1.3" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#remove-contaminants"><i class="fa fa-check"></i><b>1.1.3</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.1.4" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#global-peptide-filter"><i class="fa fa-check"></i><b>1.1.4</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.1.5" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#msms-id-filter-protein-level"><i class="fa fa-check"></i><b>1.1.5</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.1.6" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#inference-of-parsimonious-protein-set"><i class="fa fa-check"></i><b>1.1.6</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.1.7" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#remove-decoy-psms"><i class="fa fa-check"></i><b>1.1.7</b> Remove Decoy PSMs</a></li>
<li class="chapter" data-level="1.1.8" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#read-masic-output"><i class="fa fa-check"></i><b>1.1.8</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.1.9" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#filter-masic-data"><i class="fa fa-check"></i><b>1.1.9</b> Filter MASIC Data</a></li>
<li class="chapter" data-level="1.1.10" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#fetch-study-design-tables"><i class="fa fa-check"></i><b>1.1.10</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="1.1.11" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#fractions"><i class="fa fa-check"></i><b>1.1.11</b> Fractions</a></li>
<li class="chapter" data-level="1.1.12" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#samples"><i class="fa fa-check"></i><b>1.1.12</b> Samples</a></li>
<li class="chapter" data-level="1.1.13" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#references"><i class="fa fa-check"></i><b>1.1.13</b> References</a></li>
<li class="chapter" data-level="1.1.14" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#global-quant-crosstab"><i class="fa fa-check"></i><b>1.1.14</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="1.1.15" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#create-msnset-object"><i class="fa fa-check"></i><b>1.1.15</b> Create MSnSet Object</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html"><i class="fa fa-check"></i><b>1.2</b> Phosphoproteomics Data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#read-ms-gf-output"><i class="fa fa-check"></i><b>1.2.1</b> Read MS-GF+ Output</a></li>
<li class="chapter" data-level="1.2.2" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#correct-isotope-selection-error-1"><i class="fa fa-check"></i><b>1.2.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.2.3" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-non-phosphorylated-peptides"><i class="fa fa-check"></i><b>1.2.3</b> Remove Non-Phosphorylated Peptides</a></li>
<li class="chapter" data-level="1.2.4" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-contaminants-1"><i class="fa fa-check"></i><b>1.2.4</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.2.5" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#improve-phosphosite-localization"><i class="fa fa-check"></i><b>1.2.5</b> Improve Phosphosite Localization</a></li>
<li class="chapter" data-level="1.2.6" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#msms-id-filter-peptide-level"><i class="fa fa-check"></i><b>1.2.6</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.2.7" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#msms-id-filter-protein-level-1"><i class="fa fa-check"></i><b>1.2.7</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.2.8" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#inference-of-parsimonious-protein-set-1"><i class="fa fa-check"></i><b>1.2.8</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.2.9" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-decoy-psms-1"><i class="fa fa-check"></i><b>1.2.9</b> Remove Decoy PSMs</a></li>
<li class="chapter" data-level="1.2.10" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#map-sites-to-protein-sequences"><i class="fa fa-check"></i><b>1.2.10</b> Map Sites to Protein Sequences</a></li>
<li class="chapter" data-level="1.2.11" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#read-masic-output-1"><i class="fa fa-check"></i><b>1.2.11</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.2.12" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#filter-masic-data-1"><i class="fa fa-check"></i><b>1.2.12</b> Filter MASIC Data</a></li>
<li class="chapter" data-level="1.2.13" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#create-study-design-tables"><i class="fa fa-check"></i><b>1.2.13</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="1.2.14" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#phospho-create-crosstab"><i class="fa fa-check"></i><b>1.2.14</b> Create Quantitative Cross-tab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="label-free-quantification.html"><a href="label-free-quantification.html"><i class="fa fa-check"></i><b>2</b> Label-Free Quantification</a></li>
<li class="chapter" data-level="3" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>3</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="count-features-in-samples.html"><a href="count-features-in-samples.html"><i class="fa fa-check"></i><b>3.1</b> Count Features in Samples</a></li>
<li class="chapter" data-level="3.2" data-path="estimate-blood-contamination.html"><a href="estimate-blood-contamination.html"><i class="fa fa-check"></i><b>3.2</b> Estimate Blood Contamination</a></li>
<li class="chapter" data-level="3.3" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>3.3</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="heatmaps.html"><a href="heatmaps.html"><i class="fa fa-check"></i><b>4</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="4.1" data-path="expression-heatmaps.html"><a href="expression-heatmaps.html"><i class="fa fa-check"></i><b>4.1</b> Expression Heatmaps</a></li>
<li class="chapter" data-level="4.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html"><i class="fa fa-check"></i><b>4.2</b> Correlation Heatmaps</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#sample-correlation"><i class="fa fa-check"></i><b>4.2.1</b> Sample Correlation</a></li>
<li class="chapter" data-level="4.2.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#feature-correlation"><i class="fa fa-check"></i><b>4.2.2</b> Feature Correlation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html"><i class="fa fa-check"></i><b>4.3</b> Heatmap Annotation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html#modifying-default-colors"><i class="fa fa-check"></i><b>4.3.1</b> Modifying Default Colors</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="modifications.html"><a href="modifications.html"><i class="fa fa-check"></i><b>4.4</b> Modifications</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="modifications.html"><a href="modifications.html#row-and-column-labels"><i class="fa fa-check"></i><b>4.4.1</b> Row and column labels</a></li>
<li class="chapter" data-level="4.4.2" data-path="modifications.html"><a href="modifications.html#label-colors"><i class="fa fa-check"></i><b>4.4.2</b> Label colors</a></li>
<li class="chapter" data-level="4.4.3" data-path="modifications.html"><a href="modifications.html#label-specific-features"><i class="fa fa-check"></i><b>4.4.3</b> Label specific features</a></li>
<li class="chapter" data-level="4.4.4" data-path="modifications.html"><a href="modifications.html#heatmap-body-color"><i class="fa fa-check"></i><b>4.4.4</b> Heatmap body color</a></li>
<li class="chapter" data-level="4.4.5" data-path="modifications.html"><a href="modifications.html#horizontal-heatmaps"><i class="fa fa-check"></i><b>4.4.5</b> Horizontal heatmaps</a></li>
<li class="chapter" data-level="4.4.6" data-path="modifications.html"><a href="modifications.html#legends"><i class="fa fa-check"></i><b>4.4.6</b> Legends</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="linear-modeling.html"><a href="linear-modeling.html"><i class="fa fa-check"></i><b>5</b> Linear Modeling</a>
<ul>
<li class="chapter" data-level="5.1" data-path="mult-comp.html"><a href="mult-comp.html"><i class="fa fa-check"></i><b>5.1</b> Multiple Comparisons</a></li>
<li class="chapter" data-level="5.2" data-path="linear-reg.html"><a href="linear-reg.html"><i class="fa fa-check"></i><b>5.2</b> Linear Regression</a></li>
<li class="chapter" data-level="5.3" data-path="t-tests.html"><a href="t-tests.html"><i class="fa fa-check"></i><b>5.3</b> Two-Sample t-tests</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="t-tests.html"><a href="t-tests.html#t-single"><i class="fa fa-check"></i><b>5.3.1</b> One comparison</a></li>
<li class="chapter" data-level="5.3.2" data-path="t-tests.html"><a href="t-tests.html#t-mult"><i class="fa fa-check"></i><b>5.3.2</b> Multiple comparisons</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="anova.html"><a href="anova.html"><i class="fa fa-check"></i><b>5.4</b> One-Way ANOVA</a></li>
<li class="chapter" data-level="5.5" data-path="p-value-histograms.html"><a href="p-value-histograms.html"><i class="fa fa-check"></i><b>5.5</b> p-value Histograms</a></li>
<li class="chapter" data-level="5.6" data-path="volcano-plots.html"><a href="volcano-plots.html"><i class="fa fa-check"></i><b>5.6</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>6</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="annotation-databases.html"><a href="annotation-databases.html"><i class="fa fa-check"></i><b>6.1</b> Annotation Databases</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#gene-ontology"><i class="fa fa-check"></i><b>6.1.1</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="6.1.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#semantic-similarity"><i class="fa fa-check"></i><b>6.1.1.1</b> Semantic Similarity</a></li>
<li class="chapter" data-level="6.1.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#go-slim"><i class="fa fa-check"></i><b>6.1.1.2</b> GO Subsets/Slims</a></li>
</ul></li>
<li class="chapter" data-level="6.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#reactome"><i class="fa fa-check"></i><b>6.1.2</b> Reactome</a></li>
<li class="chapter" data-level="6.1.3" data-path="annotation-databases.html"><a href="annotation-databases.html#kegg"><i class="fa fa-check"></i><b>6.1.3</b> KEGG</a></li>
<li class="chapter" data-level="6.1.4" data-path="annotation-databases.html"><a href="annotation-databases.html#pfam"><i class="fa fa-check"></i><b>6.1.4</b> Pfam</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="ora.html"><a href="ora.html"><i class="fa fa-check"></i><b>6.2</b> Over-Representation Analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="ora.html"><a href="ora.html#ora-overview"><i class="fa fa-check"></i><b>6.2.1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="ora.html"><a href="ora.html#important-considerations"><i class="fa fa-check"></i>Important Considerations</a></li>
</ul></li>
<li class="chapter" data-level="6.2.2" data-path="ora.html"><a href="ora.html#ora-examples"><i class="fa fa-check"></i><b>6.2.2</b> Examples</a>
<ul>
<li class="chapter" data-level="6.2.2.1" data-path="ora.html"><a href="ora.html#ora-go"><i class="fa fa-check"></i><b>6.2.2.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="6.2.2.2" data-path="ora.html"><a href="ora.html#ora-reactome"><i class="fa fa-check"></i><b>6.2.2.2</b> Reactome</a></li>
<li class="chapter" data-level="6.2.2.3" data-path="ora.html"><a href="ora.html#ora-pfam"><i class="fa fa-check"></i><b>6.2.2.3</b> Pfam</a></li>
<li class="chapter" data-level="6.2.2.4" data-path="ora.html"><a href="ora.html#ora-other"><i class="fa fa-check"></i><b>6.2.2.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="gsea.html"><a href="gsea.html"><i class="fa fa-check"></i><b>6.3</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="gsea.html"><a href="gsea.html#gsea-overview"><i class="fa fa-check"></i><b>6.3.1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="gsea.html"><a href="gsea.html#important-considerations-1"><i class="fa fa-check"></i>Important Considerations</a></li>
</ul></li>
<li class="chapter" data-level="6.3.2" data-path="gsea.html"><a href="gsea.html#examples"><i class="fa fa-check"></i><b>6.3.2</b> Examples</a>
<ul>
<li class="chapter" data-level="6.3.2.1" data-path="gsea.html"><a href="gsea.html#gene-ontology-1"><i class="fa fa-check"></i><b>6.3.2.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="6.3.2.2" data-path="gsea.html"><a href="gsea.html#reactome-1"><i class="fa fa-check"></i><b>6.3.2.2</b> Reactome</a></li>
<li class="chapter" data-level="6.3.2.3" data-path="gsea.html"><a href="gsea.html#pfam-1"><i class="fa fa-check"></i><b>6.3.2.3</b> Pfam</a></li>
<li class="chapter" data-level="6.3.2.4" data-path="gsea.html"><a href="gsea.html#other-databases"><i class="fa fa-check"></i><b>6.3.2.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Proteomics Data Analysis in R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="global-proteomics-data" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Global Proteomics Data</h2>
<p>This pipeline shows how to process TMT data that is processed outside of PNNL’s DMS. Section <a href="phosphoproteomics-data.html#phosphoproteomics-data">1.2</a> shows how to process data from the DMS. For convenience, the results of MS-GF+ and MASIC processing are provided in a companion <code>PlexedPiperTestData</code> package. In addition, we will need <code>PlexedPiper</code> for isobaric quantification, <code>dplyr</code> to manipulate data frames, and <code>MSnbase</code> to create MSnSet objects.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="global-proteomics-data.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup</span></span>
<span id="cb1-2"><a href="global-proteomics-data.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PlexedPiper)</span>
<span id="cb1-3"><a href="global-proteomics-data.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PlexedPiperTestData)</span>
<span id="cb1-4"><a href="global-proteomics-data.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<p>The pipeline can be broken up into 4 major chunks: prepare MS/MS identifications, prepare reporter ion intensities, create a quantitative cross-tab, and create an MSnSet object.</p>
<div id="read-ms-gf-data" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Read MS-GF+ Data</h3>
<p>The first step in the preparation of the MS/MS identifications is to fetch the data. In this case, the data exists in the PlexedPiperTestData package in a local folder, so we use <code>system.file</code> to get the file path and <code>read_msgf_data</code> to read the MS-GF+ output.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="global-proteomics-data.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get file path</span></span>
<span id="cb2-2"><a href="global-proteomics-data.html#cb2-2" aria-hidden="true" tabindex="-1"></a>path_to_MSGF_results <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/global/msgf_output&quot;</span>, </span>
<span id="cb2-3"><a href="global-proteomics-data.html#cb2-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">package =</span> <span class="st">&quot;PlexedPiperTestData&quot;</span>)</span>
<span id="cb2-4"><a href="global-proteomics-data.html#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MS-GF+ data from path</span></span>
<span id="cb2-5"><a href="global-proteomics-data.html#cb2-5" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">read_msgf_data</span>(path_to_MSGF_results)</span></code></pre></div>
<p>Normally, this would display a progress bar in the console as the data is being fetched. However, the output was suppressed to save space. We can view a summary of the MSnID object with the <code>show()</code> function.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="global-proteomics-data.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  48 
## #PSMs: 1156754 at 31 % FDR
## #peptides: 511617 at 61 % FDR
## #accessions: 128378 at 98 % FDR</code></pre>
<p>This summary tells us that <code>msnid</code> consists of 4 spectrum files (datasets), and contains a total of 1,156,754 peptide-spectrum-matches (PSMs), 511,617 total peptides, and 128,378 total accessions (proteins). The reported FDR is the empirical <strong>false-discovery rate</strong>, which is calculated as the ratio of the number of false (decoy) PSMs, peptides, or accessions to their true (non-decoy) counterparts. Calculation of these counts and their FDRs is shown below.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="global-proteomics-data.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># How to calculate the counts and FDRs from the show() output</span></span>
<span id="cb5-2"><a href="global-proteomics-data.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Spectrum Files</span></span>
<span id="cb5-3"><a href="global-proteomics-data.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Count</span></span>
<span id="cb5-4"><a href="global-proteomics-data.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">psms</span>(msnid) <span class="sc">%&gt;%</span> </span>
<span id="cb5-5"><a href="global-proteomics-data.html#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Dataset) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="global-proteomics-data.html#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>() <span class="co"># 48</span></span>
<span id="cb5-7"><a href="global-proteomics-data.html#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="global-proteomics-data.html#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="do">## PSMs</span></span>
<span id="cb5-9"><a href="global-proteomics-data.html#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Count</span></span>
<span id="cb5-10"><a href="global-proteomics-data.html#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="fu">psms</span>(msnid) <span class="sc">%&gt;%</span> </span>
<span id="cb5-11"><a href="global-proteomics-data.html#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Dataset, Scan, peptide, isDecoy) <span class="sc">%&gt;%</span></span>
<span id="cb5-12"><a href="global-proteomics-data.html#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Assign intermediate to variable</span></span>
<span id="cb5-13"><a href="global-proteomics-data.html#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="st">&quot;x_psm&quot;</span>, ., <span class="at">envir =</span> <span class="fu">globalenv</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb5-14"><a href="global-proteomics-data.html#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>() <span class="co"># 1156754</span></span>
<span id="cb5-15"><a href="global-proteomics-data.html#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co"># FDR</span></span>
<span id="cb5-16"><a href="global-proteomics-data.html#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(x_psm[x_psm<span class="sc">$</span>isDecoy <span class="sc">==</span> <span class="cn">TRUE</span>, ]) <span class="sc">/</span> </span>
<span id="cb5-17"><a href="global-proteomics-data.html#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>(x_psm[x_psm<span class="sc">$</span>isDecoy <span class="sc">==</span> <span class="cn">FALSE</span>, ]) </span>
<span id="cb5-18"><a href="global-proteomics-data.html#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.3127463 = 31%</span></span>
<span id="cb5-19"><a href="global-proteomics-data.html#cb5-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-20"><a href="global-proteomics-data.html#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="do">## peptides</span></span>
<span id="cb5-21"><a href="global-proteomics-data.html#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Count</span></span>
<span id="cb5-22"><a href="global-proteomics-data.html#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="fu">psms</span>(msnid) <span class="sc">%&gt;%</span> </span>
<span id="cb5-23"><a href="global-proteomics-data.html#cb5-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(peptide, isDecoy) <span class="sc">%&gt;%</span> </span>
<span id="cb5-24"><a href="global-proteomics-data.html#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="st">&quot;x_peptide&quot;</span>, ., <span class="at">envir =</span> <span class="fu">globalenv</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb5-25"><a href="global-proteomics-data.html#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>() <span class="co"># 511617</span></span>
<span id="cb5-26"><a href="global-proteomics-data.html#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="co"># FDR</span></span>
<span id="cb5-27"><a href="global-proteomics-data.html#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(x_peptide[x_peptide<span class="sc">$</span>isDecoy <span class="sc">==</span> <span class="cn">TRUE</span>, ]) <span class="sc">/</span> </span>
<span id="cb5-28"><a href="global-proteomics-data.html#cb5-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>(x_peptide[x_peptide<span class="sc">$</span>isDecoy <span class="sc">==</span> <span class="cn">FALSE</span>, ])</span>
<span id="cb5-29"><a href="global-proteomics-data.html#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.611245 = 61%</span></span>
<span id="cb5-30"><a href="global-proteomics-data.html#cb5-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-31"><a href="global-proteomics-data.html#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="do">## accessions</span></span>
<span id="cb5-32"><a href="global-proteomics-data.html#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Count</span></span>
<span id="cb5-33"><a href="global-proteomics-data.html#cb5-33" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">accessions</span>(msnid)) <span class="co"># or</span></span>
<span id="cb5-34"><a href="global-proteomics-data.html#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="fu">psms</span>(msnid) <span class="sc">%&gt;%</span> </span>
<span id="cb5-35"><a href="global-proteomics-data.html#cb5-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(accession, isDecoy) <span class="sc">%&gt;%</span> </span>
<span id="cb5-36"><a href="global-proteomics-data.html#cb5-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="st">&quot;x_acc&quot;</span>, ., <span class="at">envir =</span> <span class="fu">globalenv</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb5-37"><a href="global-proteomics-data.html#cb5-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>() <span class="co"># 128378</span></span>
<span id="cb5-38"><a href="global-proteomics-data.html#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="co"># FDR</span></span>
<span id="cb5-39"><a href="global-proteomics-data.html#cb5-39" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(x_acc[x_acc<span class="sc">$</span>isDecoy <span class="sc">==</span> <span class="cn">TRUE</span>, ]) <span class="sc">/</span> </span>
<span id="cb5-40"><a href="global-proteomics-data.html#cb5-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nrow</span>(x_acc[x_acc<span class="sc">$</span>isDecoy <span class="sc">==</span> <span class="cn">FALSE</span>, ])</span>
<span id="cb5-41"><a href="global-proteomics-data.html#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.9827024 = 98%</span></span></code></pre></div>
<p>Now that we have an <code>MSnID</code> object, we need to process it. We begin by correcting for the isotope selection error.</p>
</div>
<div id="correct-isotope-selection-error" class="section level3" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> Correct Isotope Selection Error</h3>
<p>Carbon has two stable isotopes: <span class="math inline">\(^{12}\text{C}\)</span> and <span class="math inline">\(^{13}\text{C}\)</span>, with natural abundances of 98.93% and 1.07%, respectively. That is, we expect that about 1 out of every 100 carbon atoms is naturally going to be a <span class="math inline">\(^{13}\text{C}\)</span>, while the rest are <span class="math inline">\(^{12}\text{C}\)</span>. In larger peptides with many carbon atoms, it is more likely that at least one atom will be a <span class="math inline">\(^{13}\text{C}\)</span> than all atoms will be <span class="math inline">\(^{12}\text{C}\)</span>. In cases such as these, a non-monoisotopic ion will be selected by the instrument for fragmentation.</p>
<div class="figure" style="text-align: center">
<img src="images/MS1_non_monoisotopic.PNG" alt="MS1 spectra with peak at non-monoisotopic precursor ion." width="736" />
<p class="caption">
(#fig:MS1_peak)MS1 spectra with peak at non-monoisotopic precursor ion.
</p>
</div>
<p>In Figure @ref(fig:MS1_peak), the monoisotopic ion (m/z of 1427.29) is not the most abundant, so it is not selected as the precursor. Instead, the ion with a <span class="math inline">\(^{13}\text{C}\)</span> in place of a <span class="math inline">\(^{12}\text{C}\)</span> is selected for fragmentation. We calculate the mass difference between these two ions as the difference between the mass-to-charge ratios multiplied by the ion charge. In this case, the mass difference is 1 Dalton, or about the difference between <span class="math inline">\(^{13}\text{C}\)</span> and <span class="math inline">\(^{12}\text{C}\)</span>. (More accurately, the difference between these isotopes is 1.0033548378 Da.) While MS-GF+ is still capable of correctly identifying these peptides, the downstream calculations of mass measurement error need to be fixed because they are used for filtering later on (Section <a href="global-proteomics-data.html#global-peptide-filter">1.1.4</a>). The <code>correct_peak_selection</code> function corrects these mass measurement errors, and Figure <a href="global-proteomics-data.html#fig:mass-to-charge-diff">1.1</a> shows the distribution of the mass measurement errors before and after correction.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="global-proteomics-data.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Correct for isotope selection error</span></span>
<span id="cb6-2"><a href="global-proteomics-data.html#cb6-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">correct_peak_selection</span>(msnid)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:mass-to-charge-diff"></span>
<img src="isobaric_quantification_files/figure-html/mass-to-charge-diff-1.png" alt="Histogram of mass measurement errors before and after correction." width="672" />
<p class="caption">
Figure 1.1: Histogram of mass measurement errors before and after correction.
</p>
</div>
</div>
<div id="remove-contaminants" class="section level3" number="1.1.3">
<h3><span class="header-section-number">1.1.3</span> Remove Contaminants</h3>
<p>Now, we will remove contaminants such as the trypsin that was used for protein digestion. We can use <code>grepl</code> to search for all accessions that contain the string <code>"Contaminant"</code>. Displaying these contaminants is not necessary during processing. This is just for demonstration purposes to see what will be removed.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="global-proteomics-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># All unique contaminants</span></span>
<span id="cb7-2"><a href="global-proteomics-data.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">accessions</span>(msnid)[<span class="fu">grepl</span>(<span class="st">&quot;Contaminant&quot;</span>, <span class="fu">accessions</span>(msnid))]</span></code></pre></div>
<pre><code>##  [1] &quot;Contaminant_K2C1_HUMAN&quot;      &quot;Contaminant_K1C9_HUMAN&quot;     
##  [3] &quot;Contaminant_ALBU_HUMAN&quot;      &quot;Contaminant_ALBU_BOVIN&quot;     
##  [5] &quot;Contaminant_TRYP_PIG&quot;        &quot;Contaminant_K1C10_HUMAN&quot;    
##  [7] &quot;XXX_Contaminant_K1C9_HUMAN&quot;  &quot;Contaminant_K22E_HUMAN&quot;     
##  [9] &quot;Contaminant_Trypa3&quot;          &quot;Contaminant_Trypa5&quot;         
## [11] &quot;XXX_Contaminant_K1C10_HUMAN&quot; &quot;XXX_Contaminant_K22E_HUMAN&quot; 
## [13] &quot;XXX_Contaminant_K2C1_HUMAN&quot;  &quot;Contaminant_TRYP_BOVIN&quot;     
## [15] &quot;XXX_Contaminant_ALBU_HUMAN&quot;  &quot;XXX_Contaminant_ALBU_BOVIN&quot; 
## [17] &quot;XXX_Contaminant_TRYP_BOVIN&quot;  &quot;Contaminant_CTRB_BOVIN&quot;     
## [19] &quot;Contaminant_Trypa1&quot;          &quot;Contaminant_Trypa6&quot;         
## [21] &quot;Contaminant_CTRA_BOVIN&quot;      &quot;XXX_Contaminant_TRYP_PIG&quot;   
## [23] &quot;XXX_Contaminant_CTRB_BOVIN&quot;  &quot;XXX_Contaminant_CTRA_BOVIN&quot; 
## [25] &quot;Contaminant_Trypa2&quot;</code></pre>
<p>To remove contaminants, we use <code>apply_filter</code> with an appropriate character string that tells the function what rows to keep. In this case, we keep rows where the accession does not contain “Contaminant.” We will use <code>show</code> to see how the counts change.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="global-proteomics-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove contaminants</span></span>
<span id="cb9-2"><a href="global-proteomics-data.html#cb9-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">apply_filter</span>(msnid, <span class="st">&quot;!grepl(&#39;Contaminant&#39;, accession)&quot;</span>)</span>
<span id="cb9-3"><a href="global-proteomics-data.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  48 
## #PSMs: 1155442 at 31 % FDR
## #peptides: 511196 at 61 % FDR
## #accessions: 128353 at 98 % FDR</code></pre>
<p>We can see that the number of PSMs decreased by about 1300, peptides by ~400, and proteins by 25 (the 25 contaminants that were displayed).</p>
</div>
<div id="global-peptide-filter" class="section level3" number="1.1.4">
<h3><span class="header-section-number">1.1.4</span> MS/MS ID Filter: Peptide Level</h3>
<p>The next step is to filter the MS/MS identifications such that the empirical peptide-level FDR is less than some threshold and the number of MS/MS IDs is maximized. We will use the <span class="math inline">\(-log_{10}\)</span> of the <code>PepQValue</code> column as one of our filtering criteria and assign it to a new column in <code>psms(msnid)</code> called <code>msmsScore</code>. The <code>PepQValue</code> column is the MS-GF+ Spectrum E-value, which reflects how well the theoretical and experimental fragmentation spectra match; therefore, high values of <code>msmsScore</code> indicate a good match (see Figure <a href="global-proteomics-data.html#fig:plot-msmsScore">1.2</a>).</p>
<div class="figure" style="text-align: center"><span id="fig:plot-msmsScore"></span>
<img src="isobaric_quantification_files/figure-html/plot-msmsScore-1.png" alt="Density plot of msmsScore." width="672" />
<p class="caption">
Figure 1.2: Density plot of msmsScore.
</p>
</div>
<p></br></p>
<p>The other filtering criteria is the absolute deviation of the mass measurement error of the precursor ions in parts-per-million (ppm), which is assigned to the <code>absParentMassErrorPPM</code> column in <code>psms(msnid)</code> (see Figure <a href="global-proteomics-data.html#fig:plot-mass-error">1.3</a>).</p>
<div class="figure" style="text-align: center"><span id="fig:plot-mass-error"></span>
<img src="isobaric_quantification_files/figure-html/plot-mass-error-1.png" alt="Density plot of absParentMassErrorPPM." width="672" />
<p class="caption">
Figure 1.3: Density plot of absParentMassErrorPPM.
</p>
</div>
<p></br></p>
<p>Now, we will filter the PSMs.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="global-proteomics-data.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1% FDR filter at the peptide level</span></span>
<span id="cb11-2"><a href="global-proteomics-data.html#cb11-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">filter_msgf_data</span>(msnid,</span>
<span id="cb11-3"><a href="global-proteomics-data.html#cb11-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">level =</span> <span class="st">&quot;peptide&quot;</span>,</span>
<span id="cb11-4"><a href="global-proteomics-data.html#cb11-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fdr.max =</span> <span class="fl">0.01</span>)</span>
<span id="cb11-5"><a href="global-proteomics-data.html#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  48 
## #PSMs: 464474 at 0.45 % FDR
## #peptides: 96485 at 1 % FDR
## #accessions: 27119 at 9.2 % FDR</code></pre>
<p>We can see that filtering drastically reduces the number of PSMs, and the empirical peptide-level FDR is now 1%. However, notice that the empirical protein-level FDR is still fairly high.</p>
</div>
<div id="msms-id-filter-protein-level" class="section level3" number="1.1.5">
<h3><span class="header-section-number">1.1.5</span> MS/MS ID Filter: Protein Level</h3>
<p>Now, we need to filter proteins so that the FDR is at most 1%. A while ago, the proteomics field established the hard-and-fast two-peptides-per-protein rule. That is, the confident identification of a protein requires the confident identification of at least 2 peptides. This rule penalizes short proteins and doesn’t consider that there are some very long proteins (e.g. Titin 3.8 MDa) that easily have more then two matching peptides even in the reversed sequence. Thus, we propose to normalize the number of peptides per protein length and use that as a filtering criterion (Figure <a href="global-proteomics-data.html#fig:plot-num-pep">1.5</a>).</p>
<p>In order to get the protein lengths, we need the FASTA (pronounced FAST-AYE) file that contains the protein sequences used in the database search. The first three entries of the FASTA file are shown in Figure <a href="global-proteomics-data.html#fig:fasta-ex">1.4</a>.</p>
<div class="figure" style="text-align: center"><span id="fig:fasta-ex"></span>
<img src="images/FASTA_example_MoTrPAC.PNG" alt="First three entries of the FASTA file." width="488" />
<p class="caption">
Figure 1.4: First three entries of the FASTA file.
</p>
</div>
<p></br></p>
<p>For each protein, we divide the number of associated peptides by the length of that protein and multiply this value by 1000. This new <code>peptides_per_1000aa</code> column is used as the filter criteria.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="global-proteomics-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get path to FASTA file</span></span>
<span id="cb13-2"><a href="global-proteomics-data.html#cb13-2" aria-hidden="true" tabindex="-1"></a>path_to_FASTA <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb13-3"><a href="global-proteomics-data.html#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;extdata/Rattus_norvegicus_NCBI_RefSeq_2018-04-10.fasta.gz&quot;</span>, </span>
<span id="cb13-4"><a href="global-proteomics-data.html#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;PlexedPiperTestData&quot;</span></span>
<span id="cb13-5"><a href="global-proteomics-data.html#cb13-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-6"><a href="global-proteomics-data.html#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="global-proteomics-data.html#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute number of peptides per 1000 amino acids</span></span>
<span id="cb13-8"><a href="global-proteomics-data.html#cb13-8" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">compute_num_peptides_per_1000aa</span>(msnid, path_to_FASTA)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:plot-num-pep"></span>
<img src="isobaric_quantification_files/figure-html/plot-num-pep-1.png" alt="Density plot of peptides_per_1000aa. The plot area has been zoomed in." width="672" />
<p class="caption">
Figure 1.5: Density plot of peptides_per_1000aa. The plot area has been zoomed in.
</p>
</div>
<p></br></p>
<p>Now, we filter the proteins to 1% FDR.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="global-proteomics-data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1% FDR filter at the protein level</span></span>
<span id="cb14-2"><a href="global-proteomics-data.html#cb14-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">filter_msgf_data</span>(msnid,</span>
<span id="cb14-3"><a href="global-proteomics-data.html#cb14-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">level =</span> <span class="st">&quot;accession&quot;</span>,</span>
<span id="cb14-4"><a href="global-proteomics-data.html#cb14-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fdr.max =</span> <span class="fl">0.01</span>)</span>
<span id="cb14-5"><a href="global-proteomics-data.html#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  48 
## #PSMs: 458090 at 0.16 % FDR
## #peptides: 92036 at 0.32 % FDR
## #accessions: 15630 at 1 % FDR</code></pre>
</div>
<div id="inference-of-parsimonious-protein-set" class="section level3" number="1.1.6">
<h3><span class="header-section-number">1.1.6</span> Inference of Parsimonious Protein Set</h3>
<p>The situation when a certain peptide sequence matches multiple proteins adds complication to the downstream quantitative analysis, as it is not clear which protein this peptide is originating from. There are common ways for dealing with this. One is to simply retain uniquely matching peptides and discard shared peptides (<code>unique_only = TRUE</code>). Alternatively, assign the shared peptides to the proteins with the larger number of uniquely mapping peptides (<code>unique_only = FALSE</code>). If there is a choice between multiple proteins with equal numbers of uniquely mapping peptides, the shared peptides are assigned to the first protein according to alphanumeric order (Figure <a href="global-proteomics-data.html#fig:parsimony">1.6</a>).</p>
<!---
This step could be done prior to filtering at the accession level, but if peptides are assigned to a low-confidence protein, and that protein is removed during filtering, those peptides will be lost. Instead, it is better to filter to the set of confidently-identified proteins and then determine the parsimonious set.
--->
<div class="figure" style="text-align: center"><span id="fig:parsimony"></span>
<img src="images/parsimonious-protein-set-inference.PNG" alt="Visual explanation of the inference of the parsimonious protein set." width="528" />
<p class="caption">
Figure 1.6: Visual explanation of the inference of the parsimonious protein set.
</p>
</div>
<p></br></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="global-proteomics-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Inference of parsimonious protein set</span></span>
<span id="cb16-2"><a href="global-proteomics-data.html#cb16-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">infer_parsimonious_accessions</span>(msnid, <span class="at">unique_only =</span> <span class="cn">FALSE</span>)</span>
<span id="cb16-3"><a href="global-proteomics-data.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  48 
## #PSMs: 444999 at 0.15 % FDR
## #peptides: 90478 at 0.27 % FDR
## #accessions: 5251 at 1.1 % FDR</code></pre>
<p>Notice that the protein-level FDR increased slightly above the 1% threshold. In this case, the difference isn’t significant, so we can ignore it.</p>
<p><em>Note:</em></p>
<p>If the peptide or accession-level FDR increases significantly above 1% after inference of the parsimonious protein set, consider lowering the FDR cutoff (for example, to 0.9%) and redoing the previous processing steps. Filtering at the peptide and accession level should each be done a single time.</p>
</div>
<div id="remove-decoy-psms" class="section level3" number="1.1.7">
<h3><span class="header-section-number">1.1.7</span> Remove Decoy PSMs</h3>
<p>The final step in preparing the MS/MS identifications is to remove the decoy PSMs. We use the <code>apply_filter</code> function again and only keep entries where <code>isDecoy</code> is <code>FALSE</code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="global-proteomics-data.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove Decoy PSMs</span></span>
<span id="cb18-2"><a href="global-proteomics-data.html#cb18-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">apply_filter</span>(msnid, <span class="st">&quot;!isDecoy&quot;</span>)</span>
<span id="cb18-3"><a href="global-proteomics-data.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  48 
## #PSMs: 444338 at 0 % FDR
## #peptides: 90232 at 0 % FDR
## #accessions: 5196 at 0 % FDR</code></pre>
<p>After processing, we are left with 444,345 PSMs, 90,232 peptides, and 5,196 proteins. Table <a href="global-proteomics-data.html#tab:global-msnid-table">1.1</a> shows the first 6 rows of the processed MS-GF+ output.</p>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:global-msnid-table">Table 1.1: </span><left>First 6 rows of the processed MS-GF+ results.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:right;">
ResultID
</th>
<th style="text-align:right;">
Scan
</th>
<th style="text-align:left;">
FragMethod
</th>
<th style="text-align:right;">
SpecIndex
</th>
<th style="text-align:right;">
Charge
</th>
<th style="text-align:right;">
PrecursorMZ
</th>
<th style="text-align:right;">
DelM
</th>
<th style="text-align:right;">
DelM_PPM
</th>
<th style="text-align:right;">
MH
</th>
<th style="text-align:left;">
peptide
</th>
<th style="text-align:left;">
Protein
</th>
<th style="text-align:right;">
NTT
</th>
<th style="text-align:right;">
DeNovoScore
</th>
<th style="text-align:right;">
MSGFScore
</th>
<th style="text-align:right;">
MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
Rank_MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
EValue
</th>
<th style="text-align:right;">
QValue
</th>
<th style="text-align:right;">
PepQValue
</th>
<th style="text-align:right;">
IsotopeError
</th>
<th style="text-align:left;">
accession
</th>
<th style="text-align:right;">
calculatedMassToCharge
</th>
<th style="text-align:right;">
chargeState
</th>
<th style="text-align:right;">
experimentalMassToCharge
</th>
<th style="text-align:left;">
isDecoy
</th>
<th style="text-align:left;">
spectrumFile
</th>
<th style="text-align:right;">
spectrumID
</th>
<th style="text-align:left;">
pepSeq
</th>
<th style="text-align:right;">
msmsScore
</th>
<th style="text-align:right;">
absParentMassErrorPPM
</th>
<th style="text-align:right;">
peptides_per_1000aa
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
1862
</td>
<td style="text-align:right;">
27707
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
324
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
928.541
</td>
<td style="text-align:right;">
-0.001
</td>
<td style="text-align:right;">
-0.526
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
285
</td>
<td style="text-align:right;">
282
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
928.541
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
928.541
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27707
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.589
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
4192
</td>
<td style="text-align:right;">
27684
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
906
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
-0.002
</td>
<td style="text-align:right;">
-0.887
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
156
</td>
<td style="text-align:right;">
144
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27684
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.991
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_06_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
26263
</td>
<td style="text-align:right;">
27336
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
5187
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
118
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_06_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27336
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
1471
</td>
<td style="text-align:right;">
27096
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
415
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
-0.001
</td>
<td style="text-align:right;">
-0.591
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
157
</td>
<td style="text-align:right;">
156
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27096
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.684
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_05_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
28664
</td>
<td style="text-align:right;">
10441
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
4849
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
586.832
</td>
<td style="text-align:right;">
-0.001
</td>
<td style="text-align:right;">
-0.728
</td>
<td style="text-align:right;">
1172.659
</td>
<td style="text-align:left;">
R.AAAAADLANR.S
</td>
<td style="text-align:left;">
NP_001007804.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
124
</td>
<td style="text-align:right;">
124
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_001007804.1
</td>
<td style="text-align:right;">
586.833
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
586.832
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_05_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
10441
</td>
<td style="text-align:left;">
AAAAADLANR
</td>
<td style="text-align:right;">
2.480
</td>
<td style="text-align:right;">
0.746
</td>
<td style="text-align:right;">
34.755
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_24_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
41775
</td>
<td style="text-align:right;">
8033
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
7889
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
831.447
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
1661.886
</td>
<td style="text-align:left;">
G.AAAAAEAESGGGGGK.K
</td>
<td style="text-align:left;">
NP_001128630.1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
176
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_001128630.1
</td>
<td style="text-align:right;">
831.447
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
831.447
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_24_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
8033
</td>
<td style="text-align:left;">
AAAAAEAESGGGGGK
</td>
<td style="text-align:right;">
2.583
</td>
<td style="text-align:right;">
0.106
</td>
<td style="text-align:right;">
580.844
</td>
</tr>
</tbody>
</table>
<p></br></p>
</div>
<div id="read-masic-output" class="section level3" number="1.1.8">
<h3><span class="header-section-number">1.1.8</span> Read MASIC Output</h3>
<p>MASIC is a tool for extracting ion intensities. With proper parameter settings, it can be used for extracting TMT (or iTRAQ) reporter ion intensities. In addition, it reports a number of other helpful metrics. Notably, the interference score at the precursor ion level and the signal-to-noise ratio (S/N) at the reporter ion level (computed by Thermo software). The interference score reflects the proportion of the ion population that was isolated for fragmentation that is due to the targeted ion. In other words, <code>1 - InterferenceScore</code> is due to co-isolated species that have similar elution time and precursor ion m/z. The first step in the preparation of the reporter ion intensity data is to read the MASIC results. We use a local file path and the <code>read_masic_data</code> function. By default, the interference score is not included, so we need to set that argument to <code>TRUE</code> in order to filter the results.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="global-proteomics-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Path to MASIC data</span></span>
<span id="cb20-2"><a href="global-proteomics-data.html#cb20-2" aria-hidden="true" tabindex="-1"></a>path_to_MASIC_results <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/global/masic_output&quot;</span>, </span>
<span id="cb20-3"><a href="global-proteomics-data.html#cb20-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">package =</span> <span class="st">&quot;PlexedPiperTestData&quot;</span>)</span>
<span id="cb20-4"><a href="global-proteomics-data.html#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="global-proteomics-data.html#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MASIC data</span></span>
<span id="cb20-6"><a href="global-proteomics-data.html#cb20-6" aria-hidden="true" tabindex="-1"></a>masic_data <span class="ot">&lt;-</span> <span class="fu">read_masic_data</span>(path_to_MASIC_results, <span class="at">interference_score =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Normally, this would display progress bars in the console as the data is being fetched. However, the output was suppressed to save space.</p>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:global-masic-table">Table 1.2: </span><left>First 6 rows of the MASIC data.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:right;">
ScanNumber
</th>
<th style="text-align:left;">
Collision.Mode
</th>
<th style="text-align:right;">
ParentIonMZ
</th>
<th style="text-align:right;">
BasePeakIntensity
</th>
<th style="text-align:right;">
BasePeakMZ
</th>
<th style="text-align:right;">
ReporterIonIntensityMax
</th>
<th style="text-align:right;">
Ion_126.128
</th>
<th style="text-align:right;">
Ion_127.125
</th>
<th style="text-align:right;">
Ion_127.131
</th>
<th style="text-align:right;">
Ion_128.128
</th>
<th style="text-align:right;">
Ion_128.134
</th>
<th style="text-align:right;">
Ion_129.131
</th>
<th style="text-align:right;">
Ion_129.138
</th>
<th style="text-align:right;">
Ion_130.135
</th>
<th style="text-align:right;">
Ion_130.141
</th>
<th style="text-align:right;">
Ion_131.138
</th>
<th style="text-align:right;">
Weighted.Avg.Pct.Intensity.Correction
</th>
<th style="text-align:right;">
Ion_126.128_SignalToNoise
</th>
<th style="text-align:right;">
Ion_127.125_SignalToNoise
</th>
<th style="text-align:right;">
Ion_127.131_SignalToNoise
</th>
<th style="text-align:right;">
Ion_128.128_SignalToNoise
</th>
<th style="text-align:right;">
Ion_128.134_SignalToNoise
</th>
<th style="text-align:right;">
Ion_129.131_SignalToNoise
</th>
<th style="text-align:right;">
Ion_129.138_SignalToNoise
</th>
<th style="text-align:right;">
Ion_130.135_SignalToNoise
</th>
<th style="text-align:right;">
Ion_130.141_SignalToNoise
</th>
<th style="text-align:right;">
Ion_131.138_SignalToNoise
</th>
<th style="text-align:right;">
Ion_126.128_Resolution
</th>
<th style="text-align:right;">
Ion_127.125_Resolution
</th>
<th style="text-align:right;">
Ion_127.131_Resolution
</th>
<th style="text-align:right;">
Ion_128.128_Resolution
</th>
<th style="text-align:right;">
Ion_128.134_Resolution
</th>
<th style="text-align:right;">
Ion_129.131_Resolution
</th>
<th style="text-align:right;">
Ion_129.138_Resolution
</th>
<th style="text-align:right;">
Ion_130.135_Resolution
</th>
<th style="text-align:right;">
Ion_130.141_Resolution
</th>
<th style="text-align:right;">
Ion_131.138_Resolution
</th>
<th style="text-align:right;">
ParentIonIndex
</th>
<th style="text-align:right;">
MZ
</th>
<th style="text-align:right;">
SurveyScanNumber
</th>
<th style="text-align:right;">
OptimalPeakApexScanNumber
</th>
<th style="text-align:right;">
PeakApexOverrideParentIonIndex
</th>
<th style="text-align:right;">
CustomSICPeak
</th>
<th style="text-align:right;">
PeakScanStart
</th>
<th style="text-align:right;">
PeakScanEnd
</th>
<th style="text-align:right;">
PeakScanMaxIntensity
</th>
<th style="text-align:right;">
PeakMaxIntensity
</th>
<th style="text-align:right;">
PeakSignalToNoiseRatio
</th>
<th style="text-align:right;">
FWHMInScans
</th>
<th style="text-align:right;">
PeakArea
</th>
<th style="text-align:right;">
ParentIonIntensity
</th>
<th style="text-align:right;">
PeakBaselineNoiseLevel
</th>
<th style="text-align:right;">
PeakBaselineNoiseStDev
</th>
<th style="text-align:right;">
PeakBaselinePointsUsed
</th>
<th style="text-align:right;">
StatMomentsArea
</th>
<th style="text-align:right;">
CenterOfMassScan
</th>
<th style="text-align:right;">
PeakStDev
</th>
<th style="text-align:right;">
PeakSkew
</th>
<th style="text-align:right;">
PeakKSStat
</th>
<th style="text-align:right;">
StatMomentsDataCountUsed
</th>
<th style="text-align:right;">
InterferenceScore
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
407.74
</td>
<td style="text-align:right;">
227695.44
</td>
<td style="text-align:right;">
407.741
</td>
<td style="text-align:right;">
92236.87
</td>
<td style="text-align:right;">
70562.39
</td>
<td style="text-align:right;">
24864.62
</td>
<td style="text-align:right;">
17165.80
</td>
<td style="text-align:right;">
35625.00
</td>
<td style="text-align:right;">
92236.87
</td>
<td style="text-align:right;">
9640.23
</td>
<td style="text-align:right;">
8578.05
</td>
<td style="text-align:right;">
6996.69
</td>
<td style="text-align:right;">
11833.07
</td>
<td style="text-align:right;">
32281.34
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
71.47
</td>
<td style="text-align:right;">
25.17
</td>
<td style="text-align:right;">
17.38
</td>
<td style="text-align:right;">
36.04
</td>
<td style="text-align:right;">
93.32
</td>
<td style="text-align:right;">
9.75
</td>
<td style="text-align:right;">
8.67
</td>
<td style="text-align:right;">
7.07
</td>
<td style="text-align:right;">
11.96
</td>
<td style="text-align:right;">
32.71
</td>
<td style="text-align:right;">
44102
</td>
<td style="text-align:right;">
42700
</td>
<td style="text-align:right;">
42100
</td>
<td style="text-align:right;">
41800
</td>
<td style="text-align:right;">
44404
</td>
<td style="text-align:right;">
40500
</td>
<td style="text-align:right;">
39500
</td>
<td style="text-align:right;">
36800
</td>
<td style="text-align:right;">
41100
</td>
<td style="text-align:right;">
42302
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
407.742
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
2901600
</td>
<td style="text-align:right;">
211.000
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
50422000
</td>
<td style="text-align:right;">
2579600
</td>
<td style="text-align:right;">
13750
</td>
<td style="text-align:right;">
97562
</td>
<td style="text-align:right;">
10113
</td>
<td style="text-align:right;">
47031000
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
5.68
</td>
<td style="text-align:right;">
-0.189
</td>
<td style="text-align:right;">
0.353
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.996
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
396.72
</td>
<td style="text-align:right;">
59127.97
</td>
<td style="text-align:right;">
529.294
</td>
<td style="text-align:right;">
34294.90
</td>
<td style="text-align:right;">
23706.89
</td>
<td style="text-align:right;">
13559.32
</td>
<td style="text-align:right;">
5856.83
</td>
<td style="text-align:right;">
16322.71
</td>
<td style="text-align:right;">
34294.90
</td>
<td style="text-align:right;">
4853.11
</td>
<td style="text-align:right;">
7938.24
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1465.03
</td>
<td style="text-align:right;">
18182.27
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
26.12
</td>
<td style="text-align:right;">
14.94
</td>
<td style="text-align:right;">
6.45
</td>
<td style="text-align:right;">
17.97
</td>
<td style="text-align:right;">
37.77
</td>
<td style="text-align:right;">
5.34
</td>
<td style="text-align:right;">
8.74
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.61
</td>
<td style="text-align:right;">
19.93
</td>
<td style="text-align:right;">
42702
</td>
<td style="text-align:right;">
41100
</td>
<td style="text-align:right;">
37000
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
43404
</td>
<td style="text-align:right;">
36400
</td>
<td style="text-align:right;">
39700
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
29800
</td>
<td style="text-align:right;">
41802
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
396.718
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
2181900
</td>
<td style="text-align:right;">
19.690
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
34508000
</td>
<td style="text-align:right;">
1690600
</td>
<td style="text-align:right;">
110841
</td>
<td style="text-align:right;">
1120000
</td>
<td style="text-align:right;">
10166
</td>
<td style="text-align:right;">
31578000
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
5.59
</td>
<td style="text-align:right;">
-0.217
</td>
<td style="text-align:right;">
0.347
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.993
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
438.23
</td>
<td style="text-align:right;">
110444.82
</td>
<td style="text-align:right;">
362.224
</td>
<td style="text-align:right;">
14053.40
</td>
<td style="text-align:right;">
12459.86
</td>
<td style="text-align:right;">
11785.91
</td>
<td style="text-align:right;">
10932.51
</td>
<td style="text-align:right;">
10653.32
</td>
<td style="text-align:right;">
12328.62
</td>
<td style="text-align:right;">
5959.86
</td>
<td style="text-align:right;">
9905.82
</td>
<td style="text-align:right;">
8387.04
</td>
<td style="text-align:right;">
11166.70
</td>
<td style="text-align:right;">
14053.40
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
12.40
</td>
<td style="text-align:right;">
11.75
</td>
<td style="text-align:right;">
10.90
</td>
<td style="text-align:right;">
10.64
</td>
<td style="text-align:right;">
12.31
</td>
<td style="text-align:right;">
5.96
</td>
<td style="text-align:right;">
9.91
</td>
<td style="text-align:right;">
8.40
</td>
<td style="text-align:right;">
11.18
</td>
<td style="text-align:right;">
14.13
</td>
<td style="text-align:right;">
42006
</td>
<td style="text-align:right;">
40702
</td>
<td style="text-align:right;">
41402
</td>
<td style="text-align:right;">
40700
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
38800
</td>
<td style="text-align:right;">
40200
</td>
<td style="text-align:right;">
38900
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
41002
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
438.227
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
131
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
8255600
</td>
<td style="text-align:right;">
9.465
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
444610000
</td>
<td style="text-align:right;">
658727
</td>
<td style="text-align:right;">
872195
</td>
<td style="text-align:right;">
2620000
</td>
<td style="text-align:right;">
10129
</td>
<td style="text-align:right;">
343470000
</td>
<td style="text-align:right;">
82
</td>
<td style="text-align:right;">
22.80
</td>
<td style="text-align:right;">
-0.626
</td>
<td style="text-align:right;">
1.099
</td>
<td style="text-align:right;">
89
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
481.50
</td>
<td style="text-align:right;">
37082.72
</td>
<td style="text-align:right;">
206.466
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
481.505
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
401824
</td>
<td style="text-align:right;">
27.990
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
16244000
</td>
<td style="text-align:right;">
344491
</td>
<td style="text-align:right;">
14356
</td>
<td style="text-align:right;">
65777
</td>
<td style="text-align:right;">
10109
</td>
<td style="text-align:right;">
14899000
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
14.10
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
0.504
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
549.28
</td>
<td style="text-align:right;">
21077.05
</td>
<td style="text-align:right;">
128.129
</td>
<td style="text-align:right;">
21077.05
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
10998.67
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
21077.05
</td>
<td style="text-align:right;">
2725.50
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
6800.70
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
9.19
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
17.57
</td>
<td style="text-align:right;">
2.27
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
5.66
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
40302
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
42102
</td>
<td style="text-align:right;">
46600
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
40300
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
549.279
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
363656
</td>
<td style="text-align:right;">
0.700
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5941200
</td>
<td style="text-align:right;">
347071
</td>
<td style="text-align:right;">
519640
</td>
<td style="text-align:right;">
1990000
</td>
<td style="text-align:right;">
10109
</td>
<td style="text-align:right;">
2528900
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2.49
</td>
<td style="text-align:right;">
-0.583
</td>
<td style="text-align:right;">
0.430
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
388.72
</td>
<td style="text-align:right;">
40605.85
</td>
<td style="text-align:right;">
356.719
</td>
<td style="text-align:right;">
8087.76
</td>
<td style="text-align:right;">
6166.82
</td>
<td style="text-align:right;">
1371.27
</td>
<td style="text-align:right;">
2418.35
</td>
<td style="text-align:right;">
8087.76
</td>
<td style="text-align:right;">
5485.35
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1543.48
</td>
<td style="text-align:right;">
1943.96
</td>
<td style="text-align:right;">
7436.60
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6.92
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
2.71
</td>
<td style="text-align:right;">
9.04
</td>
<td style="text-align:right;">
6.13
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.72
</td>
<td style="text-align:right;">
2.16
</td>
<td style="text-align:right;">
8.26
</td>
<td style="text-align:right;">
40000
</td>
<td style="text-align:right;">
26400
</td>
<td style="text-align:right;">
30400
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
44300
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
28800
</td>
<td style="text-align:right;">
28500
</td>
<td style="text-align:right;">
38700
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
388.720
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
478135
</td>
<td style="text-align:right;">
28.710
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
10718000
</td>
<td style="text-align:right;">
291189
</td>
<td style="text-align:right;">
16653
</td>
<td style="text-align:right;">
142562
</td>
<td style="text-align:right;">
10142
</td>
<td style="text-align:right;">
9961100
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
8.60
</td>
<td style="text-align:right;">
-0.051
</td>
<td style="text-align:right;">
0.283
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
0.969
</td>
</tr>
</tbody>
</table>
<p></br></p>
<p>Table <a href="global-proteomics-data.html#tab:global-masic-table">1.2</a> shows the first 6 rows of <code>masic_data</code>.</p>
</div>
<div id="filter-masic-data" class="section level3" number="1.1.9">
<h3><span class="header-section-number">1.1.9</span> Filter MASIC Data</h3>
<p>The only other step in reporter ion intensity data preparation is to filter the results. Currently, we recommend keeping entries where at least 50% of the ion population is due to the targeted ion (interference score <span class="math inline">\(\geq\)</span> 0.5) and not filtering by S/N.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="global-proteomics-data.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter MASIC data</span></span>
<span id="cb21-2"><a href="global-proteomics-data.html#cb21-2" aria-hidden="true" tabindex="-1"></a>masic_data <span class="ot">&lt;-</span> <span class="fu">filter_masic_data</span>(masic_data, </span>
<span id="cb21-3"><a href="global-proteomics-data.html#cb21-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">interference_score_threshold =</span> <span class="fl">0.5</span>,</span>
<span id="cb21-4"><a href="global-proteomics-data.html#cb21-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">s2n_threshold =</span> <span class="dv">0</span>)</span></code></pre></div>
</div>
<div id="fetch-study-design-tables" class="section level3" number="1.1.10">
<h3><span class="header-section-number">1.1.10</span> Create Study Design Tables</h3>
<p>To convert from PSMs and reporter ion intensities to meaningful quantitative data, it is necessary to know what are the samples in the reporter channels and what is the intended reference channel (or combination of channels). The entire study design is captured by three tables - fractions, samples, references. With newly processed data, these typically do not exist, and must be created. The next sections show how to create these tables.</p>
</div>
<div id="fractions" class="section level3" number="1.1.11">
<h3><span class="header-section-number">1.1.11</span> Fractions</h3>
<p>The fractions table consists of two columns: <code>Dataset</code> and <code>PlexID</code>. The <code>Dataset</code> column contains all of the unique datasets from <code>msnid$Dataset</code> or <code>masic_data$Dataset</code>. The <code>PlexID</code> column contains the plex ID associated with each dataset, and is typically an “S” followed by a number (“S1,” “S2,” etc.). We can extract the plex ID from the datasets. In this case, the plex ID always comes after “<em>W</em>,” so we can use a regular expression (regex) to capture it (the first argument of <code>gsub</code>). The regex below says to capture an “S” followed by a single digit that appears after “<em>W</em>” and before an underscore.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="global-proteomics-data.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create fractions table</span></span>
<span id="cb22-2"><a href="global-proteomics-data.html#cb22-2" aria-hidden="true" tabindex="-1"></a>fractions <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Dataset =</span> <span class="fu">unique</span>(masic_data<span class="sc">$</span>Dataset)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="global-proteomics-data.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PlexID =</span> <span class="fu">gsub</span>(<span class="st">&quot;.*_W_(S</span><span class="sc">\\</span><span class="st">d{1})_.*&quot;</span>, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">1&quot;</span>, Dataset))</span></code></pre></div>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:fractions-table">Table 1.3: </span><left>First 10 rows of the fractions table.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:left;">
PlexID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_02_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_03_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_04_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_05_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_06_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_08_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_09_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_10_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
</tbody>
</table>
<p></br></p>
<p>Table <a href="global-proteomics-data.html#tab:fractions-table">1.3</a> shows the first 10 rows of <code>fractions</code>.</p>
</div>
<div id="samples" class="section level3" number="1.1.12">
<h3><span class="header-section-number">1.1.12</span> Samples</h3>
<p>The samples table contains columns <code>PlexID</code>, <code>QuantBlock</code>, <code>ReporterName</code>, <code>ReporterAlias</code>, and <code>MeasurementName</code>. The plex ID must be the same as the plex ID in the <code>fractions</code> table. <code>ReporterName</code> is the reporter ion name (“126,” “127N,” “127C,” etc.). <code>ReporterAlias</code> is the intermediate between <code>ReporterName</code> and <code>MeasurementName</code> and is used for defining the reference. <code>MeasurementName</code> determines the column names for the final cross-tab, and must be unique and begin with a letter. Finally, <code>QuantBlock</code> can be thought of as a way of defining sub-plex. In a typical TMT experiment, <code>QuantBlock</code> is always 1. In case of 5 pairwise comparisons within TMT10, there will be 5 QuantBlocks (1-5) with a reference for each <code>QuantBlock</code>.</p>
<p>For this experiment, channel 131 will serve as the reference, so we set <code>MeasurementName</code> to <code>NA</code> when <code>ReporterName</code> is <code>"131"</code>. This will make the reference channel absent from the quantitative cross-tab. In cases where reporter ion intensities are not normalized by a reference channel (reference = 1) or they are normalized by the average of select channels, do not set any <code>MeasurementName</code> to <code>NA</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="global-proteomics-data.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create samples table</span></span>
<span id="cb23-2"><a href="global-proteomics-data.html#cb23-2" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;data/MoTrPAC_pilot_TMT_labeling.txt&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="global-proteomics-data.html#cb23-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">ReporterName =</span> TMT10_channel, </span>
<span id="cb23-4"><a href="global-proteomics-data.html#cb23-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">ReporterAlias =</span> sample_ID) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="global-proteomics-data.html#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">QuantBlock =</span> <span class="dv">1</span>,</span>
<span id="cb23-6"><a href="global-proteomics-data.html#cb23-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">MeasurementName =</span> <span class="fu">ifelse</span>(ReporterAlias <span class="sc">==</span> <span class="st">&quot;ref&quot;</span>, <span class="cn">NA</span>, ReporterAlias))</span></code></pre></div>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:samples-table">Table 1.4: </span><left>First 10 rows of the samples table.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
ReporterAlias
</th>
<th style="text-align:left;">
ReporterName
</th>
<th style="text-align:left;">
PlexID
</th>
<th style="text-align:right;">
QuantBlock
</th>
<th style="text-align:left;">
MeasurementName
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
S_01
</td>
<td style="text-align:left;">
129C
</td>
<td style="text-align:left;">
S1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_01
</td>
</tr>
<tr>
<td style="text-align:left;">
S_02
</td>
<td style="text-align:left;">
129N
</td>
<td style="text-align:left;">
S1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_02
</td>
</tr>
<tr>
<td style="text-align:left;">
S_03
</td>
<td style="text-align:left;">
130C
</td>
<td style="text-align:left;">
S1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_03
</td>
</tr>
<tr>
<td style="text-align:left;">
S_04
</td>
<td style="text-align:left;">
130N
</td>
<td style="text-align:left;">
S1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_04
</td>
</tr>
<tr>
<td style="text-align:left;">
S_05
</td>
<td style="text-align:left;">
126
</td>
<td style="text-align:left;">
S2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_05
</td>
</tr>
<tr>
<td style="text-align:left;">
S_06
</td>
<td style="text-align:left;">
127C
</td>
<td style="text-align:left;">
S2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_06
</td>
</tr>
<tr>
<td style="text-align:left;">
S_07
</td>
<td style="text-align:left;">
127N
</td>
<td style="text-align:left;">
S2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_07
</td>
</tr>
<tr>
<td style="text-align:left;">
S_08
</td>
<td style="text-align:left;">
128C
</td>
<td style="text-align:left;">
S2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_08
</td>
</tr>
<tr>
<td style="text-align:left;">
S_09
</td>
<td style="text-align:left;">
128N
</td>
<td style="text-align:left;">
S2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
S_09
</td>
</tr>
<tr>
<td style="text-align:left;">
ref
</td>
<td style="text-align:left;">
131
</td>
<td style="text-align:left;">
S1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>
<p></br></p>
<p>Table <a href="global-proteomics-data.html#tab:samples-table">1.4</a> shows the first 10 rows of <code>samples</code>.</p>
</div>
<div id="references" class="section level3" number="1.1.13">
<h3><span class="header-section-number">1.1.13</span> References</h3>
<p>Reference can be a certain channel, average of multiple channels, or 1. The general form is an expression with <code>ReporterAlias</code> names as variables. It is evaluated for each <code>PlexID</code>/<code>QuantBlock</code> combination and applied to divide reporter ion intensities within corresponding <code>PlexID</code>/<code>QuantBlock</code>.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="global-proteomics-data.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create references table</span></span>
<span id="cb24-2"><a href="global-proteomics-data.html#cb24-2" aria-hidden="true" tabindex="-1"></a>references <span class="ot">&lt;-</span> samples <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="global-proteomics-data.html#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter to reference channel (ReporterName == &quot;131&quot;, ReporterAlias == &quot;ref&quot;)</span></span>
<span id="cb24-4"><a href="global-proteomics-data.html#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ReporterName <span class="sc">==</span> <span class="st">&quot;131&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="global-proteomics-data.html#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select required columns and rename ReporterAlias to Reference</span></span>
<span id="cb24-6"><a href="global-proteomics-data.html#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(PlexID, <span class="at">Reference =</span> ReporterAlias, QuantBlock)</span></code></pre></div>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:references-table">Table 1.5: </span><left>References table.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
PlexID
</th>
<th style="text-align:left;">
Reference
</th>
<th style="text-align:right;">
QuantBlock
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
ref
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S2
</td>
<td style="text-align:left;">
ref
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<p></br></p>
<p>Table <a href="global-proteomics-data.html#tab:references-table">1.5</a> shows the first 10 rows of <code>references</code>. The code to use the geometric average instead of a single channel as the reference is shown below. The geometric average is the product of the reporter ion channels to the power of (1/number of channels). For each <code>PlexID</code> group, collapse the vector of reporter ion names with <code>*</code>, surround them in parentheses, and raise to the power of (1/number of channels).</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="global-proteomics-data.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use geometric average as reference</span></span>
<span id="cb25-2"><a href="global-proteomics-data.html#cb25-2" aria-hidden="true" tabindex="-1"></a>references <span class="ot">&lt;-</span> samples <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="global-proteomics-data.html#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(PlexID, QuantBlock) <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="global-proteomics-data.html#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Reference =</span> <span class="fu">sprintf</span>(<span class="st">&quot;(%s)^(1/%d)&quot;</span>, </span>
<span id="cb25-5"><a href="global-proteomics-data.html#cb25-5" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">paste</span>(ReporterAlias, <span class="at">collapse =</span> <span class="st">&quot;*&quot;</span>), <span class="fu">n</span>()))</span>
<span id="cb25-6"><a href="global-proteomics-data.html#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="global-proteomics-data.html#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Do not normalize by reference channel (use 1 as the reference)</span></span>
<span id="cb25-8"><a href="global-proteomics-data.html#cb25-8" aria-hidden="true" tabindex="-1"></a>references <span class="ot">&lt;-</span> samples <span class="sc">%&gt;%</span> </span>
<span id="cb25-9"><a href="global-proteomics-data.html#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(PlexID, QuantBlock) <span class="sc">%&gt;%</span> </span>
<span id="cb25-10"><a href="global-proteomics-data.html#cb25-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Reference =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Now that we have the three study design tables, we should save them.</p>
<!---
TODO:
Explain how to add the study design tables to the DMS.
--->
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="global-proteomics-data.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save study design tables</span></span>
<span id="cb26-2"><a href="global-proteomics-data.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(fractions, <span class="at">file =</span> <span class="st">&quot;data/fractions.txt&quot;</span>,</span>
<span id="cb26-3"><a href="global-proteomics-data.html#cb26-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-4"><a href="global-proteomics-data.html#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(samples, <span class="at">file =</span> <span class="st">&quot;data/samples.txt&quot;</span>,</span>
<span id="cb26-5"><a href="global-proteomics-data.html#cb26-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-6"><a href="global-proteomics-data.html#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(references, <span class="at">file =</span> <span class="st">&quot;data/references.txt&quot;</span>,</span>
<span id="cb26-7"><a href="global-proteomics-data.html#cb26-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="global-quant-crosstab" class="section level3" number="1.1.14">
<h3><span class="header-section-number">1.1.14</span> Create Quantitative Cross-tab</h3>
<p>This is the step where MS/MS IDs and reporter ions are linked together and aggregated to the peptide or accession (i.e. protein) level. To retain protein IDs while aggregating to peptide level, set <code>aggregation_level &lt;- c("accession","peptide")</code>. The abundances are converted to relative abundances by dividing by the reference and then <span class="math inline">\(log_2\)</span>-transformed.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="global-proteomics-data.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set aggregation level</span></span>
<span id="cb27-2"><a href="global-proteomics-data.html#cb27-2" aria-hidden="true" tabindex="-1"></a>aggregation_level <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;accession&quot;</span>)</span>
<span id="cb27-3"><a href="global-proteomics-data.html#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create cross-tab</span></span>
<span id="cb27-4"><a href="global-proteomics-data.html#cb27-4" aria-hidden="true" tabindex="-1"></a>crosstab <span class="ot">&lt;-</span> <span class="fu">create_crosstab</span>(msnid, masic_data,</span>
<span id="cb27-5"><a href="global-proteomics-data.html#cb27-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">aggregation_level =</span> aggregation_level,</span>
<span id="cb27-6"><a href="global-proteomics-data.html#cb27-6" aria-hidden="true" tabindex="-1"></a>                            fractions, samples, references)</span></code></pre></div>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:unnamed-chunk-21">Table 1.6: </span><left>First 6 rows of the global quantitative cross-tab.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
R_01
</th>
<th style="text-align:right;">
R_02
</th>
<th style="text-align:right;">
R_03
</th>
<th style="text-align:right;">
R_04
</th>
<th style="text-align:right;">
R_05
</th>
<th style="text-align:right;">
R_06
</th>
<th style="text-align:right;">
R_07
</th>
<th style="text-align:right;">
R_08
</th>
<th style="text-align:right;">
R_09
</th>
<th style="text-align:right;">
S_01
</th>
<th style="text-align:right;">
S_02
</th>
<th style="text-align:right;">
S_03
</th>
<th style="text-align:right;">
S_04
</th>
<th style="text-align:right;">
S_05
</th>
<th style="text-align:right;">
S_06
</th>
<th style="text-align:right;">
S_07
</th>
<th style="text-align:right;">
S_08
</th>
<th style="text-align:right;">
S_09
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AP_004893.1
</td>
<td style="text-align:right;">
0.1419768
</td>
<td style="text-align:right;">
0.7628195
</td>
<td style="text-align:right;">
0.1653552
</td>
<td style="text-align:right;">
0.8662554
</td>
<td style="text-align:right;">
0.9453172
</td>
<td style="text-align:right;">
-0.8794712
</td>
<td style="text-align:right;">
-0.1912097
</td>
<td style="text-align:right;">
0.3964607
</td>
<td style="text-align:right;">
-0.2440478
</td>
<td style="text-align:right;">
-0.6460065
</td>
<td style="text-align:right;">
-1.9294467
</td>
<td style="text-align:right;">
-0.4321433
</td>
<td style="text-align:right;">
-1.2831873
</td>
<td style="text-align:right;">
-1.0271227
</td>
<td style="text-align:right;">
-0.9390945
</td>
<td style="text-align:right;">
0.4883309
</td>
<td style="text-align:right;">
-1.7148628
</td>
<td style="text-align:right;">
-0.7029685
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004894.1
</td>
<td style="text-align:right;">
0.8092676
</td>
<td style="text-align:right;">
-0.0976095
</td>
<td style="text-align:right;">
-0.3113350
</td>
<td style="text-align:right;">
0.3215692
</td>
<td style="text-align:right;">
0.2171255
</td>
<td style="text-align:right;">
-1.3515366
</td>
<td style="text-align:right;">
-0.7462995
</td>
<td style="text-align:right;">
-0.8338103
</td>
<td style="text-align:right;">
-0.2227493
</td>
<td style="text-align:right;">
-0.3678781
</td>
<td style="text-align:right;">
-0.1638689
</td>
<td style="text-align:right;">
-0.6696829
</td>
<td style="text-align:right;">
-1.2039041
</td>
<td style="text-align:right;">
-0.5124954
</td>
<td style="text-align:right;">
-0.2364175
</td>
<td style="text-align:right;">
-0.4428327
</td>
<td style="text-align:right;">
-1.3730408
</td>
<td style="text-align:right;">
-0.6711809
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004895.1
</td>
<td style="text-align:right;">
0.2078433
</td>
<td style="text-align:right;">
-0.2867209
</td>
<td style="text-align:right;">
-0.6089756
</td>
<td style="text-align:right;">
-0.1162062
</td>
<td style="text-align:right;">
-0.3840271
</td>
<td style="text-align:right;">
-0.6780284
</td>
<td style="text-align:right;">
-0.6102404
</td>
<td style="text-align:right;">
-0.3896190
</td>
<td style="text-align:right;">
-0.1548544
</td>
<td style="text-align:right;">
-1.1240967
</td>
<td style="text-align:right;">
-0.6908468
</td>
<td style="text-align:right;">
-0.6652575
</td>
<td style="text-align:right;">
-0.7140383
</td>
<td style="text-align:right;">
0.2717217
</td>
<td style="text-align:right;">
-0.1448289
</td>
<td style="text-align:right;">
-0.1200736
</td>
<td style="text-align:right;">
-0.6435709
</td>
<td style="text-align:right;">
-0.4287771
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004896.1
</td>
<td style="text-align:right;">
-0.1494849
</td>
<td style="text-align:right;">
-0.3664339
</td>
<td style="text-align:right;">
-0.7314368
</td>
<td style="text-align:right;">
-0.1742391
</td>
<td style="text-align:right;">
-0.5352280
</td>
<td style="text-align:right;">
-0.6543311
</td>
<td style="text-align:right;">
-0.6741064
</td>
<td style="text-align:right;">
-0.3994149
</td>
<td style="text-align:right;">
-0.0441485
</td>
<td style="text-align:right;">
-1.2945071
</td>
<td style="text-align:right;">
-1.0372327
</td>
<td style="text-align:right;">
-0.7060783
</td>
<td style="text-align:right;">
-0.8299749
</td>
<td style="text-align:right;">
0.1939540
</td>
<td style="text-align:right;">
-0.2274358
</td>
<td style="text-align:right;">
-0.1688422
</td>
<td style="text-align:right;">
-0.5251264
</td>
<td style="text-align:right;">
-0.4222698
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004898.1
</td>
<td style="text-align:right;">
0.0362964
</td>
<td style="text-align:right;">
0.7497227
</td>
<td style="text-align:right;">
0.4252227
</td>
<td style="text-align:right;">
0.4913660
</td>
<td style="text-align:right;">
1.1580326
</td>
<td style="text-align:right;">
-1.8439756
</td>
<td style="text-align:right;">
-0.1774225
</td>
<td style="text-align:right;">
-1.1083199
</td>
<td style="text-align:right;">
-0.4175363
</td>
<td style="text-align:right;">
0.1211536
</td>
<td style="text-align:right;">
-0.3640632
</td>
<td style="text-align:right;">
-0.3019505
</td>
<td style="text-align:right;">
-0.8291744
</td>
<td style="text-align:right;">
-0.8407749
</td>
<td style="text-align:right;">
-0.2796091
</td>
<td style="text-align:right;">
-0.4130732
</td>
<td style="text-align:right;">
-1.5747761
</td>
<td style="text-align:right;">
-0.9449498
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004899.1
</td>
<td style="text-align:right;">
0.7140968
</td>
<td style="text-align:right;">
-0.1781542
</td>
<td style="text-align:right;">
-0.3732752
</td>
<td style="text-align:right;">
0.3494902
</td>
<td style="text-align:right;">
-0.0615626
</td>
<td style="text-align:right;">
-1.0482424
</td>
<td style="text-align:right;">
-0.8052899
</td>
<td style="text-align:right;">
-0.6675429
</td>
<td style="text-align:right;">
-0.3959923
</td>
<td style="text-align:right;">
-2.1679002
</td>
<td style="text-align:right;">
-0.8550940
</td>
<td style="text-align:right;">
-0.9026145
</td>
<td style="text-align:right;">
-1.4519278
</td>
<td style="text-align:right;">
-0.3158081
</td>
<td style="text-align:right;">
-0.4056811
</td>
<td style="text-align:right;">
-0.4644758
</td>
<td style="text-align:right;">
-0.2805080
</td>
<td style="text-align:right;">
-0.9023044
</td>
</tr>
</tbody>
</table>
<p></br></p>
<p>Now that we have the cross-tab, we should save it.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="global-proteomics-data.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save cross-tab</span></span>
<span id="cb28-2"><a href="global-proteomics-data.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(crosstab, <span class="at">file =</span> <span class="st">&quot;data/global_quant_crosstab.txt&quot;</span>,</span>
<span id="cb28-3"><a href="global-proteomics-data.html#cb28-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We will also save the proteins (row names) of this cross-tab in order to demonstrate prioritized inference in Section <a href="phosphoproteomics-data.html#phosphoproteomics-data">1.2</a>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="global-proteomics-data.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save global proteins</span></span>
<span id="cb29-2"><a href="global-proteomics-data.html#cb29-2" aria-hidden="true" tabindex="-1"></a>global_proteins <span class="ot">&lt;-</span> <span class="fu">rownames</span>(crosstab)</span>
<span id="cb29-3"><a href="global-proteomics-data.html#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(global_proteins, <span class="at">file =</span> <span class="st">&quot;data/3442_global_proteins.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="create-msnset-object" class="section level3" number="1.1.15">
<h3><span class="header-section-number">1.1.15</span> Create MSnSet Object</h3>
<p>The final step is to create an MSnSet object, which is necessary for downstream data analysis tools such as the <code>plot_pca_*</code> or <code>limma_*</code> functions in MSnSet.utils. An MSnSet combines three tables into one object: <code>exprs</code>, <code>fData</code>, and <code>pData.</code> <code>exprs</code> is a matrix of protein abundance data with proteins as rows and samples as columns. <code>fData</code> is used to convert between different feature IDs such as RefSeq, UniProt accession, Entrez gene ID, gene symbol, etc. It is optional, but the rownames of <code>fData</code> must be the same as the row names of <code>exprs</code>. Lastly, <code>pData</code> contains the metadata with samples as rows, and the row names must be the same as the column names of <code>exprs</code> (same order as well). For this example, we don’t have an actual metadata table. We just know group assignment from the sample names. Samples with an “R” are part of the exeRcised group, while samples with an “S” are part of the Sedentary group. We can make a metadata table with a single column called <code>group</code>.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="global-proteomics-data.html#cb30-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">group =</span> <span class="fu">gsub</span>(<span class="st">&quot;([RS]).*&quot;</span>, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">1&quot;</span>, <span class="fu">colnames</span>(crosstab))) <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="global-proteomics-data.html#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">ifelse</span>(group <span class="sc">==</span> <span class="st">&quot;R&quot;</span>, <span class="st">&quot;exercised&quot;</span>, <span class="st">&quot;sedentary&quot;</span>))</span>
<span id="cb30-3"><a href="global-proteomics-data.html#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(crosstab)</span></code></pre></div>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:unnamed-chunk-25">Table 1.7: </span><left>Metadata for MSnSet.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
group
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
R_01
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_02
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_03
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_04
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_05
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_06
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_07
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_08
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
R_09
</td>
<td style="text-align:left;">
exercised
</td>
</tr>
<tr>
<td style="text-align:left;">
S_01
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_02
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_03
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_04
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_05
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_06
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_07
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_08
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
<tr>
<td style="text-align:left;">
S_09
</td>
<td style="text-align:left;">
sedentary
</td>
</tr>
</tbody>
</table>
<p>Since we used the sample data to create the metadata, everything is in the right order; however, we usually need to reorder the rows of the metadata. We can do this with</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="global-proteomics-data.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder rows of metadata to match sample order in crosstab</span></span>
<span id="cb31-2"><a href="global-proteomics-data.html#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Need to add drop = FALSE if the metadata is a single column</span></span>
<span id="cb31-3"><a href="global-proteomics-data.html#cb31-3" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata[<span class="fu">colnames</span>(crosstab), , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span></code></pre></div>
<p>We have the metadata and protein abundance data, so we can now create the MSnSet. We will call it <code>m1</code> and we should save it as an .RData file.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="global-proteomics-data.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create MSnSet</span></span>
<span id="cb32-2"><a href="global-proteomics-data.html#cb32-2" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> MSnbase<span class="sc">::</span><span class="fu">MSnSet</span>(<span class="at">exprs =</span> <span class="fu">as.matrix</span>(crosstab), <span class="at">pData =</span> metadata)</span>
<span id="cb32-3"><a href="global-proteomics-data.html#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">validObject</span>(m1)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>If we type the name of the MSnSet in the console, a summary will be displayed. This will show information about each of the tables, and the MSnSet will keep track of all processing done such as filtering at the bottom under “Processing information.”</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="global-proteomics-data.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display summary</span></span>
<span id="cb34-2"><a href="global-proteomics-data.html#cb34-2" aria-hidden="true" tabindex="-1"></a>m1</span></code></pre></div>
<pre><code>## MSnSet (storageMode: lockedEnvironment)
## assayData: 5178 features, 18 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: R_01 R_02 ... S_09 (18 total)
##   varLabels: group
##   varMetadata: labelDescription
## featureData: none
## experimentData: use &#39;experimentData(object)&#39;
## Annotation:  
## - - - Processing information - - -
##  MSnbase version: 2.18.0</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="global-proteomics-data.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save unprocessed MSnSet</span></span>
<span id="cb36-2"><a href="global-proteomics-data.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(m1, <span class="at">file =</span> <span class="st">&quot;data/global_msnset.RData&quot;</span>)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="isobaric-quantification-pipelines.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phosphoproteomics-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
