<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8.3 Gene Set Enrichment Analysis | Proteomics Data Analysis in R/Bioconductor</title>
  <meta name="description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="8.3 Gene Set Enrichment Analysis | Proteomics Data Analysis in R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8.3 Gene Set Enrichment Analysis | Proteomics Data Analysis in R/Bioconductor" />
  
  <meta name="twitter:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

<meta name="author" content="Tyler Sagendorf" />


<meta name="date" content="2022-05-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ora.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=".">Proteomics Data Analysis in R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="iso-global.html"><a href="iso-global.html"><i class="fa fa-check"></i><b>1</b> Isobaric Quantification: Proteomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html"><i class="fa fa-check"></i><b>1.1</b> Prepare MS/MS Identifications</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#read-ms-gf-data"><i class="fa fa-check"></i><b>1.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="1.1.2" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#correct-isotope-selection-error"><i class="fa fa-check"></i><b>1.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.1.3" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#remove-contaminants"><i class="fa fa-check"></i><b>1.1.3</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.1.4" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#global-peptide-filter"><i class="fa fa-check"></i><b>1.1.4</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.1.5" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#msms-id-filter-protein-level"><i class="fa fa-check"></i><b>1.1.5</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.1.6" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#inference-of-parsimonious-protein-set"><i class="fa fa-check"></i><b>1.1.6</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.1.7" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#remove-decoy-psms"><i class="fa fa-check"></i><b>1.1.7</b> Remove Decoy PSMs</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html"><i class="fa fa-check"></i><b>1.2</b> Prepare Reporter Ion Intensities</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html#read-masic-output"><i class="fa fa-check"></i><b>1.2.1</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.2.2" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html#filter-masic-data"><i class="fa fa-check"></i><b>1.2.2</b> Filter MASIC Data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html"><i class="fa fa-check"></i><b>1.3</b> Create Study Design Tables</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#fractions"><i class="fa fa-check"></i><b>1.3.1</b> Fractions</a></li>
<li class="chapter" data-level="1.3.2" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#samples"><i class="fa fa-check"></i><b>1.3.2</b> Samples</a></li>
<li class="chapter" data-level="1.3.3" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#global-references"><i class="fa fa-check"></i><b>1.3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="global-quant-crosstab.html"><a href="global-quant-crosstab.html"><i class="fa fa-check"></i><b>1.4</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="1.5" data-path="global-msnset.html"><a href="global-msnset.html"><i class="fa fa-check"></i><b>1.5</b> Create MSnSet</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="iso-phospho.html"><a href="iso-phospho.html"><i class="fa fa-check"></i><b>2</b> Isobaric Quantification: Phosphoproteomics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html"><i class="fa fa-check"></i><b>2.1</b> Prepare MS/MS Identifications</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#read-ms-gf-data-1"><i class="fa fa-check"></i><b>2.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="2.1.2" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#correct-isotope-selection-error-1"><i class="fa fa-check"></i><b>2.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="2.1.3" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-unmodified-peptides"><i class="fa fa-check"></i><b>2.1.3</b> Remove Unmodified Peptides</a></li>
<li class="chapter" data-level="2.1.4" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-contaminants-1"><i class="fa fa-check"></i><b>2.1.4</b> Remove Contaminants</a></li>
<li class="chapter" data-level="2.1.5" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#improve-phosphosite-localization"><i class="fa fa-check"></i><b>2.1.5</b> Improve Phosphosite Localization</a></li>
<li class="chapter" data-level="2.1.6" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#msms-id-filter-peptide-level"><i class="fa fa-check"></i><b>2.1.6</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="2.1.7" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#msms-id-filter-protein-level-1"><i class="fa fa-check"></i><b>2.1.7</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="2.1.8" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#inference-of-parsimonious-protein-set-1"><i class="fa fa-check"></i><b>2.1.8</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="2.1.9" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#map-mod-sites"><i class="fa fa-check"></i><b>2.1.9</b> Map Sites to Protein Sequences</a></li>
<li class="chapter" data-level="2.1.10" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-decoy-psms-1"><i class="fa fa-check"></i><b>2.1.10</b> Remove Decoy PSMs</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html"><i class="fa fa-check"></i><b>2.2</b> Prepare Reporter Ion Intensities</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html#read-masic-output-1"><i class="fa fa-check"></i><b>2.2.1</b> Read MASIC Output</a></li>
<li class="chapter" data-level="2.2.2" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html#filter-masic-data-1"><i class="fa fa-check"></i><b>2.2.2</b> Filter MASIC Data</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="create-study-design-tables.html"><a href="create-study-design-tables.html"><i class="fa fa-check"></i><b>2.3</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="2.4" data-path="create-quantitative-cross-tab.html"><a href="create-quantitative-cross-tab.html"><i class="fa fa-check"></i><b>2.4</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="2.5" data-path="phospho-msnset.html"><a href="phospho-msnset.html"><i class="fa fa-check"></i><b>2.5</b> Create MSnSet</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spectral-counting.html"><a href="spectral-counting.html"><i class="fa fa-check"></i><b>3</b> Spectral Counting</a></li>
<li class="chapter" data-level="4" data-path="id-conversion.html"><a href="id-conversion.html"><i class="fa fa-check"></i><b>4</b> Feature ID Conversion</a>
<ul>
<li class="chapter" data-level="4.1" data-path="conversion-with-biomart.html"><a href="conversion-with-biomart.html"><i class="fa fa-check"></i><b>4.1</b> Conversion with <code>biomaRt</code></a></li>
<li class="chapter" data-level="4.2" data-path="conversion-with-annotationhub.html"><a href="conversion-with-annotationhub.html"><i class="fa fa-check"></i><b>4.2</b> Conversion with <code>AnnotationHub</code></a></li>
<li class="chapter" data-level="4.3" data-path="conversion-using-fasta-headers.html"><a href="conversion-using-fasta-headers.html"><i class="fa fa-check"></i><b>4.3</b> Conversion Using FASTA Headers</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>5</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="count-features-in-samples.html"><a href="count-features-in-samples.html"><i class="fa fa-check"></i><b>5.1</b> Count Features in Samples</a></li>
<li class="chapter" data-level="5.2" data-path="sample-boxplots.html"><a href="sample-boxplots.html"><i class="fa fa-check"></i><b>5.2</b> Sample Boxplots</a></li>
<li class="chapter" data-level="5.3" data-path="estimate-blood-contamination.html"><a href="estimate-blood-contamination.html"><i class="fa fa-check"></i><b>5.3</b> Estimate Blood Contamination</a></li>
<li class="chapter" data-level="5.4" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>5.4</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="heatmaps.html"><a href="heatmaps.html"><i class="fa fa-check"></i><b>6</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="6.1" data-path="expression-heatmaps.html"><a href="expression-heatmaps.html"><i class="fa fa-check"></i><b>6.1</b> Expression Heatmaps</a></li>
<li class="chapter" data-level="6.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html"><i class="fa fa-check"></i><b>6.2</b> Correlation Heatmaps</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#sample-correlation"><i class="fa fa-check"></i><b>6.2.1</b> Sample Correlation</a></li>
<li class="chapter" data-level="6.2.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#feature-correlation"><i class="fa fa-check"></i><b>6.2.2</b> Feature Correlation</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html"><i class="fa fa-check"></i><b>6.3</b> Heatmap Annotation</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html#modifying-default-colors"><i class="fa fa-check"></i><b>6.3.1</b> Modifying Default Colors</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="modifications.html"><a href="modifications.html"><i class="fa fa-check"></i><b>6.4</b> Modifications</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="modifications.html"><a href="modifications.html#row-and-column-labels"><i class="fa fa-check"></i><b>6.4.1</b> Row and column labels</a></li>
<li class="chapter" data-level="6.4.2" data-path="modifications.html"><a href="modifications.html#label-colors"><i class="fa fa-check"></i><b>6.4.2</b> Label colors</a></li>
<li class="chapter" data-level="6.4.3" data-path="modifications.html"><a href="modifications.html#label-specific-features"><i class="fa fa-check"></i><b>6.4.3</b> Label specific features</a></li>
<li class="chapter" data-level="6.4.4" data-path="modifications.html"><a href="modifications.html#heatmap-body-color"><i class="fa fa-check"></i><b>6.4.4</b> Heatmap body color</a></li>
<li class="chapter" data-level="6.4.5" data-path="modifications.html"><a href="modifications.html#horizontal-heatmaps"><i class="fa fa-check"></i><b>6.4.5</b> Horizontal heatmaps</a></li>
<li class="chapter" data-level="6.4.6" data-path="modifications.html"><a href="modifications.html#legends"><i class="fa fa-check"></i><b>6.4.6</b> Legends</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="DEA.html"><a href="DEA.html"><i class="fa fa-check"></i><b>7</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="linear-reg.html"><a href="linear-reg.html"><i class="fa fa-check"></i><b>7.1</b> Linear Regression</a></li>
<li class="chapter" data-level="7.2" data-path="t-tests.html"><a href="t-tests.html"><i class="fa fa-check"></i><b>7.2</b> Two-Sample t-tests</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="t-tests.html"><a href="t-tests.html#t-single"><i class="fa fa-check"></i><b>7.2.1</b> One comparison</a></li>
<li class="chapter" data-level="7.2.2" data-path="t-tests.html"><a href="t-tests.html#t-mult"><i class="fa fa-check"></i><b>7.2.2</b> Multiple comparisons</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="anova.html"><a href="anova.html"><i class="fa fa-check"></i><b>7.3</b> One-Way ANOVA</a></li>
<li class="chapter" data-level="7.4" data-path="p-value-histograms.html"><a href="p-value-histograms.html"><i class="fa fa-check"></i><b>7.4</b> p-value Histograms</a></li>
<li class="chapter" data-level="7.5" data-path="volcano-plots.html"><a href="volcano-plots.html"><i class="fa fa-check"></i><b>7.5</b> Volcano Plots</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="volcano-plots.html"><a href="volcano-plots.html#base-plot"><i class="fa fa-check"></i><b>7.5.1</b> Base plot</a></li>
<li class="chapter" data-level="7.5.2" data-path="volcano-plots.html"><a href="volcano-plots.html#label-top-features"><i class="fa fa-check"></i><b>7.5.2</b> Label top features</a></li>
<li class="chapter" data-level="7.5.3" data-path="volcano-plots.html"><a href="volcano-plots.html#label-specific-features-1"><i class="fa fa-check"></i><b>7.5.3</b> Label specific features</a></li>
<li class="chapter" data-level="7.5.4" data-path="volcano-plots.html"><a href="volcano-plots.html#modify-point-colors"><i class="fa fa-check"></i><b>7.5.4</b> Modify point colors</a></li>
<li class="chapter" data-level="7.5.5" data-path="volcano-plots.html"><a href="volcano-plots.html#multiple-volcano-plots"><i class="fa fa-check"></i><b>7.5.5</b> Multiple volcano plots</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="upset-plots.html"><a href="upset-plots.html"><i class="fa fa-check"></i><b>7.6</b> UpSet Plots</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>8</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="annotation-databases.html"><a href="annotation-databases.html"><i class="fa fa-check"></i><b>8.1</b> Annotation Databases</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#gene-ontology"><i class="fa fa-check"></i><b>8.1.1</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="8.1.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#semantic-similarity"><i class="fa fa-check"></i><b>8.1.1.1</b> Semantic Similarity</a></li>
<li class="chapter" data-level="8.1.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#go-slim"><i class="fa fa-check"></i><b>8.1.1.2</b> GO Subsets/Slims</a></li>
</ul></li>
<li class="chapter" data-level="8.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#reactome"><i class="fa fa-check"></i><b>8.1.2</b> Reactome</a></li>
<li class="chapter" data-level="8.1.3" data-path="annotation-databases.html"><a href="annotation-databases.html#kegg"><i class="fa fa-check"></i><b>8.1.3</b> KEGG</a></li>
<li class="chapter" data-level="8.1.4" data-path="annotation-databases.html"><a href="annotation-databases.html#msigdb"><i class="fa fa-check"></i><b>8.1.4</b> MSigDB</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="ora.html"><a href="ora.html"><i class="fa fa-check"></i><b>8.2</b> Over-Representation Analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ora.html"><a href="ora.html#ora-overview"><i class="fa fa-check"></i><b>8.2.1</b> Overview</a></li>
<li class="chapter" data-level="8.2.2" data-path="ora.html"><a href="ora.html#ora-math"><i class="fa fa-check"></i><b>8.2.2</b> Mathematical Details</a></li>
<li class="chapter" data-level="8.2.3" data-path="ora.html"><a href="ora.html#ora-drawbacks"><i class="fa fa-check"></i><b>8.2.3</b> Drawbacks</a></li>
<li class="chapter" data-level="8.2.4" data-path="ora.html"><a href="ora.html#ora-examples"><i class="fa fa-check"></i><b>8.2.4</b> Examples</a>
<ul>
<li class="chapter" data-level="8.2.4.1" data-path="ora.html"><a href="ora.html#fgsea-ora"><i class="fa fa-check"></i><b>8.2.4.1</b> ORA with <code>fgsea</code></a></li>
<li class="chapter" data-level="8.2.4.2" data-path="ora.html"><a href="ora.html#ora-clustprof"><i class="fa fa-check"></i><b>8.2.4.2</b> ORA with <code>clusterProfiler</code>/<code>ReactomePA</code></a></li>
<li class="chapter" data-level="8.2.4.3" data-path="ora.html"><a href="ora.html#ora-with-gostats"><i class="fa fa-check"></i><b>8.2.4.3</b> ORA with <code>GOstats</code></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="gsea.html"><a href="gsea.html"><i class="fa fa-check"></i><b>8.3</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gsea.html"><a href="gsea.html#gsea-overview"><i class="fa fa-check"></i><b>8.3.1</b> Overview</a></li>
<li class="chapter" data-level="8.3.2" data-path="gsea.html"><a href="gsea.html#gsea-examples"><i class="fa fa-check"></i><b>8.3.2</b> Examples</a>
<ul>
<li class="chapter" data-level="8.3.2.1" data-path="gsea.html"><a href="gsea.html#gsea-with-fgsea"><i class="fa fa-check"></i><b>8.3.2.1</b> GSEA with <code>fgsea</code></a></li>
<li class="chapter" data-level="8.3.2.2" data-path="gsea.html"><a href="gsea.html#gsea-with-clusterprofilerreactomepa"><i class="fa fa-check"></i><b>8.3.2.2</b> GSEA with <code>clusterProfiler</code>/<code>ReactomePA</code></a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Proteomics Data Analysis in R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gsea" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Gene Set Enrichment Analysis<a href="gsea.html#gsea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="gsea-overview" class="section level3 hasAnchor" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Overview<a href="gsea.html#gsea-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>(More details to be added at a later date.)</strong></p>
<p><span style="color:#3366ff">Gene set enrichment analysis (GSEA) is a rank-based approach that determines whether predefined groups of genes/proteins/etc. are primarily up or down in one condition relative to another</span> <span class="citation">(<a href="#ref-mootha_pgc-1-responsive_2003" role="doc-biblioref">Vamsi K. Mootha et al., 2003</a>; <a href="#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al., 2005</a>)</span>. It is typically performed as a follow-up to differential analysis, and is preferred to ORA (Section <a href="ora.html#ora">8.2</a>).</p>
</div>
<div id="gsea-examples" class="section level3 hasAnchor" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Examples<a href="gsea.html#gsea-examples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>These examples will show how to run Fast GSEA (FGSEA) in R, which is based on the gene permutation approach <span class="citation">(<a href="#ref-korotkevich_fast_2016" role="doc-biblioref">Korotkevich et al., 2016</a>)</span>.</p>
<p>The input of FGSEA is a list of gene sets/pathways to check and a uniquely-named vector of ranking metric values sorted in descending order. The ranking metric that we will use is <span class="math inline">\(-log_{10}(\text{p-value}) * sign(\text{logFC})\)</span>, but we could have easily used t-statistics or some other metric. We will perform differential analysis on the <code>cptac_oca</code> data from the <code>MSnSet.utils</code> package and use the results to create the ranking metric vector. We start off by remapping features from RefSeq to Entrez ID.</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="gsea.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Fetch REFSEQ to ENTREZID conversion table</span></span>
<span id="cb162-2"><a href="gsea.html#cb162-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MSnID)</span>
<span id="cb162-3"><a href="gsea.html#cb162-3" aria-hidden="true" tabindex="-1"></a>conv_tbl <span class="ot">&lt;-</span> <span class="fu">fetch_conversion_table</span>(<span class="at">organism_name =</span> <span class="st">&quot;Homo sapiens&quot;</span>, </span>
<span id="cb162-4"><a href="gsea.html#cb162-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">from =</span> <span class="st">&quot;REFSEQ&quot;</span>, <span class="at">to =</span> <span class="st">&quot;ENTREZID&quot;</span>)</span>
<span id="cb162-5"><a href="gsea.html#cb162-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(conv_tbl)</span></code></pre></div>
<p>Now, we will create the differential analysis results and add the ENTREZID column.</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="gsea.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MSnSet.utils)</span>
<span id="cb163-2"><a href="gsea.html#cb163-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;cptac_oca&quot;</span>)</span>
<span id="cb163-3"><a href="gsea.html#cb163-3" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> oca.set</span>
<span id="cb163-4"><a href="gsea.html#cb163-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-5"><a href="gsea.html#cb163-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential analysis</span></span>
<span id="cb163-6"><a href="gsea.html#cb163-6" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">limma_a_b</span>(<span class="at">eset =</span> m1, </span>
<span id="cb163-7"><a href="gsea.html#cb163-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">model.str =</span> <span class="st">&quot;~ PLATINUM.STATUS&quot;</span>,</span>
<span id="cb163-8"><a href="gsea.html#cb163-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">coef.str =</span> <span class="st">&quot;PLATINUM.STATUS&quot;</span>)</span>
<span id="cb163-9"><a href="gsea.html#cb163-9" aria-hidden="true" tabindex="-1"></a><span class="co"># table(res$adj.P.Val &lt; 0.05) # 0</span></span>
<span id="cb163-10"><a href="gsea.html#cb163-10" aria-hidden="true" tabindex="-1"></a><span class="co"># hist(res$P.Value, breaks = seq(0, 1, 0.05)) # looks uniform</span></span>
<span id="cb163-11"><a href="gsea.html#cb163-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res)</span></code></pre></div>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="gsea.html#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb164-2"><a href="gsea.html#cb164-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ENTREZID column</span></span>
<span id="cb164-3"><a href="gsea.html#cb164-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb164-4"><a href="gsea.html#cb164-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REFSEQ =</span> <span class="fu">sub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.</span><span class="sc">\\</span><span class="st">d+&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(.))) <span class="sc">%&gt;%</span> </span>
<span id="cb164-5"><a href="gsea.html#cb164-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(conv_tbl, <span class="at">by =</span> <span class="st">&quot;REFSEQ&quot;</span>)</span>
<span id="cb164-6"><a href="gsea.html#cb164-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res)</span></code></pre></div>
<p>We need one ranking metric value per Entrez ID, so we will calculate the ranking metric and then take the gene-wise average. Then, we sort the genes in descending order by their ranking metric values and convert to a named vector called <code>geneList</code>.</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="gsea.html#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Ranking metric vector for GSEA</span></span>
<span id="cb165-2"><a href="gsea.html#cb165-2" aria-hidden="true" tabindex="-1"></a>geneList <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb165-3"><a href="gsea.html#cb165-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID), <span class="sc">!</span><span class="fu">is.na</span>(logFC)) <span class="sc">%&gt;%</span> </span>
<span id="cb165-4"><a href="gsea.html#cb165-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ranking_metric =</span> <span class="sc">-</span><span class="fu">log10</span>(P.Value)<span class="sc">*</span><span class="fu">sign</span>(logFC)) <span class="sc">%&gt;%</span> </span>
<span id="cb165-5"><a href="gsea.html#cb165-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ENTREZID) <span class="sc">%&gt;%</span> </span>
<span id="cb165-6"><a href="gsea.html#cb165-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">ranking_metric =</span> <span class="fu">mean</span>(ranking_metric, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb165-7"><a href="gsea.html#cb165-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>ranking_metric) <span class="sc">%&gt;%</span> <span class="co"># sort descending (important!)</span></span>
<span id="cb165-8"><a href="gsea.html#cb165-8" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">deframe</span>() <span class="co"># convert to named vector</span></span>
<span id="cb165-9"><a href="gsea.html#cb165-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geneList)</span>
<span id="cb165-10"><a href="gsea.html#cb165-10" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(geneList)</span></code></pre></div>
<p>Now that we have the uniquely-named ranking metric vector, we can proceed with FGSEA. For these examples, we will show how to perform FGSEA with the <code>fgsea</code>, <code>clusterProfiler</code>, and <code>ReactomePA</code> packages. The databases that we will cover are Gene Ontology Biological Processes and Reactome, and we will only consider gene sets/pathways with at least 15 and no more than 500 genes. The maximum gene set size, as well as the number of permutations, affect the normalized enrichment scores and p-values. For details on these different annotation databases, please see Section <a href="annotation-databases.html#annotation-databases">8.1</a> (Annotation Databases).</p>
<div id="gsea-with-fgsea" class="section level4 hasAnchor" number="8.3.2.1">
<h4><span class="header-section-number">8.3.2.1</span> GSEA with <code>fgsea</code><a href="gsea.html#gsea-with-fgsea" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To perform FGSEA with the <code>fgsea</code> package, we need a list of gene sets/pathways and the ranking metric vector. Below is one way to get the gene set list.</p>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="gsea.html#cb166-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MSigDB R package</span></span>
<span id="cb166-2"><a href="gsea.html#cb166-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb166-3"><a href="gsea.html#cb166-3" aria-hidden="true" tabindex="-1"></a>msigdbr<span class="sc">::</span><span class="fu">msigdbr_collections</span>() <span class="co"># available collections</span></span>
<span id="cb166-4"><a href="gsea.html#cb166-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to Human GO-BP sets</span></span>
<span id="cb166-5"><a href="gsea.html#cb166-5" aria-hidden="true" tabindex="-1"></a>BP_db <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, </span>
<span id="cb166-6"><a href="gsea.html#cb166-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">category =</span> <span class="st">&quot;C5&quot;</span>, <span class="at">subcategory =</span> <span class="st">&quot;GO:BP&quot;</span>)</span>
<span id="cb166-7"><a href="gsea.html#cb166-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(BP_db)</span></code></pre></div>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="gsea.html#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to a list of gene sets</span></span>
<span id="cb167-2"><a href="gsea.html#cb167-2" aria-hidden="true" tabindex="-1"></a>BP_conv <span class="ot">&lt;-</span> <span class="fu">unique</span>(BP_db[, <span class="fu">c</span>(<span class="st">&quot;entrez_gene&quot;</span>, <span class="st">&quot;gs_exact_source&quot;</span>)])</span>
<span id="cb167-3"><a href="gsea.html#cb167-3" aria-hidden="true" tabindex="-1"></a>BP_list <span class="ot">&lt;-</span> <span class="fu">split</span>(<span class="at">x =</span> BP_conv<span class="sc">$</span>entrez_gene, <span class="at">f =</span> BP_conv<span class="sc">$</span>gs_exact_source)</span>
<span id="cb167-4"><a href="gsea.html#cb167-4" aria-hidden="true" tabindex="-1"></a><span class="co"># First ~6 IDs of first 3 terms</span></span>
<span id="cb167-5"><a href="gsea.html#cb167-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="fu">head</span>(BP_list, <span class="dv">3</span>), head)</span></code></pre></div>
<p>The <code>fgseaMultilevel</code> function uses the adaptive multilevel split Monte Carlo approach described in the original FGSEA paper <span class="citation">(<a href="#ref-korotkevich_fast_2016" role="doc-biblioref">Korotkevich et al., 2016</a>)</span>.</p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="gsea.html#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="do">## GO-BP FGSEA with fgsea package</span></span>
<span id="cb168-2"><a href="gsea.html#cb168-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span>
<span id="cb168-3"><a href="gsea.html#cb168-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">99</span>)</span>
<span id="cb168-4"><a href="gsea.html#cb168-4" aria-hidden="true" tabindex="-1"></a><span class="fu">system.time</span>( <span class="co"># keep track of elapsed time</span></span>
<span id="cb168-5"><a href="gsea.html#cb168-5" aria-hidden="true" tabindex="-1"></a>  fgsea_res <span class="ot">&lt;-</span> <span class="fu">fgseaMultilevel</span>(<span class="at">pathways =</span> BP_list,</span>
<span id="cb168-6"><a href="gsea.html#cb168-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">stats =</span> geneList, </span>
<span id="cb168-7"><a href="gsea.html#cb168-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">minSize =</span> <span class="dv">15</span>, </span>
<span id="cb168-8"><a href="gsea.html#cb168-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">maxSize =</span> <span class="dv">500</span>, </span>
<span id="cb168-9"><a href="gsea.html#cb168-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">eps =</span> <span class="dv">0</span>, </span>
<span id="cb168-10"><a href="gsea.html#cb168-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">nPermSimple =</span> <span class="dv">10000</span>)</span>
<span id="cb168-11"><a href="gsea.html#cb168-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb168-12"><a href="gsea.html#cb168-12" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 rows with lowest enrichment p-values</span></span>
<span id="cb168-13"><a href="gsea.html#cb168-13" aria-hidden="true" tabindex="-1"></a>fgsea_res <span class="sc">%&gt;%</span> </span>
<span id="cb168-14"><a href="gsea.html#cb168-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add additional columns from BP_db</span></span>
<span id="cb168-15"><a href="gsea.html#cb168-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">distinct</span>(BP_db, gs_subcat, gs_exact_source, </span>
<span id="cb168-16"><a href="gsea.html#cb168-16" aria-hidden="true" tabindex="-1"></a>                     gs_name, gs_description),</span>
<span id="cb168-17"><a href="gsea.html#cb168-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;pathway&quot;</span> <span class="ot">=</span> <span class="st">&quot;gs_exact_source&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb168-18"><a href="gsea.html#cb168-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reformat descriptions</span></span>
<span id="cb168-19"><a href="gsea.html#cb168-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gs_name =</span> <span class="fu">sub</span>(<span class="st">&quot;^GOBP_&quot;</span>, <span class="st">&quot;&quot;</span>, gs_name),</span>
<span id="cb168-20"><a href="gsea.html#cb168-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">gs_name =</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, gs_name)) <span class="sc">%&gt;%</span> </span>
<span id="cb168-21"><a href="gsea.html#cb168-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span> </span>
<span id="cb168-22"><a href="gsea.html#cb168-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">8</span>)</span></code></pre></div>
<p>See <code>?fgseaMultilevel</code> for a description of the output. The output does not include term descriptions, so we had to add those ourselves.</p>
</div>
<div id="gsea-with-clusterprofilerreactomepa" class="section level4 hasAnchor" number="8.3.2.2">
<h4><span class="header-section-number">8.3.2.2</span> GSEA with <code>clusterProfiler</code>/<code>ReactomePA</code><a href="gsea.html#gsea-with-clusterprofilerreactomepa" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>See Section <a href="ora.html#ora-clustprof">8.2.4.2</a> to get a better understanding of these packages. <code>clusterProfiler</code> provides the <code>gseGO</code> and <code>gseKEGG</code> functions (among others) for FGSEA of the GO and KEGG databases, respectively. They are essentially more user-friendly wrapper functions that make use of the <code>fgsea</code> and <code>AnnotationDbi</code> package, but they tend to be much slower.</p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="gsea.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="do">## GO-BP FGSEA with clusterProfiler package</span></span>
<span id="cb169-2"><a href="gsea.html#cb169-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb169-3"><a href="gsea.html#cb169-3" aria-hidden="true" tabindex="-1"></a><span class="fu">system.time</span>( <span class="co"># keep track of elapsed time</span></span>
<span id="cb169-4"><a href="gsea.html#cb169-4" aria-hidden="true" tabindex="-1"></a>  cgsea_res <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList =</span> geneList, </span>
<span id="cb169-5"><a href="gsea.html#cb169-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb169-6"><a href="gsea.html#cb169-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">OrgDb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>, </span>
<span id="cb169-7"><a href="gsea.html#cb169-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">minGSSize =</span> <span class="dv">15</span>, </span>
<span id="cb169-8"><a href="gsea.html#cb169-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">maxGSSize =</span> <span class="dv">500</span>, </span>
<span id="cb169-9"><a href="gsea.html#cb169-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">eps =</span> <span class="dv">0</span>, </span>
<span id="cb169-10"><a href="gsea.html#cb169-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">nPermSimple =</span> <span class="dv">10000</span>, </span>
<span id="cb169-11"><a href="gsea.html#cb169-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">seed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb169-12"><a href="gsea.html#cb169-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb169-13"><a href="gsea.html#cb169-13" aria-hidden="true" tabindex="-1"></a><span class="co"># First 8 rows with lowest enrichment p-values</span></span>
<span id="cb169-14"><a href="gsea.html#cb169-14" aria-hidden="true" tabindex="-1"></a>cgsea_res<span class="sc">@</span>result <span class="sc">%&gt;%</span> </span>
<span id="cb169-15"><a href="gsea.html#cb169-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(pvalue) <span class="sc">%&gt;%</span> </span>
<span id="cb169-16"><a href="gsea.html#cb169-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">8</span>)</span></code></pre></div>
<p>Notice that the normalized enrichment scores (NES) are not quite the same as what we got when we used <code>fgseaMultiLevel</code>. This has to do with the permutations being different. To increase the precision of the NES and p-values, we can increase <code>nPermSimple</code>, though 10,000 should be more than sufficient.</p>
<p>Now, here is GSEA with the Reactome database using the <code>gsePathway</code> function from the <code>ReactomePA</code> package.</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="gsea.html#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Reactome FGSEA with ReactomePA package</span></span>
<span id="cb170-2"><a href="gsea.html#cb170-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ReactomePA)</span>
<span id="cb170-3"><a href="gsea.html#cb170-3" aria-hidden="true" tabindex="-1"></a>fgsea_react <span class="ot">&lt;-</span> <span class="fu">gsePathway</span>(<span class="at">geneList =</span> geneList, </span>
<span id="cb170-4"><a href="gsea.html#cb170-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">organism =</span> <span class="st">&quot;human&quot;</span>,</span>
<span id="cb170-5"><a href="gsea.html#cb170-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">minGSSize =</span> <span class="dv">15</span>, </span>
<span id="cb170-6"><a href="gsea.html#cb170-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">maxGSSize =</span> <span class="dv">500</span>, </span>
<span id="cb170-7"><a href="gsea.html#cb170-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">eps =</span> <span class="dv">0</span>, </span>
<span id="cb170-8"><a href="gsea.html#cb170-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">nPermSimple =</span> <span class="dv">10000</span>, </span>
<span id="cb170-9"><a href="gsea.html#cb170-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">seed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb170-10"><a href="gsea.html#cb170-10" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 rows</span></span>
<span id="cb170-11"><a href="gsea.html#cb170-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fgsea_react<span class="sc">@</span>result)</span></code></pre></div>

</div>
</div>
</div>
<!-- </div> -->
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-korotkevich_fast_2016" class="csl-entry">
Korotkevich, G., Sukhov, V., Budin, N., Shpak, B., Artyomov, M. N., and Sergushichev, A., <a href="https://doi.org/10.1101/060012">Fast Gene Set Enrichment Analysis</a>, Bioinformatics, Jun. 2016.
</div>
<div id="ref-mootha_pgc-1-responsive_2003" class="csl-entry">
Mootha, Vamsi K., Lindgren, C. M., Eriksson, K.-F., Subramanian, A., Sihag, S., Lehar, J., Puigserver, P., et al., <span>PGC</span>-1α-Responsive Genes Involved in Oxidative Phosphorylation Are Coordinately Downregulated in Human Diabetes, <em>Nature Genetics</em>, vol. <strong>34</strong>, no. 3, pp. 267–73, accessed September 19, 2021, from <a href="https://www.nature.com/articles/ng1180">https://www.nature.com/articles/ng1180</a>, July 2003. DOI: <a href="https://doi.org/10.1038/ng1180">10.1038/ng1180</a>
</div>
<div id="ref-subramanian_gene_2005" class="csl-entry">
Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., et al., Gene Set Enrichment Analysis: <span>A</span> Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles, <em>Proceedings of the National Academy of Sciences of the United States of America</em>, vol. <strong>102</strong>, no. 43, pp. 15545–50, accessed September 7, 2021, from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/</a>, October 2005. DOI: <a href="https://doi.org/10.1073/pnas.0506580102">10.1073/pnas.0506580102</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="ora.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
