<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Section 1 Isobaric Quantification Pipelines | Proteomics Data Analysis in R/Bioconductor</title>
<meta name="author" content="Tyler Sagendorf">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.9/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Proteomics Data Analysis in R/Bioconductor</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome!</a></li>
<li><a class="active" href="isobaric-quantification-pipelines.html"><span class="header-section-number">1</span> Isobaric Quantification Pipelines</a></li>
<li><a class="" href="exploratory-data-analysis.html"><span class="header-section-number">2</span> Exploratory Data Analysis</a></li>
<li><a class="" href="DEA.html"><span class="header-section-number">3</span> Differential Expression Analysis</a></li>
<li><a class="" href="pathway-analysis.html"><span class="header-section-number">4</span> Pathway Analysis</a></li>
<li><a class="" href="references-2.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="isobaric-quantification-pipelines" class="section level1" number="1">
<h1>
<span class="header-section-number">Section 1</span> Isobaric Quantification Pipelines<a class="anchor" aria-label="anchor" href="#isobaric-quantification-pipelines"><i class="fas fa-link"></i></a>
</h1>
<!---
TODO:
* Overview of isobaric labeling
--->
<p>Section summary to be added later. Please continue to subsections.</p>
<div id="global-proteomics-data" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> Global Proteomics Data<a class="anchor" aria-label="anchor" href="#global-proteomics-data"><i class="fas fa-link"></i></a>
</h2>
<p>This pipeline shows how to process TMT data that is processed outside of PNNL’s DMS. Section <a href="isobaric-quantification-pipelines.html#phosphoproteomics-data">1.2</a> shows how to process data from the DMS. For convenience, the results of MS-GF+ and MASIC processing are provided in a companion <code>PlexedPiperTestData</code> package. For this section, we need three packages: <code>PlexedPiper</code> for isobaric quantification, <code>PlexedPiperTestData</code>, and <code>dplyr</code> to manipulate data frames.</p>
<!---
For these examples, we need two packages: PlexedPiper and PlexedPiperTestData. The former is used to interface with the DMS, while the latter is tailored to isobaric quantification.
--->
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Setup</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PlexedPiper</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PlexedPiperTestData</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<div id="prepare-MS2-IDs-global" class="section level3" number="1.1.1">
<h3>
<span class="header-section-number">1.1.1</span> Prepare MS/MS Identifications<a class="anchor" aria-label="anchor" href="#prepare-MS2-IDs-global"><i class="fas fa-link"></i></a>
</h3>
<div id="read-ms-gf-data" class="section level4" number="1.1.1.1">
<h4>
<span class="header-section-number">1.1.1.1</span> Read MS-GF+ Data<a class="anchor" aria-label="anchor" href="#read-ms-gf-data"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get file path</span>
<span class="va">path_to_MSGF_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/global/msgf_output"</span>, 
                                    package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span>
<span class="co"># Read MS-GF+ data from path</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/read_msgf_data.html">read_msgf_data</a></span><span class="op">(</span><span class="va">path_to_MSGF_results</span><span class="op">)</span></code></pre></div>
<p>Normally, this would display a progress bar in the console as the data is being fetched. However, the output was suppressed to save space. We can view a summary of the MSnID object with the <code>show()</code> function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  48 
## #PSMs: 1156754 at 31 % FDR
## #peptides: 511617 at 61 % FDR
## #accessions: 128378 at 98 % FDR</code></pre>
<p><code>msnid</code> consists of 4 spectrum files (datasets), and contains a total of 1,156,754 peptide-spectrum-matches (PSMs), 511,617 total peptides, and 128,378 total accessions (proteins). The reported FDR is the empirical <strong>false-discovery rate</strong>, which is calculated as the ratio of the number of false (decoy) PSMs, peptides, or accessions to their true (non-decoy) counterparts. Calculation of these counts and their FDRs is shown below.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Calculating the counts and FDRs from the show() output ---</span>
<span class="co"># Spectrum Files:</span>
<span class="co"># Count</span>
<span class="fu">psms</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">Dataset</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># 48</span>

<span class="co"># PSMs:</span>
<span class="co"># Count</span>
<span class="fu">psms</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">Dataset</span>, <span class="va">Scan</span>, <span class="va">peptide</span>, <span class="va">isDecoy</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Assign intermediate to variable</span>
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="st">"x_psm"</span>, <span class="va">.</span>, envir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/environment.html">globalenv</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># 1156754</span>
<span class="co"># FDR</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">x_psm</span><span class="op">[</span><span class="va">x_psm</span><span class="op">$</span><span class="va">isDecoy</span> <span class="op">==</span> <span class="cn">TRUE</span>, <span class="op">]</span><span class="op">)</span> <span class="op">/</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">x_psm</span><span class="op">[</span><span class="va">x_psm</span><span class="op">$</span><span class="va">isDecoy</span> <span class="op">==</span> <span class="cn">FALSE</span>, <span class="op">]</span><span class="op">)</span> 
<span class="co"># 0.3127463 = 31%</span>

<span class="co"># peptides:</span>
<span class="co"># Count</span>
<span class="fu">psms</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">peptide</span>, <span class="va">isDecoy</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="st">"x_peptide"</span>, <span class="va">.</span>, envir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/environment.html">globalenv</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># 511617</span>
<span class="co"># FDR</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">x_peptide</span><span class="op">[</span><span class="va">x_peptide</span><span class="op">$</span><span class="va">isDecoy</span> <span class="op">==</span> <span class="cn">TRUE</span>, <span class="op">]</span><span class="op">)</span> <span class="op">/</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">x_peptide</span><span class="op">[</span><span class="va">x_peptide</span><span class="op">$</span><span class="va">isDecoy</span> <span class="op">==</span> <span class="cn">FALSE</span>, <span class="op">]</span><span class="op">)</span>
<span class="co"># 0.611245 = 61%</span>

<span class="co"># Accessions:</span>
<span class="co"># Count</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu">accessions</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span><span class="op">)</span> <span class="co"># or</span>
<span class="fu">psms</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">accession</span>, <span class="va">isDecoy</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/assign.html">assign</a></span><span class="op">(</span><span class="st">"x_acc"</span>, <span class="va">.</span>, envir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/environment.html">globalenv</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># 128378</span>
<span class="co"># FDR</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">x_acc</span><span class="op">[</span><span class="va">x_acc</span><span class="op">$</span><span class="va">isDecoy</span> <span class="op">==</span> <span class="cn">TRUE</span>, <span class="op">]</span><span class="op">)</span> <span class="op">/</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">x_acc</span><span class="op">[</span><span class="va">x_acc</span><span class="op">$</span><span class="va">isDecoy</span> <span class="op">==</span> <span class="cn">FALSE</span>, <span class="op">]</span><span class="op">)</span>
<span class="co"># 0.9827024 = 98%</span></code></pre></div>
<p>Now that we have an <code>MSnID</code> object, we need to process it.</p>
</div>
<div id="correct-isotope-selection-error" class="section level4" number="1.1.1.2">
<h4>
<span class="header-section-number">1.1.1.2</span> Correct Isotope Selection Error<a class="anchor" aria-label="anchor" href="#correct-isotope-selection-error"><i class="fas fa-link"></i></a>
</h4>
<p>Occasionally, the instrument selects a peak with +1 or more C13 atoms, rather than the monoisotopic (lowest mass) peak. While MS-FG+ is still capable of correctly identifying those, the downstream calculations of mass measurement error need to be fixed. The <code>correct_peak_selection</code> method corrects for these mass measurement errors.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Correct for isotope selection error</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">correct_peak_selection</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
</div>
<div id="remove-contaminants" class="section level4" number="1.1.1.3">
<h4>
<span class="header-section-number">1.1.1.3</span> Remove Contaminants<a class="anchor" aria-label="anchor" href="#remove-contaminants"><i class="fas fa-link"></i></a>
</h4>
<p>Now, we will remove contaminants such as the pig trypsin that was used for protein digestion. We can use <code>grepl</code> to search for all accessions that contain “Contaminant.”</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># All unique contaminants</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">$</span><span class="va">accession</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Contaminant"</span>, <span class="va">msnid</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "Contaminant_K2C1_HUMAN"      "Contaminant_K1C9_HUMAN"     
##  [3] "Contaminant_ALBU_HUMAN"      "Contaminant_ALBU_BOVIN"     
##  [5] "Contaminant_TRYP_PIG"        "Contaminant_K1C10_HUMAN"    
##  [7] "XXX_Contaminant_K1C9_HUMAN"  "Contaminant_K22E_HUMAN"     
##  [9] "Contaminant_Trypa3"          "Contaminant_Trypa5"         
## [11] "XXX_Contaminant_K1C10_HUMAN" "XXX_Contaminant_K22E_HUMAN" 
## [13] "XXX_Contaminant_K2C1_HUMAN"  "Contaminant_TRYP_BOVIN"     
## [15] "XXX_Contaminant_ALBU_HUMAN"  "XXX_Contaminant_ALBU_BOVIN" 
## [17] "XXX_Contaminant_TRYP_BOVIN"  "Contaminant_CTRB_BOVIN"     
## [19] "Contaminant_Trypa1"          "Contaminant_Trypa6"         
## [21] "Contaminant_CTRA_BOVIN"      "XXX_Contaminant_TRYP_PIG"   
## [23] "XXX_Contaminant_CTRB_BOVIN"  "XXX_Contaminant_CTRA_BOVIN" 
## [25] "Contaminant_Trypa2"</code></pre>
<p>To remove contaminants, we use <code>apply_filter</code> with an appropriate character string that tells the function what rows to keep. In this case, we keep rows where the accession does not contain “Contaminant.” We will use <code>show</code> to see how the counts change.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove contaminants</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">apply_filter</span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!grepl('Contaminant', accession)"</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  48 
## #PSMs: 1155442 at 31 % FDR
## #peptides: 511196 at 61 % FDR
## #accessions: 128353 at 98 % FDR</code></pre>
</div>
<div id="msms-id-filter-peptide-level" class="section level4" number="1.1.1.4">
<h4>
<span class="header-section-number">1.1.1.4</span> MS/MS ID Filter: Peptide Level<a class="anchor" aria-label="anchor" href="#msms-id-filter-peptide-level"><i class="fas fa-link"></i></a>
</h4>
<p>The next step is to use the <code>PepQValue</code> column from the MSnID object and the absolute deviation of the mass measurement error of parent ions (in ppm) to maximize the number of PSMs while ensuring that the empirical peptide-level FDR is at most 1%.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 1% FDR filter at the peptide level</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_msgf_data.html">filter_msgf_data</a></span><span class="op">(</span><span class="va">msnid</span>,
                          level <span class="op">=</span> <span class="st">"peptide"</span>,
                          fdr.max <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  48 
## #PSMs: 464482 at 0.45 % FDR
## #peptides: 96486 at 1 % FDR
## #accessions: 27120 at 9.2 % FDR</code></pre>
<p>We can see that filtering drastically reduces the number of PSMs, and the empirical peptide-level FDR is now 1%. However, notice that the empirical protein-level FDR is still fairly high.</p>
</div>
<div id="msms-id-filter-protein-level" class="section level4" number="1.1.1.5">
<h4>
<span class="header-section-number">1.1.1.5</span> MS/MS ID Filter: Protein Level<a class="anchor" aria-label="anchor" href="#msms-id-filter-protein-level"><i class="fas fa-link"></i></a>
</h4>
<p>A while ago, the proteomics field established the hard-and-fast two-peptides-per-protein rule. That is, we can not be confident if a protein is identified by the detection of only one peptide. This rule penalizes short proteins and doesn’t consider that there are some very long proteins (e.g. Titin 3.8 MDa) that easily have more then two matching peptides even in reversed sequence. Thus, we propose to normalize the number of peptides per protein length and use that as a filtering criterion.</p>
<p>We need the FASTA (pronounced FAST-AYE) file to get the length of each protein, which we can then use to calculate the associated number of peptides per 1000 amino acids. This new <code>peptides_per_1000aa</code> column is used to filter the MSnID object so that the empirical accession-level FDR is at most 1%.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get path to FASTA file</span>
<span class="va">path_to_FASTA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>
  <span class="st">"extdata/Rattus_norvegicus_NCBI_RefSeq_2018-04-10.fasta.gz"</span>, 
  package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span>
<span class="op">)</span>

<span class="co"># Compute number of peptides per 1000 amino acids</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/compute_num_peptides_per_1000aa.html">compute_num_peptides_per_1000aa</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">path_to_FASTA</span><span class="op">)</span>

<span class="co"># 1% FDR filter at the protein level</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_msgf_data.html">filter_msgf_data</a></span><span class="op">(</span><span class="va">msnid</span>,
                          level <span class="op">=</span> <span class="st">"accession"</span>,
                          fdr.max <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  48 
## #PSMs: 458097 at 0.16 % FDR
## #peptides: 92036 at 0.32 % FDR
## #accessions: 15630 at 1 % FDR</code></pre>
</div>
<div id="inference-of-parsimonious-protein-set" class="section level4" number="1.1.1.6">
<h4>
<span class="header-section-number">1.1.1.6</span> Inference of Parsimonious Protein Set<a class="anchor" aria-label="anchor" href="#inference-of-parsimonious-protein-set"><i class="fas fa-link"></i></a>
</h4>
<p>The situation when a certain peptide sequence matches multiple proteins adds complication to the downstream quantitative analysis, as it is not clear which protein this peptide is originating from. There are common ways for dealing with this. One is to simply retain uniquely matching peptides and discard shared peptides (<code>unique_only = TRUE</code>). Alternatively (in case of <code>unique_only = FALSE</code>) assign the shared peptides to the proteins with the larger number of uniquely mapping peptides. If there is a choice between multiple proteins with equal numbers of uniquely mapping peptides, the shared peptides are assigned to the first protein according to alphanumeric order. This step could be done prior to filtering at the accession level, but the removal of an accession will completely remove its associated peptides.</p>
<!---
TODO:
* Include a visual for inference of the parsimonious protein set.
--->
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Inference of parsimonious protein set</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">infer_parsimonious_accessions</span><span class="op">(</span><span class="va">msnid</span>, unique_only <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  48 
## #PSMs: 445006 at 0.15 % FDR
## #peptides: 90478 at 0.27 % FDR
## #accessions: 5251 at 1.1 % FDR</code></pre>
<p>Notice that the protein-level FDR increased above the acceptable threshold, so we need to reapply the filter.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 1% FDR filter at the protein level</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_msgf_data.html">filter_msgf_data</a></span><span class="op">(</span><span class="va">msnid</span>,
                          level <span class="op">=</span> <span class="st">"accession"</span>,
                          fdr.max <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  48 
## #PSMs: 444850 at 0.14 % FDR
## #peptides: 90384 at 0.25 % FDR
## #accessions: 5222 at 0.99 % FDR</code></pre>
<p>Once all filtering is done, we can remove the decoy accessions. We use the <code>apply_filter</code> function again and only keep entries where <code>isDecoy</code> is <code>FALSE</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove Decoy Accessions</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">apply_filter</span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!isDecoy"</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  48 
## #PSMs: 444213 at 0 % FDR
## #peptides: 90156 at 0 % FDR
## #accessions: 5171 at 0 % FDR</code></pre>
<p>After processing, we are left with 318,448 PSMs, 81,048 peptides, and 5,143 proteins. The empirical FDRs are the same as before, but can not be calculated because we removed the decoys.</p>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:global-msnid-table">Table 1.1: </span><left>First 6 rows of the processed MS-GF+ results.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:right;">
ResultID
</th>
<th style="text-align:right;">
Scan
</th>
<th style="text-align:left;">
FragMethod
</th>
<th style="text-align:right;">
SpecIndex
</th>
<th style="text-align:right;">
Charge
</th>
<th style="text-align:right;">
PrecursorMZ
</th>
<th style="text-align:right;">
DelM
</th>
<th style="text-align:right;">
DelM_PPM
</th>
<th style="text-align:right;">
MH
</th>
<th style="text-align:left;">
peptide
</th>
<th style="text-align:left;">
Protein
</th>
<th style="text-align:right;">
NTT
</th>
<th style="text-align:right;">
DeNovoScore
</th>
<th style="text-align:right;">
MSGFScore
</th>
<th style="text-align:right;">
MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
Rank_MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
EValue
</th>
<th style="text-align:right;">
QValue
</th>
<th style="text-align:right;">
PepQValue
</th>
<th style="text-align:right;">
IsotopeError
</th>
<th style="text-align:left;">
accession
</th>
<th style="text-align:right;">
calculatedMassToCharge
</th>
<th style="text-align:right;">
chargeState
</th>
<th style="text-align:right;">
experimentalMassToCharge
</th>
<th style="text-align:left;">
isDecoy
</th>
<th style="text-align:left;">
spectrumFile
</th>
<th style="text-align:right;">
spectrumID
</th>
<th style="text-align:left;">
pepSeq
</th>
<th style="text-align:right;">
msmsScore
</th>
<th style="text-align:right;">
absParentMassErrorPPM
</th>
<th style="text-align:right;">
peptides_per_1000aa
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
1862
</td>
<td style="text-align:right;">
27707
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
324
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
928.541
</td>
<td style="text-align:right;">
-0.001
</td>
<td style="text-align:right;">
-0.526
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
285
</td>
<td style="text-align:right;">
282
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
928.541
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
928.541
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27707
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.589
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
4192
</td>
<td style="text-align:right;">
27684
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
906
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
-0.002
</td>
<td style="text-align:right;">
-0.887
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
156
</td>
<td style="text-align:right;">
144
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27684
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.991
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_06_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
26263
</td>
<td style="text-align:right;">
27336
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
5187
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
118
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_06_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27336
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
1471
</td>
<td style="text-align:right;">
27096
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
415
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
-0.001
</td>
<td style="text-align:right;">
-0.591
</td>
<td style="text-align:right;">
1856.075
</td>
<td style="text-align:left;">
R.AAAAAAAAAAAAAAGAAGK.E
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
157
</td>
<td style="text-align:right;">
156
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_113986.1
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
619.363
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
27096
</td>
<td style="text-align:left;">
AAAAAAAAAAAAAAGAAGK
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.684
</td>
<td style="text-align:right;">
25.769
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_05_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
28664
</td>
<td style="text-align:right;">
10441
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
4849
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
586.832
</td>
<td style="text-align:right;">
-0.001
</td>
<td style="text-align:right;">
-0.728
</td>
<td style="text-align:right;">
1172.659
</td>
<td style="text-align:left;">
R.AAAAADLANR.S
</td>
<td style="text-align:left;">
NP_001007804.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
124
</td>
<td style="text-align:right;">
124
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_001007804.1
</td>
<td style="text-align:right;">
586.833
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
586.832
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S2_05_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
10441
</td>
<td style="text-align:left;">
AAAAADLANR
</td>
<td style="text-align:right;">
2.480
</td>
<td style="text-align:right;">
0.746
</td>
<td style="text-align:right;">
34.755
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_24_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
41775
</td>
<td style="text-align:right;">
8033
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
7889
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
831.447
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
1661.886
</td>
<td style="text-align:left;">
G.AAAAAEAESGGGGGK.K
</td>
<td style="text-align:left;">
NP_001128630.1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
176
</td>
<td style="text-align:right;">
76
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
NP_001128630.1
</td>
<td style="text-align:right;">
831.447
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
831.447
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_24_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
8033
</td>
<td style="text-align:left;">
AAAAAEAESGGGGGK
</td>
<td style="text-align:right;">
2.583
</td>
<td style="text-align:right;">
0.106
</td>
<td style="text-align:right;">
580.844
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>Table <a href="isobaric-quantification-pipelines.html#tab:global-msnid-table">1.1</a> shows the first 6 rows of the processed MS-GF+ output.</p>
</div>
</div>
<div id="reporter-ion-intensities" class="section level3" number="1.1.2">
<h3>
<span class="header-section-number">1.1.2</span> Prepare Reporter Ion Intensities<a class="anchor" aria-label="anchor" href="#reporter-ion-intensities"><i class="fas fa-link"></i></a>
</h3>
<div id="read-masic-output" class="section level4" number="1.1.2.1">
<h4>
<span class="header-section-number">1.1.2.1</span> Read MASIC Output<a class="anchor" aria-label="anchor" href="#read-masic-output"><i class="fas fa-link"></i></a>
</h4>
<p>MASIC is a tool for extracting ion intensities. With proper parameter settings, it can be used for extracting TMT (or iTRAQ) reporter ion intensities. In addition, it reports a number of other helpful metrics. Notably, the interference score at the parent ion level and the signal-to-noise ratio (S/N) at the reporter ion level (computed by Thermo software). The interference score reflects the proportion of the ion population that was isolated for fragmentation that is due to the targeted ion. In other words, 1 - InterferenceScore is due to co-isolated species that have similar elution time and parent ion m/z.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Path to MASIC data</span>
<span class="va">path_to_MASIC_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/global/masic_output"</span>, 
                                     package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span>

<span class="co"># Read MASIC data</span>
<span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/read_masic_data.html">read_masic_data</a></span><span class="op">(</span><span class="va">path_to_MASIC_results</span>, interference_score <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Normally, this would display two progress bars in the console as the data is being fetched. However, the output was suppressed to save space.</p>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:global-masic-table">Table 1.2: </span><left>First 6 rows of the MASIC data.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:right;">
ScanNumber
</th>
<th style="text-align:left;">
Collision.Mode
</th>
<th style="text-align:right;">
ParentIonMZ
</th>
<th style="text-align:right;">
BasePeakIntensity
</th>
<th style="text-align:right;">
BasePeakMZ
</th>
<th style="text-align:right;">
ReporterIonIntensityMax
</th>
<th style="text-align:right;">
Ion_126.128
</th>
<th style="text-align:right;">
Ion_127.125
</th>
<th style="text-align:right;">
Ion_127.131
</th>
<th style="text-align:right;">
Ion_128.128
</th>
<th style="text-align:right;">
Ion_128.134
</th>
<th style="text-align:right;">
Ion_129.131
</th>
<th style="text-align:right;">
Ion_129.138
</th>
<th style="text-align:right;">
Ion_130.135
</th>
<th style="text-align:right;">
Ion_130.141
</th>
<th style="text-align:right;">
Ion_131.138
</th>
<th style="text-align:right;">
Weighted.Avg.Pct.Intensity.Correction
</th>
<th style="text-align:right;">
Ion_126.128_SignalToNoise
</th>
<th style="text-align:right;">
Ion_127.125_SignalToNoise
</th>
<th style="text-align:right;">
Ion_127.131_SignalToNoise
</th>
<th style="text-align:right;">
Ion_128.128_SignalToNoise
</th>
<th style="text-align:right;">
Ion_128.134_SignalToNoise
</th>
<th style="text-align:right;">
Ion_129.131_SignalToNoise
</th>
<th style="text-align:right;">
Ion_129.138_SignalToNoise
</th>
<th style="text-align:right;">
Ion_130.135_SignalToNoise
</th>
<th style="text-align:right;">
Ion_130.141_SignalToNoise
</th>
<th style="text-align:right;">
Ion_131.138_SignalToNoise
</th>
<th style="text-align:right;">
Ion_126.128_Resolution
</th>
<th style="text-align:right;">
Ion_127.125_Resolution
</th>
<th style="text-align:right;">
Ion_127.131_Resolution
</th>
<th style="text-align:right;">
Ion_128.128_Resolution
</th>
<th style="text-align:right;">
Ion_128.134_Resolution
</th>
<th style="text-align:right;">
Ion_129.131_Resolution
</th>
<th style="text-align:right;">
Ion_129.138_Resolution
</th>
<th style="text-align:right;">
Ion_130.135_Resolution
</th>
<th style="text-align:right;">
Ion_130.141_Resolution
</th>
<th style="text-align:right;">
Ion_131.138_Resolution
</th>
<th style="text-align:right;">
ParentIonIndex
</th>
<th style="text-align:right;">
MZ
</th>
<th style="text-align:right;">
SurveyScanNumber
</th>
<th style="text-align:right;">
OptimalPeakApexScanNumber
</th>
<th style="text-align:right;">
PeakApexOverrideParentIonIndex
</th>
<th style="text-align:right;">
CustomSICPeak
</th>
<th style="text-align:right;">
PeakScanStart
</th>
<th style="text-align:right;">
PeakScanEnd
</th>
<th style="text-align:right;">
PeakScanMaxIntensity
</th>
<th style="text-align:right;">
PeakMaxIntensity
</th>
<th style="text-align:right;">
PeakSignalToNoiseRatio
</th>
<th style="text-align:right;">
FWHMInScans
</th>
<th style="text-align:right;">
PeakArea
</th>
<th style="text-align:right;">
ParentIonIntensity
</th>
<th style="text-align:right;">
PeakBaselineNoiseLevel
</th>
<th style="text-align:right;">
PeakBaselineNoiseStDev
</th>
<th style="text-align:right;">
PeakBaselinePointsUsed
</th>
<th style="text-align:right;">
StatMomentsArea
</th>
<th style="text-align:right;">
CenterOfMassScan
</th>
<th style="text-align:right;">
PeakStDev
</th>
<th style="text-align:right;">
PeakSkew
</th>
<th style="text-align:right;">
PeakKSStat
</th>
<th style="text-align:right;">
StatMomentsDataCountUsed
</th>
<th style="text-align:right;">
InterferenceScore
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
407.74
</td>
<td style="text-align:right;">
227695.44
</td>
<td style="text-align:right;">
407.741
</td>
<td style="text-align:right;">
92236.87
</td>
<td style="text-align:right;">
70562.39
</td>
<td style="text-align:right;">
24864.62
</td>
<td style="text-align:right;">
17165.80
</td>
<td style="text-align:right;">
35625.00
</td>
<td style="text-align:right;">
92236.87
</td>
<td style="text-align:right;">
9640.23
</td>
<td style="text-align:right;">
8578.05
</td>
<td style="text-align:right;">
6996.69
</td>
<td style="text-align:right;">
11833.07
</td>
<td style="text-align:right;">
32281.34
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
71.47
</td>
<td style="text-align:right;">
25.17
</td>
<td style="text-align:right;">
17.38
</td>
<td style="text-align:right;">
36.04
</td>
<td style="text-align:right;">
93.32
</td>
<td style="text-align:right;">
9.75
</td>
<td style="text-align:right;">
8.67
</td>
<td style="text-align:right;">
7.07
</td>
<td style="text-align:right;">
11.96
</td>
<td style="text-align:right;">
32.71
</td>
<td style="text-align:right;">
44102
</td>
<td style="text-align:right;">
42700
</td>
<td style="text-align:right;">
42100
</td>
<td style="text-align:right;">
41800
</td>
<td style="text-align:right;">
44404
</td>
<td style="text-align:right;">
40500
</td>
<td style="text-align:right;">
39500
</td>
<td style="text-align:right;">
36800
</td>
<td style="text-align:right;">
41100
</td>
<td style="text-align:right;">
42302
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
407.742
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
2901600
</td>
<td style="text-align:right;">
211.000
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
50422000
</td>
<td style="text-align:right;">
2579600
</td>
<td style="text-align:right;">
13750
</td>
<td style="text-align:right;">
97562
</td>
<td style="text-align:right;">
10113
</td>
<td style="text-align:right;">
47031000
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
5.68
</td>
<td style="text-align:right;">
-0.189
</td>
<td style="text-align:right;">
0.353
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.996
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
396.72
</td>
<td style="text-align:right;">
59127.97
</td>
<td style="text-align:right;">
529.294
</td>
<td style="text-align:right;">
34294.90
</td>
<td style="text-align:right;">
23706.89
</td>
<td style="text-align:right;">
13559.32
</td>
<td style="text-align:right;">
5856.83
</td>
<td style="text-align:right;">
16322.71
</td>
<td style="text-align:right;">
34294.90
</td>
<td style="text-align:right;">
4853.11
</td>
<td style="text-align:right;">
7938.24
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1465.03
</td>
<td style="text-align:right;">
18182.27
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
26.12
</td>
<td style="text-align:right;">
14.94
</td>
<td style="text-align:right;">
6.45
</td>
<td style="text-align:right;">
17.97
</td>
<td style="text-align:right;">
37.77
</td>
<td style="text-align:right;">
5.34
</td>
<td style="text-align:right;">
8.74
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.61
</td>
<td style="text-align:right;">
19.93
</td>
<td style="text-align:right;">
42702
</td>
<td style="text-align:right;">
41100
</td>
<td style="text-align:right;">
37000
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
43404
</td>
<td style="text-align:right;">
36400
</td>
<td style="text-align:right;">
39700
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
29800
</td>
<td style="text-align:right;">
41802
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
396.718
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
2181900
</td>
<td style="text-align:right;">
19.690
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
34508000
</td>
<td style="text-align:right;">
1690600
</td>
<td style="text-align:right;">
110841
</td>
<td style="text-align:right;">
1120000
</td>
<td style="text-align:right;">
10166
</td>
<td style="text-align:right;">
31578000
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
5.59
</td>
<td style="text-align:right;">
-0.217
</td>
<td style="text-align:right;">
0.347
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.993
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
438.23
</td>
<td style="text-align:right;">
110444.82
</td>
<td style="text-align:right;">
362.224
</td>
<td style="text-align:right;">
14053.40
</td>
<td style="text-align:right;">
12459.86
</td>
<td style="text-align:right;">
11785.91
</td>
<td style="text-align:right;">
10932.51
</td>
<td style="text-align:right;">
10653.32
</td>
<td style="text-align:right;">
12328.62
</td>
<td style="text-align:right;">
5959.86
</td>
<td style="text-align:right;">
9905.82
</td>
<td style="text-align:right;">
8387.04
</td>
<td style="text-align:right;">
11166.70
</td>
<td style="text-align:right;">
14053.40
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
12.40
</td>
<td style="text-align:right;">
11.75
</td>
<td style="text-align:right;">
10.90
</td>
<td style="text-align:right;">
10.64
</td>
<td style="text-align:right;">
12.31
</td>
<td style="text-align:right;">
5.96
</td>
<td style="text-align:right;">
9.91
</td>
<td style="text-align:right;">
8.40
</td>
<td style="text-align:right;">
11.18
</td>
<td style="text-align:right;">
14.13
</td>
<td style="text-align:right;">
42006
</td>
<td style="text-align:right;">
40702
</td>
<td style="text-align:right;">
41402
</td>
<td style="text-align:right;">
40700
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
38800
</td>
<td style="text-align:right;">
40200
</td>
<td style="text-align:right;">
38900
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
41002
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
438.227
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
131
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
8255600
</td>
<td style="text-align:right;">
9.465
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
444610000
</td>
<td style="text-align:right;">
658727
</td>
<td style="text-align:right;">
872195
</td>
<td style="text-align:right;">
2620000
</td>
<td style="text-align:right;">
10129
</td>
<td style="text-align:right;">
343470000
</td>
<td style="text-align:right;">
82
</td>
<td style="text-align:right;">
22.80
</td>
<td style="text-align:right;">
-0.626
</td>
<td style="text-align:right;">
1.099
</td>
<td style="text-align:right;">
89
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
481.50
</td>
<td style="text-align:right;">
37082.72
</td>
<td style="text-align:right;">
206.466
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
481.505
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
401824
</td>
<td style="text-align:right;">
27.990
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
16244000
</td>
<td style="text-align:right;">
344491
</td>
<td style="text-align:right;">
14356
</td>
<td style="text-align:right;">
65777
</td>
<td style="text-align:right;">
10109
</td>
<td style="text-align:right;">
14899000
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
14.10
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
0.504
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
549.28
</td>
<td style="text-align:right;">
21077.05
</td>
<td style="text-align:right;">
128.129
</td>
<td style="text-align:right;">
21077.05
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
10998.67
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
21077.05
</td>
<td style="text-align:right;">
2725.50
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
6800.70
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
9.19
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
17.57
</td>
<td style="text-align:right;">
2.27
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
5.66
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
40302
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
42102
</td>
<td style="text-align:right;">
46600
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
40300
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
549.279
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
363656
</td>
<td style="text-align:right;">
0.700
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5941200
</td>
<td style="text-align:right;">
347071
</td>
<td style="text-align:right;">
519640
</td>
<td style="text-align:right;">
1990000
</td>
<td style="text-align:right;">
10109
</td>
<td style="text-align:right;">
2528900
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
2.49
</td>
<td style="text-align:right;">
-0.583
</td>
<td style="text-align:right;">
0.430
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
hcd
</td>
<td style="text-align:right;">
388.72
</td>
<td style="text-align:right;">
40605.85
</td>
<td style="text-align:right;">
356.719
</td>
<td style="text-align:right;">
8087.76
</td>
<td style="text-align:right;">
6166.82
</td>
<td style="text-align:right;">
1371.27
</td>
<td style="text-align:right;">
2418.35
</td>
<td style="text-align:right;">
8087.76
</td>
<td style="text-align:right;">
5485.35
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1543.48
</td>
<td style="text-align:right;">
1943.96
</td>
<td style="text-align:right;">
7436.60
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6.92
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
2.71
</td>
<td style="text-align:right;">
9.04
</td>
<td style="text-align:right;">
6.13
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
1.72
</td>
<td style="text-align:right;">
2.16
</td>
<td style="text-align:right;">
8.26
</td>
<td style="text-align:right;">
40000
</td>
<td style="text-align:right;">
26400
</td>
<td style="text-align:right;">
30400
</td>
<td style="text-align:right;">
40400
</td>
<td style="text-align:right;">
44300
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
28800
</td>
<td style="text-align:right;">
28500
</td>
<td style="text-align:right;">
38700
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
388.720
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
-1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
478135
</td>
<td style="text-align:right;">
28.710
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
10718000
</td>
<td style="text-align:right;">
291189
</td>
<td style="text-align:right;">
16653
</td>
<td style="text-align:right;">
142562
</td>
<td style="text-align:right;">
10142
</td>
<td style="text-align:right;">
9961100
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
8.60
</td>
<td style="text-align:right;">
-0.051
</td>
<td style="text-align:right;">
0.283
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
0.969
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>Table <a href="isobaric-quantification-pipelines.html#tab:global-masic-table">1.2</a> shows the first 6 rows of <code>masic_data</code>.</p>
</div>
<div id="filter-masic-data" class="section level4" number="1.1.2.2">
<h4>
<span class="header-section-number">1.1.2.2</span> Filter MASIC Data<a class="anchor" aria-label="anchor" href="#filter-masic-data"><i class="fas fa-link"></i></a>
</h4>
<p>Currently, we recommend keeping entries where at least 50% of the ion population is due to the targeted ion (interference score <span class="math inline">\(\geq\)</span> 0.5) and not filtering by S/N.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Filter MASIC data</span>
<span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_masic_data.html">filter_masic_data</a></span><span class="op">(</span><span class="va">masic_data</span>, 
                                interference_score_threshold <span class="op">=</span> <span class="fl">0.5</span>,
                                s2n_threshold <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="fetch-study-design-tables" class="section level3" number="1.1.3">
<h3>
<span class="header-section-number">1.1.3</span> Create Study Design Tables<a class="anchor" aria-label="anchor" href="#fetch-study-design-tables"><i class="fas fa-link"></i></a>
</h3>
<p>To convert from PSMs and reporter ion intensities to meaningful quantitative data, it is necessary to know what are the samples in the reporter channels and what is the intended reference channel (or combination of channels). The entire study design is captured by three tables - fractions, samples, references. With newly processed data, these typically do not exist, and must be created. If the tables already exist, the code to access them is as follows.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read tables from folder:</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/study_design/fractions.txt"</span>, 
                                  package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span><span class="op">)</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/study_design/samples.txt"</span>, 
                                package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span><span class="op">)</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/study_design/references.txt"</span>, 
                                   package <span class="op">=</span> <span class="st">"PlexedPiperTestData"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># If using a data package from the DMS:</span>
<span class="va">study_design</span> <span class="op">&lt;-</span> <span class="fu">read_study_design_from_DMS</span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">study_design</span><span class="op">$</span><span class="va">fractions</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">study_design</span><span class="op">$</span><span class="va">samples</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">study_design</span><span class="op">$</span><span class="va">references</span></code></pre></div>
<div id="fractions" class="section level4" number="1.1.3.1">
<h4>
<span class="header-section-number">1.1.3.1</span> Fractions<a class="anchor" aria-label="anchor" href="#fractions"><i class="fas fa-link"></i></a>
</h4>
<p>The fractions table consists of two columns: <code>Dataset</code> and <code>PlexID</code>. The <code>Dataset</code> column contains all of the unique datasets from <code>msnid$Dataset</code> or <code>masic_data$Dataset</code>. The <code>PlexID</code> column contains the plex ID associated with each dataset, and is typically an “S” followed by a number (“S1,” “S2,” etc.). We can extract the plex ID from the datasets. In this case, the plex ID always comes after “<em>W</em>,” so we can use a regular expression (regex) to capture it (the first argument of <code>gsub</code>). The regex below says to capture an “S” followed by a single digit that appears after “<em>W</em>” and before an underscore.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create fractions table</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Dataset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">$</span><span class="va">Dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>PlexID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">".*_W_(S\\d{1})_.*"</span>, <span class="st">"\\1"</span>, <span class="va">Dataset</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:fractions-table">Table 1.3: </span><left>First 10 rows of the fractions table.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:left;">
PlexID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_01_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_02_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_03_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_04_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_05_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_06_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_07_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_08_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_09_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_W_S1_10_12Oct17_Elm_AQ-17-09-02
</td>
<td style="text-align:left;">
S1
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>Table <a href="isobaric-quantification-pipelines.html#tab:fractions-table">1.3</a> shows the first 10 rows of <code>fractions</code>.</p>
</div>
<div id="samples" class="section level4" number="1.1.3.2">
<h4>
<span class="header-section-number">1.1.3.2</span> Samples<a class="anchor" aria-label="anchor" href="#samples"><i class="fas fa-link"></i></a>
</h4>
<p>The samples table contains columns <code>PlexID</code>, <code>QuantBlock</code>, <code>ReporterName</code>, <code>ReporterAlias</code>, and <code>MeasurementName</code>. The plex ID must be the same as the plex ID in the <code>fractions</code> table. <code>ReporterName</code> is the reporter ion name (“126,” “127N,” “127C,” etc.). <code>ReporterAlias</code> is the intermediate between <code>ReporterName</code> and <code>MeasurementName</code> and is used for defining the reference. <code>MeasurementName</code> determines the column names for the final cross-tab, and must be unique and begin with a letter. <code>MeasurementName</code> is easily constructed by prepending <code>PlexID</code> to the <code>ReporterName</code>. Finally, <code>QuantBlock</code> can be thought of as a way of defining sub-plex. In a typical TMT experiment, <code>QuantBlock</code> is always 1. In case of 5 pairwise comparisons within TMT10, there will be 5 QuantBlocks (1-5) with a reference for each <code>QuantBlock</code>.</p>
<p>For this experiment, channel 131 will serve as the reference, so we set <code>MeasurementName</code> to <code>NA</code> when <code>ReporterName</code> is <code>"131"</code>. This will make the reference channel absent from the quantitative cross-tab. In cases where reporter ion intensities are not normalized by a reference channel (reference = 1) or they are normalized by the average of select channels, do not set any <code>MeasurementName</code> to <code>NA</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># TMT10 Reporter Converter table from MSnID package</span>
<span class="va">conv</span> <span class="op">&lt;-</span> <span class="va">reporter_converter</span><span class="op">$</span><span class="va">tmt10</span>
<span class="va">plexes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">$</span><span class="va">PlexID</span><span class="op">)</span>

<span class="co"># Reference channel</span>
<span class="va">ref_channel</span> <span class="op">&lt;-</span> <span class="st">"131"</span>

<span class="co"># Create samples table</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>PlexID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">plexes</span>, each <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">conv</span><span class="op">)</span><span class="op">)</span>,  
                      ReporterName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">conv</span><span class="op">$</span><span class="va">ReporterName</span>, 
                                         <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">plexes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>ReporterAlias <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"%s_%s"</span>, <span class="va">PlexID</span>, <span class="va">ReporterName</span><span class="op">)</span>,
         MeasurementName <span class="op">=</span> <span class="va">ReporterAlias</span>,
         QuantBlock <span class="op">=</span> <span class="fl">1</span>,
         <span class="co"># Comment out this next part if the reference</span>
         <span class="co"># is not one of the reporter ion channels.</span>
         MeasurementName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">ReporterName</span> <span class="op">==</span> <span class="va">ref_channel</span>,
                                  <span class="cn">NA</span>, <span class="va">MeasurementName</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:samples-table">Table 1.4: </span><left>First 10 rows of the samples table.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
PlexID
</th>
<th style="text-align:left;">
ReporterName
</th>
<th style="text-align:left;">
ReporterAlias
</th>
<th style="text-align:left;">
MeasurementName
</th>
<th style="text-align:right;">
QuantBlock
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
126
</td>
<td style="text-align:left;">
S1_126
</td>
<td style="text-align:left;">
S1_126
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
127N
</td>
<td style="text-align:left;">
S1_127N
</td>
<td style="text-align:left;">
S1_127N
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
127C
</td>
<td style="text-align:left;">
S1_127C
</td>
<td style="text-align:left;">
S1_127C
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
128N
</td>
<td style="text-align:left;">
S1_128N
</td>
<td style="text-align:left;">
S1_128N
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
128C
</td>
<td style="text-align:left;">
S1_128C
</td>
<td style="text-align:left;">
S1_128C
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
129N
</td>
<td style="text-align:left;">
S1_129N
</td>
<td style="text-align:left;">
S1_129N
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
129C
</td>
<td style="text-align:left;">
S1_129C
</td>
<td style="text-align:left;">
S1_129C
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
130N
</td>
<td style="text-align:left;">
S1_130N
</td>
<td style="text-align:left;">
S1_130N
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
130C
</td>
<td style="text-align:left;">
S1_130C
</td>
<td style="text-align:left;">
S1_130C
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
131
</td>
<td style="text-align:left;">
S1_131
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>Table <a href="isobaric-quantification-pipelines.html#tab:samples-table">1.4</a> shows the first 10 rows of <code>samples</code>.</p>
</div>
<div id="references" class="section level4" number="1.1.3.3">
<h4>
<span class="header-section-number">1.1.3.3</span> References<a class="anchor" aria-label="anchor" href="#references"><i class="fas fa-link"></i></a>
</h4>
<p>Reference can be a certain channel, average of multiple channels, or 1. The general form is an expression with <code>ReporterAlias</code> names as variables. It is evaluated for each <code>PlexID</code>/<code>QuantBlock</code> combination and applied to divide reporter ion intensities within corresponding <code>PlexID</code>/<code>QuantBlock</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create references table</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">samples</span> <span class="op">%&gt;%</span> 
  <span class="co"># Filter to reference channel</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ReporterName</span> <span class="op">==</span> <span class="va">ref_channel</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Select required columns and rename ReporterAlias to Reference</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">PlexID</span>, Reference <span class="op">=</span> <span class="va">ReporterAlias</span>, <span class="va">QuantBlock</span><span class="op">)</span></code></pre></div>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:references-table">Table 1.5: </span><left>References table.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
PlexID
</th>
<th style="text-align:left;">
Reference
</th>
<th style="text-align:right;">
QuantBlock
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
S1
</td>
<td style="text-align:left;">
S1_131
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
S2
</td>
<td style="text-align:left;">
S2_131
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>Table <a href="isobaric-quantification-pipelines.html#tab:references-table">1.5</a> shows the first 10 rows of <code>references</code>. The code to use the geometric average instead of a single channel as the reference is shown below. The geometric average is the product of the reporter ion channels to the power of (1/number of channels). For each <code>PlexID</code> group, collapse the vector of reporter ion names with <code><a href="https://rdrr.io/r/base/Arithmetic.html">*</a></code>, surround them in parentheses, and raise to the power of (1/number of channels).</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use geometric average as reference</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">samples</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">PlexID</span>, <span class="va">QuantBlock</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Reference <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"(%s)^(1/%d)"</span>, 
                                <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">ReporterAlias</span>, collapse <span class="op">=</span> <span class="st">"*"</span><span class="op">)</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Do not normalize by reference channel (use 1 as the reference)</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">samples</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">PlexID</span>, <span class="va">QuantBlock</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Reference <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p>Now that we have the three study design tables, we should save them.</p>
<!---
TODO:
Explain how to add the study design tables to the DMS.
--->
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Save study design tables</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">fractions</span>, file <span class="op">=</span> <span class="st">"fractions.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">samples</span>, file <span class="op">=</span> <span class="st">"samples.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">references</span>, file <span class="op">=</span> <span class="st">"references.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="global-quant-crosstab" class="section level3" number="1.1.4">
<h3>
<span class="header-section-number">1.1.4</span> Create Quantitative Cross-tab<a class="anchor" aria-label="anchor" href="#global-quant-crosstab"><i class="fas fa-link"></i></a>
</h3>
<p>This is the final step where MS/MS IDs and reporter ions are linked together and aggregated to the peptide or accession (i.e. protein) level. To retain protein IDs while aggregating to peptide level, set <code>aggregation_level &lt;- c("accession","peptide")</code>. The entries are <span class="math inline">\(log_2\)</span>-transformed after being normalized by the reference.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Set aggregation level</span>
<span class="va">aggregation_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"accession"</span><span class="op">)</span>
<span class="co"># Create cross-tab</span>
<span class="va">crosstab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/create_crosstab.html">create_crosstab</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">masic_data</span>,
                            aggregation_level <span class="op">=</span> <span class="va">aggregation_level</span>,
                            <span class="va">fractions</span>, <span class="va">samples</span>, <span class="va">references</span><span class="op">)</span></code></pre></div>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:unnamed-chunk-21">Table 1.6: </span><left>First 6 rows of the global quantitative cross-tab.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
S1_126
</th>
<th style="text-align:right;">
S1_127C
</th>
<th style="text-align:right;">
S1_127N
</th>
<th style="text-align:right;">
S1_128C
</th>
<th style="text-align:right;">
S1_128N
</th>
<th style="text-align:right;">
S1_129C
</th>
<th style="text-align:right;">
S1_129N
</th>
<th style="text-align:right;">
S1_130C
</th>
<th style="text-align:right;">
S1_130N
</th>
<th style="text-align:right;">
S2_126
</th>
<th style="text-align:right;">
S2_127C
</th>
<th style="text-align:right;">
S2_127N
</th>
<th style="text-align:right;">
S2_128C
</th>
<th style="text-align:right;">
S2_128N
</th>
<th style="text-align:right;">
S2_129C
</th>
<th style="text-align:right;">
S2_129N
</th>
<th style="text-align:right;">
S2_130C
</th>
<th style="text-align:right;">
S2_130N
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
AP_004893.1
</td>
<td style="text-align:right;">
0.1419768
</td>
<td style="text-align:right;">
0.7628195
</td>
<td style="text-align:right;">
0.1653552
</td>
<td style="text-align:right;">
0.8662554
</td>
<td style="text-align:right;">
0.9453172
</td>
<td style="text-align:right;">
-0.6460065
</td>
<td style="text-align:right;">
-1.9294467
</td>
<td style="text-align:right;">
-0.4321433
</td>
<td style="text-align:right;">
-1.2831873
</td>
<td style="text-align:right;">
-1.0271227
</td>
<td style="text-align:right;">
-0.9390945
</td>
<td style="text-align:right;">
0.4883309
</td>
<td style="text-align:right;">
-1.7148628
</td>
<td style="text-align:right;">
-0.7029685
</td>
<td style="text-align:right;">
-0.8794712
</td>
<td style="text-align:right;">
-0.1912097
</td>
<td style="text-align:right;">
0.3964607
</td>
<td style="text-align:right;">
-0.2440478
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004894.1
</td>
<td style="text-align:right;">
0.8092676
</td>
<td style="text-align:right;">
-0.0976095
</td>
<td style="text-align:right;">
-0.3113350
</td>
<td style="text-align:right;">
0.3215692
</td>
<td style="text-align:right;">
0.2171255
</td>
<td style="text-align:right;">
-0.3678781
</td>
<td style="text-align:right;">
-0.1638689
</td>
<td style="text-align:right;">
-0.6696829
</td>
<td style="text-align:right;">
-1.2039041
</td>
<td style="text-align:right;">
-0.5124954
</td>
<td style="text-align:right;">
-0.2364175
</td>
<td style="text-align:right;">
-0.4428327
</td>
<td style="text-align:right;">
-1.3730408
</td>
<td style="text-align:right;">
-0.6711809
</td>
<td style="text-align:right;">
-1.3515366
</td>
<td style="text-align:right;">
-0.7462995
</td>
<td style="text-align:right;">
-0.8338103
</td>
<td style="text-align:right;">
-0.2227493
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004895.1
</td>
<td style="text-align:right;">
0.2078433
</td>
<td style="text-align:right;">
-0.2867209
</td>
<td style="text-align:right;">
-0.6089756
</td>
<td style="text-align:right;">
-0.1162062
</td>
<td style="text-align:right;">
-0.3840271
</td>
<td style="text-align:right;">
-1.1240967
</td>
<td style="text-align:right;">
-0.6908468
</td>
<td style="text-align:right;">
-0.6652575
</td>
<td style="text-align:right;">
-0.7140383
</td>
<td style="text-align:right;">
0.2717217
</td>
<td style="text-align:right;">
-0.1448289
</td>
<td style="text-align:right;">
-0.1200736
</td>
<td style="text-align:right;">
-0.6435709
</td>
<td style="text-align:right;">
-0.4287771
</td>
<td style="text-align:right;">
-0.6780284
</td>
<td style="text-align:right;">
-0.6102404
</td>
<td style="text-align:right;">
-0.3896190
</td>
<td style="text-align:right;">
-0.1548544
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004896.1
</td>
<td style="text-align:right;">
-0.1494849
</td>
<td style="text-align:right;">
-0.3664339
</td>
<td style="text-align:right;">
-0.7314368
</td>
<td style="text-align:right;">
-0.1742391
</td>
<td style="text-align:right;">
-0.5352280
</td>
<td style="text-align:right;">
-1.2945071
</td>
<td style="text-align:right;">
-1.0372327
</td>
<td style="text-align:right;">
-0.7060783
</td>
<td style="text-align:right;">
-0.8299749
</td>
<td style="text-align:right;">
0.1939540
</td>
<td style="text-align:right;">
-0.2274358
</td>
<td style="text-align:right;">
-0.1688422
</td>
<td style="text-align:right;">
-0.5251264
</td>
<td style="text-align:right;">
-0.4222698
</td>
<td style="text-align:right;">
-0.6543311
</td>
<td style="text-align:right;">
-0.6741064
</td>
<td style="text-align:right;">
-0.3994149
</td>
<td style="text-align:right;">
-0.0441485
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004898.1
</td>
<td style="text-align:right;">
0.0362964
</td>
<td style="text-align:right;">
0.7497227
</td>
<td style="text-align:right;">
0.4252227
</td>
<td style="text-align:right;">
0.4913660
</td>
<td style="text-align:right;">
1.1580326
</td>
<td style="text-align:right;">
0.1211536
</td>
<td style="text-align:right;">
-0.3640632
</td>
<td style="text-align:right;">
-0.3019505
</td>
<td style="text-align:right;">
-0.8291744
</td>
<td style="text-align:right;">
-0.8407749
</td>
<td style="text-align:right;">
-0.2796091
</td>
<td style="text-align:right;">
-0.4130732
</td>
<td style="text-align:right;">
-1.5747761
</td>
<td style="text-align:right;">
-0.9449498
</td>
<td style="text-align:right;">
-1.8439756
</td>
<td style="text-align:right;">
-0.1774225
</td>
<td style="text-align:right;">
-1.1083199
</td>
<td style="text-align:right;">
-0.4175363
</td>
</tr>
<tr>
<td style="text-align:left;">
AP_004899.1
</td>
<td style="text-align:right;">
0.7140968
</td>
<td style="text-align:right;">
-0.1781542
</td>
<td style="text-align:right;">
-0.3732752
</td>
<td style="text-align:right;">
0.3494902
</td>
<td style="text-align:right;">
-0.0615626
</td>
<td style="text-align:right;">
-2.1679002
</td>
<td style="text-align:right;">
-0.8550940
</td>
<td style="text-align:right;">
-0.9026145
</td>
<td style="text-align:right;">
-1.4519278
</td>
<td style="text-align:right;">
-0.3158081
</td>
<td style="text-align:right;">
-0.4056811
</td>
<td style="text-align:right;">
-0.4644758
</td>
<td style="text-align:right;">
-0.2805080
</td>
<td style="text-align:right;">
-0.9023044
</td>
<td style="text-align:right;">
-1.0482424
</td>
<td style="text-align:right;">
-0.8052899
</td>
<td style="text-align:right;">
-0.6675429
</td>
<td style="text-align:right;">
-0.3959923
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>In order to demonstrate prioritized inference in Section <a href="isobaric-quantification-pipelines.html#phosphoproteomics-data">1.2</a>, we need to save the row names of this cross-tab.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Save protein names</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">crosstab</span><span class="op">)</span>, file <span class="op">=</span> <span class="st">"data/3442_global_protein_names.rds"</span><span class="op">)</span></code></pre></div>
<p>We should save the cross-tab as well. To do so, we need to convert the row names to a column called <code>protein</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Modify cross-tab for saving</span>
<span class="va">crosstab</span> <span class="op">&lt;-</span> <span class="va">crosstab</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"protein"</span><span class="op">)</span>

<span class="co"># Save cross-tab</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">crosstab</span>, file <span class="op">=</span> <span class="st">"data/global_quant_crosstab.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="phosphoproteomics-data" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> Phosphoproteomics Data<a class="anchor" aria-label="anchor" href="#phosphoproteomics-data"><i class="fas fa-link"></i></a>
</h2>
<p>This pipeline shows how to process data from the DMS. The number of the data package is <code>3626</code>. For this section, we need the <code>PlexedPiper</code> package for isobaric quantification and <code>PNNL.DMS.utils</code> to interface with the DMS. Also, some details will be omitted if they were already provided in Section <a href="isobaric-quantification-pipelines.html#global-proteomics-data">1.1</a>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Setup</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PNNL.DMS.utils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PlexedPiper</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span> <span class="co"># %&gt;%</span></code></pre></div>
<div id="prepare-MS2-IDs-phospho" class="section level3" number="1.2.1">
<h3>
<span class="header-section-number">1.2.1</span> Prepare MS/MS Identifications<a class="anchor" aria-label="anchor" href="#prepare-MS2-IDs-phospho"><i class="fas fa-link"></i></a>
</h3>
<div id="read-ms-gf-output" class="section level4" number="1.2.1.1">
<h4>
<span class="header-section-number">1.2.1.1</span> Read MS-GF+ Output<a class="anchor" aria-label="anchor" href="#read-ms-gf-output"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read MS-GF+ data</span>
<span class="va">data_package_num</span> <span class="op">&lt;-</span> <span class="fl">3626</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_msgf_data_from_DMS.html">read_msgf_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 612667 at 55 % FDR
## #peptides: 396540 at 75 % FDR
## #accessions: 121521 at 98 % FDR</code></pre>
</div>
<div id="remove-non-phosphorylated-peptides" class="section level4" number="1.2.1.2">
<h4>
<span class="header-section-number">1.2.1.2</span> Remove Non-Phosphorylated Peptides<a class="anchor" aria-label="anchor" href="#remove-non-phosphorylated-peptides"><i class="fas fa-link"></i></a>
</h4>
<p>In this case, the phosphorylation of an amino acid is marked by a <code><a href="https://rdrr.io/r/base/Arithmetic.html">*</a></code> appearing after the amino acid. We will not consider unmodified peptides, so we can filter them out. The <code><a href="https://rdrr.io/r/base/Arithmetic.html">*</a></code> is a special character that must be escaped with backslashes, and the backslashes must also be escaped.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove non-phosphorylated peptides</span>
<span class="co"># (peptides that do not contain a *)</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">apply_filter</span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"grepl('\\\\*', peptide)"</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 537749 at 57 % FDR
## #peptides: 353634 at 76 % FDR
## #accessions: 118817 at 98 % FDR</code></pre>
</div>
<div id="correct-isotope-selection-error-1" class="section level4" number="1.2.1.3">
<h4>
<span class="header-section-number">1.2.1.3</span> Correct Isotope Selection Error<a class="anchor" aria-label="anchor" href="#correct-isotope-selection-error-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Correct for isotope selection error</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">correct_peak_selection</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
</div>
<div id="remove-contaminants-1" class="section level4" number="1.2.1.4">
<h4>
<span class="header-section-number">1.2.1.4</span> Remove Contaminants<a class="anchor" aria-label="anchor" href="#remove-contaminants-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove contaminants</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">apply_filter</span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!grepl('Contaminant', accession)"</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 537572 at 57 % FDR
## #peptides: 353489 at 76 % FDR
## #accessions: 118797 at 98 % FDR</code></pre>
</div>
<div id="ascore" class="section level4" number="1.2.1.5">
<h4>
<span class="header-section-number">1.2.1.5</span> AScore<a class="anchor" aria-label="anchor" href="#ascore"><i class="fas fa-link"></i></a>
</h4>
<p>Phospho datasets involve AScore jobs for improving phosphosite localization. There should be one AScore job per data package. The fetched object is a data.frame that links datasets, scans and original PTM localization to newly suggested locations. Importantly, it contains <code>AScore</code> column that signifies the confidence of PTM assignment. AScore &gt; 17 is considered confident.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Filter PTMs by AScore</span>
<span class="va">ascore</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html">get_AScore_results</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">best_PTM_location_by_ascore</span><span class="op">(</span><span class="va">msnid</span>, <span class="va">ascore</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 188791 at 30 % FDR
## #peptides: 101873 at 53 % FDR
## #accessions: 90677 at 93 % FDR</code></pre>
</div>
<div id="msms-id-filter-peptide-level-1" class="section level4" number="1.2.1.6">
<h4>
<span class="header-section-number">1.2.1.6</span> MS/MS ID Filter: Peptide Level<a class="anchor" aria-label="anchor" href="#msms-id-filter-peptide-level-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 1% FDR filter at the peptide level</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_msgf_data.html">filter_msgf_data</a></span><span class="op">(</span><span class="va">msnid</span>,
                          level <span class="op">=</span> <span class="st">"peptide"</span>,
                          fdr.max <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 76103 at 0.49 % FDR
## #peptides: 23378 at 1 % FDR
## #accessions: 16090 at 4.7 % FDR</code></pre>
</div>
<div id="msms-id-filter-protein-level-1" class="section level4" number="1.2.1.7">
<h4>
<span class="header-section-number">1.2.1.7</span> MS/MS ID Filter: Protein Level<a class="anchor" aria-label="anchor" href="#msms-id-filter-protein-level-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get path to FASTA file</span>
<span class="va">path_to_FASTA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/pnnl_dms_utils.html">path_to_FASTA_used_by_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span><span class="op">)</span>

<span class="co"># Compute number of peptides per 1000 amino acids</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/compute_num_peptides_per_1000aa.html">compute_num_peptides_per_1000aa</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">path_to_FASTA</span><span class="op">)</span>

<span class="co"># 1% FDR filter at the protein level</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_msgf_data.html">filter_msgf_data</a></span><span class="op">(</span><span class="va">msnid</span>,
                          level <span class="op">=</span> <span class="st">"accession"</span>,
                          fdr.max <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 72481 at 0.12 % FDR
## #peptides: 21266 at 0.26 % FDR
## #accessions: 9424 at 0.98 % FDR</code></pre>
</div>
<div id="inference-of-parsimonious-protein-set-1" class="section level4" number="1.2.1.8">
<h4>
<span class="header-section-number">1.2.1.8</span> Inference of Parsimonious Protein Set<a class="anchor" aria-label="anchor" href="#inference-of-parsimonious-protein-set-1"><i class="fas fa-link"></i></a>
</h4>
<!---
TODO:
* Talk about using prior information from global cross-tab to improve inference of the parsimonious protein set for phospho data.
--->
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load proteins from global crosstab</span>
<span class="va">global_proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"data/3442_global_protein_names.rds"</span><span class="op">)</span>
<span class="co"># Inference of parsimonious protein set</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">infer_parsimonious_accessions</span><span class="op">(</span><span class="va">msnid</span>, unique_only <span class="op">=</span> <span class="cn">FALSE</span>,
                                       prior <span class="op">=</span> <span class="va">global_proteins</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 72481 at 0.12 % FDR
## #peptides: 21266 at 0.26 % FDR
## #accessions: 2895 at 1.6 % FDR</code></pre>
<p>Notice that the protein-level FDR increased above the acceptable threshold, so we need to reapply the filter.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 1% FDR filter at the protein level</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_msgf_data.html">filter_msgf_data</a></span><span class="op">(</span><span class="va">msnid</span>,
                          level <span class="op">=</span> <span class="st">"accession"</span>,
                          fdr.max <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 70340 at 0.075 % FDR
## #peptides: 20117 at 0.15 % FDR
## #accessions: 2351 at 0.99 % FDR</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove Decoy Accessions</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">apply_filter</span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!isDecoy"</span><span class="op">)</span>
<span class="fu">show</span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  23 
## #PSMs: 70287 at 0 % FDR
## #peptides: 20086 at 0 % FDR
## #accessions: 2328 at 0 % FDR</code></pre>
</div>
<div id="map-sites-to-protein-sequences" class="section level4" number="1.2.1.9">
<h4>
<span class="header-section-number">1.2.1.9</span> Map Sites to Protein Sequences<a class="anchor" aria-label="anchor" href="#map-sites-to-protein-sequences"><i class="fas fa-link"></i></a>
</h4>
<p>Prepare FASTA to make sure entry names in FASTA file match MSnID accessions. The plan is to make this conversion automatic. <code>map_mod_sites</code> creates number of columns describing mapping of the site/s onto the protein sequences. The most important for the user is <code>SiteID</code>.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create AAStringSet</span>
<span class="va">fst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readAAStringSet</a></span><span class="op">(</span><span class="va">path_to_FASTA</span><span class="op">)</span>
<span class="co"># Remove contaminants</span>
<span class="va">fst</span> <span class="op">&lt;-</span> <span class="va">fst</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Contaminant"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span>
<span class="co"># First 6 names</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "NP_783171.2 cathepsin R precursor [Rattus norvegicus]"                    
## [2] "NP_001101862.2 zinc finger protein ZIC 2 [Rattus norvegicus]"             
## [3] "NP_113721.4 UDP-glucuronosyltransferase 2B2 precursor [Rattus norvegicus]"
## [4] "NP_714948.1 Ly-49 stimulatory receptor 3 [Rattus norvegicus]"             
## [5] "NP_001000704.1 olfactory receptor Olr931 [Rattus norvegicus]"             
## [6] "NP_001000638.1 olfactory receptor Olr652 [Rattus norvegicus]"</code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Modify names to match accessions(msnid)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span>, split <span class="op">=</span> <span class="st">" "</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Select text before first space</span>
  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># First 6 names</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">fst</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "NP_783171.2"    "NP_001101862.2" "NP_113721.4"    "NP_714948.1"   
## [5] "NP_001000704.1" "NP_001000638.1"</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Main mapping call</span>
<span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu">map_mod_sites</span><span class="op">(</span>object <span class="op">=</span> <span class="va">msnid</span>, fasta <span class="op">=</span> <span class="va">fst</span>, 
                       accession_col <span class="op">=</span> <span class="st">"accession"</span>, 
                       peptide_mod_col <span class="op">=</span> <span class="st">"peptide"</span>, 
                       mod_char <span class="op">=</span> <span class="st">"*"</span>,
                       site_delimiter <span class="op">=</span> <span class="st">"lower"</span><span class="op">)</span></code></pre></div>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:phospho-msnid-table">Table 1.7: </span><left>First 6 rows of the processed MS-GF+ results.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:right;">
ResultID
</th>
<th style="text-align:right;">
Scan
</th>
<th style="text-align:left;">
FragMethod
</th>
<th style="text-align:right;">
SpecIndex
</th>
<th style="text-align:right;">
Charge
</th>
<th style="text-align:right;">
PrecursorMZ
</th>
<th style="text-align:right;">
DelM
</th>
<th style="text-align:right;">
DelM_PPM
</th>
<th style="text-align:right;">
MH
</th>
<th style="text-align:left;">
OriginalPeptide
</th>
<th style="text-align:left;">
Protein
</th>
<th style="text-align:right;">
NTT
</th>
<th style="text-align:right;">
DeNovoScore
</th>
<th style="text-align:right;">
MSGFScore
</th>
<th style="text-align:right;">
MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
Rank_MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
EValue
</th>
<th style="text-align:right;">
QValue
</th>
<th style="text-align:right;">
PepQValue
</th>
<th style="text-align:right;">
IsotopeError
</th>
<th style="text-align:left;">
accession
</th>
<th style="text-align:right;">
calculatedMassToCharge
</th>
<th style="text-align:right;">
chargeState
</th>
<th style="text-align:right;">
experimentalMassToCharge
</th>
<th style="text-align:left;">
isDecoy
</th>
<th style="text-align:left;">
spectrumFile
</th>
<th style="text-align:right;">
spectrumID
</th>
<th style="text-align:left;">
pepSeq
</th>
<th style="text-align:left;">
peptide
</th>
<th style="text-align:right;">
maxAScore
</th>
<th style="text-align:right;">
msmsScore
</th>
<th style="text-align:right;">
absParentMassErrorPPM
</th>
<th style="text-align:right;">
peptides_per_1000aa
</th>
<th style="text-align:left;">
First_AA
</th>
<th style="text-align:left;">
Last_AA
</th>
<th style="text-align:right;">
First_AA_First
</th>
<th style="text-align:right;">
Last_AA_First
</th>
<th style="text-align:right;">
ProtLen
</th>
<th style="text-align:left;">
ModShift
</th>
<th style="text-align:left;">
ModAAs
</th>
<th style="text-align:left;">
SiteLoc
</th>
<th style="text-align:left;">
Site
</th>
<th style="text-align:left;">
SiteCollapsed
</th>
<th style="text-align:left;">
SiteCollapsedFirst
</th>
<th style="text-align:left;">
SiteID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_06_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
12697
</td>
<td style="text-align:right;">
27321
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
2256
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1045.124
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0.858
</td>
<td style="text-align:right;">
3131.346
</td>
<td style="text-align:left;">
A.AAAAAGDS*DS*WDADTFSMEDPVRK.V
</td>
<td style="text-align:left;">
NP_001071138.1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
146
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
NP_001071138.1
</td>
<td style="text-align:right;">
1044.454
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1044.455
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_06_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
27321
</td>
<td style="text-align:left;">
AAAAAGDSDSWDADTFSMEDPVRK
</td>
<td style="text-align:left;">
A.AAAAAGDSDS*WDADT*FSMEDPVRK.V
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.057
</td>
<td style="text-align:right;">
7.722
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
28
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
259
</td>
<td style="text-align:left;">
9, 14
</td>
<td style="text-align:left;">
S, T
</td>
<td style="text-align:left;">
14, 19
</td>
<td style="text-align:left;">
S14, T19
</td>
<td style="text-align:left;">
S14,T19
</td>
<td style="text-align:left;">
S14,T19
</td>
<td style="text-align:left;">
NP_001071138.1-S14sT19t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
875
</td>
<td style="text-align:right;">
23519
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
264
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
952.144
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
1.538
</td>
<td style="text-align:right;">
2854.412
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
165
</td>
<td style="text-align:right;">
129
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
952.142
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
952.144
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
23519
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:right;">
52.349
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.625
</td>
<td style="text-align:right;">
5.305
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
262
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
262
</td>
<td style="text-align:right;">
377
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
XP_006232986.1-T250t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
12873
</td>
<td style="text-align:right;">
23508
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
2213
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.360
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
2.392
</td>
<td style="text-align:right;">
2854.412
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
122
</td>
<td style="text-align:right;">
81
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
714.358
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.360
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
23508
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:right;">
17.480
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
2.472
</td>
<td style="text-align:right;">
5.305
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
262
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
262
</td>
<td style="text-align:right;">
377
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
XP_006232986.1-T250t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S2_07_3Nov17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
2731
</td>
<td style="text-align:right;">
23697
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
502
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.610
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.706
</td>
<td style="text-align:right;">
2854.412
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
714.358
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.359
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S2_07_3Nov17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
23697
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:right;">
26.295
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.780
</td>
<td style="text-align:right;">
5.305
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
262
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
262
</td>
<td style="text-align:right;">
377
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
XP_006232986.1-T250t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
4877
</td>
<td style="text-align:right;">
21265
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
935
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
800.403
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
1.871
</td>
<td style="text-align:right;">
3196.577
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEGERDSDDALLK.M
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
194
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
799.900
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
799.901
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
21265
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEGERDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIATPGT*EGERDSDDALLK.M
</td>
<td style="text-align:right;">
6.213
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.902
</td>
<td style="text-align:right;">
10.526
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
265
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
265
</td>
<td style="text-align:right;">
380
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
NP_112621.1-T253t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
6826
</td>
<td style="text-align:right;">
21280
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
1251
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1066.532
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
-0.095
</td>
<td style="text-align:right;">
3196.577
</td>
<td style="text-align:left;">
R.AAAASAAEAGIATPGT*EGERDSDDALLK.M
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
94
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
1066.197
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1066.197
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
21280
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEGERDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIATPGT*EGERDSDDALLK.M
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
10.526
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
265
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
265
</td>
<td style="text-align:right;">
380
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
NP_112621.1-T253t
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>Table <a href="isobaric-quantification-pipelines.html#tab:phospho-msnid-table">1.7</a> shows the first 6 rows of the processed MS-GF+ output.</p>
</div>
</div>
<div id="phospho-reporter-ion-intensities" class="section level3" number="1.2.2">
<h3>
<span class="header-section-number">1.2.2</span> Prepare Reporter Ion Intensities<a class="anchor" aria-label="anchor" href="#phospho-reporter-ion-intensities"><i class="fas fa-link"></i></a>
</h3>
<div id="read-masic-output-1" class="section level4" number="1.2.2.1">
<h4>
<span class="header-section-number">1.2.2.1</span> Read MASIC Output<a class="anchor" aria-label="anchor" href="#read-masic-output-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read MASIC data</span>
<span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PNNL.DMS.utils/man/read_masic_data_from_DMS.html">read_masic_data_from_DMS</a></span><span class="op">(</span><span class="va">data_package_num</span>, 
                                       interference_score <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="filter-masic-data-1" class="section level4" number="1.2.2.2">
<h4>
<span class="header-section-number">1.2.2.2</span> Filter MASIC Data<a class="anchor" aria-label="anchor" href="#filter-masic-data-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Filter MASIC data</span>
<span class="va">masic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/filter_masic_data.html">filter_masic_data</a></span><span class="op">(</span><span class="va">masic_data</span>, 
                                interference_score_threshold <span class="op">=</span> <span class="fl">0.5</span>,
                                s2n_threshold <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="create-study-design-tables" class="section level3" number="1.2.3">
<h3>
<span class="header-section-number">1.2.3</span> Create Study Design Tables<a class="anchor" aria-label="anchor" href="#create-study-design-tables"><i class="fas fa-link"></i></a>
</h3>
<div id="fractions-1" class="section level4" number="1.2.3.1">
<h4>
<span class="header-section-number">1.2.3.1</span> Fractions<a class="anchor" aria-label="anchor" href="#fractions-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create fractions table</span>
<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Dataset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">masic_data</span><span class="op">$</span><span class="va">Dataset</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>PlexID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">".*_P_(S\\d{1})_.*"</span>, <span class="st">"\\1"</span>, <span class="va">Dataset</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="samples-1" class="section level4" number="1.2.3.2">
<h4>
<span class="header-section-number">1.2.3.2</span> Samples<a class="anchor" aria-label="anchor" href="#samples-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># TMT10 Reporter Converter table from MSnID package</span>
<span class="va">conv</span> <span class="op">&lt;-</span> <span class="va">reporter_converter</span><span class="op">$</span><span class="va">tmt10</span>
<span class="va">plexes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">$</span><span class="va">PlexID</span><span class="op">)</span>

<span class="co"># Reference channel</span>
<span class="va">ref_channel</span> <span class="op">&lt;-</span> <span class="st">"131"</span>

<span class="co"># Create samples table</span>
<span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>PlexID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">plexes</span>, each <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">conv</span><span class="op">)</span><span class="op">)</span>,  
                      ReporterName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">conv</span><span class="op">$</span><span class="va">ReporterName</span>, 
                                         <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">plexes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>ReporterAlias <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"%s_%s"</span>, <span class="va">PlexID</span>, <span class="va">ReporterName</span><span class="op">)</span>,
         MeasurementName <span class="op">=</span> <span class="va">ReporterAlias</span>,
         QuantBlock <span class="op">=</span> <span class="fl">1</span>,
         <span class="co"># Comment out this next part if the reference</span>
         <span class="co"># is not one of the reporter ion channels.</span>
         MeasurementName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">ReporterName</span> <span class="op">==</span> <span class="va">ref_channel</span>,
                                  <span class="cn">NA</span>, <span class="va">MeasurementName</span><span class="op">)</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="references-1" class="section level4" number="1.2.3.3">
<h4>
<span class="header-section-number">1.2.3.3</span> References<a class="anchor" aria-label="anchor" href="#references-1"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create references table</span>
<span class="va">references</span> <span class="op">&lt;-</span> <span class="va">samples</span> <span class="op">%&gt;%</span> 
  <span class="co"># Filter to reference channel</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ReporterName</span> <span class="op">==</span> <span class="va">ref_channel</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Select required columns and rename ReporterAlias to Reference</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">PlexID</span>, Reference <span class="op">=</span> <span class="va">ReporterAlias</span>, <span class="va">QuantBlock</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="phospho-create-crosstab" class="section level3" number="1.2.4">
<h3>
<span class="header-section-number">1.2.4</span> Create Quantitative Cross-tab<a class="anchor" aria-label="anchor" href="#phospho-create-crosstab"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Set aggregation level</span>
<span class="va">aggregation_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SiteID"</span><span class="op">)</span>
<span class="co"># Create cross-tab</span>
<span class="va">crosstab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PlexedPiper/man/create_crosstab.html">create_crosstab</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">masic_data</span>,
                            aggregation_level <span class="op">=</span> <span class="va">aggregation_level</span>,
                            <span class="va">fractions</span>, <span class="va">samples</span>, <span class="va">references</span><span class="op">)</span></code></pre></div>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:unnamed-chunk-46">Table 1.8: </span><left>First 6 rows of the phospho quantitative cross-tab.</left>
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
S1_126
</th>
<th style="text-align:right;">
S1_127C
</th>
<th style="text-align:right;">
S1_127N
</th>
<th style="text-align:right;">
S1_128C
</th>
<th style="text-align:right;">
S1_128N
</th>
<th style="text-align:right;">
S1_129C
</th>
<th style="text-align:right;">
S1_129N
</th>
<th style="text-align:right;">
S1_130C
</th>
<th style="text-align:right;">
S1_130N
</th>
<th style="text-align:right;">
S2_126
</th>
<th style="text-align:right;">
S2_127C
</th>
<th style="text-align:right;">
S2_127N
</th>
<th style="text-align:right;">
S2_128C
</th>
<th style="text-align:right;">
S2_128N
</th>
<th style="text-align:right;">
S2_129C
</th>
<th style="text-align:right;">
S2_129N
</th>
<th style="text-align:right;">
S2_130C
</th>
<th style="text-align:right;">
S2_130N
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
NP_001001512.2-S241s
</td>
<td style="text-align:right;">
-0.5441083
</td>
<td style="text-align:right;">
-0.0291611
</td>
<td style="text-align:right;">
-0.5540885
</td>
<td style="text-align:right;">
-0.1812456
</td>
<td style="text-align:right;">
-0.5675833
</td>
<td style="text-align:right;">
-0.0427948
</td>
<td style="text-align:right;">
-0.803056
</td>
<td style="text-align:right;">
-0.5236890
</td>
<td style="text-align:right;">
-1.0398773
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S242s
</td>
<td style="text-align:right;">
-0.2806018
</td>
<td style="text-align:right;">
-0.2939827
</td>
<td style="text-align:right;">
-0.3923442
</td>
<td style="text-align:right;">
-0.3242804
</td>
<td style="text-align:right;">
-0.6512914
</td>
<td style="text-align:right;">
-0.4914234
</td>
<td style="text-align:right;">
-1.210593
</td>
<td style="text-align:right;">
-0.6421375
</td>
<td style="text-align:right;">
-0.8869142
</td>
<td style="text-align:right;">
-0.3079991
</td>
<td style="text-align:right;">
-0.8072458
</td>
<td style="text-align:right;">
-0.4801256
</td>
<td style="text-align:right;">
-0.4458455
</td>
<td style="text-align:right;">
-0.7588038
</td>
<td style="text-align:right;">
-1.3958746
</td>
<td style="text-align:right;">
-0.6640721
</td>
<td style="text-align:right;">
-0.7046127
</td>
<td style="text-align:right;">
-0.3091972
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S699s
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-0.7930334
</td>
<td style="text-align:right;">
-1.2173509
</td>
<td style="text-align:right;">
-0.4608541
</td>
<td style="text-align:right;">
-1.2596689
</td>
<td style="text-align:right;">
-0.8004630
</td>
<td style="text-align:right;">
-0.8584401
</td>
<td style="text-align:right;">
-0.4918316
</td>
<td style="text-align:right;">
-0.2745069
</td>
<td style="text-align:right;">
0.2867235
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S746s
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1.4611181
</td>
<td style="text-align:right;">
-2.2975481
</td>
<td style="text-align:right;">
-1.6828088
</td>
<td style="text-align:right;">
-2.3039510
</td>
<td style="text-align:right;">
-1.8632844
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1.5981389
</td>
<td style="text-align:right;">
-2.4732843
</td>
<td style="text-align:right;">
-0.4736739
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S748s
</td>
<td style="text-align:right;">
-1.2580276
</td>
<td style="text-align:right;">
-1.3182692
</td>
<td style="text-align:right;">
-2.1938861
</td>
<td style="text-align:right;">
-2.0017857
</td>
<td style="text-align:right;">
0.0219220
</td>
<td style="text-align:right;">
-0.7086501
</td>
<td style="text-align:right;">
-1.694015
</td>
<td style="text-align:right;">
-1.4681967
</td>
<td style="text-align:right;">
-1.5795032
</td>
<td style="text-align:right;">
0.3139811
</td>
<td style="text-align:right;">
-0.6805752
</td>
<td style="text-align:right;">
-0.8132341
</td>
<td style="text-align:right;">
-1.7933771
</td>
<td style="text-align:right;">
0.0127816
</td>
<td style="text-align:right;">
0.2726023
</td>
<td style="text-align:right;">
0.2266855
</td>
<td style="text-align:right;">
0.5058189
</td>
<td style="text-align:right;">
0.9332038
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001514.1-S279s
</td>
<td style="text-align:right;">
0.0224073
</td>
<td style="text-align:right;">
-0.1412578
</td>
<td style="text-align:right;">
-0.4939068
</td>
<td style="text-align:right;">
-0.3322896
</td>
<td style="text-align:right;">
-0.6624270
</td>
<td style="text-align:right;">
0.0686814
</td>
<td style="text-align:right;">
-0.711781
</td>
<td style="text-align:right;">
-1.0613940
</td>
<td style="text-align:right;">
-1.5813449
</td>
<td style="text-align:right;">
-0.1489704
</td>
<td style="text-align:right;">
-0.4580319
</td>
<td style="text-align:right;">
-0.5889714
</td>
<td style="text-align:right;">
-0.8708282
</td>
<td style="text-align:right;">
-0.5066630
</td>
<td style="text-align:right;">
-0.7469875
</td>
<td style="text-align:right;">
-0.3374105
</td>
<td style="text-align:right;">
-0.3842025
</td>
<td style="text-align:right;">
-0.1701473
</td>
</tr>
</tbody>
</table></div>
<p></p>
<p>We will save the cross-tab for later sections.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Modify cross-tab for saving</span>
<span class="va">crosstab</span> <span class="op">&lt;-</span> <span class="va">crosstab</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"phosphosite"</span><span class="op">)</span>

<span class="co"># Save cross-tab</span>
<span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">crosstab</span>, file <span class="op">=</span> <span class="st">"data/phospho_quant_crosstab.txt"</span>,
            sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html">Welcome!</a></div>
<div class="next"><a href="exploratory-data-analysis.html"><span class="header-section-number">2</span> Exploratory Data Analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#isobaric-quantification-pipelines"><span class="header-section-number">1</span> Isobaric Quantification Pipelines</a></li>
<li>
<a class="nav-link" href="#global-proteomics-data"><span class="header-section-number">1.1</span> Global Proteomics Data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#prepare-MS2-IDs-global"><span class="header-section-number">1.1.1</span> Prepare MS/MS Identifications</a></li>
<li><a class="nav-link" href="#reporter-ion-intensities"><span class="header-section-number">1.1.2</span> Prepare Reporter Ion Intensities</a></li>
<li><a class="nav-link" href="#fetch-study-design-tables"><span class="header-section-number">1.1.3</span> Create Study Design Tables</a></li>
<li><a class="nav-link" href="#global-quant-crosstab"><span class="header-section-number">1.1.4</span> Create Quantitative Cross-tab</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#phosphoproteomics-data"><span class="header-section-number">1.2</span> Phosphoproteomics Data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#prepare-MS2-IDs-phospho"><span class="header-section-number">1.2.1</span> Prepare MS/MS Identifications</a></li>
<li><a class="nav-link" href="#phospho-reporter-ion-intensities"><span class="header-section-number">1.2.2</span> Prepare Reporter Ion Intensities</a></li>
<li><a class="nav-link" href="#create-study-design-tables"><span class="header-section-number">1.2.3</span> Create Study Design Tables</a></li>
<li><a class="nav-link" href="#phospho-create-crosstab"><span class="header-section-number">1.2.4</span> Create Quantitative Cross-tab</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial/blob/master/isobaric_quantification.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial/edit/master/isobaric_quantification.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Proteomics Data Analysis in R/Bioconductor</strong>" was written by Tyler Sagendorf. It was last built on October 17, 2021.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
