<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Section 2 Label-Free Quantification | Proteomics Data Analysis in R/Bioconductor</title>
  <meta name="description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Section 2 Label-Free Quantification | Proteomics Data Analysis in R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 2 Label-Free Quantification | Proteomics Data Analysis in R/Bioconductor" />
  
  <meta name="twitter:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

<meta name="author" content="Tyler Sagendorf" />


<meta name="date" content="2022-01-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="phosphoproteomics-data.html"/>
<link rel="next" href="exploratory-data-analysis.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=".">Proteomics Data Analysis in R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="isobaric-quantification-pipelines.html"><a href="isobaric-quantification-pipelines.html"><i class="fa fa-check"></i><b>1</b> Isobaric Quantification Pipelines</a>
<ul>
<li class="chapter" data-level="1.1" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html"><i class="fa fa-check"></i><b>1.1</b> Global Proteomics Data</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#read-ms-gf-data"><i class="fa fa-check"></i><b>1.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="1.1.2" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#correct-isotope-selection-error"><i class="fa fa-check"></i><b>1.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.1.3" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#remove-contaminants"><i class="fa fa-check"></i><b>1.1.3</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.1.4" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#global-peptide-filter"><i class="fa fa-check"></i><b>1.1.4</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.1.5" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#msms-id-filter-protein-level"><i class="fa fa-check"></i><b>1.1.5</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.1.6" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#inference-of-parsimonious-protein-set"><i class="fa fa-check"></i><b>1.1.6</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.1.7" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#remove-decoy-psms"><i class="fa fa-check"></i><b>1.1.7</b> Remove Decoy PSMs</a></li>
<li class="chapter" data-level="1.1.8" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#read-masic-output"><i class="fa fa-check"></i><b>1.1.8</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.1.9" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#filter-masic-data"><i class="fa fa-check"></i><b>1.1.9</b> Filter MASIC Data</a></li>
<li class="chapter" data-level="1.1.10" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#fetch-study-design-tables"><i class="fa fa-check"></i><b>1.1.10</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="1.1.11" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#fractions"><i class="fa fa-check"></i><b>1.1.11</b> Fractions</a></li>
<li class="chapter" data-level="1.1.12" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#samples"><i class="fa fa-check"></i><b>1.1.12</b> Samples</a></li>
<li class="chapter" data-level="1.1.13" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#references"><i class="fa fa-check"></i><b>1.1.13</b> References</a></li>
<li class="chapter" data-level="1.1.14" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#global-quant-crosstab"><i class="fa fa-check"></i><b>1.1.14</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="1.1.15" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#create-msnset-object"><i class="fa fa-check"></i><b>1.1.15</b> Create MSnSet Object</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html"><i class="fa fa-check"></i><b>1.2</b> Phosphoproteomics Data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#read-ms-gf-output"><i class="fa fa-check"></i><b>1.2.1</b> Read MS-GF+ Output</a></li>
<li class="chapter" data-level="1.2.2" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#correct-isotope-selection-error-1"><i class="fa fa-check"></i><b>1.2.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.2.3" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-non-phosphorylated-peptides"><i class="fa fa-check"></i><b>1.2.3</b> Remove Non-Phosphorylated Peptides</a></li>
<li class="chapter" data-level="1.2.4" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-contaminants-1"><i class="fa fa-check"></i><b>1.2.4</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.2.5" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#improve-phosphosite-localization"><i class="fa fa-check"></i><b>1.2.5</b> Improve Phosphosite Localization</a></li>
<li class="chapter" data-level="1.2.6" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#msms-id-filter-peptide-level"><i class="fa fa-check"></i><b>1.2.6</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.2.7" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#msms-id-filter-protein-level-1"><i class="fa fa-check"></i><b>1.2.7</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.2.8" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#inference-of-parsimonious-protein-set-1"><i class="fa fa-check"></i><b>1.2.8</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.2.9" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-decoy-psms-1"><i class="fa fa-check"></i><b>1.2.9</b> Remove Decoy PSMs</a></li>
<li class="chapter" data-level="1.2.10" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#map-sites-to-protein-sequences"><i class="fa fa-check"></i><b>1.2.10</b> Map Sites to Protein Sequences</a></li>
<li class="chapter" data-level="1.2.11" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#read-masic-output-1"><i class="fa fa-check"></i><b>1.2.11</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.2.12" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#filter-masic-data-1"><i class="fa fa-check"></i><b>1.2.12</b> Filter MASIC Data</a></li>
<li class="chapter" data-level="1.2.13" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#create-study-design-tables"><i class="fa fa-check"></i><b>1.2.13</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="1.2.14" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#phospho-create-crosstab"><i class="fa fa-check"></i><b>1.2.14</b> Create Quantitative Cross-tab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="label-free-quantification.html"><a href="label-free-quantification.html"><i class="fa fa-check"></i><b>2</b> Label-Free Quantification</a></li>
<li class="chapter" data-level="3" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>3</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="count-features-in-samples.html"><a href="count-features-in-samples.html"><i class="fa fa-check"></i><b>3.1</b> Count Features in Samples</a></li>
<li class="chapter" data-level="3.2" data-path="estimate-blood-contamination.html"><a href="estimate-blood-contamination.html"><i class="fa fa-check"></i><b>3.2</b> Estimate Blood Contamination</a></li>
<li class="chapter" data-level="3.3" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>3.3</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="heatmaps.html"><a href="heatmaps.html"><i class="fa fa-check"></i><b>4</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="4.1" data-path="expression-heatmaps.html"><a href="expression-heatmaps.html"><i class="fa fa-check"></i><b>4.1</b> Expression Heatmaps</a></li>
<li class="chapter" data-level="4.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html"><i class="fa fa-check"></i><b>4.2</b> Correlation Heatmaps</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#sample-correlation"><i class="fa fa-check"></i><b>4.2.1</b> Sample Correlation</a></li>
<li class="chapter" data-level="4.2.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#feature-correlation"><i class="fa fa-check"></i><b>4.2.2</b> Feature Correlation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html"><i class="fa fa-check"></i><b>4.3</b> Heatmap Annotation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html#modifying-default-colors"><i class="fa fa-check"></i><b>4.3.1</b> Modifying Default Colors</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="modifications.html"><a href="modifications.html"><i class="fa fa-check"></i><b>4.4</b> Modifications</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="modifications.html"><a href="modifications.html#row-and-column-labels"><i class="fa fa-check"></i><b>4.4.1</b> Row and column labels</a></li>
<li class="chapter" data-level="4.4.2" data-path="modifications.html"><a href="modifications.html#label-colors"><i class="fa fa-check"></i><b>4.4.2</b> Label colors</a></li>
<li class="chapter" data-level="4.4.3" data-path="modifications.html"><a href="modifications.html#label-specific-features"><i class="fa fa-check"></i><b>4.4.3</b> Label specific features</a></li>
<li class="chapter" data-level="4.4.4" data-path="modifications.html"><a href="modifications.html#heatmap-body-color"><i class="fa fa-check"></i><b>4.4.4</b> Heatmap body color</a></li>
<li class="chapter" data-level="4.4.5" data-path="modifications.html"><a href="modifications.html#horizontal-heatmaps"><i class="fa fa-check"></i><b>4.4.5</b> Horizontal heatmaps</a></li>
<li class="chapter" data-level="4.4.6" data-path="modifications.html"><a href="modifications.html#legends"><i class="fa fa-check"></i><b>4.4.6</b> Legends</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="linear-modeling.html"><a href="linear-modeling.html"><i class="fa fa-check"></i><b>5</b> Linear Modeling</a>
<ul>
<li class="chapter" data-level="5.1" data-path="mult-comp.html"><a href="mult-comp.html"><i class="fa fa-check"></i><b>5.1</b> Multiple Comparisons</a></li>
<li class="chapter" data-level="5.2" data-path="linear-reg.html"><a href="linear-reg.html"><i class="fa fa-check"></i><b>5.2</b> Linear Regression</a></li>
<li class="chapter" data-level="5.3" data-path="t-tests.html"><a href="t-tests.html"><i class="fa fa-check"></i><b>5.3</b> Two-Sample t-tests</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="t-tests.html"><a href="t-tests.html#t-single"><i class="fa fa-check"></i><b>5.3.1</b> One comparison</a></li>
<li class="chapter" data-level="5.3.2" data-path="t-tests.html"><a href="t-tests.html#t-mult"><i class="fa fa-check"></i><b>5.3.2</b> Multiple comparisons</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="anova.html"><a href="anova.html"><i class="fa fa-check"></i><b>5.4</b> One-Way ANOVA</a></li>
<li class="chapter" data-level="5.5" data-path="p-value-histograms.html"><a href="p-value-histograms.html"><i class="fa fa-check"></i><b>5.5</b> p-value Histograms</a></li>
<li class="chapter" data-level="5.6" data-path="volcano-plots.html"><a href="volcano-plots.html"><i class="fa fa-check"></i><b>5.6</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>6</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="annotation-databases.html"><a href="annotation-databases.html"><i class="fa fa-check"></i><b>6.1</b> Annotation Databases</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#gene-ontology"><i class="fa fa-check"></i><b>6.1.1</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="6.1.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#semantic-similarity"><i class="fa fa-check"></i><b>6.1.1.1</b> Semantic Similarity</a></li>
<li class="chapter" data-level="6.1.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#go-slim"><i class="fa fa-check"></i><b>6.1.1.2</b> GO Subsets/Slims</a></li>
</ul></li>
<li class="chapter" data-level="6.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#reactome"><i class="fa fa-check"></i><b>6.1.2</b> Reactome</a></li>
<li class="chapter" data-level="6.1.3" data-path="annotation-databases.html"><a href="annotation-databases.html#kegg"><i class="fa fa-check"></i><b>6.1.3</b> KEGG</a></li>
<li class="chapter" data-level="6.1.4" data-path="annotation-databases.html"><a href="annotation-databases.html#pfam"><i class="fa fa-check"></i><b>6.1.4</b> Pfam</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="ora.html"><a href="ora.html"><i class="fa fa-check"></i><b>6.2</b> Over-Representation Analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="ora.html"><a href="ora.html#ora-overview"><i class="fa fa-check"></i><b>6.2.1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="ora.html"><a href="ora.html#important-considerations"><i class="fa fa-check"></i>Important Considerations</a></li>
</ul></li>
<li class="chapter" data-level="6.2.2" data-path="ora.html"><a href="ora.html#ora-examples"><i class="fa fa-check"></i><b>6.2.2</b> Examples</a>
<ul>
<li class="chapter" data-level="6.2.2.1" data-path="ora.html"><a href="ora.html#ora-go"><i class="fa fa-check"></i><b>6.2.2.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="6.2.2.2" data-path="ora.html"><a href="ora.html#ora-reactome"><i class="fa fa-check"></i><b>6.2.2.2</b> Reactome</a></li>
<li class="chapter" data-level="6.2.2.3" data-path="ora.html"><a href="ora.html#ora-pfam"><i class="fa fa-check"></i><b>6.2.2.3</b> Pfam</a></li>
<li class="chapter" data-level="6.2.2.4" data-path="ora.html"><a href="ora.html#ora-other"><i class="fa fa-check"></i><b>6.2.2.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="gsea.html"><a href="gsea.html"><i class="fa fa-check"></i><b>6.3</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="gsea.html"><a href="gsea.html#gsea-overview"><i class="fa fa-check"></i><b>6.3.1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="gsea.html"><a href="gsea.html#important-considerations-1"><i class="fa fa-check"></i>Important Considerations</a></li>
</ul></li>
<li class="chapter" data-level="6.3.2" data-path="gsea.html"><a href="gsea.html#examples"><i class="fa fa-check"></i><b>6.3.2</b> Examples</a>
<ul>
<li class="chapter" data-level="6.3.2.1" data-path="gsea.html"><a href="gsea.html#gene-ontology-1"><i class="fa fa-check"></i><b>6.3.2.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="6.3.2.2" data-path="gsea.html"><a href="gsea.html#reactome-1"><i class="fa fa-check"></i><b>6.3.2.2</b> Reactome</a></li>
<li class="chapter" data-level="6.3.2.3" data-path="gsea.html"><a href="gsea.html#pfam-1"><i class="fa fa-check"></i><b>6.3.2.3</b> Pfam</a></li>
<li class="chapter" data-level="6.3.2.4" data-path="gsea.html"><a href="gsea.html#other-databases"><i class="fa fa-check"></i><b>6.3.2.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Proteomics Data Analysis in R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="label-free-quantification" class="section level1" number="2">
<h1><span class="header-section-number">Section 2</span> Label-Free Quantification</h1>
<p>This is a generic spectral counting script for MS-GF+ Human/UniProt searches. Only modify the lines that change the data package number and the name of the final .xlsx file that will be saved, unless you know what you are doing.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="label-free-quantification.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Required packages</span></span>
<span id="cb68-2"><a href="label-free-quantification.html#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PNNL.DMS.utils)</span>
<span id="cb68-3"><a href="label-free-quantification.html#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MSnID)</span>
<span id="cb68-4"><a href="label-free-quantification.html#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PlexedPiper)</span>
<span id="cb68-5"><a href="label-free-quantification.html#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MSnbase)</span>
<span id="cb68-6"><a href="label-free-quantification.html#cb68-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(writexl)</span>
<span id="cb68-7"><a href="label-free-quantification.html#cb68-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb68-8"><a href="label-free-quantification.html#cb68-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="label-free-quantification.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Data package number</span></span>
<span id="cb69-2"><a href="label-free-quantification.html#cb69-2" aria-hidden="true" tabindex="-1"></a>data_package_num <span class="ot">&lt;-</span> <span class="dv">3987</span></span>
<span id="cb69-3"><a href="label-free-quantification.html#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Name of the final file to save</span></span>
<span id="cb69-4"><a href="label-free-quantification.html#cb69-4" aria-hidden="true" tabindex="-1"></a>file_name <span class="ot">&lt;-</span> <span class="st">&quot;data/3987_spectral_counts.xlsx&quot;</span></span></code></pre></div>
<p>Do not modify anything below unless you know what you are doing.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="label-free-quantification.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MS-GF+ results from the DMS</span></span>
<span id="cb70-2"><a href="label-free-quantification.html#cb70-2" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">read_msgf_data_from_DMS</span>(<span class="at">data_package_num =</span> data_package_num)</span>
<span id="cb70-3"><a href="label-free-quantification.html#cb70-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-4"><a href="label-free-quantification.html#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to 1% FDR at the peptide level</span></span>
<span id="cb70-5"><a href="label-free-quantification.html#cb70-5" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">filter_msgf_data</span>(m, <span class="at">level =</span> <span class="st">&quot;peptide&quot;</span>, <span class="at">fdr.max =</span> <span class="fl">0.01</span>)</span>
<span id="cb70-6"><a href="label-free-quantification.html#cb70-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-7"><a href="label-free-quantification.html#cb70-7" aria-hidden="true" tabindex="-1"></a><span class="co"># UniProt to gene symbol conversion table</span></span>
<span id="cb70-8"><a href="label-free-quantification.html#cb70-8" aria-hidden="true" tabindex="-1"></a>conv_tab <span class="ot">&lt;-</span> <span class="fu">fetch_conversion_table</span>(<span class="at">organism_name =</span> <span class="st">&quot;Homo sapiens&quot;</span>, </span>
<span id="cb70-9"><a href="label-free-quantification.html#cb70-9" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">from =</span> <span class="st">&quot;UNIPROT&quot;</span>, <span class="at">to =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span></code></pre></div>
<p>When running <code>fetch_conversion_table</code>, if a prompt appears that requires an answer, type <code>yes</code> and press enter.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="label-free-quantification.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Modify accessions column of psms to use gene symbols</span></span>
<span id="cb71-2"><a href="label-free-quantification.html#cb71-2" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">remap_accessions</span>(m, conv_tab, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">|([^|-]+)(-</span><span class="sc">\\</span><span class="st">d+)?</span><span class="sc">\\</span><span class="st">|&quot;</span>)</span>
<span id="cb71-3"><a href="label-free-quantification.html#cb71-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-4"><a href="label-free-quantification.html#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Do the same remapping to the FASTA file</span></span>
<span id="cb71-5"><a href="label-free-quantification.html#cb71-5" aria-hidden="true" tabindex="-1"></a>fst_path <span class="ot">&lt;-</span> <span class="fu">path_to_FASTA_used_by_DMS</span>(<span class="at">data_package_num =</span> data_package_num)</span>
<span id="cb71-6"><a href="label-free-quantification.html#cb71-6" aria-hidden="true" tabindex="-1"></a>fst_path_2 <span class="ot">&lt;-</span> <span class="fu">remap_fasta_entry_names</span>(</span>
<span id="cb71-7"><a href="label-free-quantification.html#cb71-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">path_to_FASTA =</span> fst_path, <span class="at">conversion_table =</span> conv_tab, </span>
<span id="cb71-8"><a href="label-free-quantification.html#cb71-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">extraction_pttrn =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">|([^|-]+)(-</span><span class="sc">\\</span><span class="st">d+)?</span><span class="sc">\\</span><span class="st">|&quot;</span></span>
<span id="cb71-9"><a href="label-free-quantification.html#cb71-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb71-10"><a href="label-free-quantification.html#cb71-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-11"><a href="label-free-quantification.html#cb71-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute the number of amino acids per 1000 and use that to filter</span></span>
<span id="cb71-12"><a href="label-free-quantification.html#cb71-12" aria-hidden="true" tabindex="-1"></a><span class="co"># to 1% FDR at the protein level</span></span>
<span id="cb71-13"><a href="label-free-quantification.html#cb71-13" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">compute_num_peptides_per_1000aa</span>(m, fst_path_2)</span>
<span id="cb71-14"><a href="label-free-quantification.html#cb71-14" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">filter_msgf_data</span>(m, <span class="st">&quot;accession&quot;</span>, <span class="at">fdr.max =</span> <span class="fl">0.01</span>)</span>
<span id="cb71-15"><a href="label-free-quantification.html#cb71-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-16"><a href="label-free-quantification.html#cb71-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Parsimonious protein inference</span></span>
<span id="cb71-17"><a href="label-free-quantification.html#cb71-17" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">infer_parsimonious_accessions</span>(m)</span>
<span id="cb71-18"><a href="label-free-quantification.html#cb71-18" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(m) <span class="co"># Assessment of filtering quality</span></span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  8 
## #PSMs: 87392 at 0.044 % FDR
## #peptides: 21634 at 0.12 % FDR
## #accessions: 2421 at 1 % FDR</code></pre>
<p>The results look reasonable, so we will continue on to spectral counting.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="label-free-quantification.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove decoys</span></span>
<span id="cb73-2"><a href="label-free-quantification.html#cb73-2" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">apply_filter</span>(m, <span class="st">&quot;!isDecoy&quot;</span>)</span>
<span id="cb73-3"><a href="label-free-quantification.html#cb73-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-4"><a href="label-free-quantification.html#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert m to an MSnSet</span></span>
<span id="cb73-5"><a href="label-free-quantification.html#cb73-5" aria-hidden="true" tabindex="-1"></a>msnset <span class="ot">&lt;-</span> <span class="fu">as</span>(m, <span class="st">&quot;MSnSet&quot;</span>)</span>
<span id="cb73-6"><a href="label-free-quantification.html#cb73-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-7"><a href="label-free-quantification.html#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Spectral counting:</span></span>
<span id="cb73-8"><a href="label-free-quantification.html#cb73-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Within each accession group, sum the values within columns.</span></span>
<span id="cb73-9"><a href="label-free-quantification.html#cb73-9" aria-hidden="true" tabindex="-1"></a>msnset <span class="ot">&lt;-</span> <span class="fu">combineFeatures</span>(msnset,</span>
<span id="cb73-10"><a href="label-free-quantification.html#cb73-10" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">fData</span>(msnset)<span class="sc">$</span>accession,</span>
<span id="cb73-11"><a href="label-free-quantification.html#cb73-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">redundancy.handler =</span> <span class="st">&quot;multiple&quot;</span>,</span>
<span id="cb73-12"><a href="label-free-quantification.html#cb73-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">method =</span> <span class="st">&quot;sum&quot;</span>,</span>
<span id="cb73-13"><a href="label-free-quantification.html#cb73-13" aria-hidden="true" tabindex="-1"></a>                          <span class="at">cv =</span> <span class="cn">FALSE</span>)</span>
<span id="cb73-14"><a href="label-free-quantification.html#cb73-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-15"><a href="label-free-quantification.html#cb73-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort features from most to least abundant</span></span>
<span id="cb73-16"><a href="label-free-quantification.html#cb73-16" aria-hidden="true" tabindex="-1"></a>tot_count <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">exprs</span>(msnset))</span>
<span id="cb73-17"><a href="label-free-quantification.html#cb73-17" aria-hidden="true" tabindex="-1"></a>msnset <span class="ot">&lt;-</span> msnset[<span class="fu">order</span>(<span class="sc">-</span>tot_count), ]</span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="label-free-quantification.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save exprs as an .xlsx file</span></span>
<span id="cb74-2"><a href="label-free-quantification.html#cb74-2" aria-hidden="true" tabindex="-1"></a>msnset <span class="sc">%&gt;%</span></span>
<span id="cb74-3"><a href="label-free-quantification.html#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">exprs</span>() <span class="sc">%&gt;%</span></span>
<span id="cb74-4"><a href="label-free-quantification.html#cb74-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb74-5"><a href="label-free-quantification.html#cb74-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb74-6"><a href="label-free-quantification.html#cb74-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_xlsx</span>(<span class="at">path =</span> file_name)</span></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="phosphoproteomics-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="exploratory-data-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
