<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.4 Modifications | Proteomics Data Analysis in R/Bioconductor</title>
  <meta name="description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="6.4 Modifications | Proteomics Data Analysis in R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.4 Modifications | Proteomics Data Analysis in R/Bioconductor" />
  
  <meta name="twitter:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

<meta name="author" content="Tyler Sagendorf" />


<meta name="date" content="2022-02-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="heatmap-annotation.html"/>
<link rel="next" href="DEA.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=".">Proteomics Data Analysis in R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="iso-global.html"><a href="iso-global.html"><i class="fa fa-check"></i><b>1</b> Isobaric Quantification: Proteomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html"><i class="fa fa-check"></i><b>1.1</b> Prepare MS/MS Identifications</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#read-ms-gf-data"><i class="fa fa-check"></i><b>1.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="1.1.2" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#correct-isotope-selection-error"><i class="fa fa-check"></i><b>1.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.1.3" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#remove-contaminants"><i class="fa fa-check"></i><b>1.1.3</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.1.4" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#global-peptide-filter"><i class="fa fa-check"></i><b>1.1.4</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.1.5" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#msms-id-filter-protein-level"><i class="fa fa-check"></i><b>1.1.5</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.1.6" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#inference-of-parsimonious-protein-set"><i class="fa fa-check"></i><b>1.1.6</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.1.7" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#remove-decoy-psms"><i class="fa fa-check"></i><b>1.1.7</b> Remove Decoy PSMs</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html"><i class="fa fa-check"></i><b>1.2</b> Prepare Reporter Ion Intensities</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html#read-masic-output"><i class="fa fa-check"></i><b>1.2.1</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.2.2" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html#filter-masic-data"><i class="fa fa-check"></i><b>1.2.2</b> Filter MASIC Data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html"><i class="fa fa-check"></i><b>1.3</b> Create Study Design Tables</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#fractions"><i class="fa fa-check"></i><b>1.3.1</b> Fractions</a></li>
<li class="chapter" data-level="1.3.2" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#samples"><i class="fa fa-check"></i><b>1.3.2</b> Samples</a></li>
<li class="chapter" data-level="1.3.3" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#global-references"><i class="fa fa-check"></i><b>1.3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="global-quant-crosstab.html"><a href="global-quant-crosstab.html"><i class="fa fa-check"></i><b>1.4</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="1.5" data-path="global-msnset.html"><a href="global-msnset.html"><i class="fa fa-check"></i><b>1.5</b> Create MSnSet</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="iso-phospho.html"><a href="iso-phospho.html"><i class="fa fa-check"></i><b>2</b> Isobaric Quantification: Phosphoproteomics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html"><i class="fa fa-check"></i><b>2.1</b> Prepare MS/MS Identifications</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#read-ms-gf-data-1"><i class="fa fa-check"></i><b>2.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="2.1.2" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#correct-isotope-selection-error-1"><i class="fa fa-check"></i><b>2.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="2.1.3" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-unmodified-peptides"><i class="fa fa-check"></i><b>2.1.3</b> Remove Unmodified Peptides</a></li>
<li class="chapter" data-level="2.1.4" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-contaminants-1"><i class="fa fa-check"></i><b>2.1.4</b> Remove Contaminants</a></li>
<li class="chapter" data-level="2.1.5" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#improve-phosphosite-localization"><i class="fa fa-check"></i><b>2.1.5</b> Improve Phosphosite Localization</a></li>
<li class="chapter" data-level="2.1.6" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#msms-id-filter-peptide-level"><i class="fa fa-check"></i><b>2.1.6</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="2.1.7" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#msms-id-filter-protein-level-1"><i class="fa fa-check"></i><b>2.1.7</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="2.1.8" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#inference-of-parsimonious-protein-set-1"><i class="fa fa-check"></i><b>2.1.8</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="2.1.9" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#map-mod-sites"><i class="fa fa-check"></i><b>2.1.9</b> Map Sites to Protein Sequences</a></li>
<li class="chapter" data-level="2.1.10" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-decoy-psms-1"><i class="fa fa-check"></i><b>2.1.10</b> Remove Decoy PSMs</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html"><i class="fa fa-check"></i><b>2.2</b> Prepare Reporter Ion Intensities</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html#read-masic-output-1"><i class="fa fa-check"></i><b>2.2.1</b> Read MASIC Output</a></li>
<li class="chapter" data-level="2.2.2" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html#filter-masic-data-1"><i class="fa fa-check"></i><b>2.2.2</b> Filter MASIC Data</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="create-study-design-tables.html"><a href="create-study-design-tables.html"><i class="fa fa-check"></i><b>2.3</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="2.4" data-path="create-quantitative-cross-tab.html"><a href="create-quantitative-cross-tab.html"><i class="fa fa-check"></i><b>2.4</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="2.5" data-path="phospho-msnset.html"><a href="phospho-msnset.html"><i class="fa fa-check"></i><b>2.5</b> Create MSnSet</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spectral-counting.html"><a href="spectral-counting.html"><i class="fa fa-check"></i><b>3</b> Spectral Counting</a></li>
<li class="chapter" data-level="4" data-path="id-conversion.html"><a href="id-conversion.html"><i class="fa fa-check"></i><b>4</b> Feature ID Conversion</a>
<ul>
<li class="chapter" data-level="4.1" data-path="conversion-with-biomart.html"><a href="conversion-with-biomart.html"><i class="fa fa-check"></i><b>4.1</b> Conversion with <code>biomaRt</code></a></li>
<li class="chapter" data-level="4.2" data-path="conversion-with-annotationhub.html"><a href="conversion-with-annotationhub.html"><i class="fa fa-check"></i><b>4.2</b> Conversion with <code>AnnotationHub</code></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>5</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="count-features-in-samples.html"><a href="count-features-in-samples.html"><i class="fa fa-check"></i><b>5.1</b> Count Features in Samples</a></li>
<li class="chapter" data-level="5.2" data-path="sample-boxplots.html"><a href="sample-boxplots.html"><i class="fa fa-check"></i><b>5.2</b> Sample Boxplots</a></li>
<li class="chapter" data-level="5.3" data-path="estimate-blood-contamination.html"><a href="estimate-blood-contamination.html"><i class="fa fa-check"></i><b>5.3</b> Estimate Blood Contamination</a></li>
<li class="chapter" data-level="5.4" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>5.4</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="heatmaps.html"><a href="heatmaps.html"><i class="fa fa-check"></i><b>6</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="6.1" data-path="expression-heatmaps.html"><a href="expression-heatmaps.html"><i class="fa fa-check"></i><b>6.1</b> Expression Heatmaps</a></li>
<li class="chapter" data-level="6.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html"><i class="fa fa-check"></i><b>6.2</b> Correlation Heatmaps</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#sample-correlation"><i class="fa fa-check"></i><b>6.2.1</b> Sample Correlation</a></li>
<li class="chapter" data-level="6.2.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#feature-correlation"><i class="fa fa-check"></i><b>6.2.2</b> Feature Correlation</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html"><i class="fa fa-check"></i><b>6.3</b> Heatmap Annotation</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html#modifying-default-colors"><i class="fa fa-check"></i><b>6.3.1</b> Modifying Default Colors</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="modifications.html"><a href="modifications.html"><i class="fa fa-check"></i><b>6.4</b> Modifications</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="modifications.html"><a href="modifications.html#row-and-column-labels"><i class="fa fa-check"></i><b>6.4.1</b> Row and column labels</a></li>
<li class="chapter" data-level="6.4.2" data-path="modifications.html"><a href="modifications.html#label-colors"><i class="fa fa-check"></i><b>6.4.2</b> Label colors</a></li>
<li class="chapter" data-level="6.4.3" data-path="modifications.html"><a href="modifications.html#label-specific-features"><i class="fa fa-check"></i><b>6.4.3</b> Label specific features</a></li>
<li class="chapter" data-level="6.4.4" data-path="modifications.html"><a href="modifications.html#heatmap-body-color"><i class="fa fa-check"></i><b>6.4.4</b> Heatmap body color</a></li>
<li class="chapter" data-level="6.4.5" data-path="modifications.html"><a href="modifications.html#horizontal-heatmaps"><i class="fa fa-check"></i><b>6.4.5</b> Horizontal heatmaps</a></li>
<li class="chapter" data-level="6.4.6" data-path="modifications.html"><a href="modifications.html#legends"><i class="fa fa-check"></i><b>6.4.6</b> Legends</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="DEA.html"><a href="DEA.html"><i class="fa fa-check"></i><b>7</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="linear-reg.html"><a href="linear-reg.html"><i class="fa fa-check"></i><b>7.1</b> Linear Regression</a></li>
<li class="chapter" data-level="7.2" data-path="t-tests.html"><a href="t-tests.html"><i class="fa fa-check"></i><b>7.2</b> Two-Sample t-tests</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="t-tests.html"><a href="t-tests.html#t-single"><i class="fa fa-check"></i><b>7.2.1</b> One comparison</a></li>
<li class="chapter" data-level="7.2.2" data-path="t-tests.html"><a href="t-tests.html#t-mult"><i class="fa fa-check"></i><b>7.2.2</b> Multiple comparisons</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="anova.html"><a href="anova.html"><i class="fa fa-check"></i><b>7.3</b> One-Way ANOVA</a></li>
<li class="chapter" data-level="7.4" data-path="p-value-histograms.html"><a href="p-value-histograms.html"><i class="fa fa-check"></i><b>7.4</b> p-value Histograms</a></li>
<li class="chapter" data-level="7.5" data-path="volcano-plots.html"><a href="volcano-plots.html"><i class="fa fa-check"></i><b>7.5</b> Volcano Plots</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="volcano-plots.html"><a href="volcano-plots.html#base-plot"><i class="fa fa-check"></i><b>7.5.1</b> Base plot</a></li>
<li class="chapter" data-level="7.5.2" data-path="volcano-plots.html"><a href="volcano-plots.html#label-top-features"><i class="fa fa-check"></i><b>7.5.2</b> Label top features</a></li>
<li class="chapter" data-level="7.5.3" data-path="volcano-plots.html"><a href="volcano-plots.html#label-specific-features-1"><i class="fa fa-check"></i><b>7.5.3</b> Label specific features</a></li>
<li class="chapter" data-level="7.5.4" data-path="volcano-plots.html"><a href="volcano-plots.html#modify-point-colors"><i class="fa fa-check"></i><b>7.5.4</b> Modify point colors</a></li>
<li class="chapter" data-level="7.5.5" data-path="volcano-plots.html"><a href="volcano-plots.html#multiple-volcano-plots"><i class="fa fa-check"></i><b>7.5.5</b> Multiple volcano plots</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="upset-plots.html"><a href="upset-plots.html"><i class="fa fa-check"></i><b>7.6</b> UpSet Plots</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>8</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="annotation-databases.html"><a href="annotation-databases.html"><i class="fa fa-check"></i><b>8.1</b> Annotation Databases</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#gene-ontology"><i class="fa fa-check"></i><b>8.1.1</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="8.1.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#semantic-similarity"><i class="fa fa-check"></i><b>8.1.1.1</b> Semantic Similarity</a></li>
<li class="chapter" data-level="8.1.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#go-slim"><i class="fa fa-check"></i><b>8.1.1.2</b> GO Subsets/Slims</a></li>
</ul></li>
<li class="chapter" data-level="8.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#reactome"><i class="fa fa-check"></i><b>8.1.2</b> Reactome</a></li>
<li class="chapter" data-level="8.1.3" data-path="annotation-databases.html"><a href="annotation-databases.html#kegg"><i class="fa fa-check"></i><b>8.1.3</b> KEGG</a></li>
<li class="chapter" data-level="8.1.4" data-path="annotation-databases.html"><a href="annotation-databases.html#pfam"><i class="fa fa-check"></i><b>8.1.4</b> Pfam</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="ora.html"><a href="ora.html"><i class="fa fa-check"></i><b>8.2</b> Over-Representation Analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ora.html"><a href="ora.html#ora-examples"><i class="fa fa-check"></i><b>8.2.1</b> Examples</a>
<ul>
<li class="chapter" data-level="8.2.1.1" data-path="ora.html"><a href="ora.html#ora-go"><i class="fa fa-check"></i><b>8.2.1.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="8.2.1.2" data-path="ora.html"><a href="ora.html#ora-reactome"><i class="fa fa-check"></i><b>8.2.1.2</b> Reactome</a></li>
<li class="chapter" data-level="8.2.1.3" data-path="ora.html"><a href="ora.html#ora-pfam"><i class="fa fa-check"></i><b>8.2.1.3</b> Pfam</a></li>
<li class="chapter" data-level="8.2.1.4" data-path="ora.html"><a href="ora.html#ora-other"><i class="fa fa-check"></i><b>8.2.1.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="gsea.html"><a href="gsea.html"><i class="fa fa-check"></i><b>8.3</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gsea.html"><a href="gsea.html#examples"><i class="fa fa-check"></i><b>8.3.1</b> Examples</a>
<ul>
<li class="chapter" data-level="8.3.1.1" data-path="gsea.html"><a href="gsea.html#gene-ontology-1"><i class="fa fa-check"></i><b>8.3.1.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="8.3.1.2" data-path="gsea.html"><a href="gsea.html#reactome-1"><i class="fa fa-check"></i><b>8.3.1.2</b> Reactome</a></li>
<li class="chapter" data-level="8.3.1.3" data-path="gsea.html"><a href="gsea.html#pfam-1"><i class="fa fa-check"></i><b>8.3.1.3</b> Pfam</a></li>
<li class="chapter" data-level="8.3.1.4" data-path="gsea.html"><a href="gsea.html#gsea-other"><i class="fa fa-check"></i><b>8.3.1.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Proteomics Data Analysis in R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="modifications" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Modifications</h2>
<p><code>heatmap_args</code> and <code>anno_args</code> are used to modify the heatmaps: changing the row and column labels, labeling specific features, changing the colors of labels, changing font size, adding different types of annotations, splitting rows or columns into groups, etc. The <a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/">ComplexHeatmap Complete Reference</a> goes more into detail about each of these modifications, but we will cover a few of them in this section.</p>
<p>We will use a random subset of the MSnSet to explore some of these modifications. This is just so we can see the row and column names more easily.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="modifications.html#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">99</span>)</span>
<span id="cb102-2"><a href="modifications.html#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="co"># subset to 25 features and 40 samples</span></span>
<span id="cb102-3"><a href="modifications.html#cb102-3" aria-hidden="true" tabindex="-1"></a>m_sub <span class="ot">&lt;-</span> m[<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(m), <span class="at">size =</span> <span class="dv">25</span>), </span>
<span id="cb102-4"><a href="modifications.html#cb102-4" aria-hidden="true" tabindex="-1"></a>           <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(m), <span class="at">size =</span> <span class="dv">40</span>)]</span></code></pre></div>
<div id="row-and-column-labels" class="section level3" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Row and column labels</h3>
<p>By default, the row and column labels are the row and column names of the matrix passed to <code>Heatmap</code>. In the case of the default expression heatmap, the row names are the <code>featureNames</code> of the MSnSet and the column names are the <code>sampleNames</code>. We can instead use any column in <code>fData</code> and any column in <code>pData</code> to label the rows and columns, respectively. Duplicate labels are also allowed, which is especially useful if there are multiple peptides that map to the same protein, multiple proteins that map to the same gene, etc.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="modifications.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Default row labels - peptides</span></span>
<span id="cb103-2"><a href="modifications.html#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m_sub, <span class="at">show_row_names =</span> <span class="cn">TRUE</span>, </span>
<span id="cb103-3"><a href="modifications.html#cb103-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">heatmap_title =</span> <span class="st">&quot;Default Row Labels&quot;</span>)</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-9-1.png" width="75%" style="display: block; margin: auto;" /></p>
<p>Suppose, for example, we want to label the rows by the Protein column of <code>fData(m_sub)</code>. We can do this by modifying the <code>row_labels</code> argument of <code>ComplexHeatmap::Heatmap</code> with the <code>heatmap_args</code> list. We will also reduce the font size of the labels so that they do not overlap.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="modifications.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Label rows with proteins and change font size</span></span>
<span id="cb104-2"><a href="modifications.html#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m_sub, <span class="at">show_row_names =</span> <span class="cn">TRUE</span>, </span>
<span id="cb104-3"><a href="modifications.html#cb104-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">heatmap_title =</span> <span class="st">&quot;New Row Labels&quot;</span>,</span>
<span id="cb104-4"><a href="modifications.html#cb104-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">heatmap_args =</span> <span class="fu">list</span>(<span class="at">row_labels =</span> <span class="fu">fData</span>(m_sub)[[<span class="st">&quot;Protein&quot;</span>]],</span>
<span id="cb104-5"><a href="modifications.html#cb104-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="co"># Change font size of row labels</span></span>
<span id="cb104-6"><a href="modifications.html#cb104-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>)))</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-10-1.png" width="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="label-colors" class="section level3" number="6.4.2">
<h3><span class="header-section-number">6.4.2</span> Label colors</h3>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="modifications.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># One color</span></span>
<span id="cb105-2"><a href="modifications.html#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m_sub, <span class="at">show_row_names =</span> <span class="cn">TRUE</span>, </span>
<span id="cb105-3"><a href="modifications.html#cb105-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">heatmap_args =</span> <span class="fu">list</span>(<span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">&quot;orange&quot;</span>)))</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-11-1.png" width="75%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="modifications.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Multiple colors</span></span>
<span id="cb106-2"><a href="modifications.html#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="co"># If peptide begins with &quot;A&quot;, color it red</span></span>
<span id="cb106-3"><a href="modifications.html#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="co"># If peptide begins with &quot;G&quot;, color it blue</span></span>
<span id="cb106-4"><a href="modifications.html#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Otherwise, color it black</span></span>
<span id="cb106-5"><a href="modifications.html#cb106-5" aria-hidden="true" tabindex="-1"></a>row_colors <span class="ot">&lt;-</span> <span class="fu">featureNames</span>(m_sub) <span class="sc">%&gt;%</span> </span>
<span id="cb106-6"><a href="modifications.html#cb106-6" aria-hidden="true" tabindex="-1"></a>  {<span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">&quot;^A&quot;</span>, .) <span class="sc">~</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb106-7"><a href="modifications.html#cb106-7" aria-hidden="true" tabindex="-1"></a>             <span class="fu">grepl</span>(<span class="st">&quot;^G&quot;</span>, .) <span class="sc">~</span> <span class="st">&quot;blue&quot;</span>,</span>
<span id="cb106-8"><a href="modifications.html#cb106-8" aria-hidden="true" tabindex="-1"></a>             <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;black&quot;</span>)}</span>
<span id="cb106-9"><a href="modifications.html#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m_sub, <span class="at">show_row_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb106-10"><a href="modifications.html#cb106-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">heatmap_args =</span> <span class="fu">list</span>(</span>
<span id="cb106-11"><a href="modifications.html#cb106-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> row_colors)))</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-12-1.png" width="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="label-specific-features" class="section level3" number="6.4.3">
<h3><span class="header-section-number">6.4.3</span> Label specific features</h3>
<p>We can use mark annotation to label specific features or samples. For this example, we will label all non-human proteins. We need the indices of the proteins to label and the column in <code>fData</code> used to select these labels.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="modifications.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Indices of non-human proteins</span></span>
<span id="cb107-2"><a href="modifications.html#cb107-2" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;HUMAN&quot;</span>, <span class="fu">fData</span>(m)[[<span class="st">&quot;Protein&quot;</span>]]))</span>
<span id="cb107-3"><a href="modifications.html#cb107-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-4"><a href="modifications.html#cb107-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Row annotation object. The name can be anything, so we just use anno</span></span>
<span id="cb107-5"><a href="modifications.html#cb107-5" aria-hidden="true" tabindex="-1"></a>ra <span class="ot">&lt;-</span> <span class="fu">rowAnnotation</span>(<span class="at">anno =</span> <span class="fu">anno_mark</span>(<span class="at">at =</span> idx, </span>
<span id="cb107-6"><a href="modifications.html#cb107-6" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">labels =</span> <span class="fu">fData</span>(m)[[<span class="st">&quot;Protein&quot;</span>]][idx]))</span>
<span id="cb107-7"><a href="modifications.html#cb107-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-8"><a href="modifications.html#cb107-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmap with labels for select features</span></span>
<span id="cb107-9"><a href="modifications.html#cb107-9" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m, <span class="at">heatmap_args =</span> <span class="fu">list</span>(<span class="at">row_labels =</span> <span class="fu">fData</span>(m)[[<span class="st">&quot;Protein&quot;</span>]],</span>
<span id="cb107-10"><a href="modifications.html#cb107-10" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">right_annotation =</span> ra))</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-13-1.png" width="75%" style="display: block; margin: auto;" />
<!---
### Split rows or columns


```r
# Split row dendrogram into 3 groups and column dengrogram into 2 groups
complex_heatmap(m, color_range = c(-1.5, 1.5), 
heatmap_args = list(row_split = 3, column_split = 2))
```

<img src="heatmaps_files/figure-html/unnamed-chunk-14-1.png" width="75%" style="display: block; margin: auto;" />



```r
# Do not label split groups
complex_heatmap(m, color_range = c(-1.5, 1.5), 
heatmap_args = list(
row_split = 3, column_split = 2, 
row_title = NULL, column_title = NULL))
```

<img src="heatmaps_files/figure-html/unnamed-chunk-15-1.png" width="75%" style="display: block; margin: auto;" />

---></p>
</div>
<div id="heatmap-body-color" class="section level3" number="6.4.4">
<h3><span class="header-section-number">6.4.4</span> Heatmap body color</h3>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="modifications.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m, <span class="at">heatmap_args =</span> <span class="fu">list</span>(<span class="at">col =</span> circlize<span class="sc">::</span><span class="fu">colorRamp2</span>(</span>
<span id="cb108-2"><a href="modifications.html#cb108-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">breaks =</span> <span class="fu">c</span>(<span class="fu">min</span>(<span class="fu">exprs</span>(m), <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="dv">0</span>, </span>
<span id="cb108-3"><a href="modifications.html#cb108-3" aria-hidden="true" tabindex="-1"></a>             <span class="fu">max</span>(<span class="fu">exprs</span>(m), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb108-4"><a href="modifications.html#cb108-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;purple&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;orange&quot;</span>)</span>
<span id="cb108-5"><a href="modifications.html#cb108-5" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb108-6"><a href="modifications.html#cb108-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-16-1.png" width="75%" style="display: block; margin: auto;" /></p>
<p>If changing the colors of the heatmap body, <code>color_range</code> will not work. Instead, the breaks and colors need to be modified so that the minimum value and the lower limit are the same color and the maximum value and the upper limit are the same color.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="modifications.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m, <span class="at">heatmap_args =</span> <span class="fu">list</span>(<span class="at">col =</span> circlize<span class="sc">::</span><span class="fu">colorRamp2</span>(</span>
<span id="cb109-2"><a href="modifications.html#cb109-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">breaks =</span> <span class="fu">c</span>(<span class="fu">min</span>(<span class="fu">exprs</span>(m), <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb109-3"><a href="modifications.html#cb109-3" aria-hidden="true" tabindex="-1"></a>             <span class="sc">-</span><span class="fl">1.5</span>, <span class="dv">0</span>, <span class="fl">1.5</span>, <span class="co"># add color limits</span></span>
<span id="cb109-4"><a href="modifications.html#cb109-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">max</span>(<span class="fu">exprs</span>(m), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb109-5"><a href="modifications.html#cb109-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;purple&quot;</span>, <span class="st">&quot;purple&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;orange&quot;</span>)</span>
<span id="cb109-6"><a href="modifications.html#cb109-6" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb109-7"><a href="modifications.html#cb109-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-17-1.png" width="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="horizontal-heatmaps" class="section level3" number="6.4.5">
<h3><span class="header-section-number">6.4.5</span> Horizontal heatmaps</h3>
<p>To create a horizontal heatmap, we need to take the transpose of the MSnSet. <em>This will switch the phenoData and featureData</em>. We will also reverse the order of the rows prior to the transpose so that the heatmap would appear to be rotated 90 degrees if samples were not clustered.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="modifications.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Rotate MSnSet</span></span>
<span id="cb110-2"><a href="modifications.html#cb110-2" aria-hidden="true" tabindex="-1"></a>m_rot <span class="ot">&lt;-</span> <span class="fu">t</span>(m[, <span class="fu">ncol</span>(m)<span class="sc">:</span><span class="dv">1</span>])</span>
<span id="cb110-3"><a href="modifications.html#cb110-3" aria-hidden="true" tabindex="-1"></a>m_rot</span></code></pre></div>
<pre><code>## MSnSet (storageMode: lockedEnvironment)
## assayData: 236 features, 300 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: NTVISVFGASGDLAK TFPALFGLFR ... LLAEPVPGIK (300 total)
##   varLabels: Organism Protein Peptide isSpike
##   varMetadata: labelDescription
## featureData
##   featureNames: QC.24 QC.23 ... X03_C_B (236 total)
##   fvarLabels: Sample isQC ... Age (12 total)
##   fvarMetadata: labelDescription
## experimentData: use &#39;experimentData(object)&#39;
## Annotation:  
## - - - Processing information - - -
## Subset [308,236][300,236] Fri Nov  9 15:22:49 2018 
## Subset [300,236][300,236] Fri Feb 11 23:08:12 2022 
## MSnSet transposed [Fri Feb 11 23:08:12 2022] 
##  MSnbase version: 2.7.11</code></pre>
<p>We will annotate rows to show that it worked.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="modifications.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Horizontal heatmap with row annotations</span></span>
<span id="cb112-2"><a href="modifications.html#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m_rot, <span class="at">anno_row =</span> <span class="st">&quot;Age&quot;</span>)</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-19-1.png" width="75%" style="display: block; margin: auto;" /></p>
</div>
<div id="legends" class="section level3" number="6.4.6">
<h3><span class="header-section-number">6.4.6</span> Legends</h3>
<p>Modifying legends with <code>draw_args</code>, <code>heatmap_args</code>, and <code>anno_args</code>. Below, we change the direction of the heatmap legend and continuous annotation legends to horizontal, change position of annotation titles to top left, and set the number of rows for discrete annotation legends to 1. We also change the width of the “Age” legend to 28 mm, move the legends to the bottom of the heatmap, and set the space between the legends to 10 mm.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="modifications.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Horizontal legends at the bottom</span></span>
<span id="cb113-2"><a href="modifications.html#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="fu">complex_heatmap</span>(m, <span class="at">anno_column =</span> <span class="fu">c</span>(<span class="st">&quot;Type&quot;</span>, <span class="st">&quot;Age&quot;</span>), </span>
<span id="cb113-3"><a href="modifications.html#cb113-3" aria-hidden="true" tabindex="-1"></a>                <span class="co"># horizontal heatmap legend</span></span>
<span id="cb113-4"><a href="modifications.html#cb113-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">heatmap_args =</span> <span class="fu">list</span>(</span>
<span id="cb113-5"><a href="modifications.html#cb113-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>)</span>
<span id="cb113-6"><a href="modifications.html#cb113-6" aria-hidden="true" tabindex="-1"></a>                ),</span>
<span id="cb113-7"><a href="modifications.html#cb113-7" aria-hidden="true" tabindex="-1"></a>                <span class="co"># horizontal annotation legend</span></span>
<span id="cb113-8"><a href="modifications.html#cb113-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">anno_args =</span> <span class="fu">list</span>(</span>
<span id="cb113-9"><a href="modifications.html#cb113-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">annotation_legend_param =</span> <span class="fu">list</span>(</span>
<span id="cb113-10"><a href="modifications.html#cb113-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">title_position =</span> <span class="st">&quot;lefttop&quot;</span>,</span>
<span id="cb113-11"><a href="modifications.html#cb113-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">legend_width =</span> <span class="fu">unit</span>(<span class="dv">28</span>, <span class="st">&quot;mm&quot;</span>),</span>
<span id="cb113-12"><a href="modifications.html#cb113-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>, <span class="co"># for continuous legends</span></span>
<span id="cb113-13"><a href="modifications.html#cb113-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">nrow =</span> <span class="dv">1</span> <span class="co"># for discrete legends</span></span>
<span id="cb113-14"><a href="modifications.html#cb113-14" aria-hidden="true" tabindex="-1"></a>                  )</span>
<span id="cb113-15"><a href="modifications.html#cb113-15" aria-hidden="true" tabindex="-1"></a>                ),</span>
<span id="cb113-16"><a href="modifications.html#cb113-16" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Place legends at bottom of heatmap</span></span>
<span id="cb113-17"><a href="modifications.html#cb113-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">draw_args =</span> <span class="fu">list</span>(<span class="at">heatmap_legend_side =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb113-18"><a href="modifications.html#cb113-18" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">legend_gap =</span> <span class="fu">unit</span>(<span class="dv">10</span>, <span class="st">&quot;mm&quot;</span>))</span>
<span id="cb113-19"><a href="modifications.html#cb113-19" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="heatmaps_files/figure-html/unnamed-chunk-20-1.png" width="75%" style="display: block; margin: auto;" /></p>

</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="heatmap-annotation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="DEA.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
