<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8.2 Over-Representation Analysis | Proteomics Data Analysis in R/Bioconductor</title>
  <meta name="description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="8.2 Over-Representation Analysis | Proteomics Data Analysis in R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8.2 Over-Representation Analysis | Proteomics Data Analysis in R/Bioconductor" />
  
  <meta name="twitter:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

<meta name="author" content="Tyler Sagendorf" />


<meta name="date" content="2022-05-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="annotation-databases.html"/>
<link rel="next" href="gsea.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=".">Proteomics Data Analysis in R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="iso-global.html"><a href="iso-global.html"><i class="fa fa-check"></i><b>1</b> Isobaric Quantification: Proteomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html"><i class="fa fa-check"></i><b>1.1</b> Prepare MS/MS Identifications</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#read-ms-gf-data"><i class="fa fa-check"></i><b>1.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="1.1.2" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#correct-isotope-selection-error"><i class="fa fa-check"></i><b>1.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.1.3" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#remove-contaminants"><i class="fa fa-check"></i><b>1.1.3</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.1.4" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#global-peptide-filter"><i class="fa fa-check"></i><b>1.1.4</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.1.5" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#msms-id-filter-protein-level"><i class="fa fa-check"></i><b>1.1.5</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.1.6" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#inference-of-parsimonious-protein-set"><i class="fa fa-check"></i><b>1.1.6</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.1.7" data-path="prepare-msms-identifications.html"><a href="prepare-msms-identifications.html#remove-decoy-psms"><i class="fa fa-check"></i><b>1.1.7</b> Remove Decoy PSMs</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html"><i class="fa fa-check"></i><b>1.2</b> Prepare Reporter Ion Intensities</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html#read-masic-output"><i class="fa fa-check"></i><b>1.2.1</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.2.2" data-path="prepare-reporter-ion-intensities.html"><a href="prepare-reporter-ion-intensities.html#filter-masic-data"><i class="fa fa-check"></i><b>1.2.2</b> Filter MASIC Data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html"><i class="fa fa-check"></i><b>1.3</b> Create Study Design Tables</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#fractions"><i class="fa fa-check"></i><b>1.3.1</b> Fractions</a></li>
<li class="chapter" data-level="1.3.2" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#samples"><i class="fa fa-check"></i><b>1.3.2</b> Samples</a></li>
<li class="chapter" data-level="1.3.3" data-path="fetch-study-design-tables.html"><a href="fetch-study-design-tables.html#global-references"><i class="fa fa-check"></i><b>1.3.3</b> References</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="global-quant-crosstab.html"><a href="global-quant-crosstab.html"><i class="fa fa-check"></i><b>1.4</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="1.5" data-path="global-msnset.html"><a href="global-msnset.html"><i class="fa fa-check"></i><b>1.5</b> Create MSnSet</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="iso-phospho.html"><a href="iso-phospho.html"><i class="fa fa-check"></i><b>2</b> Isobaric Quantification: Phosphoproteomics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html"><i class="fa fa-check"></i><b>2.1</b> Prepare MS/MS Identifications</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#read-ms-gf-data-1"><i class="fa fa-check"></i><b>2.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="2.1.2" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#correct-isotope-selection-error-1"><i class="fa fa-check"></i><b>2.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="2.1.3" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-unmodified-peptides"><i class="fa fa-check"></i><b>2.1.3</b> Remove Unmodified Peptides</a></li>
<li class="chapter" data-level="2.1.4" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-contaminants-1"><i class="fa fa-check"></i><b>2.1.4</b> Remove Contaminants</a></li>
<li class="chapter" data-level="2.1.5" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#improve-phosphosite-localization"><i class="fa fa-check"></i><b>2.1.5</b> Improve Phosphosite Localization</a></li>
<li class="chapter" data-level="2.1.6" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#msms-id-filter-peptide-level"><i class="fa fa-check"></i><b>2.1.6</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="2.1.7" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#msms-id-filter-protein-level-1"><i class="fa fa-check"></i><b>2.1.7</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="2.1.8" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#inference-of-parsimonious-protein-set-1"><i class="fa fa-check"></i><b>2.1.8</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="2.1.9" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#map-mod-sites"><i class="fa fa-check"></i><b>2.1.9</b> Map Sites to Protein Sequences</a></li>
<li class="chapter" data-level="2.1.10" data-path="prepare-msms-identifications-1.html"><a href="prepare-msms-identifications-1.html#remove-decoy-psms-1"><i class="fa fa-check"></i><b>2.1.10</b> Remove Decoy PSMs</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html"><i class="fa fa-check"></i><b>2.2</b> Prepare Reporter Ion Intensities</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html#read-masic-output-1"><i class="fa fa-check"></i><b>2.2.1</b> Read MASIC Output</a></li>
<li class="chapter" data-level="2.2.2" data-path="prepare-reporter-ion-intensities-1.html"><a href="prepare-reporter-ion-intensities-1.html#filter-masic-data-1"><i class="fa fa-check"></i><b>2.2.2</b> Filter MASIC Data</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="create-study-design-tables.html"><a href="create-study-design-tables.html"><i class="fa fa-check"></i><b>2.3</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="2.4" data-path="create-quantitative-cross-tab.html"><a href="create-quantitative-cross-tab.html"><i class="fa fa-check"></i><b>2.4</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="2.5" data-path="phospho-msnset.html"><a href="phospho-msnset.html"><i class="fa fa-check"></i><b>2.5</b> Create MSnSet</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="spectral-counting.html"><a href="spectral-counting.html"><i class="fa fa-check"></i><b>3</b> Spectral Counting</a></li>
<li class="chapter" data-level="4" data-path="id-conversion.html"><a href="id-conversion.html"><i class="fa fa-check"></i><b>4</b> Feature ID Conversion</a>
<ul>
<li class="chapter" data-level="4.1" data-path="conversion-with-biomart.html"><a href="conversion-with-biomart.html"><i class="fa fa-check"></i><b>4.1</b> Conversion with <code>biomaRt</code></a></li>
<li class="chapter" data-level="4.2" data-path="conversion-with-annotationhub.html"><a href="conversion-with-annotationhub.html"><i class="fa fa-check"></i><b>4.2</b> Conversion with <code>AnnotationHub</code></a></li>
<li class="chapter" data-level="4.3" data-path="conversion-using-fasta-headers.html"><a href="conversion-using-fasta-headers.html"><i class="fa fa-check"></i><b>4.3</b> Conversion Using FASTA Headers</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>5</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="count-features-in-samples.html"><a href="count-features-in-samples.html"><i class="fa fa-check"></i><b>5.1</b> Count Features in Samples</a></li>
<li class="chapter" data-level="5.2" data-path="sample-boxplots.html"><a href="sample-boxplots.html"><i class="fa fa-check"></i><b>5.2</b> Sample Boxplots</a></li>
<li class="chapter" data-level="5.3" data-path="estimate-blood-contamination.html"><a href="estimate-blood-contamination.html"><i class="fa fa-check"></i><b>5.3</b> Estimate Blood Contamination</a></li>
<li class="chapter" data-level="5.4" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>5.4</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="heatmaps.html"><a href="heatmaps.html"><i class="fa fa-check"></i><b>6</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="6.1" data-path="expression-heatmaps.html"><a href="expression-heatmaps.html"><i class="fa fa-check"></i><b>6.1</b> Expression Heatmaps</a></li>
<li class="chapter" data-level="6.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html"><i class="fa fa-check"></i><b>6.2</b> Correlation Heatmaps</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#sample-correlation"><i class="fa fa-check"></i><b>6.2.1</b> Sample Correlation</a></li>
<li class="chapter" data-level="6.2.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#feature-correlation"><i class="fa fa-check"></i><b>6.2.2</b> Feature Correlation</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html"><i class="fa fa-check"></i><b>6.3</b> Heatmap Annotation</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html#modifying-default-colors"><i class="fa fa-check"></i><b>6.3.1</b> Modifying Default Colors</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="modifications.html"><a href="modifications.html"><i class="fa fa-check"></i><b>6.4</b> Modifications</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="modifications.html"><a href="modifications.html#row-and-column-labels"><i class="fa fa-check"></i><b>6.4.1</b> Row and column labels</a></li>
<li class="chapter" data-level="6.4.2" data-path="modifications.html"><a href="modifications.html#label-colors"><i class="fa fa-check"></i><b>6.4.2</b> Label colors</a></li>
<li class="chapter" data-level="6.4.3" data-path="modifications.html"><a href="modifications.html#label-specific-features"><i class="fa fa-check"></i><b>6.4.3</b> Label specific features</a></li>
<li class="chapter" data-level="6.4.4" data-path="modifications.html"><a href="modifications.html#heatmap-body-color"><i class="fa fa-check"></i><b>6.4.4</b> Heatmap body color</a></li>
<li class="chapter" data-level="6.4.5" data-path="modifications.html"><a href="modifications.html#horizontal-heatmaps"><i class="fa fa-check"></i><b>6.4.5</b> Horizontal heatmaps</a></li>
<li class="chapter" data-level="6.4.6" data-path="modifications.html"><a href="modifications.html#legends"><i class="fa fa-check"></i><b>6.4.6</b> Legends</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="DEA.html"><a href="DEA.html"><i class="fa fa-check"></i><b>7</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="linear-reg.html"><a href="linear-reg.html"><i class="fa fa-check"></i><b>7.1</b> Linear Regression</a></li>
<li class="chapter" data-level="7.2" data-path="t-tests.html"><a href="t-tests.html"><i class="fa fa-check"></i><b>7.2</b> Two-Sample t-tests</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="t-tests.html"><a href="t-tests.html#t-single"><i class="fa fa-check"></i><b>7.2.1</b> One comparison</a></li>
<li class="chapter" data-level="7.2.2" data-path="t-tests.html"><a href="t-tests.html#t-mult"><i class="fa fa-check"></i><b>7.2.2</b> Multiple comparisons</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="anova.html"><a href="anova.html"><i class="fa fa-check"></i><b>7.3</b> One-Way ANOVA</a></li>
<li class="chapter" data-level="7.4" data-path="p-value-histograms.html"><a href="p-value-histograms.html"><i class="fa fa-check"></i><b>7.4</b> p-value Histograms</a></li>
<li class="chapter" data-level="7.5" data-path="volcano-plots.html"><a href="volcano-plots.html"><i class="fa fa-check"></i><b>7.5</b> Volcano Plots</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="volcano-plots.html"><a href="volcano-plots.html#base-plot"><i class="fa fa-check"></i><b>7.5.1</b> Base plot</a></li>
<li class="chapter" data-level="7.5.2" data-path="volcano-plots.html"><a href="volcano-plots.html#label-top-features"><i class="fa fa-check"></i><b>7.5.2</b> Label top features</a></li>
<li class="chapter" data-level="7.5.3" data-path="volcano-plots.html"><a href="volcano-plots.html#label-specific-features-1"><i class="fa fa-check"></i><b>7.5.3</b> Label specific features</a></li>
<li class="chapter" data-level="7.5.4" data-path="volcano-plots.html"><a href="volcano-plots.html#modify-point-colors"><i class="fa fa-check"></i><b>7.5.4</b> Modify point colors</a></li>
<li class="chapter" data-level="7.5.5" data-path="volcano-plots.html"><a href="volcano-plots.html#multiple-volcano-plots"><i class="fa fa-check"></i><b>7.5.5</b> Multiple volcano plots</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="upset-plots.html"><a href="upset-plots.html"><i class="fa fa-check"></i><b>7.6</b> UpSet Plots</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>8</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="annotation-databases.html"><a href="annotation-databases.html"><i class="fa fa-check"></i><b>8.1</b> Annotation Databases</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#gene-ontology"><i class="fa fa-check"></i><b>8.1.1</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="8.1.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#semantic-similarity"><i class="fa fa-check"></i><b>8.1.1.1</b> Semantic Similarity</a></li>
<li class="chapter" data-level="8.1.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#go-slim"><i class="fa fa-check"></i><b>8.1.1.2</b> GO Subsets/Slims</a></li>
</ul></li>
<li class="chapter" data-level="8.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#reactome"><i class="fa fa-check"></i><b>8.1.2</b> Reactome</a></li>
<li class="chapter" data-level="8.1.3" data-path="annotation-databases.html"><a href="annotation-databases.html#kegg"><i class="fa fa-check"></i><b>8.1.3</b> KEGG</a></li>
<li class="chapter" data-level="8.1.4" data-path="annotation-databases.html"><a href="annotation-databases.html#msigdb"><i class="fa fa-check"></i><b>8.1.4</b> MSigDB</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="ora.html"><a href="ora.html"><i class="fa fa-check"></i><b>8.2</b> Over-Representation Analysis</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="ora.html"><a href="ora.html#ora-overview"><i class="fa fa-check"></i><b>8.2.1</b> Overview</a></li>
<li class="chapter" data-level="8.2.2" data-path="ora.html"><a href="ora.html#ora-math"><i class="fa fa-check"></i><b>8.2.2</b> Mathematical Details</a></li>
<li class="chapter" data-level="8.2.3" data-path="ora.html"><a href="ora.html#ora-drawbacks"><i class="fa fa-check"></i><b>8.2.3</b> Drawbacks</a></li>
<li class="chapter" data-level="8.2.4" data-path="ora.html"><a href="ora.html#ora-examples"><i class="fa fa-check"></i><b>8.2.4</b> Examples</a>
<ul>
<li class="chapter" data-level="8.2.4.1" data-path="ora.html"><a href="ora.html#fgsea-ora"><i class="fa fa-check"></i><b>8.2.4.1</b> ORA with <code>fgsea</code></a></li>
<li class="chapter" data-level="8.2.4.2" data-path="ora.html"><a href="ora.html#ora-clustprof"><i class="fa fa-check"></i><b>8.2.4.2</b> ORA with <code>clusterProfiler</code>/<code>ReactomePA</code></a></li>
<li class="chapter" data-level="8.2.4.3" data-path="ora.html"><a href="ora.html#ora-with-gostats"><i class="fa fa-check"></i><b>8.2.4.3</b> ORA with <code>GOstats</code></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="gsea.html"><a href="gsea.html"><i class="fa fa-check"></i><b>8.3</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="gsea.html"><a href="gsea.html#gsea-overview"><i class="fa fa-check"></i><b>8.3.1</b> Overview</a></li>
<li class="chapter" data-level="8.3.2" data-path="gsea.html"><a href="gsea.html#gsea-examples"><i class="fa fa-check"></i><b>8.3.2</b> Examples</a>
<ul>
<li class="chapter" data-level="8.3.2.1" data-path="gsea.html"><a href="gsea.html#gsea-with-fgsea"><i class="fa fa-check"></i><b>8.3.2.1</b> GSEA with <code>fgsea</code></a></li>
<li class="chapter" data-level="8.3.2.2" data-path="gsea.html"><a href="gsea.html#gsea-with-clusterprofilerreactomepa"><i class="fa fa-check"></i><b>8.3.2.2</b> GSEA with <code>clusterProfiler</code>/<code>ReactomePA</code></a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Proteomics Data Analysis in R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ora" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Over-Representation Analysis<a href="ora.html#ora" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="ora-overview" class="section level3 hasAnchor" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Overview<a href="ora.html#ora-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><span style="color:#3366ff">Over-representation analysis (ORA) is used to determine which <em>a priori</em> defined gene sets are more present (over-represented) in a subset of “interesting” genes than what would be expected by chance</span> <span class="citation">(<a href="#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al., 2009</a>)</span>.</p>
<p>For example, if 10% of all genes being considered are “interesting” (statistically different between conditions, clustered together, etc.), we expect that about 10% of every gene set will be in this “interesting” group of genes. If any gene sets are significantly more present in this group, we would say they are over-represented. Note that <strong>we are not limited to analyzing genes</strong>, though that is how most databases are set up. So long as we have a feature-to-feature-set map, we can perform ORA.</p>
<p><strong>I recommend ORA only when GSEA is not appropriate</strong>, such as for analyzing clusters (from k-means clustering, WGCNA, Mclust, etc.), where the only information available are the group designations. See Section <a href="ora.html#ora-drawbacks">8.2.3</a> (ORA Drawbacks) for details.</p>
</div>
<div id="ora-math" class="section level3 hasAnchor" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Mathematical Details<a href="ora.html#ora-math" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For each gene set, an enrichment p-value is calculated using the Binomial distribution, Hypergeometric distribution, the Fisher exact test, or the Chi-square test. Although this list is not all-encompassing, these are the most popular statistical methods <span class="citation">(<a href="#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al., 2009</a>)</span>. Below is the formula for calculating the enrichment p-value for a particular gene set using the Hypergeometric distribution.</p>
<div class="math">
<p><span class="math display">\[
P(X\geq x) = 1 - P(X \leq x-1) = 1 - \sum\limits_{i=0}^{x-1}\frac{\hphantom{}{M \choose i }{N - M \choose n-i}}{N \choose n}
\]</span></p>
</div>
<p>In this equation, <span class="math inline">\(N\)</span> is the number of background genes, <span class="math inline">\(n\)</span> is the number of “interesting” genes, <span class="math inline">\(M\)</span> is the number of genes that are annotated to a particular gene set <span class="math inline">\(S\)</span>, and <span class="math inline">\(x\)</span> is the number of “interesting” genes that are annotated to <span class="math inline">\(S\)</span>. The numerator of the sum is the number of samples of <span class="math inline">\(n\)</span> genes that can be taken from a population of <span class="math inline">\(N\)</span> genes where exactly <span class="math inline">\(i\)</span> of the genes are annotated to <span class="math inline">\(S\)</span> and <span class="math inline">\(n-i\)</span> are not annotated to <span class="math inline">\(S\)</span>. The denominator of the sum is the total number of samples of size <span class="math inline">\(n\)</span> that can be taken from a population of size <span class="math inline">\(N\)</span>.</p>
<p>For example, suppose we have a list of 8000 genes, of which 400 are members of the same cluster <span class="math inline">\(C\)</span>. Also suppose that 100 of the 8000 genes are annotated to a particular gene set <span class="math inline">\(S\)</span>. Of these 100 genes, 20 are members of <span class="math inline">\(C\)</span>. The probability that 20 or more (up to 100) genes annotated to <span class="math inline">\(S\)</span> are in cluster <span class="math inline">\(C\)</span> by chance is given by</p>
<div class="math">
<p><span class="math display">\[
P(X\geq 20) = 1 - P(X \leq 19) = 1-\sum \limits_{i=0}^{19}\frac{\hphantom{}{100 \choose i}{8000 - 100 \choose 400-i}}{8000 \choose 400} = 7.88 \times 10^{-8}
\]</span></p>
</div>
<p>That is, it is extremely unlikely that 20 of the 100 genes from this set are grouped in cluster <span class="math inline">\(C\)</span> by chance (at least, prior to adjustment for multiple comparisons). The code to calculate this p-value is</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="ora.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">phyper</span>(<span class="at">q =</span> <span class="dv">20</span> <span class="sc">-</span> <span class="dv">1</span>, <span class="at">m =</span> <span class="dv">400</span>, <span class="at">n =</span> <span class="dv">8000</span> <span class="sc">-</span> <span class="dv">400</span>, <span class="at">k =</span> <span class="dv">100</span>, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>After a p-value has been calculated for each of the applicable gene sets, a multiple testing adjustment should be applied.</p>
</div>
<div id="ora-drawbacks" class="section level3 hasAnchor" number="8.2.3">
<h3><span class="header-section-number">8.2.3</span> Drawbacks<a href="ora.html#ora-drawbacks" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>ORA is not recommended as a follow-up to differential-expression analysis for the reasons below. Use GSEA instead, if appropriate.</p>
<ol style="list-style-type: decimal">
<li><p>The choice of the threshold for statistical significance and the multiple comparison adjustment method can greatly impact the analysis <span class="citation">(<a href="#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al., 2009</a>)</span>.</p></li>
<li><p>ORA fails to incorporate direction of change. (Are the genes in a given set mainly up or down-regulated in one condition relative to another?). It is NOT a good idea to split DEA results by the direction of change and apply ORA to the resulting subsets, unless you are specifically asking “which gene sets are over-represented when we only consider genes that are up- or down-regulated?”</p></li>
<li><p>If few genes are in the “interesting” group, ORA may not yield useful or reliable results. For example, suppose 30 out of 8000 genes are “interesting”. 100 of the genes are annotated to a particular gene set, of which 3 are “interesting”. The associated Hypergeometric p-value is 0.006, and this set would be considered significantly over-represented at the 0.01 level (at least, prior to p-value adjustment); however, if only 2 of the genes in this set are “interesting”, this p-value increases 10-fold to 0.0536 and is no longer significant even at the 0.05 level.</p></li>
<li><p>ORA can not be used if the input contains duplicates. For example, a single feature can not be a member of two or more groups or present multiple times in the same group. This usually happens when attempting to perform gene-level ORA on protein-level differential analysis results, and can lead to artificial over-representation if genes are counted multiple times. Instead, use GSEA and summarize the ranking metric at the gene level (take the average).</p></li>
</ol>
</div>
<div id="ora-examples" class="section level3 hasAnchor" number="8.2.4">
<h3><span class="header-section-number">8.2.4</span> Examples<a href="ora.html#ora-examples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For these examples, we will show how to perform ORA with the <code>fgsea</code>, <code>clusterProfiler</code>, <code>ReactomePA</code>, and <code>GOstats</code> packages on clustering results. The databases that we will cover are Gene Ontology Biological Processes and Reactome, and we will only consider gene sets/pathways with at least 15 and no more than 300 genes. For details on these different annotation databases, please see Section <a href="annotation-databases.html#annotation-databases">8.1</a> (Annotation Databases).</p>
<p>We will use the <code>gcSample</code> data from <code>clusterProfiler</code> and treat the entire list as the <strong>gene universe/background</strong>. Each gene is represented by a human Entrez gene ID, which is the default keytype used by the <code>clusterProfiler</code> functions (and the only keytype compatible with <code>ReactomePA::enrichPathway</code>).</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="ora.html#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb154-2"><a href="ora.html#cb154-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;gcSample&quot;</span>) <span class="co"># data for examples</span></span></code></pre></div>
<p>Since the genes in <code>gcSample</code> are not unique, we will subset to unique genes for the sake of these examples.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="ora.html#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Need to remove duplicates for the examples</span></span>
<span id="cb155-2"><a href="ora.html#cb155-2" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unlist</span>(gcSample)</span>
<span id="cb155-3"><a href="ora.html#cb155-3" aria-hidden="true" tabindex="-1"></a>universe <span class="ot">&lt;-</span> all_genes[Biobase<span class="sc">::</span><span class="fu">isUnique</span>(all_genes)] <span class="co"># all unique genes</span></span>
<span id="cb155-4"><a href="ora.html#cb155-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-5"><a href="ora.html#cb155-5" aria-hidden="true" tabindex="-1"></a><span class="co"># List with only unique genes</span></span>
<span id="cb155-6"><a href="ora.html#cb155-6" aria-hidden="true" tabindex="-1"></a>gcUnique <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gcSample, <span class="cf">function</span>(group_i) {</span>
<span id="cb155-7"><a href="ora.html#cb155-7" aria-hidden="true" tabindex="-1"></a>  group_i[group_i <span class="sc">%in%</span> universe]</span>
<span id="cb155-8"><a href="ora.html#cb155-8" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<div id="fgsea-ora" class="section level4 hasAnchor" number="8.2.4.1">
<h4><span class="header-section-number">8.2.4.1</span> ORA with <code>fgsea</code><a href="ora.html#fgsea-ora" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>fgsea</code> package can be used to perform over-representation analysis with the <code>fora</code> function, which applies the hypergeometric test. It requires a list of gene sets or pathways, a vector of “interesting” genes to test, and a gene universe vector. For clustering results, we can run this function in <code>lapply</code> to test each cluster. For this example, we will perform ORA on the non-redundant Gene Ontology biological processes (GO-BP) sets from MSigDB, but any database can be used.</p>
<p>We first need to get the list of gene sets; we do this with the <code>msigdbr</code> package (Section <a href="annotation-databases.html#msigdb">8.1.4</a>). We use <code>msigdbr_collections</code> to determine the category and subcategory and <code>msigdbr</code> to fetch the data.</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="ora.html#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MSigDB R package</span></span>
<span id="cb156-2"><a href="ora.html#cb156-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb156-3"><a href="ora.html#cb156-3" aria-hidden="true" tabindex="-1"></a>msigdbr<span class="sc">::</span><span class="fu">msigdbr_collections</span>() <span class="co"># available collections</span></span>
<span id="cb156-4"><a href="ora.html#cb156-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to Human GO-BP sets</span></span>
<span id="cb156-5"><a href="ora.html#cb156-5" aria-hidden="true" tabindex="-1"></a>BP_db <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, </span>
<span id="cb156-6"><a href="ora.html#cb156-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">category =</span> <span class="st">&quot;C5&quot;</span>, <span class="at">subcategory =</span> <span class="st">&quot;GO:BP&quot;</span>)</span>
<span id="cb156-7"><a href="ora.html#cb156-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(BP_db)</span></code></pre></div>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="ora.html#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to a list of gene sets</span></span>
<span id="cb157-2"><a href="ora.html#cb157-2" aria-hidden="true" tabindex="-1"></a>BP_conv <span class="ot">&lt;-</span> <span class="fu">unique</span>(BP_db[, <span class="fu">c</span>(<span class="st">&quot;entrez_gene&quot;</span>, <span class="st">&quot;gs_exact_source&quot;</span>)])</span>
<span id="cb157-3"><a href="ora.html#cb157-3" aria-hidden="true" tabindex="-1"></a>BP_list <span class="ot">&lt;-</span> <span class="fu">split</span>(<span class="at">x =</span> BP_conv<span class="sc">$</span>entrez_gene, <span class="at">f =</span> BP_conv<span class="sc">$</span>gs_exact_source)</span>
<span id="cb157-4"><a href="ora.html#cb157-4" aria-hidden="true" tabindex="-1"></a><span class="co"># First ~6 IDs of first 3 terms</span></span>
<span id="cb157-5"><a href="ora.html#cb157-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="fu">head</span>(BP_list, <span class="dv">3</span>), head)</span></code></pre></div>
<p>We have all the required input, so we can move on to ORA. In order to perform ORA on each cluster, we can wrap <code>fora</code> in <code>lapply</code>.</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="ora.html#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Cluster GO-BP ORA with fgsea package</span></span>
<span id="cb158-2"><a href="ora.html#cb158-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span>
<span id="cb158-3"><a href="ora.html#cb158-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb158-4"><a href="ora.html#cb158-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-5"><a href="ora.html#cb158-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For each cluster i, perform ORA</span></span>
<span id="cb158-6"><a href="ora.html#cb158-6" aria-hidden="true" tabindex="-1"></a>fgsea_ora <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(gcUnique), <span class="cf">function</span>(i) {</span>
<span id="cb158-7"><a href="ora.html#cb158-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fora</span>(<span class="at">pathways =</span> BP_list, </span>
<span id="cb158-8"><a href="ora.html#cb158-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">genes =</span> gcUnique[[i]], <span class="co"># genes in cluster i</span></span>
<span id="cb158-9"><a href="ora.html#cb158-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">universe =</span> universe, <span class="co"># all genes</span></span>
<span id="cb158-10"><a href="ora.html#cb158-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">minSize =</span> <span class="dv">15</span>, </span>
<span id="cb158-11"><a href="ora.html#cb158-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">maxSize =</span> <span class="dv">500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb158-12"><a href="ora.html#cb158-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">cluster =</span> <span class="fu">names</span>(gcUnique)[i]) <span class="co"># add cluster column</span></span>
<span id="cb158-13"><a href="ora.html#cb158-13" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span> </span>
<span id="cb158-14"><a href="ora.html#cb158-14" aria-hidden="true" tabindex="-1"></a>  data.table<span class="sc">::</span><span class="fu">rbindlist</span>() <span class="sc">%&gt;%</span> <span class="co"># combine tables</span></span>
<span id="cb158-15"><a href="ora.html#cb158-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb158-16"><a href="ora.html#cb158-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(cluster, padj) <span class="sc">%&gt;%</span> </span>
<span id="cb158-17"><a href="ora.html#cb158-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add additional columns from BP_db</span></span>
<span id="cb158-18"><a href="ora.html#cb158-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="fu">distinct</span>(BP_db, gs_subcat, gs_exact_source, </span>
<span id="cb158-19"><a href="ora.html#cb158-19" aria-hidden="true" tabindex="-1"></a>                     gs_name, gs_description),</span>
<span id="cb158-20"><a href="ora.html#cb158-20" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;pathway&quot;</span> <span class="ot">=</span> <span class="st">&quot;gs_exact_source&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb158-21"><a href="ora.html#cb158-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reformat descriptions</span></span>
<span id="cb158-22"><a href="ora.html#cb158-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gs_name =</span> <span class="fu">sub</span>(<span class="st">&quot;^GOBP_&quot;</span>, <span class="st">&quot;&quot;</span>, gs_name),</span>
<span id="cb158-23"><a href="ora.html#cb158-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">gs_name =</span> <span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>, gs_name))</span>
<span id="cb158-24"><a href="ora.html#cb158-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-25"><a href="ora.html#cb158-25" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 rows</span></span>
<span id="cb158-26"><a href="ora.html#cb158-26" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fgsea_ora)</span></code></pre></div>
<p>See <code>?fora</code> for a description of the output. The base output does not include term descriptions, so we had to add those ourselves.</p>
</div>
<div id="ora-clustprof" class="section level4 hasAnchor" number="8.2.4.2">
<h4><span class="header-section-number">8.2.4.2</span> ORA with <code>clusterProfiler</code>/<code>ReactomePA</code><a href="ora.html#ora-clustprof" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><code>clusterProfiler</code> and <code>ReactomePA</code> are convenience packages that use <code>fgsea</code> and <code>AnnotationDbi</code> as the basis for their pathway analysis functions. The functions are a bit more user-friendly, as the user does not need to fetch the gene lists themselves, but they are much slower, and the resulting objects are not simple data.frames.</p>
<p><code>clusterProfiler</code> provides the <code>enrichGO</code> and <code>enrichKEGG</code> functions for GO and KEGG ORA, respectively (among others). <code>ReactomePA</code> provides the <code>enrichPathway</code> function for Reactome pathway ORA. For other databases, the <code>clusterProfiler::enricher</code> function can be used (though this is slower than just using <code>fgsea::fora</code> with a list of gene sets). To perform ORA on clustering results, we use <code>clusterProfiler::compareCluster</code> and tell it to use <code>enrichGO</code> as the ORA function.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="ora.html#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Cluster GO-BP ORA with clusterProfiler package</span></span>
<span id="cb159-2"><a href="ora.html#cb159-2" aria-hidden="true" tabindex="-1"></a>cp_ora <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb159-3"><a href="ora.html#cb159-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">geneClusters =</span> gcUnique, </span>
<span id="cb159-4"><a href="ora.html#cb159-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">fun =</span> <span class="st">&quot;enrichGO&quot;</span>, <span class="co"># ORA function to apply to each cluster</span></span>
<span id="cb159-5"><a href="ora.html#cb159-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Arguments below are passed to enrichGO</span></span>
<span id="cb159-6"><a href="ora.html#cb159-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>, </span>
<span id="cb159-7"><a href="ora.html#cb159-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">&quot;ENTREZID&quot;</span>, </span>
<span id="cb159-8"><a href="ora.html#cb159-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, <span class="co"># BP, CC, MF, or ALL for all ontologies</span></span>
<span id="cb159-9"><a href="ora.html#cb159-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb159-10"><a href="ora.html#cb159-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">qvalueCutoff =</span> <span class="dv">1</span>, <span class="co"># do not filter by q-value</span></span>
<span id="cb159-11"><a href="ora.html#cb159-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="co"># p-values are adjusted within clusters</span></span>
<span id="cb159-12"><a href="ora.html#cb159-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">universe =</span> universe, <span class="co"># all genes</span></span>
<span id="cb159-13"><a href="ora.html#cb159-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">15</span>, </span>
<span id="cb159-14"><a href="ora.html#cb159-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxGSSize =</span> <span class="dv">500</span></span>
<span id="cb159-15"><a href="ora.html#cb159-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb159-16"><a href="ora.html#cb159-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-17"><a href="ora.html#cb159-17" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 entries sorted by cluster and p-value</span></span>
<span id="cb159-18"><a href="ora.html#cb159-18" aria-hidden="true" tabindex="-1"></a>cp_ora<span class="sc">@</span>compareClusterResult <span class="sc">%&gt;%</span> </span>
<span id="cb159-19"><a href="ora.html#cb159-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Cluster, pvalue) <span class="sc">%&gt;%</span> </span>
<span id="cb159-20"><a href="ora.html#cb159-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>()</span></code></pre></div>
<p>Unlike the <code>fgsea::fora</code> results, these include the description of each term. As for the other columns: <code>GeneRatio</code> is the same as <code>overlap</code> (from the <code>fora</code> results) divided by the cluster size, <code>BgRatio</code> is the set size divided by the universe size, <code>pvalue</code> is the raw p-value, <code>p.adjust</code> is the BH-adjusted p-value, <code>qvalue</code> is the q-value, <code>geneID</code> is the same as <code>overlapGenes</code> from <code>fora</code>, and <code>Count</code> is the overlap size.</p>
<p>Below is an example of how to perform Reactome ORA with <code>ReactomePA::enrichPathway</code>.</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="ora.html#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Reactome ORA with ReactomePA package</span></span>
<span id="cb160-2"><a href="ora.html#cb160-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ReactomePA)</span>
<span id="cb160-3"><a href="ora.html#cb160-3" aria-hidden="true" tabindex="-1"></a>react_ora <span class="ot">&lt;-</span> <span class="fu">compareCluster</span>(</span>
<span id="cb160-4"><a href="ora.html#cb160-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">geneClusters =</span> gcUnique, </span>
<span id="cb160-5"><a href="ora.html#cb160-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">fun =</span> <span class="st">&quot;enrichPathway&quot;</span>, <span class="co"># ORA function to apply to each cluster</span></span>
<span id="cb160-6"><a href="ora.html#cb160-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Arguments below are passed to enrichPathway</span></span>
<span id="cb160-7"><a href="ora.html#cb160-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">organism =</span> <span class="st">&quot;human&quot;</span>,</span>
<span id="cb160-8"><a href="ora.html#cb160-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="dv">1</span>, <span class="co"># Do not filter by p-value</span></span>
<span id="cb160-9"><a href="ora.html#cb160-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">qvalueCutoff =</span> <span class="dv">1</span>, <span class="co"># Do not filter by q-value</span></span>
<span id="cb160-10"><a href="ora.html#cb160-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="co"># p-values are adjusted within clusters</span></span>
<span id="cb160-11"><a href="ora.html#cb160-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">universe =</span> universe, <span class="co"># all genes</span></span>
<span id="cb160-12"><a href="ora.html#cb160-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">15</span>, </span>
<span id="cb160-13"><a href="ora.html#cb160-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxGSSize =</span> <span class="dv">500</span></span>
<span id="cb160-14"><a href="ora.html#cb160-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb160-15"><a href="ora.html#cb160-15" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 rows</span></span>
<span id="cb160-16"><a href="ora.html#cb160-16" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(react_ora<span class="sc">@</span>compareClusterResult)</span></code></pre></div>
</div>
<div id="ora-with-gostats" class="section level4 hasAnchor" number="8.2.4.3">
<h4><span class="header-section-number">8.2.4.3</span> ORA with <code>GOstats</code><a href="ora.html#ora-with-gostats" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In the previous ORA examples, the Hypergeometric test is performed independently for each gene set; however, this does not capture the relationship between GO terms (described in Section <a href="annotation-databases.html#gene-ontology">8.1.1</a>). Since “each GO term inherits all annotations from its more specific descendants,” results tend to be redundant (except when using MSigDB), as they include directly-related GO terms with a high degree of overlap <span class="citation">(<a href="#ref-falcon_using_2007" role="doc-biblioref">S. Falcon et al., 2007</a>)</span>. One way to handle this is with a procedure that conditions on the GO structure, like the one described by S. Falcon and R. Gentleman <span class="citation">(<a href="#ref-falcon_using_2007" role="doc-biblioref">2007</a>)</span>:</p>
<blockquote>
<p>Given a subgraph of one of the three GO ontologies [BP, MF, or CC], we test the leaves of the graph, that is, those terms with no child terms. Before testing the terms whose children have already been tested, we remove all genes annotated at significant children [<code>pvalueCutoff = 0.05</code> in the code below] from the parent’s gene list. This continues until all terms have been tested.</p>
</blockquote>
<p>This approach is implemented in the <code>GOstats</code> package by setting <code>conditional = TRUE</code> when creating a new object of class <code>GOHyperGParams</code> (See <code>help("GOHyperGParams-class", package = "Category")</code> for more details). Below, we will perform GO-BP ORA just for the first cluster of <code>gcUnique</code> because it is time-consuming, but this could be wrapped in <code>lapply</code> to get results for each cluster (like in Section <a href="ora.html#fgsea-ora">8.2.4.1</a>).</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="ora.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Cluster GO-BP ORA with GOstats package</span></span>
<span id="cb161-2"><a href="ora.html#cb161-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GOstats)</span>
<span id="cb161-3"><a href="ora.html#cb161-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb161-4"><a href="ora.html#cb161-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb161-5"><a href="ora.html#cb161-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For cluster 1, perform conditional ORA</span></span>
<span id="cb161-6"><a href="ora.html#cb161-6" aria-hidden="true" tabindex="-1"></a>gostats_ora <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="at">Class =</span> <span class="st">&quot;GOHyperGParams&quot;</span>,</span>
<span id="cb161-7"><a href="ora.html#cb161-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ontology =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb161-8"><a href="ora.html#cb161-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">geneIds =</span> gcUnique[[<span class="dv">1</span>]],</span>
<span id="cb161-9"><a href="ora.html#cb161-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">universeGeneIds =</span> universe,</span>
<span id="cb161-10"><a href="ora.html#cb161-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">annotation =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>,</span>
<span id="cb161-11"><a href="ora.html#cb161-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb161-12"><a href="ora.html#cb161-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">testDirection =</span> <span class="st">&quot;over&quot;</span>,</span>
<span id="cb161-13"><a href="ora.html#cb161-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">conditional =</span> <span class="cn">TRUE</span>, <span class="co"># condition on GO structure</span></span>
<span id="cb161-14"><a href="ora.html#cb161-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">minSizeCutoff =</span> <span class="dv">15</span>,</span>
<span id="cb161-15"><a href="ora.html#cb161-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">maxSizeCutoff =</span> <span class="dv">500</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb161-16"><a href="ora.html#cb161-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hyperGTest</span>() <span class="sc">%&gt;%</span> <span class="co"># Hypergeometric testing</span></span>
<span id="cb161-17"><a href="ora.html#cb161-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>() <span class="sc">%&gt;%</span> <span class="co"># extract results</span></span>
<span id="cb161-18"><a href="ora.html#cb161-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># adjust p-values</span></span>
<span id="cb161-19"><a href="ora.html#cb161-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Padj =</span> <span class="fu">p.adjust</span>(Pvalue, <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb161-20"><a href="ora.html#cb161-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb161-21"><a href="ora.html#cb161-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Padj)</span>
<span id="cb161-22"><a href="ora.html#cb161-22" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 rows</span></span>
<span id="cb161-23"><a href="ora.html#cb161-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gostats_ora)</span></code></pre></div>
<p>If we compare this table to the results from the other packages, we see that only <a href="GO:0007600" class="uri">GO:0007600</a> and <a href="GO:0061844" class="uri">GO:0061844</a> from the top 6 terms made it into the above table.</p>
</div>
</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-falcon_using_2007" class="csl-entry">
Falcon, S. and Gentleman, R., Using <span>GOstats</span> to Test Gene Lists for <span>GO</span> Term Association, <em>Bioinformatics</em>, vol. <strong>23</strong>, no. 2, pp. 257–58, accessed May 5, 2022, from <a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btl567">https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btl567</a>, January 2007. DOI: <a href="https://doi.org/10.1093/bioinformatics/btl567">10.1093/bioinformatics/btl567</a>
</div>
<div id="ref-huang_bioinformatics_2009" class="csl-entry">
Huang, D. W., Sherman, B. T. and Lempicki, R. A., Bioinformatics Enrichment Tools: Paths Toward the Comprehensive Functional Analysis of Large Gene Lists, <em>Nucleic Acids Research</em>, vol. <strong>37</strong>, no. 1, pp. 1–13, accessed September 7, 2021, from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615629/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615629/</a>, January 2009. DOI: <a href="https://doi.org/10.1093/nar/gkn923">10.1093/nar/gkn923</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="annotation-databases.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gsea.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
