<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Section 4 Pathway Analysis | Proteomics Data Analysis in R/Bioconductor</title>
<meta name="author" content="Tyler Sagendorf">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Proteomics Data Analysis in R/Bioconductor</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome!</a></li>
<li><a class="" href="isobaric-quantification-pipelines.html"><span class="header-section-number">1</span> Isobaric Quantification Pipelines</a></li>
<li><a class="" href="exploratory-data-analysis.html"><span class="header-section-number">2</span> Exploratory Data Analysis</a></li>
<li><a class="" href="DEA.html"><span class="header-section-number">3</span> Differential Expression Analysis</a></li>
<li><a class="active" href="pathway-analysis.html"><span class="header-section-number">4</span> Pathway Analysis</a></li>
<li><a class="" href="references-1.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="pathway-analysis" class="section level1" number="4">
<h1>
<span class="header-section-number">Section 4</span> Pathway Analysis<a class="anchor" aria-label="anchor" href="#pathway-analysis"><i class="fas fa-link"></i></a>
</h1>
<p>In Section <a href="DEA.html#DEA">3</a>, we covered analysis at the individual feature level (protein, peptide, phosphoprotein, etc.). While DEA is useful, it is not without its own set of shortcomings. For instance, there may be no features that pass the significance threshold after correcting for multiple hypothesis testing. Alternatively, there may be many features that are statistically significant, and interpreting this list can be tedious and “prone to investigator bias toward a hypothesis of interest” <span class="citation">(<a href="references-1.html#ref-maleki_gene_2020" role="doc-biblioref">Maleki et al., 2020</a>)</span>. Another issue is that single-feature analysis fails to detect subtle, yet coordinated changes in groups of related features <span class="citation">(<a href="references-1.html#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al., 2005</a>)</span>.</p>
<p>In order to address these, and other, issues, pathway analysis instead examines <em>a priori</em> defined <strong>gene sets</strong>—groups of genes that participate in the same biological pathway, share the same cellular location, etc. In this section, we will explore some common annotation databases, as well as two pathway analysis methods: Over-Representation Analysis (ORA) and Gene Set Enrichment Analysis (GSEA).</p>
<div id="annotation-databases" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Annotation Databases<a class="anchor" aria-label="anchor" href="#annotation-databases"><i class="fas fa-link"></i></a>
</h2>
<p>In this section, we will explore some of the common annotation databases used for pathway analysis.</p>
<div id="gene-ontology" class="section level3" number="4.1.1">
<h3>
<span class="header-section-number">4.1.1</span> Gene Ontology<a class="anchor" aria-label="anchor" href="#gene-ontology"><i class="fas fa-link"></i></a>
</h3>
<p>The Gene Ontology (GO) database is divided into three separate domains: Biological Process, Cellular Component, and Molecular Function (see the <a href="http://geneontology.org/docs/ontology-documentation/" title="Gene Ontology overview">Gene Ontology overview</a> for more details regarding each domain). Each domain is structured as a directed acyclic graph (DAG) where nodes are terms and edges are the <a href="http://geneontology.org/docs/ontology-relations/#:~:text=Main%20relations%20used%20in%20GO" title="Main relations used in GO">relations</a> between the terms (part of, is a, has part, regulates). Nodes can be connected to multiple child and parent nodes, where the group of genes annotated to a child node is a subset of those that are annotated to its parent node(s) <span class="citation">(<a href="references-1.html#ref-noauthor_relations_2021" role="doc-biblioref">2021</a>; <a href="references-1.html#ref-goeman_multiple_2008" role="doc-biblioref">Goeman et al., 2008</a>)</span>.</p>
<div id="semantic-similarity" class="section level4" number="4.1.1.1">
<h4>
<span class="header-section-number">4.1.1.1</span> Semantic Similarity<a class="anchor" aria-label="anchor" href="#semantic-similarity"><i class="fas fa-link"></i></a>
</h4>
<p>Due to the DAG structure of each domain, there is often redundancy in pathway analysis results. For example, suppose terms <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006119" title="oxidative phosphorylation">GO:0006119</a>, <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009060" title="aerobic respiration">GO:0009060</a>, and <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046034" title="ATP metabolic process">GO:0046034</a> are significantly over-represented biological processes. <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009060" title="aerobic respiration">GO:0009060</a> and <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046034" title="ATP metabolic process">GO:0046034</a> are the parent terms of <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006119" title="oxidative phosphorylation">GO:0006119</a>. Due to this relationship, the terms likely provide much of the same information, so the inclusion of all three terms in the output is unnecessary. In order to resolve this redundancy, we can calculate the <strong>semantic similarity</strong> between pairs of GO terms, which “assesses the likeness in meaning of two concepts” <span class="citation">(<a href="references-1.html#ref-pesquita_semantic_2017" role="doc-biblioref">Pesquita, 2017</a>)</span>. Basically, if two terms are highly related, we can use some other criteria (such as adjusted p-value or level in the DAG) to retain only one of the terms. Below, we use the <code>GOSemSim</code> package to calculate the semantic similarity between the terms.</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Calculate semantic similarity between GO terms</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">GOSemSim</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>

<span class="co"># GO DATA for measuring semantic similarity.</span>
<span class="co"># keytype is "ENTREZID" by default and </span>
<span class="co"># information content is calculated (computeIC = TRUE)</span>
<span class="va">semData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GOSemSim/man/godata.html">godata</a></span><span class="op">(</span>OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>, ont <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span>
<span class="va">terms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GO:0006119"</span>, <span class="st">"GO:0009060"</span>, <span class="st">"GO:0046034"</span><span class="op">)</span>
<span class="co"># measure = "Rel" is the default for clusterProfiler::simplify</span>
<span class="co"># See code for clusterProfiler:::simplify_internal</span>
<span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GOSemSim/man/mgoSim.html">mgoSim</a></span><span class="op">(</span>GO1 <span class="op">=</span> <span class="va">terms</span>, GO2 <span class="op">=</span> <span class="va">terms</span>, semData <span class="op">=</span> <span class="va">semData</span>, 
              measure <span class="op">=</span> <span class="st">"Rel"</span>, combine <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> </code></pre></div>
<div class="inline-table"><table class="table table-condensed" style="width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:sem-sim-table">Table 4.1: </span>Semantic Similarity of select GO terms
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
<a href="GO:0006119" class="uri">GO:0006119</a>
</th>
<th style="text-align:right;">
<a href="GO:0009060" class="uri">GO:0009060</a>
</th>
<th style="text-align:right;">
<a href="GO:0046034" class="uri">GO:0046034</a>
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006119" class="uri">GO:0006119</a>
</td>
<td style="text-align:right;">
0.998
</td>
<td style="text-align:right;">
0.754
</td>
<td style="text-align:right;">
0.936
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009060" class="uri">GO:0009060</a>
</td>
<td style="text-align:right;">
0.754
</td>
<td style="text-align:right;">
0.999
</td>
<td style="text-align:right;">
0.128
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046034" class="uri">GO:0046034</a>
</td>
<td style="text-align:right;">
0.936
</td>
<td style="text-align:right;">
0.128
</td>
<td style="text-align:right;">
0.997
</td>
</tr>
</tbody>
</table></div>
<p>If <code>measure</code> is <code>"Lin"</code>, <code>"Jiang"</code>, or <code>"Wang"</code>, the semantic similarity of a term with itself will be 1. This is not true for the other methods.</p>
<p>We can see from Table <a href="pathway-analysis.html#tab:sem-sim-table">4.1</a> that <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009060" title="aerobic respiration">GO:0009060</a> and <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046034" title="ATP metabolic process">GO:0046034</a> have low semantic similarity, while <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006119" title="oxidative phosphorylation">GO:0006119</a> is highly similar to its parent terms. This makes sense because the parent terms are not related/connected in the DAG.</p>
<p>Now that we have the semantic similarities, we can remove redundant terms. <code>clusterProfiler</code> has a function called <code>simplify</code> that will calculate semantic similarity and remove terms. By default, if there are two terms with a semantic similarity greater than 0.7, <code>simplify</code> retains the term with the lowest adjusted p-value. See <a href="https://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/" title="use simplify to remove redundancy of enriched GO terms">this post</a> by Guangchuang Yu for more details on <code><a href="https://rdrr.io/pkg/clusterProfiler/man/simplify-methods.html">clusterProfiler::simplify</a></code>.</p>
</div>
<div id="go-subsetsslims" class="section level4" number="4.1.1.2">
<h4>
<span class="header-section-number">4.1.1.2</span> GO Subsets/Slims<a class="anchor" aria-label="anchor" href="#go-subsetsslims"><i class="fas fa-link"></i></a>
</h4>
<p>Another way to handle the redundancy of GO terms is to use a <a href="http://geneontology.org/docs/go-subset-guide/" title="Guide to GO subsets">GO slim</a>, which is a subset of more general or research-relevant terms from the GO. GO slims can be <a href="http://geneontology.org/docs/download-ontology/#subsets" title="Download the ontology">downloaded</a> or the <code>biomaRt</code> package can be used to access GO slim accessions.</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create human GO slim</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomaRt</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span> <span class="co"># gcSample data</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>

<span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,
                dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span>
<span class="co"># Uncomment to determine which attributes to select in getBM()</span>
<span class="co"># View(listAttributes(mart)) </span>

<span class="co"># The GO slim columns are goslim_goa_accession and goslim_goa_description.</span>
<span class="co"># We will map from the Entrez IDs in gcSample to these attributes.</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span>
<span class="va">universe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span><span class="op">)</span>
<span class="va">GO_slim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html">getBM</a></span><span class="op">(</span>filters <span class="op">=</span> <span class="st">"entrezgene_id"</span>,
                 attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"entrezgene_id"</span>,
                                <span class="st">"goslim_goa_accession"</span>,
                                <span class="st">"goslim_goa_description"</span><span class="op">)</span>,
                 values <span class="op">=</span> <span class="va">universe</span>, <span class="co"># Subset to these Entrez IDs</span>
                 mart <span class="op">=</span> <span class="va">mart</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Convert entrezgene_id from integer to character</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html">mutate_all</a></span><span class="op">(</span><span class="va">as.character</span><span class="op">)</span></code></pre></div>
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
entrezgene_id
</th>
<th style="text-align:left;">
goslim_goa_accession
</th>
<th style="text-align:left;">
goslim_goa_description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0003674" style="     ">GO:0003674</a>
</td>
<td style="text-align:left;">
molecular_function
</td>
</tr>
<tr>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0016810" style="     ">GO:0016810</a>
</td>
<td style="text-align:left;">
hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds
</td>
</tr>
<tr>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0008150" style="     ">GO:0008150</a>
</td>
<td style="text-align:left;">
biological_process
</td>
</tr>
<tr>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009058" style="     ">GO:0009058</a>
</td>
<td style="text-align:left;">
biosynthetic process
</td>
</tr>
<tr>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0044281" style="     ">GO:0044281</a>
</td>
<td style="text-align:left;">
small molecule metabolic process
</td>
</tr>
<tr>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0034641" style="     ">GO:0034641</a>
</td>
<td style="text-align:left;">
cellular nitrogen compound metabolic process
</td>
</tr>
</tbody>
</table></div>
<p>Unfortunately, not every GO accession maps to a domain when we use <code>biomaRt</code> (unsure why this is the case), so we won’t be able to separate the terms. However, there are two ways that we can still use these GO slim accessions. Either follow the steps for using <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">clusterProfiler::enricher</a></code> (shown in the Pfam subsection of Section <a href="pathway-analysis.html#ora-examples">4.2.2</a>) with conversion tables created from <code>org.Hs.eg.db</code> that have been subset to the GO slim accessions, or remove any non GO slim accessions from the final results and readjust the remaining p-values (the easier approach).</p>
</div>
</div>
<div id="reactome" class="section level3" number="4.1.2">
<h3>
<span class="header-section-number">4.1.2</span> Reactome<a class="anchor" aria-label="anchor" href="#reactome"><i class="fas fa-link"></i></a>
</h3>
<p><a href="https://reactome.org/">Home - Reactome Pathway Database</a></p>
</div>
<div id="kegg" class="section level3" number="4.1.3">
<h3>
<span class="header-section-number">4.1.3</span> KEGG<a class="anchor" aria-label="anchor" href="#kegg"><i class="fas fa-link"></i></a>
</h3>
<p><a href="https://www.genome.jp/kegg/">KEGG: Kyoto Encyclopedia of Genes and Genomes</a></p>
</div>
<div id="pfam" class="section level3" number="4.1.4">
<h3>
<span class="header-section-number">4.1.4</span> Pfam<a class="anchor" aria-label="anchor" href="#pfam"><i class="fas fa-link"></i></a>
</h3>
<p><a href="http://pfam.xfam.org/">Pfam: Home page</a>
<a href="https://pfam-docs.readthedocs.io/en/latest/">Pfam Documentation</a>
<a href="https://watermark.silverchair.com/140755.pdf?token=AQECAHi208BE49Ooan9kkhW_Ercy7Dm3ZL_9Cf3qfKAc485ysgAAAvYwggLyBgkqhkiG9w0BBwagggLjMIIC3wIBADCCAtgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQM8AYKH-bv0m2nHfTQAgEQgIICqYjz7XAVxAREVHGvJofH3peUmhTifphKa4BaiK7Xw4XUV41B2Eolc7IBL6Q9Oh0MAVCD0yHlxqBSwJ3cETaSRp60T6Bk9Cl6nUyjTUpq-4f8YW1pUgOxvd7OCLkE11Twa60Y9T6JFyIUuCxHs6UgbRBAbwNZ-6HG9-WRHxiWzCSf6SbrH74ouzlsabLh1OZMSoJ3dfbNrBYRz-bkonlmHUBASLKee1YA6Eg4EavR1Qa07g5FwDJhGVUgDw4IyhkyGAucfJTwl5_FxFGrs8-FEQ01LZqQDzZLsL0zcH7NsD8dB6PaWcP-lrH0TQpRK-WHhSC2WmL6rEoV_zwzUE9faUkLRzbg82siXNFUhgR9cwpe_ycHF-ffmIVMjpPWYvmMzrk4YfMfp_A0OsNbu0S7vSL7gqJ-uD2HyeJ8XkAZ1HXDjBKGHyQmQX6otAHfdNHbHUMl8GaFmgUPDkfj9swuUeQ0udhKRRssJNF8cqrs-xDzNQnB3-cGhi0bfV7PT1CZxGBMhlC4UsTu0hLZUfMxozxXfDjLY0kZEu66lTJL7CYGru4JCx7qFHvaQZg3icjHLOAiwov7v9-CimdXepQrZxZQ588N2ZlgsZ5f0xiZ5qdGkGwDyhdFmLPj2f_RFXRe5TuCWLU1DTkX5NyIUYGwGJECn6n_qSsA37n73ek--_eBPV-F6GBYQe1oRWP2S5SQTxP0r3yPyh7dIlqcqQxWozlsew1fGCMjRoUv1e3SFU7LZ0j8bUSXXEAgOxjjOBSxWu1QzgjLa-MRV0IQ61tDXfGdJhSuuZSxRwlxTbpZKQWfaS3s1jFU9Hm-B-9CvhmJQgb7elOeD0cbpVMeM03ebaK-NgPA0dX5slcIe1--BtOdKoKjepE7--RS4Z4Bls4ZI0d0valj-zsHiw">profile Hidden Markov Models</a></p>
</div>
</div>
<div id="ora" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> Over-Representation Analysis<a class="anchor" aria-label="anchor" href="#ora"><i class="fas fa-link"></i></a>
</h2>
<div id="ora-overview" class="section level3" number="4.2.1">
<h3>
<span class="header-section-number">4.2.1</span> Overview<a class="anchor" aria-label="anchor" href="#ora-overview"><i class="fas fa-link"></i></a>
</h3>
<p>Over-Representation Analysis (ORA) is used to determine which <em>a priori</em> defined gene sets are more present (over-represented) in a subset of “interesting” genes than what would be expected by chance <span class="citation">(<a href="references-1.html#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al., 2009</a>)</span>. Essentially, it identifies the gene sets—rather than the individual genes—that are significantly different between two conditions.</p>
<p>For each gene set, an enrichment p-value is calculated using the Binomial distribution, Hypergeometric distribution, the Fisher exact test, or the Chi-square test. Although this list is not all-encompassing, these are the most popular statistical methods <span class="citation">(<a href="references-1.html#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al., 2009</a>)</span>. Below is the formula for calculating the enrichment p-value for a particular gene set using the Hypergeometric distribution.</p>
<div class="math">
<p><span class="math display">\[
P(X\geq x) = 1 - P(X \leq x-1) = 1 - \sum\limits_{i=0}^{x-1}\frac{\hphantom{}{M \choose i }{N - M \choose n-i}}{N \choose n}
\]</span></p>
</div>
<p>In this equation, <span class="math inline">\(N\)</span> is the number of background genes, <span class="math inline">\(n\)</span> is the number of “interesting” (i.e. statistically-significant) genes, <span class="math inline">\(M\)</span> is the number of genes that are annotated to a particular gene set <span class="math inline">\(S\)</span>, and <span class="math inline">\(x\)</span> is the number of “interesting” genes that are annotated to <span class="math inline">\(S\)</span>. The numerator of the sum is the number of samples of <span class="math inline">\(n\)</span> genes that can be taken from a population of <span class="math inline">\(N\)</span> genes where exactly <span class="math inline">\(i\)</span> of the genes are annotated to <span class="math inline">\(S\)</span> and <span class="math inline">\(n-i\)</span> are not annotated to <span class="math inline">\(S\)</span>. The denominator of the sum is the total number of samples of size <span class="math inline">\(n\)</span> that can be taken from a population of size <span class="math inline">\(N\)</span>.</p>
<p>For example, suppose we have a list of 8000 genes, of which 400 are differentially expressed. Also suppose that 100 of the 8000 genes are annotated to a particular gene set <span class="math inline">\(S\)</span>. Of these 100 genes, 20 are differentially expressed. The probability that 20 or more (up to 100) genes annotated to <span class="math inline">\(S\)</span> are differentially expressed by chance is given by</p>
<div class="math">
<p><span class="math display">\[
P(X\geq 20) = 1 - P(X \leq 19) = 1-\sum \limits_{i=0}^{19}\frac{\hphantom{}{100 \choose i}{8000 - 100 \choose 400-i}}{8000 \choose 400} = 7.88 \times 10^{-8}
\]</span></p>
</div>
<p>That is, it is extremely unlikely that 20 of the 100 genes from this set are significantly differentially expressed by chance (at least, prior to adjustment for multiple comparisons). The code to calculate this p-value is</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/Hypergeometric.html">phyper</a></span><span class="op">(</span>q <span class="op">=</span> <span class="fl">20</span> <span class="op">-</span> <span class="fl">1</span>, m <span class="op">=</span> <span class="fl">400</span>, n <span class="op">=</span> <span class="fl">8000</span> <span class="op">-</span> <span class="fl">400</span>, k <span class="op">=</span> <span class="fl">100</span>, lower.tail <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>After a p-value has been calculated for each of the applicable gene sets, a multiple comparison adjustment should be performed.</p>
</div>
<div id="ora-examples" class="section level3" number="4.2.2">
<h3>
<span class="header-section-number">4.2.2</span> Examples<a class="anchor" aria-label="anchor" href="#ora-examples"><i class="fas fa-link"></i></a>
</h3>
<p>For these examples, we will show how to perform ORA with the GOstats, clusterProfiler, and ReactomePA packages. The databases that we will cover are Gene Ontology, Reactome, and Pfam. For details on these different annotation databases, please see Section <a href="pathway-analysis.html#annotation-databases">4.1</a>.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Setup</span>
<span class="co"># Required packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">clusterProfiler</a></span><span class="op">)</span> <span class="co"># GO, KEGG, and custom ORA</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">ReactomePA</a></span><span class="op">)</span> <span class="co"># Reactome ORA</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GOstats</span><span class="op">)</span> <span class="co"># GO, KEGG, and Pfam ORA</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PFAM.db</span><span class="op">)</span> <span class="co"># Map Pfam IDs to descriptions</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span> <span class="co"># Human annotation database</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<p>Normally, we would use a DEA table to create two character vectors: one for the significantly expressed genes and the other for all genes that were tested (referred to as the “background” or “universe”); instead, we will use the <code>gcSample</code> data that comes with clusterProfiler and treat the eighth cluster as our vector of significant genes and the entire list as the gene universe. Each gene is represented by a human Entrez gene ID, which is the default keytype used by the clusterProfiler functions and the only keytype compatible with <code><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html">ReactomePA::enrichPathway</a></code>.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"gcSample"</span><span class="op">)</span> <span class="co"># Data for examples</span>
<span class="va">sig_genes</span> <span class="op">&lt;-</span> <span class="va">gcSample</span><span class="op">[[</span><span class="fl">8</span><span class="op">]</span><span class="op">]</span> <span class="co"># significant genes</span>
<span class="va">universe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">gcSample</span><span class="op">)</span><span class="op">)</span> <span class="co"># universe</span></code></pre></div>
<p>It is important to note that the genes should be unique from the start. The terms between any two clusters of <code>gcSample</code> may overlap, so we must use <code>unique</code> for the sake of these examples. <em>If your DEA results are not gene-centric, do NOT use ORA. Instead, switch to GSEA and summarize the ranking metric in some way to make it gene-centric (i.e. take the average, min, max, etc. of the metrics for each gene group)</em>.</p>
<div id="gene-ontology-1" class="section level4" number="4.2.2.1">
<h4>
<span class="header-section-number">4.2.2.1</span> Gene Ontology<a class="anchor" aria-label="anchor" href="#gene-ontology-1"><i class="fas fa-link"></i></a>
</h4>
<p>We will first use the clusterProfiler package to test which biological processes are over-represented in the set of interesting genes. For this example, we will only consider gene sets of size 20 to 500. In order to test either molecular functions, cellular components, or all three ontologies at once, set <code>ont</code> to <code>"MF"</code>, <code>"CC"</code>, or <code>"ALL"</code>, respectively.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## GO BP ORA with clusterProfiler</span>
<span class="co"># This takes a while</span>
<span class="va">cp_ora_go</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enrichGO.html">enrichGO</a></span><span class="op">(</span>
  gene <span class="op">=</span> <span class="va">sig_genes</span>, 
  OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>, 
  keyType <span class="op">=</span> <span class="st">"ENTREZID"</span>, 
  ont <span class="op">=</span> <span class="st">"BP"</span>, <span class="co"># BP, CC, MF, or ALL for all ontologies</span>
  pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,
  qvalueCutoff <span class="op">=</span> <span class="fl">1</span>, <span class="co"># Do not filter by q-value</span>
  pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>, 
  universe <span class="op">=</span> <span class="va">universe</span>, 
  minGSSize <span class="op">=</span> <span class="fl">20</span>, 
  maxGSSize <span class="op">=</span> <span class="fl">500</span>, 
  readable <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># Convert Entrez ID to gene symbol</span>
<span class="op">)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:20em; ">
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:cp-ora-go-table">Table 4.2: </span>Top significantly over-represented biological processes from enrichGO output.
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Description
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
GeneRatio
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
BgRatio
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.adjust
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
geneID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046034" style="     ">GO:0046034</a>
</td>
<td style="text-align:left;">
ATP metabolic process
</td>
<td style="text-align:left;">
24/232
</td>
<td style="text-align:left;">
96/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/PGK1/TPI1…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006119" style="     ">GO:0006119</a>
</td>
<td style="text-align:left;">
oxidative phosphorylation
</td>
<td style="text-align:left;">
16/232
</td>
<td style="text-align:left;">
49/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/COX8A/COX7C…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006402" style="     ">GO:0006402</a>
</td>
<td style="text-align:left;">
mRNA catabolic process
</td>
<td style="text-align:left;">
22/232
</td>
<td style="text-align:left;">
93/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/RPL13A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:1902600" style="     ">GO:1902600</a>
</td>
<td style="text-align:left;">
proton transmembrane transport
</td>
<td style="text-align:left;">
12/232
</td>
<td style="text-align:left;">
28/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/COX8A/COX7C…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0016071" style="     ">GO:0016071</a>
</td>
<td style="text-align:left;">
mRNA metabolic process
</td>
<td style="text-align:left;">
35/232
</td>
<td style="text-align:left;">
208/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/RPL13A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006091" style="     ">GO:0006091</a>
</td>
<td style="text-align:left;">
generation of precursor metabolites and energy
</td>
<td style="text-align:left;">
29/232
</td>
<td style="text-align:left;">
159/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/PGK1/TPI1…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009060" style="     ">GO:0009060</a>
</td>
<td style="text-align:left;">
aerobic respiration
</td>
<td style="text-align:left;">
12/232
</td>
<td style="text-align:left;">
32/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1e-04
</td>
<td style="text-align:left;">
COX4I1/MDH1/HIF1A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006401" style="     ">GO:0006401</a>
</td>
<td style="text-align:left;">
RNA catabolic process
</td>
<td style="text-align:left;">
22/232
</td>
<td style="text-align:left;">
104/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1e-04
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/RPL13A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0010608" style="     ">GO:0010608</a>
</td>
<td style="text-align:left;">
posttranscriptional regulation of gene expression
</td>
<td style="text-align:left;">
28/232
</td>
<td style="text-align:left;">
168/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3e-04
</td>
<td style="text-align:left;">
HNRNPU/MATR3/YWHAZ…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0042773" style="     ">GO:0042773</a>
</td>
<td style="text-align:left;">
ATP synthesis coupled electron transport
</td>
<td style="text-align:left;">
11/232
</td>
<td style="text-align:left;">
31/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3e-04
</td>
<td style="text-align:left;">
COX4I1/COX8A/COX7C…
</td>
</tr>
</tbody>
</table></div>
</div>
<p></p>
<p>There were 4436 biological processes that were tested. Of these, 2049 passed the size filter, and only 182 were significantly over-represented after multiple testing correction. The top 10 are shown in Table <a href="pathway-analysis.html#tab:cp-ora-go-table">4.2</a>.</p>
<p><strong>Remove Redundant GO Terms</strong></p>
<p>By default, if two terms have a semantic similarity above 0.7, <code><a href="https://rdrr.io/pkg/clusterProfiler/man/simplify-methods.html">clusterProfiler::simplify</a></code> retains the more significantly over-represented term.</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove redundant GO terms with simplify</span>
<span class="va">cp_ora_go_sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/simplify-methods.html">simplify</a></span><span class="op">(</span><span class="va">cp_ora_go</span><span class="op">)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:20em; ">
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:simplify-cp-ora-bp-table">Table 4.3: </span>Simplified top significantly over-represented biological processes from enrichGO output.
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Description
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
GeneRatio
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
BgRatio
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.adjust
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
geneID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046034" style="     ">GO:0046034</a>
</td>
<td style="text-align:left;">
ATP metabolic process
</td>
<td style="text-align:left;">
24/232
</td>
<td style="text-align:left;">
96/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/PGK1/TPI1…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006119" style="     ">GO:0006119</a>
</td>
<td style="text-align:left;">
oxidative phosphorylation
</td>
<td style="text-align:left;">
16/232
</td>
<td style="text-align:left;">
49/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/COX8A/COX7C…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006402" style="     ">GO:0006402</a>
</td>
<td style="text-align:left;">
mRNA catabolic process
</td>
<td style="text-align:left;">
22/232
</td>
<td style="text-align:left;">
93/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/RPL13A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:1902600" style="     ">GO:1902600</a>
</td>
<td style="text-align:left;">
proton transmembrane transport
</td>
<td style="text-align:left;">
12/232
</td>
<td style="text-align:left;">
28/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/COX8A/COX7C…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0016071" style="     ">GO:0016071</a>
</td>
<td style="text-align:left;">
mRNA metabolic process
</td>
<td style="text-align:left;">
35/232
</td>
<td style="text-align:left;">
208/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/RPL13A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006091" style="     ">GO:0006091</a>
</td>
<td style="text-align:left;">
generation of precursor metabolites and energy
</td>
<td style="text-align:left;">
29/232
</td>
<td style="text-align:left;">
159/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/PGK1/TPI1…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0010608" style="     ">GO:0010608</a>
</td>
<td style="text-align:left;">
posttranscriptional regulation of gene expression
</td>
<td style="text-align:left;">
28/232
</td>
<td style="text-align:left;">
168/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3e-04
</td>
<td style="text-align:left;">
HNRNPU/MATR3/YWHAZ…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009205" style="     ">GO:0009205</a>
</td>
<td style="text-align:left;">
purine ribonucleoside triphosphate metabolic process
</td>
<td style="text-align:left;">
9/232
</td>
<td style="text-align:left;">
21/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4e-04
</td>
<td style="text-align:left;">
RAN/ENO1/ATP5F1B…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0034097" style="     ">GO:0034097</a>
</td>
<td style="text-align:left;">
response to cytokine
</td>
<td style="text-align:left;">
45/232
</td>
<td style="text-align:left;">
357/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4e-04
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/SLC25A5…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0071345" style="     ">GO:0071345</a>
</td>
<td style="text-align:left;">
cellular response to cytokine stimulus
</td>
<td style="text-align:left;">
42/232
</td>
<td style="text-align:left;">
325/3652
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4e-04
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/SLC25A5…
</td>
</tr>
</tbody>
</table></div>
</div>
<p></p>
<p>Notice that <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009060" title="aerobic respiration">GO:0009060</a> is not present in Table <a href="pathway-analysis.html#tab:simplify-cp-ora-bp-table">4.3</a> anymore, since it was highly similar to <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006119" title="oxidative phosphorylation">GO:0006119</a>, and the latter was more significant. After removing redundant GO terms, 96 pass the significance threshold.</p>
<p>Instead of retaining terms based on adjusted p-value, we could ignore significance and instead retain either parent or child terms (<code>select_fun = min</code> and <code>select_fun = max</code>, respectively). The code to do so is provided <a href="https://github.com/YuLab-SMU/clusterProfiler/issues/372" title="New feature: simplify to parent GO terms #372">here</a>.</p>
<p>Another alternative would be to use a gene ID to <a href="http://geneontology.org/docs/go-subset-guide/" title="Guide to GO subsets">GO slim</a> ID conversion table. See the Semantic Similarity subsection of Section <a href="pathway-analysis.html#annotation-databases">4.1</a> for more details.</p>
<p>Now, we will perform GO ORA with the GOstats package. We begin by creating an object of class <code>GOHyperGParams</code> and passing it to <code>hyperGTest</code> for calculation of Hypergeometric p-values. We do not filter by p-value, since adjusted p-values are not provided and we must calculate them ourselves.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## GO BP ORA with GOstats</span>
<span class="va">hyperg_ora_go</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span>
  Class <span class="op">=</span> <span class="st">"GOHyperGParams"</span>,
  ontology <span class="op">=</span> <span class="st">"BP"</span>,
  geneIds <span class="op">=</span> <span class="va">sig_genes</span>,
  universeGeneIds <span class="op">=</span> <span class="va">universe</span>,
  annotation <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">1</span>, <span class="co"># Do not filter by p-value</span>
  testDirection <span class="op">=</span> <span class="st">"over"</span>,
  conditional <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GOstats/man/hyperGTest.html">hyperGTest</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Hypergeometric testing</span></code></pre></div>
<p>Now, we need to filter by gene set size (limit 20 to 500), adjust the p-values, and filter based on the adjusted p-values.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_go_ora</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">hyperg_ora_go</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Filter by size</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Size</span> <span class="op">&gt;=</span> <span class="fl">20</span>, <span class="va">Size</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Adjust p-values</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>p.adjust <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">Pvalue</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Filter and sort by adjusted p-values</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">p.adjust</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">p.adjust</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:20em; ">
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:cp-go-ora-table">Table 4.4: </span>Over-represented biological processes obtained using the GOstats package.
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
GOBPID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Term
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OddsRatio
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
ExpCount
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Count
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Size
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.adjust
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046034" style="     ">GO:0046034</a>
</td>
<td style="text-align:left;">
ATP metabolic process
</td>
<td style="text-align:right;">
5.3654
</td>
<td style="text-align:right;">
6.0986
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
96
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006119" style="     ">GO:0006119</a>
</td>
<td style="text-align:left;">
oxidative phosphorylation
</td>
<td style="text-align:right;">
7.6027
</td>
<td style="text-align:right;">
3.1128
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006402" style="     ">GO:0006402</a>
</td>
<td style="text-align:left;">
mRNA catabolic process
</td>
<td style="text-align:right;">
4.9415
</td>
<td style="text-align:right;">
5.9080
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
93
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:1902600" style="     ">GO:1902600</a>
</td>
<td style="text-align:left;">
proton transmembrane transport
</td>
<td style="text-align:right;">
11.6045
</td>
<td style="text-align:right;">
1.7788
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0016071" style="     ">GO:0016071</a>
</td>
<td style="text-align:left;">
mRNA metabolic process
</td>
<td style="text-align:right;">
3.3346
</td>
<td style="text-align:right;">
13.2136
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
208
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006091" style="     ">GO:0006091</a>
</td>
<td style="text-align:left;">
generation of precursor metabolites and energy
</td>
<td style="text-align:right;">
3.6154
</td>
<td style="text-align:right;">
10.1008
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
159
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009060" style="     ">GO:0009060</a>
</td>
<td style="text-align:left;">
aerobic respiration
</td>
<td style="text-align:right;">
9.2727
</td>
<td style="text-align:right;">
2.0329
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
32
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006401" style="     ">GO:0006401</a>
</td>
<td style="text-align:left;">
RNA catabolic process
</td>
<td style="text-align:right;">
4.2646
</td>
<td style="text-align:right;">
6.6068
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0010608" style="     ">GO:0010608</a>
</td>
<td style="text-align:left;">
posttranscriptional regulation of gene expression
</td>
<td style="text-align:right;">
3.2157
</td>
<td style="text-align:right;">
10.6725
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
168
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3e-04
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0042773" style="     ">GO:0042773</a>
</td>
<td style="text-align:left;">
ATP synthesis coupled electron transport
</td>
<td style="text-align:right;">
8.4615
</td>
<td style="text-align:right;">
1.9693
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3e-04
</td>
</tr>
</tbody>
</table></div>
</div>
<p></p>
<p>There were 4435 biological processes that were tested. Of these, 2049 passed the size filter, and only 182 were significantly over-represented after multiple testing correction. The top 10 are shown in Table <a href="pathway-analysis.html#tab:cp-go-ora-table">4.4</a>. The total number of biological processes that were tested is 1 less than the number reported from the <code>enrichGO</code> results. This is because <code>enrichGO</code> includes the “biological_process” root <a href="https://www.ebi.ac.uk/QuickGO/term/GO:0008150" title="biological process">GO:0008150</a>.</p>
<p><strong>Remove Redundant GO Terms</strong></p>
<p>While there is no function akin to <code><a href="https://rdrr.io/pkg/clusterProfiler/man/simplify-methods.html">clusterProfiler::simplify</a></code> to process results, there is an argument called <code>conditional</code>. Setting <code>conditional</code> to <code>TRUE</code> tells <code>hyperGTest</code> to use the term-term relationships when calculating p-values <span class="citation">(<a href="references-1.html#ref-falcon_how_2021" role="doc-biblioref">Falcon et al., 2021</a>)</span>. This approach is actually more akin to Modular Enrichment Analysis (MEA) than Over-Representation Analysis (also called Singular Enrichment Analysis; <span class="citation"><a href="references-1.html#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al.</a> (<a href="references-1.html#ref-huang_bioinformatics_2009" role="doc-biblioref">2009</a>)</span>).</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Conditional GO BP ORA with GOstats</span>
<span class="va">hyperg_ora_go_sim</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span>
  Class <span class="op">=</span> <span class="st">"GOHyperGParams"</span>,
  ontology <span class="op">=</span> <span class="st">"BP"</span>,
  geneIds <span class="op">=</span> <span class="va">sig_genes</span>,
  universeGeneIds <span class="op">=</span> <span class="va">universe</span>,
  annotation <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">1</span>, <span class="co"># Do not filter by p-value</span>
  testDirection <span class="op">=</span> <span class="st">"over"</span>,
  conditional <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># Use structure of GO graph</span>
<span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GOstats/man/hyperGTest.html">hyperGTest</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Hypergeometric testing</span></code></pre></div>
<p>We will process the results the same as before and compare.</p>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:20em; ">
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:cp-go-ora-simple-table">Table 4.5: </span>Simplified over-represented biological processes obtained using the GOstats package.
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
GOBPID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Term
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OddsRatio
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
ExpCount
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Count
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Size
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.adjust
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046034" style="     ">GO:0046034</a>
</td>
<td style="text-align:left;">
ATP metabolic process
</td>
<td style="text-align:right;">
7.4290
</td>
<td style="text-align:right;">
2.3261
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0011
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0042773" style="     ">GO:0042773</a>
</td>
<td style="text-align:left;">
ATP synthesis coupled electron transport
</td>
<td style="text-align:right;">
8.4615
</td>
<td style="text-align:right;">
1.9693
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0011
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0043043" style="     ">GO:0043043</a>
</td>
<td style="text-align:left;">
peptide biosynthetic process
</td>
<td style="text-align:right;">
3.2288
</td>
<td style="text-align:right;">
9.0208
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
142
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0028
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0046364" style="     ">GO:0046364</a>
</td>
<td style="text-align:left;">
monosaccharide biosynthetic process
</td>
<td style="text-align:right;">
6.7593
</td>
<td style="text-align:right;">
2.2870
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0028
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009145" style="     ">GO:0009145</a>
</td>
<td style="text-align:left;">
purine nucleoside triphosphate biosynthetic process
</td>
<td style="text-align:right;">
10.1429
</td>
<td style="text-align:right;">
1.2705
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009201" style="     ">GO:0009201</a>
</td>
<td style="text-align:left;">
ribonucleoside triphosphate biosynthetic process
</td>
<td style="text-align:right;">
8.6888
</td>
<td style="text-align:right;">
1.3976
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0083
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0006094" style="     ">GO:0006094</a>
</td>
<td style="text-align:left;">
gluconeogenesis
</td>
<td style="text-align:right;">
6.5618
</td>
<td style="text-align:right;">
1.8981
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
1e-04
</td>
<td style="text-align:right;">
0.0127
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0009141" style="     ">GO:0009141</a>
</td>
<td style="text-align:left;">
nucleoside triphosphate metabolic process
</td>
<td style="text-align:right;">
6.2336
</td>
<td style="text-align:right;">
1.9693
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
1e-04
</td>
<td style="text-align:right;">
0.0130
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0034655" style="     ">GO:0034655</a>
</td>
<td style="text-align:left;">
nucleobase-containing compound catabolic process
</td>
<td style="text-align:right;">
2.8081
</td>
<td style="text-align:right;">
9.2114
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
145
</td>
<td style="text-align:right;">
1e-04
</td>
<td style="text-align:right;">
0.0130
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.ebi.ac.uk/QuickGO/term/GO:0048002" style="     ">GO:0048002</a>
</td>
<td style="text-align:left;">
antigen processing and presentation of peptide antigen
</td>
<td style="text-align:right;">
3.8578
</td>
<td style="text-align:right;">
4.4469
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
70
</td>
<td style="text-align:right;">
1e-04
</td>
<td style="text-align:right;">
0.0130
</td>
</tr>
</tbody>
</table></div>
</div>
<p></p>
<p>Now, only 2953 terms showed up in the summary. Of these, 1445 passed the size filter, and 31 were significantly over-represented after multiple testing correction. The top 10 are shown in Table <a href="pathway-analysis.html#tab:cp-go-ora-simple-table">4.5</a>. The major difference between these results and the ones from <code>enrichGO</code> after using <code>simplify</code> is that p-value adjustment is not affected by <code>simplify</code>, as redundant GO terms are removed after. The conditional method is actually more akin to performing modular enrichment analysis (MEA) than ORA (also called singular enrichment analysis)</p>
</div>
<div id="reactome-1" class="section level4" number="4.2.2.2">
<h4>
<span class="header-section-number">4.2.2.2</span> Reactome<a class="anchor" aria-label="anchor" href="#reactome-1"><i class="fas fa-link"></i></a>
</h4>
<p>GOstats does not have a dedicated class to test for Reactome pathway over-representation, but we can use ReactomePA. For this example, we will only consider pathways of size 20 to 500. <code>enrichPathway</code> only accepts Entrez gene IDs as input. In cases where the Entrez ID is not readily available, we must convert to them. Alternatively, we could use the custom ORA function <code>enricher</code>, which allows us to use any ID type (use demonstrated in Pfam example).</p>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Reactome ORA with ReactomePA</span>
<span class="va">cp_ora_reactome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ReactomePA/man/enrichPathway.html">enrichPathway</a></span><span class="op">(</span>
  gene <span class="op">=</span> <span class="va">sig_genes</span>, 
  organism <span class="op">=</span> <span class="st">"human"</span>, 
  pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>, 
  pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>, 
  qvalueCutoff <span class="op">=</span> <span class="fl">1</span>, <span class="co"># Do not filter by q-value</span>
  universe <span class="op">=</span> <span class="va">universe</span>, 
  minGSSize <span class="op">=</span> <span class="fl">20</span>, 
  maxGSSize <span class="op">=</span> <span class="fl">500</span>,
  readable <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># Convert Entrez ID to gene symbol</span>
<span class="op">)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:20em; ">
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:cp-ora-react-table">Table 4.6: </span>Top significantly over-represented Reactome pathways from enrichPathway output.
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Description
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
GeneRatio
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
BgRatio
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.adjust
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
geneID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-8953897" style="     ">R-HSA-8953897</a>
</td>
<td style="text-align:left;">
Cellular responses to external stimuli
</td>
<td style="text-align:left;">
39/193
</td>
<td style="text-align:left;">
185/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/RPL13A/SKP1…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-2262752" style="     ">R-HSA-2262752</a>
</td>
<td style="text-align:left;">
Cellular responses to stress
</td>
<td style="text-align:left;">
37/193
</td>
<td style="text-align:left;">
181/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/RPL13A/SKP1…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-163200" style="     ">R-HSA-163200</a>
</td>
<td style="text-align:left;">
Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.
</td>
<td style="text-align:left;">
15/193
</td>
<td style="text-align:left;">
37/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/COX8A/COX7C…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-5663205" style="     ">R-HSA-5663205</a>
</td>
<td style="text-align:left;">
Infectious disease
</td>
<td style="text-align:left;">
38/193
</td>
<td style="text-align:left;">
212/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
PRKAR1A/SLC25A5/RPL13A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-9711123" style="     ">R-HSA-9711123</a>
</td>
<td style="text-align:left;">
Cellular response to chemical stress
</td>
<td style="text-align:left;">
18/193
</td>
<td style="text-align:left;">
61/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/SKP1/PSMB7…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-9707564" style="     ">R-HSA-9707564</a>
</td>
<td style="text-align:left;">
Cytoprotection by HMOX1
</td>
<td style="text-align:left;">
16/193
</td>
<td style="text-align:left;">
50/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/SKP1/PSMB7…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-1428517" style="     ">R-HSA-1428517</a>
</td>
<td style="text-align:left;">
The citric acid (TCA) cycle and respiratory electron transport
</td>
<td style="text-align:left;">
17/193
</td>
<td style="text-align:left;">
57/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
COX4I1/LDHB/COX8A…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-72766" style="     ">R-HSA-72766</a>
</td>
<td style="text-align:left;">
Translation
</td>
<td style="text-align:left;">
17/193
</td>
<td style="text-align:left;">
58/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
RPL13A/RPL29/KARS1…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-5628897" style="     ">R-HSA-5628897</a>
</td>
<td style="text-align:left;">
TP53 Regulates Metabolic Genes
</td>
<td style="text-align:left;">
12/193
</td>
<td style="text-align:left;">
34/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1e-04
</td>
<td style="text-align:left;">
COX4I1/YWHAZ/YWHAQ…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://reactome.org/content/detail/R-HSA-8953854" style="     ">R-HSA-8953854</a>
</td>
<td style="text-align:left;">
Metabolism of RNA
</td>
<td style="text-align:left;">
31/193
</td>
<td style="text-align:left;">
181/2548
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3e-04
</td>
<td style="text-align:left;">
HNRNPU/YWHAZ/RPL13A…
</td>
</tr>
</tbody>
</table></div>
</div>
<p></p>
<p>There were 919 Reactome pathways that were tested. Of these, 388 passed the size filter, and 155 were significantly over-represented after multiple testing correction. The top 10 are shown in Table <a href="pathway-analysis.html#tab:cp-ora-react-table">4.6</a>.</p>
<!---
#### KEGG {#ora-kegg}


```r
## KEGG ORA with clusterProfiler
cp_ora_kegg <- enrichKEGG(
gene = sig_genes,
organism = "rno", # Rattus norvegicus
keyType = "ncbi-proteinid",
pvalueCutoff = 0.05,
pAdjustMethod = "BH",
universe = universe,
minGSSize = 20,
maxGSSize = 500,
qvalueCutoff = 1,
use_internal_data = FALSE
)
```
--->
</div>
<div id="pfam-1" class="section level4" number="4.2.2.3">
<h4>
<span class="header-section-number">4.2.2.3</span> Pfam<a class="anchor" aria-label="anchor" href="#pfam-1"><i class="fas fa-link"></i></a>
</h4>
<!---
First, we will perform Pfam ORA with the GOstats package using the org.Hs.eg.db human annotation database; however, it is a good idea to check the number of Pfam entries that are provided by this database before doing so. Ideally, it should be close to the number shown at the top of the <a href="http://pfam.xfam.org">main Pfam page</a>. We can accomplish this with functions in AnnotationDbi.


```r
# Map from Entrez gene to Pfam ID
ENTREZ2PFAM <- AnnotationDbi::select(
org.Hs.eg.db, keys = keys(org.Hs.eg.db, keytype = "ENTREZID"),
columns = c("ENTREZID", "PFAM"), keytype = "ENTREZID"
)

# Nice table
kable(head(ENTREZ2PFAM)) %>% 
kable_styling(full_width = FALSE, font_size = 12)
```

<table class="table" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> ENTREZID </th>
   <th style="text-align:left;"> PFAM </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> 1 </td>
   <td style="text-align:left;"> PF13895 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> PF17791 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> PF07703 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> PF07678 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> PF00207 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 2 </td>
   <td style="text-align:left;"> PF17789 </td>
  </tr>
</tbody>
</table>

In this data frame, there are only 6186 Pfam entries. This is not even close to the most up-to-date number. We will continue for the sake of demonstration, but the results are unlikely to be reliable. 
--->
<p>While there is no dedicated ORA function for Pfam entries in clusterProfiler, there is a general ORA function called <code>enricher</code> that can be used with any term to gene conversion table. This function is also much faster than the dedicated <code>enrichGO</code> and <code>enrichPathway</code> functions, so it may be preferred if running many separate over-representation analyses. The conversion table is generated with the biomaRt package as follows.</p>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># NOT RUN ---</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomaRt</span><span class="op">)</span>

<span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,
                dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/View.html">View</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listAttributes.html">listAttributes</a></span><span class="op">(</span><span class="va">mart</span><span class="op">)</span><span class="op">)</span> <span class="co"># Determine which columns to return</span>

<span class="va">conv_tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html">getBM</a></span><span class="op">(</span>filters <span class="op">=</span> <span class="st">"entrezgene_id"</span>,
                  attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"entrezgene_id"</span>,
                                 <span class="st">"pfam"</span><span class="op">)</span>,
                  values <span class="op">=</span> <span class="va">universe</span>,
                  mart <span class="op">=</span> <span class="va">mart</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html">mutate_all</a></span><span class="op">(</span><span class="va">as.character</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">conv_tbl</span>, file <span class="op">=</span> <span class="st">"data/pfam_conv_tbl.rds"</span><span class="op">)</span></code></pre></div>
<p>In addition to the term to gene conversion table, we can supply a term to name table that maps Pfam IDs to human-readable descriptions. While org.Hs.eg.db does not provide Pfam descriptions, the <a href="https://doi.org/doi:10.18129/B9.bioc.PFAM.db" title="PFAM.db: A set of protein ID mappings for PFAM">PFAM.db</a> annotation data package has a Bimap with this information. Now that we have the two conversion tables, we will perform ORA and limit the output to Pfam entries of size 15 to 500.</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Entrez to Pfam ID mapping file created with biomaRt package</span>
<span class="va">TERM2GENE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"data/pfam_conv_tbl.rds"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Remove blank terms</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">pfam</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Reorder and rename columns. </span>
  <span class="co"># ! These specific column names must be used in this exact order</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>term <span class="op">=</span> <span class="va">pfam</span>, gene <span class="op">=</span> <span class="va">entrezgene_id</span><span class="op">)</span>

<span class="co"># Pfam IDs to descriptions</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">PFAM.db</span><span class="op">)</span>
<span class="va">TERM2NAME</span> <span class="op">&lt;-</span> <span class="fu">as.data.frame.Bimap</span><span class="op">(</span><span class="va">PFAMDE</span><span class="op">)</span>
<span class="co"># ! These specific column names must be used in this exact order</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">TERM2NAME</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"term"</span>, <span class="st">"name"</span><span class="op">)</span>

<span class="co">## Pfam ORA with clusterProfiler</span>
<span class="va">cp_ora_pfam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">enricher</a></span><span class="op">(</span>
  gene <span class="op">=</span> <span class="va">sig_genes</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,
  pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,
  universe <span class="op">=</span> <span class="va">universe</span>,
  minGSSize <span class="op">=</span> <span class="fl">15</span>,
  maxGSSize <span class="op">=</span> <span class="fl">500</span>,
  qvalueCutoff <span class="op">=</span> <span class="fl">1</span>,
  TERM2GENE <span class="op">=</span> <span class="va">TERM2GENE</span>,
  TERM2NAME <span class="op">=</span> <span class="va">TERM2NAME</span>
<span class="op">)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:20em; ">
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:cp-ora-pfam-table">Table 4.7: </span>Top Pfam entries from enricher output.
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Description
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
GeneRatio
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
BgRatio
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.adjust
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
geneID
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF01391" style="     ">PF01391</a>
</td>
<td style="text-align:left;">
Collagen triple helix repeat (20 copies)
</td>
<td style="text-align:left;">
7/236
</td>
<td style="text-align:left;">
31/3675
</td>
<td style="text-align:right;">
0.0029
</td>
<td style="text-align:right;">
0.0687
</td>
<td style="text-align:left;">
1293/1281/1277…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00076" style="     ">PF00076</a>
</td>
<td style="text-align:left;">
RNA recognition motif. (a.k.a. RRM, RBD, or RNP domain)
</td>
<td style="text-align:left;">
7/236
</td>
<td style="text-align:left;">
55/3675
</td>
<td style="text-align:right;">
0.0599
</td>
<td style="text-align:right;">
0.5363
</td>
<td style="text-align:left;">
1153/3182/5042…
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00092" style="     ">PF00092</a>
</td>
<td style="text-align:left;">
von Willebrand factor type A domain
</td>
<td style="text-align:left;">
3/236
</td>
<td style="text-align:left;">
15/3675
</td>
<td style="text-align:right;">
0.0670
</td>
<td style="text-align:right;">
0.5363
</td>
<td style="text-align:left;">
1293/1292/1291
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00071" style="     ">PF00071</a>
</td>
<td style="text-align:left;">
Ras family
</td>
<td style="text-align:left;">
3/236
</td>
<td style="text-align:left;">
23/3675
</td>
<td style="text-align:right;">
0.1804
</td>
<td style="text-align:right;">
0.9022
</td>
<td style="text-align:left;">
5901/6009/388
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF07679" style="     ">PF07679</a>
</td>
<td style="text-align:left;">
Immunoglobulin I-set domain
</td>
<td style="text-align:left;">
3/236
</td>
<td style="text-align:left;">
27/3675
</td>
<td style="text-align:right;">
0.2489
</td>
<td style="text-align:right;">
0.9022
</td>
<td style="text-align:left;">
23022/3490/25878
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF13499" style="     ">PF13499</a>
</td>
<td style="text-align:left;">
EF-hand domain pair
</td>
<td style="text-align:left;">
2/236
</td>
<td style="text-align:left;">
20/3675
</td>
<td style="text-align:right;">
0.3711
</td>
<td style="text-align:right;">
0.9022
</td>
<td style="text-align:left;">
6717/80303
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00271" style="     ">PF00271</a>
</td>
<td style="text-align:left;">
Helicase conserved C-terminal domain
</td>
<td style="text-align:left;">
2/236
</td>
<td style="text-align:left;">
24/3675
</td>
<td style="text-align:right;">
0.4623
</td>
<td style="text-align:right;">
0.9022
</td>
<td style="text-align:left;">
1973/10521
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00412" style="     ">PF00412</a>
</td>
<td style="text-align:left;">
LIM domain
</td>
<td style="text-align:left;">
2/236
</td>
<td style="text-align:left;">
24/3675
</td>
<td style="text-align:right;">
0.4623
</td>
<td style="text-align:right;">
0.9022
</td>
<td style="text-align:left;">
1396/9124
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF13855" style="     ">PF13855</a>
</td>
<td style="text-align:left;">
Leucine rich repeat
</td>
<td style="text-align:left;">
3/236
</td>
<td style="text-align:left;">
42/3675
</td>
<td style="text-align:right;">
0.5123
</td>
<td style="text-align:right;">
0.9022
</td>
<td style="text-align:left;">
4060/1634/25878
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00046" style="     ">PF00046</a>
</td>
<td style="text-align:left;">
Homeodomain
</td>
<td style="text-align:left;">
2/236
</td>
<td style="text-align:left;">
27/3675
</td>
<td style="text-align:right;">
0.5254
</td>
<td style="text-align:right;">
0.9022
</td>
<td style="text-align:left;">
5087/29956
</td>
</tr>
</tbody>
</table></div>
</div>
<p></p>
<p>There were 275 Pfam entries that were tested. Of these, 24 passed the size filter, and 0 were significantly over-represented after multiple testing correction. While none were significant, the top 10 are shown in Table <a href="pathway-analysis.html#tab:cp-ora-pfam-table">4.7</a> so that we can compare it to the results from <code>hyperGTest</code>.</p>
<p>Now, we will perform Pfam ORA with the GOstats package. We begin by constructing a new object of class <code>PFAMHyperGParams</code> and then passing it to <code>hyperGTest</code> for calculation of Hypergeometric p-values. We do not filter by p-value, since adjusted p-values are not provided and we must calculate them ourselves.</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Pfam ORA with GOstats</span>
<span class="va">hyperg_pfam_ora</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span>
  Class <span class="op">=</span> <span class="st">"PFAMHyperGParams"</span>,
  geneIds <span class="op">=</span> <span class="va">sig_genes</span>,
  universeGeneIds <span class="op">=</span> <span class="va">universe</span>,
  annotation <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,
  pvalueCutoff <span class="op">=</span> <span class="fl">1</span>,
  testDirection <span class="op">=</span> <span class="st">"over"</span>
<span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/GOstats/man/hyperGTest.html">hyperGTest</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Hypergeometric testing</span></code></pre></div>
<p>Now, we need to filter by gene set size (limit 15 to 500) and adjust the p-values. Normally, we would also filter based on the adjusted p-values, but we will skip this step so that we can compare the output with that of <code><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html">clusterProfiler::enricher</a></code>.</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_pfam_ora</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">hyperg_pfam_ora</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Filter by Pfam entry size</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Size</span> <span class="op">&gt;=</span> <span class="fl">15</span>, <span class="va">Size</span> <span class="op">&lt;=</span> <span class="fl">500</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Adjust p-values</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>p.adjust <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">Pvalue</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="co"># Sort by adjusted p-values</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">p.adjust</span><span class="op">)</span></code></pre></div>
<p>The last step is to add the human-readable Pfam descriptions. We will use the <code>TERM2NAME</code> data frame from before and just change the column names for easy joining.</p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Pfam entries to descriptions</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">TERM2NAME</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"PFAMID"</span>, <span class="st">"Description"</span><span class="op">)</span>
<span class="va">res_pfam_ora</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">res_pfam_ora</span>, <span class="va">TERM2NAME</span><span class="op">)</span></code></pre></div>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:20em; ">
<div class="inline-table"><table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:gostats-pfam-ora-table">Table 4.8: </span>Top Pfam entries obtained using the GOstats package.
</caption>
<thead><tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PFAMID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Description
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
OddsRatio
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
ExpCount
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Count
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Size
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
Pvalue
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
p.adjust
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF01391" style="     ">PF01391</a>
</td>
<td style="text-align:left;">
Collagen triple helix repeat (20 copies)
</td>
<td style="text-align:right;">
4.5163
</td>
<td style="text-align:right;">
1.9368
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0024
</td>
<td style="text-align:right;">
0.0530
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00076" style="     ">PF00076</a>
</td>
<td style="text-align:left;">
RNA recognition motif. (a.k.a. RRM, RBD, or RNP domain)
</td>
<td style="text-align:right;">
2.1478
</td>
<td style="text-align:right;">
3.5509
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
55
</td>
<td style="text-align:right;">
0.0613
</td>
<td style="text-align:right;">
0.4980
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00092" style="     ">PF00092</a>
</td>
<td style="text-align:left;">
von Willebrand factor type A domain
</td>
<td style="text-align:right;">
3.6568
</td>
<td style="text-align:right;">
0.9684
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
0.0679
</td>
<td style="text-align:right;">
0.4980
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00071" style="     ">PF00071</a>
</td>
<td style="text-align:left;">
Ras family
</td>
<td style="text-align:right;">
2.1888
</td>
<td style="text-align:right;">
1.4849
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
0.1824
</td>
<td style="text-align:right;">
0.8976
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF07679" style="     ">PF07679</a>
</td>
<td style="text-align:left;">
Immunoglobulin I-set domain
</td>
<td style="text-align:right;">
1.9016
</td>
<td style="text-align:right;">
1.6786
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
0.2338
</td>
<td style="text-align:right;">
0.8976
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF13499" style="     ">PF13499</a>
</td>
<td style="text-align:left;">
EF-hand domain pair
</td>
<td style="text-align:right;">
1.8182
</td>
<td style="text-align:right;">
1.1621
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
0.3256
</td>
<td style="text-align:right;">
0.8976
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00271" style="     ">PF00271</a>
</td>
<td style="text-align:left;">
Helicase conserved C-terminal domain
</td>
<td style="text-align:right;">
1.3200
</td>
<td style="text-align:right;">
1.5495
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
0.4652
</td>
<td style="text-align:right;">
0.8976
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00412" style="     ">PF00412</a>
</td>
<td style="text-align:left;">
LIM domain
</td>
<td style="text-align:right;">
1.3200
</td>
<td style="text-align:right;">
1.5495
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
0.4652
</td>
<td style="text-align:right;">
0.8976
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF13855" style="     ">PF13855</a>
</td>
<td style="text-align:left;">
Leucine rich repeat
</td>
<td style="text-align:right;">
1.1458
</td>
<td style="text-align:right;">
2.6470
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
0.4995
</td>
<td style="text-align:right;">
0.8976
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="http://pfam.xfam.org/family/PF00595" style="     ">PF00595</a>
</td>
<td style="text-align:left;">
PDZ domain
</td>
<td style="text-align:right;">
1.2092
</td>
<td style="text-align:right;">
1.6786
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
0.5079
</td>
<td style="text-align:right;">
0.8976
</td>
</tr>
</tbody>
</table></div>
</div>
<p></p>
<p>260 Pfam entries were tested. Of these, 16 passed the size filter, and none were significantly over-represented after multiple testing correction. The top 10 are shown in Table <a href="pathway-analysis.html#tab:gostats-pfam-ora-table">4.8</a>.</p>
<p>Notice that this table and Table <a href="pathway-analysis.html#tab:cp-ora-pfam-table">4.7</a> are different. For one, the sizes of the entries are not always the same. For example, <a href="http://pfam.xfam.org/family/PF01391" title="Collagen triple helix repeat (20 copies)">PF01391</a> has size 31 according to the mapping file created from the <code>biomaRt</code> package, but only size 30 according to <code>org.Hs.eg.db</code>. This leads to differences in the number of total Pfam entries that remain after filtering by size and, subsequently, differences in the adjusted p-values.</p>
</div>
</div>
<div id="considerations" class="section level3" number="4.2.3">
<h3>
<span class="header-section-number">4.2.3</span> Considerations<a class="anchor" aria-label="anchor" href="#considerations"><i class="fas fa-link"></i></a>
</h3>
<ol style="list-style-type: decimal">
<li>The choice of the threshold for statistical significance and the multiple comparison adjustment method can greatly impact the analysis <span class="citation">(<a href="references-1.html#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al., 2009</a>)</span>.</li>
<li>ORA fails to incorporate direction of gene regulation. (Are the genes in a given set mainly up or down-regulated?). It is not a good idea to split DEA results by the sign of the logFC and apply ORA to the two sets. Use GSEA instead.
<!--- 
TODO:
Fact-check #3
--->
</li>
<li>If few genes are differentially expressed, ORA is unlikely to yield useful or reliable results. For example, suppose 30 out of 8000 genes are significant. 100 of the genes are annotated to a particular gene set, of which 3 are significant. The associated Hypergeometric p-value is 0.006, and this set would be considered significantly over-represented at the 0.01 level (at least, prior to adjustment for multiple comparisons); however, if only 2 of the genes in this set are significant, this p-value increases 10-fold to 0.0536 and is no longer significant even at the 0.05 level.</li>
<li>If the DEA results are not gene-centric (i.e. DEA was performed at the protein or phosphosite level), then there may be cases where two proteins are associated with the same gene, but only one is significantly differentially abundant. In this case, there is no way to categorize the gene as significant or not, so ORA should not be used. The other problem that could arise if the DEA results are not gene-centric is that the same gene may be counted as significant multiple times, which leads to artificial over-representation.</li>
</ol>
</div>
</div>
<div id="gsea" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> Gene Set Enrichment Analysis<a class="anchor" aria-label="anchor" href="#gsea"><i class="fas fa-link"></i></a>
</h2>
<div id="gsea-overview" class="section level3" number="4.3.1">
<h3>
<span class="header-section-number">4.3.1</span> Overview<a class="anchor" aria-label="anchor" href="#gsea-overview"><i class="fas fa-link"></i></a>
</h3>
<p>Gene Set Enrichment Analysis (GSEA) employs a “no-cutoff” strategy that utilizes some experimental value (such as fold change, the moderated t-statistic, or Z-Score) to rank a list of <span class="math inline">\(N\)</span> genes in descending order. Using this ranked list <span class="math inline">\(L\)</span>, the values of the ranking metric, and an <em>a priori</em> defined gene set <span class="math inline">\(S\)</span>, we can calculate an enrichment score. This is done by “walking down the list <span class="math inline">\(L\)</span>, increasing a running-sum statistic when we encounter a gene in <span class="math inline">\(S\)</span> and decreasing it when we encounter genes not in <span class="math inline">\(S\)</span>.” The maximum deviation from zero of this running-sum statistic is the enrichment score for <span class="math inline">\(S\)</span>, denoted by <span class="math inline">\(ES(S)\)</span>. If the genes in <span class="math inline">\(S\)</span> are randomly distributed throughout <span class="math inline">\(L\)</span>, <span class="math inline">\(ES(S)\)</span> will be relatively small; however, if they are not randomly distributed (i.e. primarily located near either end of <span class="math inline">\(L\)</span>), then <span class="math inline">\(ES(S)\)</span> will be relatively large <span class="citation">(<a href="references-1.html#ref-mootha_pgc-1-responsive_2003" role="doc-biblioref">Mootha et al., 2003</a>; <a href="references-1.html#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al., 2005</a>)</span>.</p>
<p>To assess the statistical significance of each <span class="math inline">\(ES(S)\)</span>, we permute the phenotype labels in the expression data and recalculate the ranking metric and subsequent enrichment score <span class="math inline">\(ES_{\text{NULL}}\)</span>. Alternatively, we could permute the genes: for a given <span class="math inline">\(S\)</span> of size <span class="math inline">\(k\)</span>, we randomly select <span class="math inline">\(k\)</span> genes from the list <span class="math inline">\(L\)</span> and calculate an enrichment score <span class="math inline">\(ES_{\text{NULL}}\)</span>. This is the method implemented by <a href="https://doi.org/doi:10.18129/B9.bioc.fgsea" title="fgsea: Fast Gene Set Enrichment Analysis">fgsea</a> and, consequently, <a href="https://doi.org/doi:10.18129/B9.bioc.clusterProfiler" title="clusterProfiler: A universal enrichment tool for interpreting omics data">clusterProfiler</a>, though it is “not strictly accurate” because it does not preserve the correlations between genes <span class="citation">(<a href="references-1.html#ref-subramanian_gene_2005" role="doc-biblioref">Subramanian et al., 2005</a>)</span>. Regardless of the permutation approach, it is repeated one thousand to tens of thousands of times to produce an empirical null distribution. Then, we define the nominal p-value as the proportion of <span class="math inline">\(ES_{\text{NULL}}\)</span> that are more extreme than <span class="math inline">\(ES(S)\)</span>. Once a p-value has been obtained for every gene set, we apply an appropriate adjustment for multiple hypothesis testing. The phenotype randomization approach is summarized in <a href="https://www.nature.com/articles/ng1180/figures/1" title="Figure 1: Schematic overview of GSEA.">Figure 1: Schematic overview of GSEA</a> in the paper by <span class="citation"><a href="references-1.html#ref-mootha_pgc-1-responsive_2003" role="doc-biblioref">Mootha et al.</a> (<a href="references-1.html#ref-mootha_pgc-1-responsive_2003" role="doc-biblioref">2003</a>)</span>. The mathematical details of GSEA are provided below.</p>
<p>Let <span class="math inline">\(x = \{x_1, ..., x_N : x_i &gt; x_j \ \forall i &lt; j\}\)</span> be the vector of ranking metric values. Also let <span class="math inline">\(S\)</span> be a particular gene set of size <span class="math inline">\(k\)</span> and <span class="math inline">\(NS = \sum\limits_{i \in S}|x_i|\)</span>. We define the value of the running sum at gene <span class="math inline">\(i\)</span> as</p>
<div class="math">
<p><span class="math display">\[
r_i = \begin{cases}
0 &amp; \text{if } i = 0, \\
r_{i-1} + \frac{1}{NS}|x_i| &amp; \text{if } 1 \leq i \leq N \text{ and } i \in S, \\
r_{i-1} - \frac{1}{N-k} &amp; \text{if } 1 \leq i \leq N \text{ and } i \not\in S
\end{cases}
\]</span></p>
</div>
<p><span class="math inline">\(ES(S)\)</span> is the largest (in terms of absolute value) element of <span class="math inline">\(r_N\)</span>. This notation is a slightly modified version of what is presented in the paper by <span class="citation"><a href="references-1.html#ref-korotkevich_fast_2016" role="doc-biblioref">Korotkevich et al.</a> (<a href="references-1.html#ref-korotkevich_fast_2016" role="doc-biblioref">2016</a>)</span>.</p>
</div>
<div id="examples" class="section level3" number="4.3.2">
<h3>
<span class="header-section-number">4.3.2</span> Examples<a class="anchor" aria-label="anchor" href="#examples"><i class="fas fa-link"></i></a>
</h3>
<div id="gene-ontology-2" class="section level4" number="4.3.2.1">
<h4>
<span class="header-section-number">4.3.2.1</span> Gene Ontology<a class="anchor" aria-label="anchor" href="#gene-ontology-2"><i class="fas fa-link"></i></a>
</h4>
</div>
<div id="reactome-2" class="section level4" number="4.3.2.2">
<h4>
<span class="header-section-number">4.3.2.2</span> Reactome<a class="anchor" aria-label="anchor" href="#reactome-2"><i class="fas fa-link"></i></a>
</h4>
</div>
<div id="pfam-2" class="section level4" number="4.3.2.3">
<h4>
<span class="header-section-number">4.3.2.3</span> Pfam<a class="anchor" aria-label="anchor" href="#pfam-2"><i class="fas fa-link"></i></a>
</h4>
</div>
</div>
<div id="considerations-1" class="section level3" number="4.3.3">
<h3>
<span class="header-section-number">4.3.3</span> Considerations<a class="anchor" aria-label="anchor" href="#considerations-1"><i class="fas fa-link"></i></a>
</h3>
<ol style="list-style-type: decimal">
<li>GSEA is not influenced by an arbitrary cutoff for statistical significance. This is especially useful when only a few features pass this threshold after adjustment for multiple testing, and it also means that it does not depend on the choice of p-value adjustment.</li>
<li>All genes in an experiment are allowed to “contribute to the enrichment analysis in differing degrees” <span class="citation">(<a href="references-1.html#ref-huang_bioinformatics_2009" role="doc-biblioref">Huang et al., 2009</a>)</span>. This addresses the point brought up at the beginning of Section <a href="pathway-analysis.html#pathway-analysis">4</a> that biological changes may be the result of small changes in many related genes. GSEA allows for the detection of such changes.</li>
<li>The sign of the enrichment score tell us in which category a set of genes is primarily up- or down-regulated. For example, suppose we test the difference between the means of two groups for each gene using the <a href="https://doi.org/doi:10.18129/B9.bioc.limma" title="limma: Linear Models for Microarray Data">limma</a> package. Since ranking metrics incorporate the direction of change, a positive enrichment score would mean that the majority of genes in that particular set are up-regulated in A relative to B/down-regulated in B relative to A.</li>
<li>The choice of ranking metric is important.
<!---
Talk more about ranking metrics and include sources!!!
--->
</li>
</ol>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="DEA.html"><span class="header-section-number">3</span> Differential Expression Analysis</a></div>
<div class="next"><a href="references-1.html">References</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#pathway-analysis"><span class="header-section-number">4</span> Pathway Analysis</a></li>
<li>
<a class="nav-link" href="#annotation-databases"><span class="header-section-number">4.1</span> Annotation Databases</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#gene-ontology"><span class="header-section-number">4.1.1</span> Gene Ontology</a></li>
<li><a class="nav-link" href="#reactome"><span class="header-section-number">4.1.2</span> Reactome</a></li>
<li><a class="nav-link" href="#kegg"><span class="header-section-number">4.1.3</span> KEGG</a></li>
<li><a class="nav-link" href="#pfam"><span class="header-section-number">4.1.4</span> Pfam</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#ora"><span class="header-section-number">4.2</span> Over-Representation Analysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#ora-overview"><span class="header-section-number">4.2.1</span> Overview</a></li>
<li><a class="nav-link" href="#ora-examples"><span class="header-section-number">4.2.2</span> Examples</a></li>
<li><a class="nav-link" href="#considerations"><span class="header-section-number">4.2.3</span> Considerations</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#gsea"><span class="header-section-number">4.3</span> Gene Set Enrichment Analysis</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#gsea-overview"><span class="header-section-number">4.3.1</span> Overview</a></li>
<li><a class="nav-link" href="#examples"><span class="header-section-number">4.3.2</span> Examples</a></li>
<li><a class="nav-link" href="#considerations-1"><span class="header-section-number">4.3.3</span> Considerations</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial/blob/master/pathway_analysis.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial/edit/master/pathway_analysis.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Proteomics Data Analysis in R/Bioconductor</strong>" was written by Tyler Sagendorf. It was last built on October 21, 2021.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
