<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1.2 Phosphoproteomics Data | Proteomics Data Analysis in R/Bioconductor</title>
  <meta name="description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="1.2 Phosphoproteomics Data | Proteomics Data Analysis in R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1.2 Phosphoproteomics Data | Proteomics Data Analysis in R/Bioconductor" />
  
  <meta name="twitter:description" content="This is a tutorial for proteomics data analysis in R that utilizes packages developed by researchers at PNNL and from Bioconductor." />
  

<meta name="author" content="Tyler Sagendorf" />


<meta name="date" content="2022-01-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="global-proteomics-data.html"/>
<link rel="next" href="label-free-quantification.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=".">Proteomics Data Analysis in R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome!</a></li>
<li class="chapter" data-level="1" data-path="isobaric-quantification-pipelines.html"><a href="isobaric-quantification-pipelines.html"><i class="fa fa-check"></i><b>1</b> Isobaric Quantification Pipelines</a>
<ul>
<li class="chapter" data-level="1.1" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html"><i class="fa fa-check"></i><b>1.1</b> Global Proteomics Data</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#read-ms-gf-data"><i class="fa fa-check"></i><b>1.1.1</b> Read MS-GF+ Data</a></li>
<li class="chapter" data-level="1.1.2" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#correct-isotope-selection-error"><i class="fa fa-check"></i><b>1.1.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.1.3" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#remove-contaminants"><i class="fa fa-check"></i><b>1.1.3</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.1.4" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#global-peptide-filter"><i class="fa fa-check"></i><b>1.1.4</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.1.5" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#msms-id-filter-protein-level"><i class="fa fa-check"></i><b>1.1.5</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.1.6" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#inference-of-parsimonious-protein-set"><i class="fa fa-check"></i><b>1.1.6</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.1.7" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#remove-decoy-psms"><i class="fa fa-check"></i><b>1.1.7</b> Remove Decoy PSMs</a></li>
<li class="chapter" data-level="1.1.8" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#read-masic-output"><i class="fa fa-check"></i><b>1.1.8</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.1.9" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#filter-masic-data"><i class="fa fa-check"></i><b>1.1.9</b> Filter MASIC Data</a></li>
<li class="chapter" data-level="1.1.10" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#fetch-study-design-tables"><i class="fa fa-check"></i><b>1.1.10</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="1.1.11" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#fractions"><i class="fa fa-check"></i><b>1.1.11</b> Fractions</a></li>
<li class="chapter" data-level="1.1.12" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#samples"><i class="fa fa-check"></i><b>1.1.12</b> Samples</a></li>
<li class="chapter" data-level="1.1.13" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#references"><i class="fa fa-check"></i><b>1.1.13</b> References</a></li>
<li class="chapter" data-level="1.1.14" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#global-quant-crosstab"><i class="fa fa-check"></i><b>1.1.14</b> Create Quantitative Cross-tab</a></li>
<li class="chapter" data-level="1.1.15" data-path="global-proteomics-data.html"><a href="global-proteomics-data.html#create-msnset-object"><i class="fa fa-check"></i><b>1.1.15</b> Create MSnSet Object</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html"><i class="fa fa-check"></i><b>1.2</b> Phosphoproteomics Data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#read-ms-gf-output"><i class="fa fa-check"></i><b>1.2.1</b> Read MS-GF+ Output</a></li>
<li class="chapter" data-level="1.2.2" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#correct-isotope-selection-error-1"><i class="fa fa-check"></i><b>1.2.2</b> Correct Isotope Selection Error</a></li>
<li class="chapter" data-level="1.2.3" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-non-phosphorylated-peptides"><i class="fa fa-check"></i><b>1.2.3</b> Remove Non-Phosphorylated Peptides</a></li>
<li class="chapter" data-level="1.2.4" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-contaminants-1"><i class="fa fa-check"></i><b>1.2.4</b> Remove Contaminants</a></li>
<li class="chapter" data-level="1.2.5" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#improve-phosphosite-localization"><i class="fa fa-check"></i><b>1.2.5</b> Improve Phosphosite Localization</a></li>
<li class="chapter" data-level="1.2.6" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#msms-id-filter-peptide-level"><i class="fa fa-check"></i><b>1.2.6</b> MS/MS ID Filter: Peptide Level</a></li>
<li class="chapter" data-level="1.2.7" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#msms-id-filter-protein-level-1"><i class="fa fa-check"></i><b>1.2.7</b> MS/MS ID Filter: Protein Level</a></li>
<li class="chapter" data-level="1.2.8" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#inference-of-parsimonious-protein-set-1"><i class="fa fa-check"></i><b>1.2.8</b> Inference of Parsimonious Protein Set</a></li>
<li class="chapter" data-level="1.2.9" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#remove-decoy-psms-1"><i class="fa fa-check"></i><b>1.2.9</b> Remove Decoy PSMs</a></li>
<li class="chapter" data-level="1.2.10" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#map-sites-to-protein-sequences"><i class="fa fa-check"></i><b>1.2.10</b> Map Sites to Protein Sequences</a></li>
<li class="chapter" data-level="1.2.11" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#read-masic-output-1"><i class="fa fa-check"></i><b>1.2.11</b> Read MASIC Output</a></li>
<li class="chapter" data-level="1.2.12" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#filter-masic-data-1"><i class="fa fa-check"></i><b>1.2.12</b> Filter MASIC Data</a></li>
<li class="chapter" data-level="1.2.13" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#create-study-design-tables"><i class="fa fa-check"></i><b>1.2.13</b> Create Study Design Tables</a></li>
<li class="chapter" data-level="1.2.14" data-path="phosphoproteomics-data.html"><a href="phosphoproteomics-data.html#phospho-create-crosstab"><i class="fa fa-check"></i><b>1.2.14</b> Create Quantitative Cross-tab</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="label-free-quantification.html"><a href="label-free-quantification.html"><i class="fa fa-check"></i><b>2</b> Label-Free Quantification</a></li>
<li class="chapter" data-level="3" data-path="exploratory-data-analysis.html"><a href="exploratory-data-analysis.html"><i class="fa fa-check"></i><b>3</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="count-features-in-samples.html"><a href="count-features-in-samples.html"><i class="fa fa-check"></i><b>3.1</b> Count Features in Samples</a></li>
<li class="chapter" data-level="3.2" data-path="estimate-blood-contamination.html"><a href="estimate-blood-contamination.html"><i class="fa fa-check"></i><b>3.2</b> Estimate Blood Contamination</a></li>
<li class="chapter" data-level="3.3" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>3.3</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="heatmaps.html"><a href="heatmaps.html"><i class="fa fa-check"></i><b>4</b> Heatmaps</a>
<ul>
<li class="chapter" data-level="4.1" data-path="expression-heatmaps.html"><a href="expression-heatmaps.html"><i class="fa fa-check"></i><b>4.1</b> Expression Heatmaps</a></li>
<li class="chapter" data-level="4.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html"><i class="fa fa-check"></i><b>4.2</b> Correlation Heatmaps</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#sample-correlation"><i class="fa fa-check"></i><b>4.2.1</b> Sample Correlation</a></li>
<li class="chapter" data-level="4.2.2" data-path="correlation-heatmaps.html"><a href="correlation-heatmaps.html#feature-correlation"><i class="fa fa-check"></i><b>4.2.2</b> Feature Correlation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html"><i class="fa fa-check"></i><b>4.3</b> Heatmap Annotation</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="heatmap-annotation.html"><a href="heatmap-annotation.html#modifying-default-colors"><i class="fa fa-check"></i><b>4.3.1</b> Modifying Default Colors</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="additional-modifications.html"><a href="additional-modifications.html"><i class="fa fa-check"></i><b>4.4</b> Additional Modifications</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="additional-modifications.html"><a href="additional-modifications.html#change-row-or-column-labels"><i class="fa fa-check"></i><b>4.4.1</b> Change row or column labels</a></li>
<li class="chapter" data-level="4.4.2" data-path="additional-modifications.html"><a href="additional-modifications.html#change-label-colors"><i class="fa fa-check"></i><b>4.4.2</b> Change label colors</a></li>
<li class="chapter" data-level="4.4.3" data-path="additional-modifications.html"><a href="additional-modifications.html#label-specific-features"><i class="fa fa-check"></i><b>4.4.3</b> Label specific features</a></li>
<li class="chapter" data-level="4.4.4" data-path="additional-modifications.html"><a href="additional-modifications.html#change-heatmap-body-color"><i class="fa fa-check"></i><b>4.4.4</b> Change heatmap body color</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="DEA.html"><a href="DEA.html"><i class="fa fa-check"></i><b>5</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="linear-regression.html"><a href="linear-regression.html"><i class="fa fa-check"></i><b>5.1</b> Linear Regression</a></li>
<li class="chapter" data-level="5.2" data-path="one-way-anova.html"><a href="one-way-anova.html"><i class="fa fa-check"></i><b>5.2</b> One-Way ANOVA</a></li>
<li class="chapter" data-level="5.3" data-path="t-tests.html"><a href="t-tests.html"><i class="fa fa-check"></i><b>5.3</b> t-tests</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="t-tests.html"><a href="t-tests.html#one-comparison"><i class="fa fa-check"></i><b>5.3.1</b> One Comparison</a></li>
<li class="chapter" data-level="5.3.2" data-path="t-tests.html"><a href="t-tests.html#multiple-comparisons"><i class="fa fa-check"></i><b>5.3.2</b> Multiple Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="p-value-histograms.html"><a href="p-value-histograms.html"><i class="fa fa-check"></i><b>5.4</b> p-value Histograms</a></li>
<li class="chapter" data-level="5.5" data-path="volcano-plots.html"><a href="volcano-plots.html"><i class="fa fa-check"></i><b>5.5</b> Volcano Plots</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>6</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="annotation-databases.html"><a href="annotation-databases.html"><i class="fa fa-check"></i><b>6.1</b> Annotation Databases</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#gene-ontology"><i class="fa fa-check"></i><b>6.1.1</b> Gene Ontology</a>
<ul>
<li class="chapter" data-level="6.1.1.1" data-path="annotation-databases.html"><a href="annotation-databases.html#semantic-similarity"><i class="fa fa-check"></i><b>6.1.1.1</b> Semantic Similarity</a></li>
<li class="chapter" data-level="6.1.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#go-slim"><i class="fa fa-check"></i><b>6.1.1.2</b> GO Subsets/Slims</a></li>
</ul></li>
<li class="chapter" data-level="6.1.2" data-path="annotation-databases.html"><a href="annotation-databases.html#reactome"><i class="fa fa-check"></i><b>6.1.2</b> Reactome</a></li>
<li class="chapter" data-level="6.1.3" data-path="annotation-databases.html"><a href="annotation-databases.html#kegg"><i class="fa fa-check"></i><b>6.1.3</b> KEGG</a></li>
<li class="chapter" data-level="6.1.4" data-path="annotation-databases.html"><a href="annotation-databases.html#pfam"><i class="fa fa-check"></i><b>6.1.4</b> Pfam</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="ora.html"><a href="ora.html"><i class="fa fa-check"></i><b>6.2</b> Over-Representation Analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="ora.html"><a href="ora.html#ora-overview"><i class="fa fa-check"></i><b>6.2.1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="ora.html"><a href="ora.html#important-considerations"><i class="fa fa-check"></i>Important Considerations</a></li>
</ul></li>
<li class="chapter" data-level="6.2.2" data-path="ora.html"><a href="ora.html#ora-examples"><i class="fa fa-check"></i><b>6.2.2</b> Examples</a>
<ul>
<li class="chapter" data-level="6.2.2.1" data-path="ora.html"><a href="ora.html#ora-go"><i class="fa fa-check"></i><b>6.2.2.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="6.2.2.2" data-path="ora.html"><a href="ora.html#ora-reactome"><i class="fa fa-check"></i><b>6.2.2.2</b> Reactome</a></li>
<li class="chapter" data-level="6.2.2.3" data-path="ora.html"><a href="ora.html#ora-pfam"><i class="fa fa-check"></i><b>6.2.2.3</b> Pfam</a></li>
<li class="chapter" data-level="6.2.2.4" data-path="ora.html"><a href="ora.html#ora-other"><i class="fa fa-check"></i><b>6.2.2.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="gsea.html"><a href="gsea.html"><i class="fa fa-check"></i><b>6.3</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="gsea.html"><a href="gsea.html#gsea-overview"><i class="fa fa-check"></i><b>6.3.1</b> Overview</a>
<ul>
<li class="chapter" data-level="" data-path="gsea.html"><a href="gsea.html#important-considerations-1"><i class="fa fa-check"></i>Important Considerations</a></li>
</ul></li>
<li class="chapter" data-level="6.3.2" data-path="gsea.html"><a href="gsea.html#examples"><i class="fa fa-check"></i><b>6.3.2</b> Examples</a>
<ul>
<li class="chapter" data-level="6.3.2.1" data-path="gsea.html"><a href="gsea.html#gene-ontology-1"><i class="fa fa-check"></i><b>6.3.2.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="6.3.2.2" data-path="gsea.html"><a href="gsea.html#reactome-1"><i class="fa fa-check"></i><b>6.3.2.2</b> Reactome</a></li>
<li class="chapter" data-level="6.3.2.3" data-path="gsea.html"><a href="gsea.html#pfam-1"><i class="fa fa-check"></i><b>6.3.2.3</b> Pfam</a></li>
<li class="chapter" data-level="6.3.2.4" data-path="gsea.html"><a href="gsea.html#other-databases"><i class="fa fa-check"></i><b>6.3.2.4</b> Other Databases</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Proteomics Data Analysis in R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phosphoproteomics-data" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Phosphoproteomics Data</h2>
<p>This pipeline shows how to process phosphoproteomics data from the DMS. Phosphoproteomics deals with proteins that have been phosphorylated. Phosphorylation is a type of reversible post-translational modification (PTM; a protein modification that occurs after protein synthesis) in which a protein kinase attaches a phosphate group to an amino acid residue. This change in a protein’s structure can completely alter aspects such as its biological activity and protein-protein interactions, and “abnormal phosphorylation is now recognized as a cause or consequence of many human diseases” <span class="citation">(<a href="#ref-cohen_origins_2002" role="doc-biblioref">Cohen, 2002</a>)</span>. The most commonly phosphorylated amino acids are serine (S), threonine (T), and tyrosine (Y).</p>
<p>For this section, we will use data package number <code>3626</code>. We will need the <code>PlexedPiper</code> package for isobaric quantification and <code>PNNL.DMS.utils</code> to interface with the DMS. Also, details will be omitted if they were already provided in Section <a href="global-proteomics-data.html#global-proteomics-data">1.1</a>.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="phosphoproteomics-data.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup</span></span>
<span id="cb37-2"><a href="phosphoproteomics-data.html#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PNNL.DMS.utils)</span>
<span id="cb37-3"><a href="phosphoproteomics-data.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PlexedPiper)</span>
<span id="cb37-4"><a href="phosphoproteomics-data.html#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb37-5"><a href="phosphoproteomics-data.html#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr) <span class="co"># %&gt;%</span></span>
<span id="cb37-6"><a href="phosphoproteomics-data.html#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MSnbase)</span></code></pre></div>
<div id="read-ms-gf-output" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Read MS-GF+ Output</h3>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="phosphoproteomics-data.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MS-GF+ data</span></span>
<span id="cb38-2"><a href="phosphoproteomics-data.html#cb38-2" aria-hidden="true" tabindex="-1"></a>data_package_num <span class="ot">&lt;-</span> <span class="dv">3626</span></span>
<span id="cb38-3"><a href="phosphoproteomics-data.html#cb38-3" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">read_msgf_data_from_DMS</span>(data_package_num)</span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="phosphoproteomics-data.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 612667 at 55 % FDR
## #peptides: 396540 at 75 % FDR
## #accessions: 121521 at 98 % FDR</code></pre>
</div>
<div id="correct-isotope-selection-error-1" class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Correct Isotope Selection Error</h3>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="phosphoproteomics-data.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Correct for isotope selection error</span></span>
<span id="cb41-2"><a href="phosphoproteomics-data.html#cb41-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">correct_peak_selection</span>(msnid)</span></code></pre></div>
</div>
<div id="remove-non-phosphorylated-peptides" class="section level3" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> Remove Non-Phosphorylated Peptides</h3>
<p>In this case, the phosphorylation of an amino acid is marked by a <code>*</code> inserted into the sequence after said amino acid. We will not consider unmodified peptides, so we can filter out peptides that do not contain this symbol with <code>apply_filter</code>. The <code>*</code> is a special character that must be escaped with backslashes, and the backslashes must also be escaped, since they are enclosed within a nested string (<code>"''"</code>).</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="phosphoproteomics-data.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove non-phosphorylated peptides</span></span>
<span id="cb42-2"><a href="phosphoproteomics-data.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co"># (peptides that do not contain a *)</span></span>
<span id="cb42-3"><a href="phosphoproteomics-data.html#cb42-3" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">apply_filter</span>(msnid, <span class="st">&quot;grepl(&#39;</span><span class="sc">\\\\</span><span class="st">*&#39;, peptide)&quot;</span>)</span>
<span id="cb42-4"><a href="phosphoproteomics-data.html#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 537749 at 57 % FDR
## #peptides: 353634 at 76 % FDR
## #accessions: 118817 at 98 % FDR</code></pre>
</div>
<div id="remove-contaminants-1" class="section level3" number="1.2.4">
<h3><span class="header-section-number">1.2.4</span> Remove Contaminants</h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="phosphoproteomics-data.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove contaminants</span></span>
<span id="cb44-2"><a href="phosphoproteomics-data.html#cb44-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">apply_filter</span>(msnid, <span class="st">&quot;!grepl(&#39;Contaminant&#39;, accession)&quot;</span>)</span>
<span id="cb44-3"><a href="phosphoproteomics-data.html#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 537572 at 57 % FDR
## #peptides: 353489 at 76 % FDR
## #accessions: 118797 at 98 % FDR</code></pre>
</div>
<div id="improve-phosphosite-localization" class="section level3" number="1.2.5">
<h3><span class="header-section-number">1.2.5</span> Improve Phosphosite Localization</h3>
<p>Phospho datasets involve AScore jobs for improving phosphosite localization. There should be one AScore job per data package. If the AScore job does not exist, see <a href="https://prismwiki.pnl.gov/wiki/AScore_Job_Creation">AScore Job Creation</a> for how to set it up. The fetched object is a data.frame that links datasets, scans and original PTM localization to newly suggested locations. Importantly, it contains <code>AScore</code> column that signifies the confidence of PTM assignment. AScore &gt; 17 is considered confident.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="phosphoproteomics-data.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter PTMs by Ascore</span></span>
<span id="cb46-2"><a href="phosphoproteomics-data.html#cb46-2" aria-hidden="true" tabindex="-1"></a>ascore <span class="ot">&lt;-</span> <span class="fu">get_AScore_results</span>(data_package_num)</span>
<span id="cb46-3"><a href="phosphoproteomics-data.html#cb46-3" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">best_PTM_location_by_ascore</span>(msnid, ascore)</span></code></pre></div>
<!---
Why are some of the original sequences in ascore not phosphorylated? (AScore = -1)
--->
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="phosphoproteomics-data.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 188791 at 30 % FDR
## #peptides: 101873 at 53 % FDR
## #accessions: 90677 at 93 % FDR</code></pre>
</div>
<div id="msms-id-filter-peptide-level" class="section level3" number="1.2.6">
<h3><span class="header-section-number">1.2.6</span> MS/MS ID Filter: Peptide Level</h3>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="phosphoproteomics-data.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1% FDR filter at the peptide level</span></span>
<span id="cb49-2"><a href="phosphoproteomics-data.html#cb49-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">filter_msgf_data</span>(msnid,</span>
<span id="cb49-3"><a href="phosphoproteomics-data.html#cb49-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">level =</span> <span class="st">&quot;peptide&quot;</span>,</span>
<span id="cb49-4"><a href="phosphoproteomics-data.html#cb49-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fdr.max =</span> <span class="fl">0.01</span>)</span>
<span id="cb49-5"><a href="phosphoproteomics-data.html#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 76103 at 0.49 % FDR
## #peptides: 23378 at 1 % FDR
## #accessions: 16090 at 4.7 % FDR</code></pre>
</div>
<div id="msms-id-filter-protein-level-1" class="section level3" number="1.2.7">
<h3><span class="header-section-number">1.2.7</span> MS/MS ID Filter: Protein Level</h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="phosphoproteomics-data.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get path to FASTA file</span></span>
<span id="cb51-2"><a href="phosphoproteomics-data.html#cb51-2" aria-hidden="true" tabindex="-1"></a>path_to_FASTA <span class="ot">&lt;-</span> <span class="fu">path_to_FASTA_used_by_DMS</span>(data_package_num)</span>
<span id="cb51-3"><a href="phosphoproteomics-data.html#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="phosphoproteomics-data.html#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute number of peptides per 1000 amino acids</span></span>
<span id="cb51-5"><a href="phosphoproteomics-data.html#cb51-5" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">compute_num_peptides_per_1000aa</span>(msnid, path_to_FASTA)</span>
<span id="cb51-6"><a href="phosphoproteomics-data.html#cb51-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-7"><a href="phosphoproteomics-data.html#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 1% FDR filter at the protein level</span></span>
<span id="cb51-8"><a href="phosphoproteomics-data.html#cb51-8" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">filter_msgf_data</span>(msnid,</span>
<span id="cb51-9"><a href="phosphoproteomics-data.html#cb51-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">level =</span> <span class="st">&quot;accession&quot;</span>,</span>
<span id="cb51-10"><a href="phosphoproteomics-data.html#cb51-10" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fdr.max =</span> <span class="fl">0.01</span>)</span>
<span id="cb51-11"><a href="phosphoproteomics-data.html#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 72481 at 0.12 % FDR
## #peptides: 21266 at 0.26 % FDR
## #accessions: 9424 at 0.98 % FDR</code></pre>
</div>
<div id="inference-of-parsimonious-protein-set-1" class="section level3" number="1.2.8">
<h3><span class="header-section-number">1.2.8</span> Inference of Parsimonious Protein Set</h3>
<!---
TODO:
* Talk about using prior information from global cross-tab to improve inference of the parsimonious protein set for phospho data.
--->
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="phosphoproteomics-data.html#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load proteins from global crosstab</span></span>
<span id="cb53-2"><a href="phosphoproteomics-data.html#cb53-2" aria-hidden="true" tabindex="-1"></a>global_proteins <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/3442_global_protein_names.rds&quot;</span>)</span>
<span id="cb53-3"><a href="phosphoproteomics-data.html#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Inference of parsimonious protein set</span></span>
<span id="cb53-4"><a href="phosphoproteomics-data.html#cb53-4" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">infer_parsimonious_accessions</span>(msnid, <span class="at">unique_only =</span> <span class="cn">FALSE</span>,</span>
<span id="cb53-5"><a href="phosphoproteomics-data.html#cb53-5" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">prior =</span> global_proteins)</span>
<span id="cb53-6"><a href="phosphoproteomics-data.html#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 72481 at 0.12 % FDR
## #peptides: 21266 at 0.26 % FDR
## #accessions: 2895 at 1.6 % FDR</code></pre>
</div>
<div id="remove-decoy-psms-1" class="section level3" number="1.2.9">
<h3><span class="header-section-number">1.2.9</span> Remove Decoy PSMs</h3>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="phosphoproteomics-data.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove Decoy PSMs</span></span>
<span id="cb55-2"><a href="phosphoproteomics-data.html#cb55-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">apply_filter</span>(msnid, <span class="st">&quot;!isDecoy&quot;</span>)</span>
<span id="cb55-3"><a href="phosphoproteomics-data.html#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(msnid)</span></code></pre></div>
<pre><code>## MSnID object
## Working directory: &quot;.&quot;
## #Spectrum Files:  23 
## #PSMs: 72391 at 0 % FDR
## #peptides: 21211 at 0 % FDR
## #accessions: 2849 at 0 % FDR</code></pre>
</div>
<div id="map-sites-to-protein-sequences" class="section level3" number="1.2.10">
<h3><span class="header-section-number">1.2.10</span> Map Sites to Protein Sequences</h3>
<p>Prepare FASTA to make sure entry names in FASTA file match MSnID accessions. The plan is to make this conversion automatic. <code>map_mod_sites</code> creates a number of columns describing mapping of the sites onto the protein sequences. The most important for the user is <code>SiteID</code>.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="phosphoproteomics-data.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create AAStringSet</span></span>
<span id="cb57-2"><a href="phosphoproteomics-data.html#cb57-2" aria-hidden="true" tabindex="-1"></a>fst <span class="ot">&lt;-</span> <span class="fu">readAAStringSet</span>(path_to_FASTA)</span>
<span id="cb57-3"><a href="phosphoproteomics-data.html#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove contaminants</span></span>
<span id="cb57-4"><a href="phosphoproteomics-data.html#cb57-4" aria-hidden="true" tabindex="-1"></a>fst <span class="ot">&lt;-</span> fst[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;Contaminant&quot;</span>, <span class="fu">names</span>(fst)), ]</span>
<span id="cb57-5"><a href="phosphoproteomics-data.html#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 names</span></span>
<span id="cb57-6"><a href="phosphoproteomics-data.html#cb57-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">names</span>(fst))</span></code></pre></div>
<pre><code>## [1] &quot;NP_783171.2 cathepsin R precursor [Rattus norvegicus]&quot;                    
## [2] &quot;NP_001101862.2 zinc finger protein ZIC 2 [Rattus norvegicus]&quot;             
## [3] &quot;NP_113721.4 UDP-glucuronosyltransferase 2B2 precursor [Rattus norvegicus]&quot;
## [4] &quot;NP_714948.1 Ly-49 stimulatory receptor 3 [Rattus norvegicus]&quot;             
## [5] &quot;NP_001000704.1 olfactory receptor Olr931 [Rattus norvegicus]&quot;             
## [6] &quot;NP_001000638.1 olfactory receptor Olr652 [Rattus norvegicus]&quot;</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="phosphoproteomics-data.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Modify names to match accessions(msnid)</span></span>
<span id="cb59-2"><a href="phosphoproteomics-data.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(fst) <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(<span class="fu">names</span>(fst), <span class="at">split =</span> <span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb59-3"><a href="phosphoproteomics-data.html#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select text before first space</span></span>
<span id="cb59-4"><a href="phosphoproteomics-data.html#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(<span class="cf">function</span>(x) x[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb59-5"><a href="phosphoproteomics-data.html#cb59-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>()</span>
<span id="cb59-6"><a href="phosphoproteomics-data.html#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="co"># First 6 names</span></span>
<span id="cb59-7"><a href="phosphoproteomics-data.html#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">names</span>(fst))</span></code></pre></div>
<pre><code>## [1] &quot;NP_783171.2&quot;    &quot;NP_001101862.2&quot; &quot;NP_113721.4&quot;    &quot;NP_714948.1&quot;   
## [5] &quot;NP_001000704.1&quot; &quot;NP_001000638.1&quot;</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="phosphoproteomics-data.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Main mapping call</span></span>
<span id="cb61-2"><a href="phosphoproteomics-data.html#cb61-2" aria-hidden="true" tabindex="-1"></a>msnid <span class="ot">&lt;-</span> <span class="fu">map_mod_sites</span>(<span class="at">object =</span> msnid, <span class="at">fasta =</span> fst, </span>
<span id="cb61-3"><a href="phosphoproteomics-data.html#cb61-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">accession_col =</span> <span class="st">&quot;accession&quot;</span>, </span>
<span id="cb61-4"><a href="phosphoproteomics-data.html#cb61-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">peptide_mod_col =</span> <span class="st">&quot;peptide&quot;</span>, </span>
<span id="cb61-5"><a href="phosphoproteomics-data.html#cb61-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mod_char =</span> <span class="st">&quot;*&quot;</span>,</span>
<span id="cb61-6"><a href="phosphoproteomics-data.html#cb61-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">site_delimiter =</span> <span class="st">&quot;lower&quot;</span>)</span></code></pre></div>
<table class="table table-hover table-condensed" style="font-size: 12px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:phospho-msnid-table">Table 1.8: </span><left>First 6 rows of the processed MS-GF+ results.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:right;">
ResultID
</th>
<th style="text-align:right;">
Scan
</th>
<th style="text-align:left;">
FragMethod
</th>
<th style="text-align:right;">
SpecIndex
</th>
<th style="text-align:right;">
Charge
</th>
<th style="text-align:right;">
PrecursorMZ
</th>
<th style="text-align:right;">
DelM
</th>
<th style="text-align:right;">
DelM_PPM
</th>
<th style="text-align:right;">
MH
</th>
<th style="text-align:left;">
OriginalPeptide
</th>
<th style="text-align:left;">
Protein
</th>
<th style="text-align:right;">
NTT
</th>
<th style="text-align:right;">
DeNovoScore
</th>
<th style="text-align:right;">
MSGFScore
</th>
<th style="text-align:right;">
MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
Rank_MSGFDB_SpecEValue
</th>
<th style="text-align:right;">
EValue
</th>
<th style="text-align:right;">
QValue
</th>
<th style="text-align:right;">
PepQValue
</th>
<th style="text-align:right;">
IsotopeError
</th>
<th style="text-align:left;">
accession
</th>
<th style="text-align:right;">
calculatedMassToCharge
</th>
<th style="text-align:right;">
chargeState
</th>
<th style="text-align:right;">
experimentalMassToCharge
</th>
<th style="text-align:left;">
isDecoy
</th>
<th style="text-align:left;">
spectrumFile
</th>
<th style="text-align:right;">
spectrumID
</th>
<th style="text-align:left;">
pepSeq
</th>
<th style="text-align:left;">
peptide
</th>
<th style="text-align:right;">
maxAScore
</th>
<th style="text-align:right;">
msmsScore
</th>
<th style="text-align:right;">
absParentMassErrorPPM
</th>
<th style="text-align:right;">
peptides_per_1000aa
</th>
<th style="text-align:left;">
First_AA
</th>
<th style="text-align:left;">
Last_AA
</th>
<th style="text-align:right;">
First_AA_First
</th>
<th style="text-align:right;">
Last_AA_First
</th>
<th style="text-align:right;">
ProtLen
</th>
<th style="text-align:left;">
ModShift
</th>
<th style="text-align:left;">
ModAAs
</th>
<th style="text-align:left;">
SiteLoc
</th>
<th style="text-align:left;">
Site
</th>
<th style="text-align:left;">
SiteCollapsed
</th>
<th style="text-align:left;">
SiteCollapsedFirst
</th>
<th style="text-align:left;">
SiteID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_06_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
12697
</td>
<td style="text-align:right;">
27321
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
2256
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1045.124
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0.858
</td>
<td style="text-align:right;">
3131.346
</td>
<td style="text-align:left;">
A.AAAAAGDS*DS*WDADTFSMEDPVRK.V
</td>
<td style="text-align:left;">
NP_001071138.1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
146
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
NP_001071138.1
</td>
<td style="text-align:right;">
1044.454
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1044.455
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_06_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
27321
</td>
<td style="text-align:left;">
AAAAAGDSDSWDADTFSMEDPVRK
</td>
<td style="text-align:left;">
A.AAAAAGDSDS*WDADT*FSMEDPVRK.V
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.057
</td>
<td style="text-align:right;">
7.722
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
28
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
259
</td>
<td style="text-align:left;">
9, 14
</td>
<td style="text-align:left;">
S, T
</td>
<td style="text-align:left;">
14, 19
</td>
<td style="text-align:left;">
S14, T19
</td>
<td style="text-align:left;">
S14,T19
</td>
<td style="text-align:left;">
S14,T19
</td>
<td style="text-align:left;">
NP_001071138.1-S14sT19t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
875
</td>
<td style="text-align:right;">
23519
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
264
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
952.144
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
1.538
</td>
<td style="text-align:right;">
2854.412
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
165
</td>
<td style="text-align:right;">
129
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
952.142
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
952.144
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
23519
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:right;">
52.349
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.625
</td>
<td style="text-align:right;">
5.305
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
262
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
262
</td>
<td style="text-align:right;">
377
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
XP_006232986.1-T250t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
12873
</td>
<td style="text-align:right;">
23508
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
2213
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.360
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
2.392
</td>
<td style="text-align:right;">
2854.412
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
122
</td>
<td style="text-align:right;">
81
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
714.358
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.360
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
23508
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:right;">
17.480
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
2.472
</td>
<td style="text-align:right;">
5.305
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
262
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
262
</td>
<td style="text-align:right;">
377
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
XP_006232986.1-T250t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S2_07_3Nov17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
2731
</td>
<td style="text-align:right;">
23697
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
502
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.610
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.706
</td>
<td style="text-align:right;">
2854.412
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
135
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
XP_006232986.1
</td>
<td style="text-align:right;">
714.358
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
714.359
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S2_07_3Nov17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
23697
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEDSDDALLK.M
</td>
<td style="text-align:right;">
26.295
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.780
</td>
<td style="text-align:right;">
5.305
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
262
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
262
</td>
<td style="text-align:right;">
377
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
T250
</td>
<td style="text-align:left;">
XP_006232986.1-T250t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
4877
</td>
<td style="text-align:right;">
21265
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
935
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
800.403
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
1.871
</td>
<td style="text-align:right;">
3196.577
</td>
<td style="text-align:left;">
R.AAAASAAEAGIAT*PGTEGERDSDDALLK.M
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
194
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
799.900
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
799.901
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
21265
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEGERDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIATPGT*EGERDSDDALLK.M
</td>
<td style="text-align:right;">
6.213
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
1.902
</td>
<td style="text-align:right;">
10.526
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
265
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
265
</td>
<td style="text-align:right;">
380
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
NP_112621.1-T253t
</td>
</tr>
<tr>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
6826
</td>
<td style="text-align:right;">
21280
</td>
<td style="text-align:left;">
HCD
</td>
<td style="text-align:right;">
1251
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1066.532
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
-0.095
</td>
<td style="text-align:right;">
3196.577
</td>
<td style="text-align:left;">
R.AAAASAAEAGIATPGT*EGERDSDDALLK.M
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
200
</td>
<td style="text-align:right;">
94
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
NP_112621.1
</td>
<td style="text-align:right;">
1066.197
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1066.197
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:left;">
MoTrPAC_Pilot_TMT_P_S1_07_DIL_28Oct17_Elm_AQ-17-10-03
</td>
<td style="text-align:right;">
21280
</td>
<td style="text-align:left;">
AAAASAAEAGIATPGTEGERDSDDALLK
</td>
<td style="text-align:left;">
R.AAAASAAEAGIATPGT*EGERDSDDALLK.M
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
Inf
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
10.526
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
265
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
265
</td>
<td style="text-align:right;">
380
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:left;">
253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
T253
</td>
<td style="text-align:left;">
NP_112621.1-T253t
</td>
</tr>
</tbody>
</table>
<p></br></p>
<p>Table <a href="phosphoproteomics-data.html#tab:phospho-msnid-table">1.8</a> shows the first 6 rows of the processed MS-GF+ output.</p>
</div>
<div id="read-masic-output-1" class="section level3" number="1.2.11">
<h3><span class="header-section-number">1.2.11</span> Read MASIC Output</h3>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="phosphoproteomics-data.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MASIC data</span></span>
<span id="cb62-2"><a href="phosphoproteomics-data.html#cb62-2" aria-hidden="true" tabindex="-1"></a>masic_data <span class="ot">&lt;-</span> <span class="fu">read_masic_data_from_DMS</span>(data_package_num, </span>
<span id="cb62-3"><a href="phosphoproteomics-data.html#cb62-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">interference_score =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="filter-masic-data-1" class="section level3" number="1.2.12">
<h3><span class="header-section-number">1.2.12</span> Filter MASIC Data</h3>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="phosphoproteomics-data.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter MASIC data</span></span>
<span id="cb63-2"><a href="phosphoproteomics-data.html#cb63-2" aria-hidden="true" tabindex="-1"></a>masic_data <span class="ot">&lt;-</span> <span class="fu">filter_masic_data</span>(masic_data, </span>
<span id="cb63-3"><a href="phosphoproteomics-data.html#cb63-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">interference_score_threshold =</span> <span class="fl">0.5</span>,</span>
<span id="cb63-4"><a href="phosphoproteomics-data.html#cb63-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">s2n_threshold =</span> <span class="dv">0</span>)</span></code></pre></div>
</div>
<div id="create-study-design-tables" class="section level3" number="1.2.13">
<h3><span class="header-section-number">1.2.13</span> Create Study Design Tables</h3>
<p>If study design tables are on the DMS, they can be accessed in the following way.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="phosphoproteomics-data.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read study design tables from DMS</span></span>
<span id="cb64-2"><a href="phosphoproteomics-data.html#cb64-2" aria-hidden="true" tabindex="-1"></a>study_design <span class="ot">&lt;-</span> <span class="fu">read_study_design_from_DMS</span>(data_package_num)</span>
<span id="cb64-3"><a href="phosphoproteomics-data.html#cb64-3" aria-hidden="true" tabindex="-1"></a>fractions <span class="ot">&lt;-</span> study_design<span class="sc">$</span>fractions</span>
<span id="cb64-4"><a href="phosphoproteomics-data.html#cb64-4" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> study_design<span class="sc">$</span>samples</span>
<span id="cb64-5"><a href="phosphoproteomics-data.html#cb64-5" aria-hidden="true" tabindex="-1"></a>references <span class="ot">&lt;-</span> study_design<span class="sc">$</span>references</span></code></pre></div>
<p>While the study design tables are not on the DMS, we already created them in Section <a href="global-proteomics-data.html#fetch-study-design-tables">1.1.10</a>. We just need to recreate the fractions table because the dataset names are different.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="phosphoproteomics-data.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read tables from folder</span></span>
<span id="cb65-2"><a href="phosphoproteomics-data.html#cb65-2" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;data/samples.txt&quot;</span>)</span>
<span id="cb65-3"><a href="phosphoproteomics-data.html#cb65-3" aria-hidden="true" tabindex="-1"></a>references <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;data/references.txt&quot;</span>)</span>
<span id="cb65-4"><a href="phosphoproteomics-data.html#cb65-4" aria-hidden="true" tabindex="-1"></a>fractions <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Dataset =</span> <span class="fu">unique</span>(masic_data<span class="sc">$</span>Dataset)) <span class="sc">%&gt;%</span> </span>
<span id="cb65-5"><a href="phosphoproteomics-data.html#cb65-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">PlexID =</span> <span class="fu">gsub</span>(<span class="st">&quot;.*_P_(S</span><span class="sc">\\</span><span class="st">d{1})_.*&quot;</span>, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">1&quot;</span>, Dataset))</span></code></pre></div>
</div>
<div id="phospho-create-crosstab" class="section level3" number="1.2.14">
<h3><span class="header-section-number">1.2.14</span> Create Quantitative Cross-tab</h3>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="phosphoproteomics-data.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set aggregation level</span></span>
<span id="cb66-2"><a href="phosphoproteomics-data.html#cb66-2" aria-hidden="true" tabindex="-1"></a>aggregation_level <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SiteID&quot;</span>)</span>
<span id="cb66-3"><a href="phosphoproteomics-data.html#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create cross-tab</span></span>
<span id="cb66-4"><a href="phosphoproteomics-data.html#cb66-4" aria-hidden="true" tabindex="-1"></a>crosstab <span class="ot">&lt;-</span> <span class="fu">create_crosstab</span>(msnid, masic_data,</span>
<span id="cb66-5"><a href="phosphoproteomics-data.html#cb66-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">aggregation_level =</span> aggregation_level,</span>
<span id="cb66-6"><a href="phosphoproteomics-data.html#cb66-6" aria-hidden="true" tabindex="-1"></a>                            fractions, samples, references)</span></code></pre></div>
<table class="table table-hover table-condensed" style="font-size: 12px; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
<span id="tab:unnamed-chunk-50">Table 1.9: </span><left>First 6 rows of the phospho quantitative cross-tab.</left>
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
R_01
</th>
<th style="text-align:right;">
R_02
</th>
<th style="text-align:right;">
R_03
</th>
<th style="text-align:right;">
R_04
</th>
<th style="text-align:right;">
R_05
</th>
<th style="text-align:right;">
R_06
</th>
<th style="text-align:right;">
R_07
</th>
<th style="text-align:right;">
R_08
</th>
<th style="text-align:right;">
R_09
</th>
<th style="text-align:right;">
S_01
</th>
<th style="text-align:right;">
S_02
</th>
<th style="text-align:right;">
S_03
</th>
<th style="text-align:right;">
S_04
</th>
<th style="text-align:right;">
S_05
</th>
<th style="text-align:right;">
S_06
</th>
<th style="text-align:right;">
S_07
</th>
<th style="text-align:right;">
S_08
</th>
<th style="text-align:right;">
S_09
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
NP_001000283.1-Y132yS137s
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-0.7653521
</td>
<td style="text-align:right;">
-0.5428785
</td>
<td style="text-align:right;">
-0.2119856
</td>
<td style="text-align:right;">
-0.1239261
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.0038421
</td>
<td style="text-align:right;">
-0.2425074
</td>
<td style="text-align:right;">
-0.7416622
</td>
<td style="text-align:right;">
-0.7497171
</td>
<td style="text-align:right;">
-0.4879016
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001064.1-Y129y
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1.0408865
</td>
<td style="text-align:right;">
-0.2976259
</td>
<td style="text-align:right;">
-0.7301538
</td>
<td style="text-align:right;">
-0.1385762
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.1594274
</td>
<td style="text-align:right;">
-0.6587720
</td>
<td style="text-align:right;">
0.1932351
</td>
<td style="text-align:right;">
-0.6370354
</td>
<td style="text-align:right;">
-0.4461276
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S241s
</td>
<td style="text-align:right;">
-0.5441083
</td>
<td style="text-align:right;">
-0.0291611
</td>
<td style="text-align:right;">
-0.5540885
</td>
<td style="text-align:right;">
-0.1812456
</td>
<td style="text-align:right;">
-0.5675833
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-0.0427948
</td>
<td style="text-align:right;">
-0.803056
</td>
<td style="text-align:right;">
-0.5236890
</td>
<td style="text-align:right;">
-1.0398773
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S242s
</td>
<td style="text-align:right;">
-0.2806018
</td>
<td style="text-align:right;">
-0.2939827
</td>
<td style="text-align:right;">
-0.3923442
</td>
<td style="text-align:right;">
-0.3242804
</td>
<td style="text-align:right;">
-0.6512914
</td>
<td style="text-align:right;">
-1.3958746
</td>
<td style="text-align:right;">
-0.6640721
</td>
<td style="text-align:right;">
-0.7046127
</td>
<td style="text-align:right;">
-0.3091972
</td>
<td style="text-align:right;">
-0.4914234
</td>
<td style="text-align:right;">
-1.210593
</td>
<td style="text-align:right;">
-0.6421375
</td>
<td style="text-align:right;">
-0.8869142
</td>
<td style="text-align:right;">
-0.3079991
</td>
<td style="text-align:right;">
-0.8072458
</td>
<td style="text-align:right;">
-0.4801256
</td>
<td style="text-align:right;">
-0.4458455
</td>
<td style="text-align:right;">
-0.7588038
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S699s
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-0.8584401
</td>
<td style="text-align:right;">
-0.4918316
</td>
<td style="text-align:right;">
-0.2745069
</td>
<td style="text-align:right;">
0.2867235
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-0.7930334
</td>
<td style="text-align:right;">
-1.2173509
</td>
<td style="text-align:right;">
-0.4608541
</td>
<td style="text-align:right;">
-1.2596689
</td>
<td style="text-align:right;">
-0.8004630
</td>
</tr>
<tr>
<td style="text-align:left;">
NP_001001512.2-S746s
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1.5981389
</td>
<td style="text-align:right;">
-2.4732843
</td>
<td style="text-align:right;">
-0.4736739
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
-1.4611181
</td>
<td style="text-align:right;">
-2.2975481
</td>
<td style="text-align:right;">
-1.6828088
</td>
<td style="text-align:right;">
-2.3039510
</td>
<td style="text-align:right;">
-1.8632844
</td>
</tr>
</tbody>
</table>
<p></br></p>
<p>We will save the cross-tab for later sections.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="phosphoproteomics-data.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Modify cross-tab for saving</span></span>
<span id="cb67-2"><a href="phosphoproteomics-data.html#cb67-2" aria-hidden="true" tabindex="-1"></a>crosstab <span class="ot">&lt;-</span> crosstab <span class="sc">%&gt;%</span> </span>
<span id="cb67-3"><a href="phosphoproteomics-data.html#cb67-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb67-4"><a href="phosphoproteomics-data.html#cb67-4" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SiteID&quot;</span>)</span>
<span id="cb67-5"><a href="phosphoproteomics-data.html#cb67-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-6"><a href="phosphoproteomics-data.html#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Save cross-tab</span></span>
<span id="cb67-7"><a href="phosphoproteomics-data.html#cb67-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(crosstab, <span class="at">file =</span> <span class="st">&quot;data/phosphosite_quant_crosstab.txt&quot;</span>,</span>
<span id="cb67-8"><a href="phosphoproteomics-data.html#cb67-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>

</div>
</div>
<!-- </div> -->
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-cohen_origins_2002" class="csl-entry">
Cohen, P., The Origins of Protein Phosphorylation, <em>Nature Cell Biology</em>, vol. <strong>4</strong>, no. 5, pp. E127–30, accessed October 21, 2021, from <a href="https://www.nature.com/articles/ncb0502-e127">https://www.nature.com/articles/ncb0502-e127</a>, May 2002. DOI: <a href="https://doi.org/10.1038/ncb0502-e127">10.1038/ncb0502-e127</a>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="global-proteomics-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="label-free-quantification.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
