<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>References | Proteomics Data Analysis in R/Bioconductor</title>
<meta name="author" content="Tyler Sagendorf">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Proteomics Data Analysis in R/Bioconductor</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome!</a></li>
<li><a class="" href="isobaric-quantification-pipelines.html"><span class="header-section-number">1</span> Isobaric Quantification Pipelines</a></li>
<li><a class="" href="exploratory-data-analysis.html"><span class="header-section-number">2</span> Exploratory Data Analysis</a></li>
<li><a class="" href="DEA.html"><span class="header-section-number">3</span> Differential Expression Analysis</a></li>
<li><a class="" href="pathway-analysis.html"><span class="header-section-number">4</span> Pathway Analysis</a></li>
<li><a class="active" href="references-2.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="references-2" class="section level1 unnumbered">
<h1>References<a class="anchor" aria-label="anchor" href="#references-2"><i class="fas fa-link"></i></a>
</h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-R-rmarkdown" class="csl-entry">
Allaire, J., Xie, Y., McPherson, J., Luraschi, J., Ushey, K., Atkins, A., Wickham, H., Cheng, J., Chang, W. and Iannone, R., <em>Rmarkdown: Dynamic Documents for r</em>, from <a href="https://CRAN.R-project.org/package=rmarkdown">https://CRAN.R-project.org/package=rmarkdown</a>, 2021.
</div>
<div id="ref-R-org.Hs.eg.db" class="csl-entry">
Carlson, M., <em>Org.hs.eg.db: Genome Wide Annotation for Human</em>, 2021.
</div>
<div id="ref-R-PFAM.db" class="csl-entry">
Carlson, M., Liu, T.-Y., Lin, C., Falcon, S., Zhang, J. and MacDonald, J. W., <em>PFAM.db: A Set of Protein ID Mappings for PFAM</em>, 2021.
</div>
<div id="ref-R-biomaRt" class="csl-entry">
Durinck, S. and Huber, W., <em>biomaRt: Interface to BioMart Databases (i.e. Ensembl)</em>, 2021.
</div>
<div id="ref-biomaRt2005" class="csl-entry">
Durinck, S., Moreau, Y., Kasprzyk, A., Davis, S., De Moor, B., Brazma, A. and Huber, W., BioMart and Bioconductor: A Powerful Link Between Biological Databases and Microarray Data Analysis, <em>Bioinformatics</em>, vol. <strong>21</strong>, pp. 3439–40, 2005.
</div>
<div id="ref-biomaRt2009" class="csl-entry">
Durinck, S., Spellman, P. T., Birney, E. and Huber, W., Mapping Identifiers for the Integration of Genomic Datasets with the r/Bioconductor Package biomaRt, <em>Nature Protocols</em>, vol. <strong>4</strong>, pp. 1184–91, 2009.
</div>
<div id="ref-falcon_how_2021" class="csl-entry">
Falcon, S. and Gentleman, R., How <span>To</span> <span>Use</span> <span>GOstats</span> <span>Testing</span> <span>Gene</span> <span>Lists</span> for <span>GO</span> <span>Term</span> <span>Association</span>, p. 10, accessed October 11, 2021, from <a href="https://bioconductor.org/packages/release/bioc/vignettes/GOstats/inst/doc/GOstatsHyperG.pdf">https://bioconductor.org/packages/release/bioc/vignettes/GOstats/inst/doc/GOstatsHyperG.pdf</a>, May 2021. DOI: <a href="https://doi.org/10.18129/B9.bioc.GOstats">10.18129/B9.bioc.GOstats</a>
</div>
<div id="ref-GOstats2007" class="csl-entry">
Falcon, S. and Gentleman, R., Using <span>G</span><span>O</span>stats to Test Gene Lists for <span>G</span><span>O</span> Term Association., <em>Bioinformatics</em>, vol. <strong>23</strong>, no. 2, pp. 257–58, 2007.
</div>
<div id="ref-R-GOstats" class="csl-entry">
Gentleman, R., <em>GOstats: Tools for Manipulating GO and Microarrays</em>, 2021.
</div>
<div id="ref-goeman_multiple_2008" class="csl-entry">
Goeman, J. J. and Mansmann, U., Multiple Testing on the Directed Acyclic Graph of Gene Ontology, <em>Bioinformatics</em>, vol. <strong>24</strong>, no. 4, pp. 537–44, accessed October 11, 2021, from <a href="https://doi.org/10.1093/bioinformatics/btm628">https://doi.org/10.1093/bioinformatics/btm628</a>, February 2008. DOI: <a href="https://doi.org/10.1093/bioinformatics/btm628">10.1093/bioinformatics/btm628</a>
</div>
<div id="ref-huang_bioinformatics_2009" class="csl-entry">
Huang, D. W., Sherman, B. T. and Lempicki, R. A., Bioinformatics Enrichment Tools: Paths Toward the Comprehensive Functional Analysis of Large Gene Lists, <em>Nucleic Acids Research</em>, vol. <strong>37</strong>, no. 1, pp. 1–13, accessed September 7, 2021, from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615629/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2615629/</a>, January 2009. DOI: <a href="https://doi.org/10.1093/nar/gkn923">10.1093/nar/gkn923</a>
</div>
<div id="ref-korotkevich_fast_2016" class="csl-entry">
Korotkevich, G., Sukhov, V., Budin, N., Shpak, B., Artyomov, M. N., and Sergushichev, A., Fast Gene Set Enrichment Analysis, Bioinformatics, Jun. 2016.
</div>
<div id="ref-maleki_gene_2020" class="csl-entry">
Maleki, F., Ovens, K., Hogan, D. J. and Kusalik, A. J., Gene <span>Set</span> <span>Analysis</span>: <span>Challenges</span>, <span>Opportunities</span>, and <span>Future</span> <span>Research</span>, <em>Frontiers in Genetics</em>, vol. <strong>11</strong>, p. 654, accessed September 7, 2021, from <a href="https://www.frontiersin.org/article/10.3389/fgene.2020.00654">https://www.frontiersin.org/article/10.3389/fgene.2020.00654</a>, 2020. DOI: <a href="https://doi.org/10.3389/fgene.2020.00654">10.3389/fgene.2020.00654</a>
</div>
<div id="ref-mootha_pgc-1-responsive_2003" class="csl-entry">
Mootha, V. K., Lindgren, C. M., Eriksson, K.-F., Subramanian, A., Sihag, S., Lehar, J., Puigserver, P., et al., <span>PGC</span>-1α-Responsive Genes Involved in Oxidative Phosphorylation Are Coordinately Downregulated in Human Diabetes, <em>Nature Genetics</em>, vol. <strong>34</strong>, no. 3, pp. 267–73, accessed September 19, 2021, from <a href="https://www.nature.com/articles/ng1180">https://www.nature.com/articles/ng1180</a>, July 2003. DOI: <a href="https://doi.org/10.1038/ng1180">10.1038/ng1180</a>
</div>
<div id="ref-R-tibble" class="csl-entry">
Müller, K. and Wickham, H., <em>Tibble: Simple Data Frames</em>, from <a href="https://CRAN.R-project.org/package=tibble">https://CRAN.R-project.org/package=tibble</a>, 2021.
</div>
<div id="ref-R-Biostrings" class="csl-entry">
Pagès, H., Aboyoun, P., Gentleman, R. and DebRoy, S., <em>Biostrings: Efficient Manipulation of Biological Strings</em>, from <a href="https://bioconductor.org/packages/Biostrings">https://bioconductor.org/packages/Biostrings</a>, 2021.
</div>
<div id="ref-pesquita_semantic_2017" class="csl-entry">
Pesquita, C., Semantic <span>Similarity</span> in the <span>Gene</span> <span>Ontology</span>, in <em>The <span>Gene</span> <span>Ontology</span> <span>Handbook</span></em>, C. Dessimoz and N. Škunca, Eds., New York, NY: Springer, accessed October 7, 2021, from <a href="https://doi.org/10.1007/978-1-4939-3743-1_12">https://doi.org/10.1007/978-1-4939-3743-1_12</a>, pp. 161–73, 2017.
</div>
<div id="ref-R-MSnID" class="csl-entry">
Petyuk, V., <em>MSnID: Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications</em>, 2021a.
</div>
<div id="ref-R-MSnSet.utils" class="csl-entry">
Petyuk, V., <em>MSnSet.utils: Miscellaneous Functions for Proteomics Data Analysis</em>, 2015.
</div>
<div id="ref-R-PlexedPiperTestData" class="csl-entry">
Petyuk, V., <em>PlexedPiperTestData: Example of MASIC and MS-GF+ Ouput</em>, 2021b.
</div>
<div id="ref-R-PNNL.DMS.utils" class="csl-entry">
Petyuk, V., Nestor, M. and Moon, J., <em>PNNL.DMS.utils: Tools for Accessing PNNL’s Data Management System</em>, 2021.
</div>
<div id="ref-noauthor_relations_2021" class="csl-entry">
Relations in the <span>Gene</span> <span>Ontology</span>, <em>Gene Ontology Resource</em>, October 2021.
</div>
<div id="ref-limma2015" class="csl-entry">
Ritchie, M. E., Phipson, B., Wu, D., Hu, Y., Law, C. W., Shi, W. and Smyth, G. K., <span class="nocase">limma</span> Powers Differential Expression Analyses for <span>RNA</span>-Sequencing and Microarray Studies, <em>Nucleic Acids Research</em>, vol. <strong>43</strong>, no. 7, p. e47, 2015. DOI: <a href="https://doi.org/10.1093/nar/gkv007">10.1093/nar/gkv007</a>
</div>
<div id="ref-R-limma" class="csl-entry">
Smyth, G., Hu, Y., Ritchie, M., Silver, J., Wettenhall, J., McCarthy, D., Wu, D., et al., <em>Limma: Linear Models for Microarray Data</em>, from <a href="http://bioinf.wehi.edu.au/limma">http://bioinf.wehi.edu.au/limma</a>, 2021.
</div>
<div id="ref-subramanian_gene_2005" class="csl-entry">
Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., et al., Gene Set Enrichment Analysis: <span>A</span> Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles, <em>Proceedings of the National Academy of Sciences of the United States of America</em>, vol. <strong>102</strong>, no. 43, pp. 15545–50, accessed September 7, 2021, from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/</a>, October 2005. DOI: <a href="https://doi.org/10.1073/pnas.0506580102">10.1073/pnas.0506580102</a>
</div>
<div id="ref-R-PlexedPiper" class="csl-entry">
vladislav.petyuk@pnnl.gov, V. P., <em>PlexedPiper: Pipeline for Isobaric Quantification.</em>, 2021.
</div>
<div id="ref-R-downlit" class="csl-entry">
Wickham, H., <em>Downlit: Syntax Highlighting and Automatic Linking</em>, from <a href="https://CRAN.R-project.org/package=downlit">https://CRAN.R-project.org/package=downlit</a>, 2020.
</div>
<div id="ref-ggplot22016" class="csl-entry">
Wickham, H., <em>Ggplot2: Elegant Graphics for Data Analysis</em>, Springer-Verlag New York, from <a href="https://ggplot2.tidyverse.org">https://ggplot2.tidyverse.org</a>, 2016.
</div>
<div id="ref-R-ggplot2" class="csl-entry">
Wickham, H., Chang, W., Henry, L., Pedersen, T. L., Takahashi, K., Wilke, C., Woo, K., Yutani, H. and Dunnington, D., <em>Ggplot2: Create Elegant Data Visualisations Using the Grammar of Graphics</em>, from <a href="https://CRAN.R-project.org/package=ggplot2">https://CRAN.R-project.org/package=ggplot2</a>, 2021.
</div>
<div id="ref-R-dplyr" class="csl-entry">
Wickham, H., François, R., Henry, L. and Müller, K., <em>Dplyr: A Grammar of Data Manipulation</em>, from <a href="https://CRAN.R-project.org/package=dplyr">https://CRAN.R-project.org/package=dplyr</a>, 2021.
</div>
<div id="ref-R-scales" class="csl-entry">
Wickham, H. and Seidel, D., <em>Scales: Scale Functions for Visualization</em>, from <a href="https://CRAN.R-project.org/package=scales">https://CRAN.R-project.org/package=scales</a>, 2020.
</div>
<div id="ref-R-bookdown" class="csl-entry">
Xie, Y., <em>Bookdown: Authoring Books and Technical Documents with r Markdown</em>, from <a href="https://CRAN.R-project.org/package=bookdown">https://CRAN.R-project.org/package=bookdown</a>, 2021a.
</div>
<div id="ref-bookdown2016" class="csl-entry">
Xie, Y., <em>Bookdown: Authoring Books and Technical Documents with <span>R</span> Markdown</em>, Boca Raton, Florida: Chapman; Hall/CRC, from <a href="https://bookdown.org/yihui/bookdown">https://bookdown.org/yihui/bookdown</a>, 2016.
</div>
<div id="ref-knitr2015" class="csl-entry">
Xie, Y., <em>Dynamic Documents with <span>R</span> and Knitr</em>, Boca Raton, Florida: Chapman; Hall/CRC, from <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>, 2015.
</div>
<div id="ref-knitr2014" class="csl-entry">
Xie, Y., Knitr: A Comprehensive Tool for Reproducible Research in <span>R</span>, in <em>Implementing Reproducible Computational Research</em>, V. Stodden F. Leisch and R. D. Peng, Eds., Chapman; Hall/CRC, from <a href="http://www.crcpress.com/product/isbn/9781466561595">http://www.crcpress.com/product/isbn/9781466561595</a>, 2014.
</div>
<div id="ref-R-knitr" class="csl-entry">
Xie, Y., <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>, from <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>, 2021b.
</div>
<div id="ref-rmarkdown2018" class="csl-entry">
Xie, Y., Allaire, J. J. and Grolemund, G., <em>R Markdown: The Definitive Guide</em>, Boca Raton, Florida: Chapman; Hall/CRC, from <a href="https://bookdown.org/yihui/rmarkdown">https://bookdown.org/yihui/rmarkdown</a>, 2018.
</div>
<div id="ref-rmarkdown2020" class="csl-entry">
Xie, Y., Dervieux, C. and Riederer, E., <em>R Markdown Cookbook</em>, Boca Raton, Florida: Chapman; Hall/CRC, from <a href="https://bookdown.org/yihui/rmarkdown-cookbook">https://bookdown.org/yihui/rmarkdown-cookbook</a>, 2020.
</div>
<div id="ref-R-clusterProfiler" class="csl-entry">
Yu, G., <em>clusterProfiler: Statistical Analysis and Visualization of Functional Profiles for Genes and Gene Clusters</em>, from <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">https://yulab-smu.top/biomedical-knowledge-mining-book/</a>, 2021a.
</div>
<div id="ref-GOSemSim2020" class="csl-entry">
Yu, G., Gene Ontology Semantic Similarity Analysis Using GOSemSim, <em>Methods in Molecular Biology</em>, vol. <strong>2117</strong>, pp. 207–15, 2020. DOI: <a href="https://doi.org/10.1007/978-1-0716-0301-7_11">10.1007/978-1-0716-0301-7_11</a>
</div>
<div id="ref-R-GOSemSim" class="csl-entry">
Yu, G., <em>GOSemSim: GO-Terms Semantic Similarity Measures</em>, from <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">https://yulab-smu.top/biomedical-knowledge-mining-book/</a>, 2021b.
</div>
<div id="ref-R-ReactomePA" class="csl-entry">
Yu, G., <em>ReactomePA: Reactome Pathway Analysis</em>, from <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/">https://yulab-smu.top/biomedical-knowledge-mining-book/</a>, 2021c.
</div>
<div id="ref-ReactomePA2016" class="csl-entry">
Yu, G. and He, Q.-Y., ReactomePA: An r/Bioconductor Package for Reactome Pathway Analysis and Visualization, <em>Molecular BioSystems</em>, vol. <strong>12</strong>, no. 12, pp. 477–79, from <a href="http://pubs.rsc.org/en/Content/ArticleLanding/2015/MB/C5MB00663E">http://pubs.rsc.org/en/Content/ArticleLanding/2015/MB/C5MB00663E</a>, 2016. DOI: <a href="https://doi.org/10.1039/C5MB00663E">10.1039/C5MB00663E</a>
</div>
<div id="ref-GOSemSim2010" class="csl-entry">
Yu, G., Li, F., Qin, Y., Bo, X., Wu, Y. and Wang, S., GOSemSim: An r Package for Measuring Semantic Similarity Among GO Terms and Gene Products, <em>Bioinformatics</em>, vol. <strong>26</strong>, no. 7, pp. 976–78, 2010. DOI: <a href="https://doi.org/10.1093/bioinformatics/btq064">10.1093/bioinformatics/btq064</a>
</div>
<div id="ref-clusterProfiler2012" class="csl-entry">
Yu, G., Wang, L.-G., Han, Y. and He, Q.-Y., clusterProfiler: An r Package for Comparing Biological Themes Among Gene Clusters, <em>OMICS: A Journal of Integrative Biology</em>, vol. <strong>16</strong>, no. 5, pp. 284–87, 2012. DOI: <a href="https://doi.org/10.1089/omi.2011.0118">10.1089/omi.2011.0118</a>
</div>
<div id="ref-R-kableExtra" class="csl-entry">
Zhu, H., <em>kableExtra: Construct Complex Table with Kable and Pipe Syntax</em>, from <a href="https://CRAN.R-project.org/package=kableExtra">https://CRAN.R-project.org/package=kableExtra</a>, 2021.
</div>
<div id="ref-zyla_ranking_2017" class="csl-entry">
Zyla, J., Marczyk, M., Weiner, J. and Polanska, J., Ranking Metrics in Gene Set Enrichment Analysis: Do They Matter?, <em>BMC Bioinformatics</em>, vol. <strong>18</strong>, no. 1, p. 256, accessed September 19, 2021, from <a href="https://doi.org/10.1186/s12859-017-1674-0">https://doi.org/10.1186/s12859-017-1674-0</a>, May 2017. DOI: <a href="https://doi.org/10.1186/s12859-017-1674-0">10.1186/s12859-017-1674-0</a>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="pathway-analysis.html"><span class="header-section-number">4</span> Pathway Analysis</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#references-2">References</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial/blob/master/references.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/PNNL-Comp-Mass-Spec/proteomics-data-analysis-tutorial/edit/master/references.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Proteomics Data Analysis in R/Bioconductor</strong>" was written by Tyler Sagendorf. It was last built on October 19, 2021.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
